---
title: "Multi-omics Mouse Embryo DBiT-Seq"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Multi-omics Mouse Embryo DBiT-Seq}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# 1 Dataset explanation

The dataset was created by [Liu, et al 2020](https://www.sciencedirect.com/science/article/pii/S0092867420313908?via%3Dihub) and downloaded from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE137986. For running this tutorial, we will use the dataset E10 Whole (50 Î¼m) 2.


# 2 Start Giotto

```{r, eval=FALSE}
# Ensure Giotto Suite is installed
if(!"Giotto" %in% installed.packages()) {
  pak::pkg_install("drieslab/Giotto")
}
library(Giotto)

# Ensure the Python environment for Giotto has been installed
genv_exists = checkGiottoEnvironment()
if(!genv_exists){
  # The following command need only be run once to install the Giotto environment
  installGiottoEnvironment()
}
```

# 3 Create Giotto object

Read expression matrix 22846 genes, 22 proteins, 901 cells

```{r, eval = FALSE}
## RNA
rna_expression = read.table("data/GSE137986_RAW/GSM4189613_0702cL.tsv.gz", 
                            sep = "\t",
                            header = TRUE)
rownames(rna_expression) = rna_expression$X
```

```{r, eval = FALSE}
## Protein
protein_expression = read.table("data/GSE137986_RAW/GSM4202307_0702aL.tsv.gz", 
                                sep = "\t",
                                header = TRUE)
rownames(protein_expression) = protein_expression$X
```

Transpose matrix

```{r, eval = FALSE}
rna_expression = t(rna_expression[-1])
protein_expression = t(protein_expression[-1])
```

Get the spatial coordinates

```{r, eval = FALSE}
spatial_coords = data.frame(cell_ID = colnames(rna_expression))
spatial_coords = cbind(spatial_coords, 
                       tidyr::separate(spatial_coords, cell_ID, c("x","y"), sep = "x"))

spatial_coords$x = as.numeric(spatial_coords$x)
spatial_coords$y = as.numeric(spatial_coords$y)*-1
```

Create the Giotto object

```{r, eval = FALSE}
save_dir = 'results'
instrs = createGiottoInstructions(save_dir = save_dir,
                                  save_plot = TRUE,
                                  show_plot = TRUE)

giottoObject = createGiottoObject(expression = list(raw = rna_expression,
                                                    raw = protein_expression),
                                  expression_feat = c('rna', 'protein'),
                                  spatial_locs = spatial_coords,
                                  instructions = instrs)
```

# 4 Processing

## Filtering

```{r, eval = FALSE}
## RNA
giottoObject = filterGiotto(gobject = giottoObject,
                            expression_threshold = 1,
                            feat_det_in_min_cells = 1,
                            min_det_feats_per_cell = 1,
                            expression_values = 'raw',
                            verbose = TRUE)

## Protein
giottoObject = filterGiotto(gobject = giottoObject,
                            spat_unit = 'cell',
                            feat_type = 'protein',
                            expression_threshold = 1,
                            feat_det_in_min_cells = 1,
                            min_det_feats_per_cell = 1,
                            expression_values = 'raw',
                            verbose = TRUE)
```

## Normalization

```{r, eval = FALSE}
## RNA
giottoObject = normalizeGiotto(gobject = giottoObject, 
                               scalefactor = 6000, 
                               verbose = TRUE)

## Protein
giottoObject = normalizeGiotto(gobject = giottoObject, 
                               spat_unit = 'cell',
                               feat_type = 'protein',
                               scalefactor = 6000, 
                               verbose = TRUE)
```

## Statistics

```{r, eval = FALSE}
## RNA
giottoObject = addStatistics(gobject = giottoObject)

spatPlot2D(giottoObject,
           spat_unit = 'cell',
           feat_type = 'rna',
           cell_color = "nr_feats",
           color_as_factor = FALSE,
           point_size = 3.5)

spatPlot2D(giottoObject,
           cell_color = "total_expr",
           color_as_factor = FALSE,
           point_size = 3.5)
```

![](images/mouse_embryo_dbitseq/0-spatPlot2D.png)
![](images/mouse_embryo_dbitseq/1-spatPlot2D.png)


```{r, eval = FALSE}
## Protein
giottoObject = addStatistics(gobject = giottoObject,
                             spat_unit = 'cell',
                             feat_type = 'protein')

spatPlot2D(giottoObject,
           spat_unit = 'cell',
           feat_type = 'protein',
           cell_color = "total_expr",
           color_as_factor = FALSE,
           point_size = 3.5)
```

![](images/mouse_embryo_dbitseq/2-spatPlot2D.png)

# 5 Dimention Reduction

## Identify highly variable features (HVF)

```{r, eval = FALSE}
giottoObject = calculateHVF(gobject = giottoObject)
```

![](images/mouse_embryo_dbitseq/3-HVFplot.png)


## PCA

```{r, eval = FALSE}
# RNA
giottoObject <- runPCA(gobject = giottoObject)
```

```{r, eval = FALSE}
screePlot(giottoObject, ncp = 30)
```

![](images/mouse_embryo_dbitseq/4-screePlot.png)

```{r, eval = FALSE}
plotPCA(gobject = giottoObject)
```

![](images/mouse_embryo_dbitseq/5-PCA.png)

```{r, eval = FALSE}
# Protein
giottoObject <- runPCA(gobject = giottoObject,
                       spat_unit = 'cell',
                       feat_type = 'protein')
```

```{r, eval = FALSE}
screePlot(giottoObject, 
          spat_unit = 'cell',
          feat_type = 'protein',
          ncp = 30)
```

![](images/mouse_embryo_dbitseq/6-screePlot.png)

```{r, eval = FALSE}
plotPCA(gobject = giottoObject,
        spat_unit = 'cell',
        feat_type = 'protein')
```

![](images/mouse_embryo_dbitseq/7-PCA.png)

# 6 Clustering

## UMAP

```{r, eval = FALSE}
# RNA
giottoObject <- runUMAP(giottoObject, 
                        dimensions_to_use = 1:10)
```

```{r, eval = FALSE}
plotUMAP(gobject = giottoObject)
```

![](images/mouse_embryo_dbitseq/8-UMAP.png)

```{r, eval = FALSE}
# Protein
giottoObject <- runUMAP(giottoObject, 
                        spat_unit = 'cell',
                        feat_type = 'protein',
                        dimensions_to_use = 1:10)
```

```{r, eval = FALSE}
plotUMAP(gobject = giottoObject,
         spat_unit = 'cell',
         feat_type = 'protein')
```

![](images/mouse_embryo_dbitseq/9-UMAP.png)

## Create shared nearest network (sNN) and perform leiden clustering

```{r, eval = FALSE}
# RNA
giottoObject <- createNearestNetwork(gobject = giottoObject, 
                                     dimensions_to_use = 1:10, 
                                     k = 30)

giottoObject <- doLeidenCluster(gobject = giottoObject, 
                                resolution = 1,
                                n_iterations = 1000)

# Protein
giottoObject <- createNearestNetwork(gobject = giottoObject, 
                                     spat_unit = 'cell', 
                                     feat_type = 'protein', 
                                     dimensions_to_use = 1:10, 
                                     k = 30)

giottoObject <- doLeidenCluster(gobject = giottoObject, 
                                spat_unit = 'cell', 
                                feat_type = 'protein', 
                                resolution = 1,
                                n_iterations = 1000)
```

## Visualize UMAP cluster results

```{r, eval = FALSE}
# RNA
plotUMAP(gobject = giottoObject, 
         cell_color = 'leiden_clus', 
         show_NN_network = FALSE, 
         point_size = 2,
         title = "",
         axis_text = 14,
         axis_title = 18,
         legend_text = 14)
```

![](images/mouse_embryo_dbitseq/10-UMAP.png)

```{r, eval = FALSE}
# Protein
plotUMAP(gobject = giottoObject, 
         spat_unit = 'cell', 
         feat_type = 'protein',
         cell_color = 'leiden_clus', 
         show_NN_network = FALSE, 
         point_size = 2,
         title = "",
         axis_text = 14,
         axis_title = 18,
         legend_text = 14)
```

![](images/mouse_embryo_dbitseq/11-UMAP.png)

## Visualize  spatial results

```{r, eval = FALSE}
# RNA
spatPlot2D(gobject = giottoObject, 
           show_image = FALSE,
           cell_color = 'leiden_clus',
           point_size = 3.5,
           title = "",
           axis_text = 14,
           axis_title = 18,
           legend_text = 14)
```

![](images/mouse_embryo_dbitseq/12-spatPlot2D.png)

```{r, eval = FALSE}
# Protein
spatPlot2D(gobject = giottoObject, 
           spat_unit = 'cell', 
           feat_type = 'protein',
           show_image = FALSE,
           cell_color = 'leiden_clus',
           point_size = 3.5,
           title = "",
           axis_text = 14,
           axis_title = 18,
           legend_text = 14)
```

![](images/mouse_embryo_dbitseq/13-spatPlot2D.png)

# 7 Multi-omics integration

```{r, eval = FALSE}
# RNA
giottoObject <- createNearestNetwork(gobject = giottoObject, 
                                     type = 'kNN',
                                     dimensions_to_use = 1:10, 
                                     k = 20)

# Protein
giottoObject <- createNearestNetwork(gobject = giottoObject, 
                                     spat_unit = 'cell', 
                                     feat_type = 'protein', 
                                     type = 'kNN',
                                     dimensions_to_use = 1:10, 
                                     k = 20)
```

```{r, eval = FALSE}
giottoObject <- runWNN(giottoObject,
                       spat_unit = "cell",
                       modality_1 = "rna",
                       modality_2 = "protein",
                       pca_name_modality_1 = "pca",
                       pca_name_modality_2 = "protein.pca",
                       k = 20,
                       verbose = TRUE)
```

```{r, eval = FALSE}
giottoObject <- runIntegratedUMAP(giottoObject,
                                  modality1 = "rna",
                                  modality2 = "protein",
                                  spread = 7,
                                  min_dist = 1,
                                  force = FALSE)
```

```{r, eval = FALSE}
giottoObject <- doLeidenCluster(gobject = giottoObject,
                                spat_unit = "cell",
                                feat_type = "rna",
                                nn_network_to_use = "kNN",
                                network_name = "integrated_kNN",
                                name = "integrated_leiden_clus",
                                resolution = 1)
```

```{r, eval = FALSE}
plotUMAP(gobject = giottoObject,
         spat_unit = "cell",
         feat_type = "rna",
         cell_color = 'integrated_leiden_clus',
         dim_reduction_name = "integrated.umap",
         point_size = 2,
         title = "",
         axis_text = 14,
         axis_title = 18,
         legend_text = 14)
```

![](images/mouse_embryo_dbitseq/14-UMAP.png)

```{r, eval = FALSE}
spatPlot2D(giottoObject,
           spat_unit = "cell",
           feat_type = "rna",
           cell_color = "integrated_leiden_clus",
           point_size = 3.5,
           show_image = FALSE,
           title = "",
           axis_text = 14,
           axis_title = 18,
           legend_text = 14)
```

![](images/mouse_embryo_dbitseq/15-spatPlot2D.png)

# 8 Deconvolution

We used the scRNAseq from [Cao et al., 2019](https://www.nature.com/articles/s41586-019-0969-x) as reference. J. Cao, M. Spielmann, X. Qiu, X. Huang, D.M. Ibrahim, A.J. Hill, F. Zhang, S. Mundlos, L. Christiansen, F.J. Steemers, et al. The single-cell transcriptional landscape of mammalian organogenesis

## Preparation of the scRNAseq

```{r, eval = FALSE}
gene_count_cleaned_sampled_100k <- readRDS("gene_count_cleaned_sampled_100k.RDS")

cell_annotation = data.table::fread("data/scrnaseq/cell_annotate.csv")
cell_annotation = cell_annotation[,c("sample", "Total_mRNAs", "num_genes_expressed", "Main_cell_type")]
colnames(cell_annotation)[1] = "cell_ID"

cell_annotation = cell_annotation[cell_annotation$cell_ID %in% colnames(gene_count_cleaned_sampled_100k),]
```

```{r, eval = FALSE}
sc_giotto = createGiottoObject(expression = gene_count_cleaned_sampled_100k)
sc_giotto = addCellMetadata(sc_giotto,
                            new_metadata = cell_annotation)
```

```{r, eval = FALSE}
sc_giotto = normalizeGiotto(sc_giotto, 
                            log_norm = FALSE,
                            scale_feats = FALSE, 
                            scale_cells = FALSE)
```

Find markergenes

```{r, eval = FALSE}
markers_scran <- findMarkers_one_vs_all(gobject = sc_giotto,
                                        method = "scran",
                                        expression_values = "normalized",
                                        cluster_column = 'Main_cell_type',
                                        min_feats = 3)
markergenes_scran <- unique(markers_scran[, head(.SD, 30), by = "cluster"][["feats"]])
```

Create DWLS matrix

```{r, eval=FALSE}
DWLS_matrix_direct <- makeSignMatrixDWLSfromMatrix(
  matrix = getExpression(sc_giotto,
                         values = "normalized",
                         output = "matrix"),
  cell_type = pDataDT(sc_giotto)$Main_cell_type,
  sign_gene = markergenes_scran)
```

Fix gene names

```{r, eval = FALSE}
sc_gene_names = read.csv("data/scrnaseq/GSE119945_gene_annotate.csv")

ENSMUS_names = rownames(DWLS_matrix_direct)
sc_gene_names = sc_gene_names[sc_gene_names$gene_id %in% ENSMUS_names,]

rownames(DWLS_matrix_direct) = sc_gene_names$gene_short_name
```

## Run deconvolution

```{r, eval=FALSE}
# run DWLS using integrated leiden clusters
giottoObject <- runDWLSDeconv(gobject = giottoObject,
                              sign_matrix = DWLS_matrix_direct,
                              cluster_column = "integrated_leiden_clus")
```

Plot DWLS deconvolution result

```{r, eval = FALSE}
# Plot DWLS deconvolution result with Pie plots dataset 1
spatDeconvPlot(giottoObject,
               show_image = FALSE,
               radius = 0.5,
               return_plot = TRUE,
               save_plot = TRUE,
               save_param = list(save_name = "integrated_deconvolution"),
               title = "",
               axis_text = 14,
               axis_title = 18,
               legend_text = 0,
               background_color = "black")
```

![](images/mouse_embryo_dbitseq/integrated_deconvolution.png)

# 9 Session Info

```{r, eval = FALSE}
sessionInfo()
```

```{r, eval = FALSE}
R version 4.3.2 (2023-10-31)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Giotto_4.0.2      GiottoClass_0.1.3

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3          rstudioapi_0.15.0           jsonlite_1.8.8             
  [4] magrittr_2.0.3              magick_2.8.2                farver_2.1.1               
  [7] rmarkdown_2.25              zlibbioc_1.48.0             ragg_1.2.7                 
 [10] vctrs_0.6.5                 DelayedMatrixStats_1.24.0   GiottoUtils_0.1.4          
 [13] RCurl_1.98-1.14             terra_1.7-71                htmltools_0.5.7            
 [16] S4Arrays_1.2.0              curl_5.2.0                  BiocNeighbors_1.20.0       
 [19] SparseArray_1.2.3           parallelly_1.36.0           desc_1.4.3                 
 [22] igraph_2.0.1.1              lifecycle_1.0.4             pkgconfig_2.0.3            
 [25] rsvd_1.0.5                  Matrix_1.6-5                R6_2.5.1                   
 [28] fastmap_1.1.1               GenomeInfoDbData_1.2.11     MatrixGenerics_1.14.0      
 [31] future_1.33.1               digest_0.6.34               colorspace_2.1-0           
 [34] S4Vectors_0.40.2            ps_1.7.6                    dqrng_0.3.2                
 [37] irlba_2.3.5.1               textshaping_0.3.7           GenomicRanges_1.54.1       
 [40] beachmat_2.18.0             labeling_0.4.3              progressr_0.14.0           
 [43] RcppZiggurat_0.1.6          fansi_1.0.6                 polyclip_1.10-6            
 [46] abind_1.4-5                 compiler_4.3.2              remotes_2.4.2.1            
 [49] withr_3.0.0                 backports_1.4.1             BiocParallel_1.36.0        
 [52] pkgbuild_1.4.3              ggforce_0.4.1               MASS_7.3-60.0.1            
 [55] DelayedArray_0.28.0         rjson_0.2.21                bluster_1.12.0             
 [58] gtools_3.9.5                GiottoVisuals_0.1.4         tools_4.3.2                
 [61] scatterpie_0.2.1            future.apply_1.11.1         glue_1.7.0                 
 [64] quadprog_1.5-8              dbscan_1.1-12               callr_3.7.3                
 [67] grid_4.3.2                  checkmate_2.3.1             cluster_2.1.6              
 [70] generics_0.1.3              gtable_0.3.4                tidyr_1.3.1                
 [73] data.table_1.15.0           BiocSingular_1.18.0         ScaledMatrix_1.10.0        
 [76] metapod_1.10.0              utf8_1.2.4                  XVector_0.42.0             
 [79] BiocGenerics_0.48.1         ggrepel_0.9.5               pillar_1.9.0               
 [82] limma_3.58.1                tweenr_2.0.2                dplyr_1.1.4                
 [85] lattice_0.22-5              FNN_1.1.4                   tidyselect_1.2.0           
 [88] SingleCellExperiment_1.24.0 locfit_1.5-9.8              scuttle_1.12.0             
 [91] knitr_1.45                  IRanges_2.36.0              edgeR_4.0.0                
 [94] SummarizedExperiment_1.32.0 stats4_4.3.2                xfun_0.42                  
 [97] Biobase_2.62.0              statmod_1.5.0               matrixStats_1.2.0          
[100] ggfun_0.1.4                 yaml_2.3.8                  evaluate_0.23              
[103] codetools_0.2-19            tibble_3.2.1                colorRamp2_0.1.0           
[106] cli_3.6.2                   uwot_0.1.16                 RcppParallel_5.1.7         
[109] reticulate_1.35.0           systemfonts_1.0.5           munsell_0.5.0              
[112] processx_3.8.3              Rcpp_1.0.12                 GenomeInfoDb_1.38.5        
[115] globals_0.16.2              png_0.1-8                   parallel_4.3.2             
[118] Rfast_2.1.0                 ggplot2_3.4.4               scran_1.30.0               
[121] sparseMatrixStats_1.14.0    bitops_1.0-7                listenv_0.9.1              
[124] SpatialExperiment_1.12.0    scales_1.3.0                purrr_1.0.2                
[127] crayon_1.5.2                rlang_1.1.3                 cowplot_1.1.3  
```



