<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>cellProximityEnrichmentSpots — cellProximityEnrichmentSpots • Giotto</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="cellProximityEnrichmentSpots — cellProximityEnrichmentSpots"><meta property="og:description" content="Compute cell-cell interaction enrichment for spots
(observed vs expected)"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul></li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    <li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/interoperability_seurat.html">Seurat</a>
    </li>
    <li>
      <a href="../articles/bento.html">Integrate Bento Analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
    <li>
      <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul></li>
<li>
  <a></a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>cellProximityEnrichmentSpots</h1>
    <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/R/spatial_interaction_spot.R" class="external-link"><code>R/spatial_interaction_spot.R</code></a></small>
    <div class="hidden name"><code>cellProximityEnrichmentSpots.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Compute cell-cell interaction enrichment for spots
(observed vs expected)</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">cellProximityEnrichmentSpots</span><span class="op">(</span></span>
<span>  <span class="va">gobject</span>,</span>
<span>  spat_unit <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  feat_type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  spatial_network_name <span class="op">=</span> <span class="st">"spatial_network"</span>,</span>
<span>  cluster_column <span class="op">=</span> <span class="st">"cell_ID"</span>,</span>
<span>  cells_in_spot <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  number_of_simulations <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  adjust_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"fdr"</span>, <span class="st">"bonferroni"</span>, <span class="st">"BH"</span>, <span class="st">"holm"</span>, <span class="st">"hochberg"</span>, <span class="st">"hommel"</span>,</span>
<span>    <span class="st">"BY"</span><span class="op">)</span>,</span>
<span>  set_seed <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  seed_number <span class="op">=</span> <span class="fl">1234</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>gobject</dt>
<dd><p>giotto object</p></dd>


<dt>spat_unit</dt>
<dd><p>spatial unit (e.g. 'cell')</p></dd>


<dt>feat_type</dt>
<dd><p>feature type (e.g. 'rna')</p></dd>


<dt>spatial_network_name</dt>
<dd><p>name of spatial network to use</p></dd>


<dt>cluster_column</dt>
<dd><p>name of column to use for clusters</p></dd>


<dt>cells_in_spot</dt>
<dd><p>cell number in each spot</p></dd>


<dt>number_of_simulations</dt>
<dd><p>number of simulations to create expected
observations</p></dd>


<dt>adjust_method</dt>
<dd><p>method to adjust p.values
(e.g. "none", "fdr", "bonferroni","BH","holm", "hochberg", "hommel","BY")</p></dd>


<dt>set_seed</dt>
<dd><p>use of seed. Default = TRUE</p></dd>


<dt>seed_number</dt>
<dd><p>seed number to use. Default = 1234</p></dd>


<dt>verbose</dt>
<dd><p>be verbose</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>List of cell Proximity scores (CPscores) in data.table format.
The first
data.table (raw_sim_table) shows the raw observations of both the original
and simulated networks. The second data.table (enrichm_res) shows the
enrichment results.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Spatial proximity enrichment or depletion between pairs of cell
types is calculated by calculating the observed over the expected frequency
of cell-cell proximity interactions. The expected frequency is the average
frequency calculated from a number of spatial network simulations. Each
individual simulation is obtained by reshuffling the cell type labels of
each node (spot) in the spatial network.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">GiottoData</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/loadGiottoMini.html" class="external-link">loadGiottoMini</a></span><span class="op">(</span><span class="st">"visium"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1. read Giotto object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2. read Giotto feature information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3. read Giotto spatial information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3.1 read Giotto spatial shape information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> cell_spatInfo_spatVector.shp</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> cell</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3.2 read Giotto spatial centroid information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> cell</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3.3 read Giotto spatial overlap information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No overlaps were found, overlap loading will be</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  skipped</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4. read Giotto image information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> a giotto python environment was found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using python path:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  "/Users/yuanlab/Library/r-miniconda/envs/giotto_env/bin/pythonw"</span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span><span class="va">g</span>,</span></span>
<span class="r-in"><span>cluster_column <span class="op">=</span> <span class="st">"leiden_clus"</span>, min_feats <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using 'Scran' to detect marker feats. If used in published</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       research, please cite: Lun ATL, McCarthy DJ, Marioni JC (2016).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       'A step-by-step workflow for low-level analysis of single-cell RNA-seq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       data with Bioconductor.'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>       F1000Res., 5, 2122. doi: 10.12688/f1000research.9501.2. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> start with cluster  1start with cluster  2start with cluster  3start with cluster  4start with cluster  5start with cluster  6start with cluster  7</span>
<span class="r-in"><span><span class="va">sign_gene</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">feats</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sign_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sign_gene</span><span class="op">)</span><span class="op">*</span><span class="fl">8</span>, mean  <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sign_gene</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sign_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sign_gene</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sign_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"celltype_"</span>,<span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in dimnames(x) &lt;- dn:</span> length of 'dimnames' [2] not equal to array extent</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="runDWLSDeconv.html">runDWLSDeconv</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">g</span>, sign_matrix <span class="op">=</span> <span class="va">sign_matrix</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>no non-missing arguments to min; returning Inf</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>no non-missing arguments to max; returning -Inf</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>no non-missing arguments to max; returning -Inf</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>NaNs produced</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in seq_len(ceiling(log2(max(wsScaledMinusInf)))):</span> argument must be coercible to non-negative integer</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">cellProximityEnrichmentSpots</span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">g</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $raw_sim_table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Index: &lt;unified_int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          unified_int type_int          V1    external</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               &lt;char&gt;   &lt;char&gt;       &lt;num&gt;       &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    1:         Astrocytes--Astrocytes     homo   77.097645   77.097645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    2:            Astrocytes--Neurons   hetero  368.902614  342.075257</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    3:             Astrocytes--Oligos   hetero  258.428639  238.459464</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    4:     Astrocytes--PeripheralGlia   hetero   15.075425   13.983535</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    5:               Neurons--Neurons     homo 1572.742242 1572.742242</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ---                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1006:                Neurons--Oligos   hetero 1159.793815 1158.830010</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1007:        Neurons--PeripheralGlia   hetero   53.120837   53.097420</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1008:                 Oligos--Oligos     homo  291.758384  291.758384</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1009:         Oligos--PeripheralGlia   hetero   24.678681   24.664819</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1010: PeripheralGlia--PeripheralGlia     homo    1.249734    1.249734</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          internal        orig    round</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;num&gt;      &lt;char&gt;   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    1:  0.00000000    original original</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    2: 26.82735755    original original</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    3: 19.96917482    original original</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    4:  1.09188982    original original</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    5:  0.00000000    original original</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ---                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1006:  0.96380446 simulations   sim100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1007:  0.02341696 simulations   sim100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1008:  0.00000000 simulations   sim100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1009:  0.01386250 simulations   sim100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1010:  0.00000000 simulations   sim100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $enrichm_res</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        unified_int type_int    original simulations    enrichm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             &lt;fctr&gt;   &lt;char&gt;       &lt;num&gt;       &lt;num&gt;      &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:         Astrocytes--Astrocytes     homo   77.097645   43.908466  0.7982916</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: PeripheralGlia--PeripheralGlia     homo    2.741436    1.238117  0.7413068</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:     Astrocytes--PeripheralGlia   hetero   15.075425    9.601463  0.6005935</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                 Oligos--Oligos     homo  436.276740  296.264218  0.5568008</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:         Oligos--PeripheralGlia   hetero   34.552973   25.350889  0.4321185</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             Astrocytes--Oligos   hetero  258.428639  208.237077  0.3101992</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               Neurons--Neurons     homo 1572.742242 1352.935773  0.2170400</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            Astrocytes--Neurons   hetero  368.902614  439.918806 -0.2533675</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:        Neurons--PeripheralGlia   hetero   41.364766   52.414225 -0.3343591</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                Neurons--Oligos   hetero  900.814538 1153.717050 -0.3566367</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p_higher_orig p_lower_orig p.adj_higher p.adj_lower   PI_value int_ranking</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;num&gt;        &lt;num&gt;        &lt;num&gt;       &lt;num&gt;      &lt;num&gt;       &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             0            1            0           1  1.5965832           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             0            1            0           1  1.4826135           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             0            1            0           1  1.2011870           3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             0            1            0           1  1.1136016           4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             0            1            0           1  0.8642371           5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             0            1            0           1  0.6203985           6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             0            1            0           1  0.4340799           7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             1            0            1           0 -0.5067350           8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:             1            0            1           0 -0.6687182           9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:             1            0            1           0 -0.7132734          10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

