<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>perform cellpose segmentation — doCellposeSegmentation • Giotto</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="perform cellpose segmentation — doCellposeSegmentation"><meta property="og:description" content="perform the Giotto Wrapper of cellpose segmentation. This is for a model inference to generate segmentation mask file from input image.
main parameters needed"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul></li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    <li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/interoperability_seurat.html">Seurat</a>
    </li>
    <li>
      <a href="../articles/bento.html">Integrate Bento Analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
    <li>
      <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Scalability</li>
    <li>
      <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul></li>
<li>
  <a></a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>perform cellpose segmentation</h1>
    <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/R/cell_segmentation.R" class="external-link"><code>R/cell_segmentation.R</code></a></small>
    <div class="hidden name"><code>doCellposeSegmentation.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>perform the Giotto Wrapper of cellpose segmentation. This is for a model inference to generate segmentation mask file from input image.
main parameters needed</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">doCellposeSegmentation</span><span class="op">(</span></span>
<span>  python_env <span class="op">=</span> <span class="st">"giotto_cellpose"</span>,</span>
<span>  <span class="va">image_dir</span>,</span>
<span>  <span class="va">mask_output</span>,</span>
<span>  channel_1 <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  channel_2 <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  model_name <span class="op">=</span> <span class="st">"cyto3"</span>,</span>
<span>  batch_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  resample <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  channel_axis <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  z_axis <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  normalize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  invert <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  rescale <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  diameter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  flow_threshold <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>  cellprob_threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  do_3D <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  anisotropy <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stitch_threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  min_size <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  niter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  augment <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  tile <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  tile_overlap <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  bsize <span class="op">=</span> <span class="fl">224</span>,</span>
<span>  interp <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  compute_masks <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  progress <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-image-dir">image_dir<a class="anchor" aria-label="anchor" href="#arg-image-dir"></a></dt>
<dd><p>character, required. Provide a path to a gray scale or a three channel image.</p></dd>


<dt id="arg-mask-output">mask_output<a class="anchor" aria-label="anchor" href="#arg-mask-output"></a></dt>
<dd><p>required. Provide a path to the output mask file.</p></dd>


<dt id="arg-channel-">channel_1<a class="anchor" aria-label="anchor" href="#arg-channel-"></a></dt>
<dd><p>channel number for cytoplasm, default to 0(gray scale)</p></dd>


<dt id="arg-channel-">channel_2<a class="anchor" aria-label="anchor" href="#arg-channel-"></a></dt>
<dd><p>channel number for Nuclei, default to 0(gray scale)</p></dd>


<dt id="arg-model-name">model_name<a class="anchor" aria-label="anchor" href="#arg-model-name"></a></dt>
<dd><p>Name of the model to run inference. Default to 'cyto3', if you want to run cutomized trained model, place your model file in ~/.cellpose/models and specify your model name.</p></dd>


<dt id="arg-batch-size">batch_size<a class="anchor" aria-label="anchor" href="#arg-batch-size"></a></dt>
<dd><p>Cellpose Parameter, Number of 224x224 patches to run simultaneously on the GPU. Can make smaller or bigger depending on GPU memory usage. Defaults to 8.</p></dd>


<dt id="arg-resample">resample<a class="anchor" aria-label="anchor" href="#arg-resample"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-channel-axis">channel_axis<a class="anchor" aria-label="anchor" href="#arg-channel-axis"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-z-axis">z_axis<a class="anchor" aria-label="anchor" href="#arg-z-axis"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-normalize">normalize<a class="anchor" aria-label="anchor" href="#arg-normalize"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-invert">invert<a class="anchor" aria-label="anchor" href="#arg-invert"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-rescale">rescale<a class="anchor" aria-label="anchor" href="#arg-rescale"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-diameter">diameter<a class="anchor" aria-label="anchor" href="#arg-diameter"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-flow-threshold">flow_threshold<a class="anchor" aria-label="anchor" href="#arg-flow-threshold"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-cellprob-threshold">cellprob_threshold<a class="anchor" aria-label="anchor" href="#arg-cellprob-threshold"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-do-d">do_3D<a class="anchor" aria-label="anchor" href="#arg-do-d"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-anisotropy">anisotropy<a class="anchor" aria-label="anchor" href="#arg-anisotropy"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-stitch-threshold">stitch_threshold<a class="anchor" aria-label="anchor" href="#arg-stitch-threshold"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-min-size">min_size<a class="anchor" aria-label="anchor" href="#arg-min-size"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-niter">niter<a class="anchor" aria-label="anchor" href="#arg-niter"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-augment">augment<a class="anchor" aria-label="anchor" href="#arg-augment"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-tile">tile<a class="anchor" aria-label="anchor" href="#arg-tile"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-tile-overlap">tile_overlap<a class="anchor" aria-label="anchor" href="#arg-tile-overlap"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-bsize">bsize<a class="anchor" aria-label="anchor" href="#arg-bsize"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-interp">interp<a class="anchor" aria-label="anchor" href="#arg-interp"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-compute-masks">compute_masks<a class="anchor" aria-label="anchor" href="#arg-compute-masks"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-progress">progress<a class="anchor" aria-label="anchor" href="#arg-progress"></a></dt>
<dd><p>Cellpose Parameter</p></dd>


<dt id="arg-python-path">python_path<a class="anchor" aria-label="anchor" href="#arg-python-path"></a></dt>
<dd><p>python environment with cellpose installed. default = "giotto_cellpose".</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>No return variable, as this will write directly to output path provided.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># example code</span></span></span>
<span class="r-in"><span><span class="fu">doCellposeSegmentation</span><span class="op">(</span>image_dir <span class="op">=</span> <span class="va">input_image</span>, mask_output <span class="op">=</span> <span class="va">output</span>, channel_1 <span class="op">=</span> <span class="fl">2</span>, channel_2 <span class="op">=</span> <span class="fl">1</span>, model_name <span class="op">=</span> <span class="st">'cyto3'</span>,batch_size<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in .os_py_path(envname = path):</span> Assertion on 'path' failed: Directory '/home/runner/.local/share/r-miniconda' does not exist.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

