<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>initHMRF_V2 — initHMRF_V2 • Giotto</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="initHMRF_V2 — initHMRF_V2"><meta property="og:description" content="Run initialization for HMRF model"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul></li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Process Giotto</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactivity</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul></li>
    <li class="divider">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu"><li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul></li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul></li>
<li>
  <a></a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>initHMRF_V2</h1>
    <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/R/python_hmrf.R" class="external-link"><code>R/python_hmrf.R</code></a></small>
    <div class="hidden name"><code>initHMRF_V2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run initialization for HMRF model</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">initHMRF_V2</span><span class="op">(</span></span>
<span>  <span class="va">gobject</span>,</span>
<span>  spat_unit <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  feat_type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"scaled"</span>, <span class="st">"normalized"</span>, <span class="st">"custom"</span><span class="op">)</span>,</span>
<span>  spatial_network_name <span class="op">=</span> <span class="st">"Delaunay_network"</span>,</span>
<span>  use_spatial_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"binSpect"</span>, <span class="st">"silhouetteRank"</span><span class="op">)</span>,</span>
<span>  use_score <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_list_from_top <span class="op">=</span> <span class="fl">2500</span>,</span>
<span>  filter_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"elbow"</span><span class="op">)</span>,</span>
<span>  user_gene_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  use_pca <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use_pca_dim <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,</span>
<span>  gene_samples <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  gene_sampling_rate <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  gene_sampling_seed <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  use_metagene <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_metagene <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  top_metagene <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  existing_spatial_enrichm_to_use <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  use_neighborhood_composition <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  spatial_network_name_for_neighborhood <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  metadata_to_use <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hmrf_seed <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  cl.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"km"</span>, <span class="st">"leiden"</span>, <span class="st">"louvain"</span><span class="op">)</span>,</span>
<span>  resolution.cl <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  k <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  tolerance <span class="op">=</span> <span class="fl">1e-05</span>,</span>
<span>  zscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"rowcol"</span>, <span class="st">"colrow"</span><span class="op">)</span>,</span>
<span>  nstart <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  factor_step <span class="op">=</span> <span class="fl">1.05</span>,</span>
<span>  python_path <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-gobject">gobject<a class="anchor" aria-label="anchor" href="#arg-gobject"></a></dt>
<dd><p>giotto object</p></dd>


<dt id="arg-spat-unit">spat_unit<a class="anchor" aria-label="anchor" href="#arg-spat-unit"></a></dt>
<dd><p>spatial unit</p></dd>


<dt id="arg-feat-type">feat_type<a class="anchor" aria-label="anchor" href="#arg-feat-type"></a></dt>
<dd><p>feature type</p></dd>


<dt id="arg-expression-values">expression_values<a class="anchor" aria-label="anchor" href="#arg-expression-values"></a></dt>
<dd><p>expression values to use</p></dd>


<dt id="arg-spatial-network-name">spatial_network_name<a class="anchor" aria-label="anchor" href="#arg-spatial-network-name"></a></dt>
<dd><p>name of spatial network to use for HMRF</p></dd>


<dt id="arg-use-spatial-genes">use_spatial_genes<a class="anchor" aria-label="anchor" href="#arg-use-spatial-genes"></a></dt>
<dd><p>which of Giotto's spatial genes to use</p></dd>


<dt id="arg-use-score">use_score<a class="anchor" aria-label="anchor" href="#arg-use-score"></a></dt>
<dd><p>use score as gene selection criterion
(applies when use_spatial_genes=silhouetteRank)</p></dd>


<dt id="arg-gene-list-from-top">gene_list_from_top<a class="anchor" aria-label="anchor" href="#arg-gene-list-from-top"></a></dt>
<dd><p>total spatial genes before sampling</p></dd>


<dt id="arg-filter-method">filter_method<a class="anchor" aria-label="anchor" href="#arg-filter-method"></a></dt>
<dd><p>filter genes by top or by elbow method, prior to
sampling</p></dd>


<dt id="arg-user-gene-list">user_gene_list<a class="anchor" aria-label="anchor" href="#arg-user-gene-list"></a></dt>
<dd><p>user-specified genes (optional)</p></dd>


<dt id="arg-use-pca">use_pca<a class="anchor" aria-label="anchor" href="#arg-use-pca"></a></dt>
<dd><p>if PCA is used on the spatial gene expression value for
clustering</p></dd>


<dt id="arg-use-pca-dim">use_pca_dim<a class="anchor" aria-label="anchor" href="#arg-use-pca-dim"></a></dt>
<dd><p>dimensions of the PCs of the selected expression</p></dd>


<dt id="arg-gene-samples">gene_samples<a class="anchor" aria-label="anchor" href="#arg-gene-samples"></a></dt>
<dd><p>number of spatial gene subset to use for HMRF</p></dd>


<dt id="arg-gene-sampling-rate">gene_sampling_rate<a class="anchor" aria-label="anchor" href="#arg-gene-sampling-rate"></a></dt>
<dd><p>parameter (1-50) controlling proportion of gene
samples from different module when sampling, 1 corresponding to equal gene
samples between different modules; 50 corresponding to gene samples
proportional to module size.</p></dd>


<dt id="arg-gene-sampling-seed">gene_sampling_seed<a class="anchor" aria-label="anchor" href="#arg-gene-sampling-seed"></a></dt>
<dd><p>random number seed to sample spatial genes</p></dd>


<dt id="arg-use-metagene">use_metagene<a class="anchor" aria-label="anchor" href="#arg-use-metagene"></a></dt>
<dd><p>if metagene expression is used for clustering</p></dd>


<dt id="arg-cluster-metagene">cluster_metagene<a class="anchor" aria-label="anchor" href="#arg-cluster-metagene"></a></dt>
<dd><p>number of metagenes to use</p></dd>


<dt id="arg-top-metagene">top_metagene<a class="anchor" aria-label="anchor" href="#arg-top-metagene"></a></dt>
<dd><p>= number of genes in each cluster for the metagene
calculation</p></dd>


<dt id="arg-existing-spatial-enrichm-to-use">existing_spatial_enrichm_to_use<a class="anchor" aria-label="anchor" href="#arg-existing-spatial-enrichm-to-use"></a></dt>
<dd><p>name of existing spatial enrichment
result to use</p></dd>


<dt id="arg-use-neighborhood-composition">use_neighborhood_composition<a class="anchor" aria-label="anchor" href="#arg-use-neighborhood-composition"></a></dt>
<dd><p>if neighborhood composition is used for
hmrf</p></dd>


<dt id="arg-spatial-network-name-for-neighborhood">spatial_network_name_for_neighborhood<a class="anchor" aria-label="anchor" href="#arg-spatial-network-name-for-neighborhood"></a></dt>
<dd><p>spatial network used to
calculate neighborhood composition</p></dd>


<dt id="arg-metadata-to-use">metadata_to_use<a class="anchor" aria-label="anchor" href="#arg-metadata-to-use"></a></dt>
<dd><p>metadata used to calculate neighborhood composition</p></dd>


<dt id="arg-hmrf-seed">hmrf_seed<a class="anchor" aria-label="anchor" href="#arg-hmrf-seed"></a></dt>
<dd><p>random number seed to generate initial mean vector of HMRF
model</p></dd>


<dt id="arg-cl-method">cl.method<a class="anchor" aria-label="anchor" href="#arg-cl-method"></a></dt>
<dd><p>clustering method to calculate the initial mean vector,
selecting from 'km', 'leiden', or 'louvain'</p></dd>


<dt id="arg-resolution-cl">resolution.cl<a class="anchor" aria-label="anchor" href="#arg-resolution-cl"></a></dt>
<dd><p>resolution of Leiden or Louvain clustering</p></dd>


<dt id="arg-k">k<a class="anchor" aria-label="anchor" href="#arg-k"></a></dt>
<dd><p>number of HMRF domains</p></dd>


<dt id="arg-tolerance">tolerance<a class="anchor" aria-label="anchor" href="#arg-tolerance"></a></dt>
<dd><p>error tolerance threshold</p></dd>


<dt id="arg-zscore">zscore<a class="anchor" aria-label="anchor" href="#arg-zscore"></a></dt>
<dd><p>type of zscore to use</p></dd>


<dt id="arg-nstart">nstart<a class="anchor" aria-label="anchor" href="#arg-nstart"></a></dt>
<dd><p>number of Kmeans initializations from which to select the
best initialization</p></dd>


<dt id="arg-factor-step">factor_step<a class="anchor" aria-label="anchor" href="#arg-factor-step"></a></dt>
<dd><p>dampened factor step</p></dd>


<dt id="arg-python-path">python_path<a class="anchor" aria-label="anchor" href="#arg-python-path"></a></dt>
<dd><p>python_path</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>initialized HMRF</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function is the initialization step of HMRF domain clustering. First,
user specify which of Giotto's spatial genes to run,
through use_spatial_genes. Spatial genes have been stored in the gene
metadata table. A first pass of genes will filter genes that
are not significantly spatial, as determined by filter_method. If
filter_method is none, then top 2500 (gene_list_from_top) genes
ranked by pvalue are considered spatial. If filter_method is elbow, then the
exact cutoff is determined by the elbow in
the -log10 P-value vs. gene rank plot. Second, users have a few options to
decrease the dimension of the spatial genes for
clustering, listed with selection priority:
   1. use PCA of the spatial gene expressions (selected by use_pca)
   2. use metagene expressions (selected by use_metagene)
   3. sampling to select 500 spatial genes (controlled by gene_samples).
Third, once spatial genes are finalized, we are using clustering method to
initialize HMRF.
Instead of select spatial genes for domain clustering, HMRF method could
also applied on unit neighborhood composition of any group
membership(such as cell types), specified by parameter:
use_neighborhood_composition,  spatial_network_name_for_neighborhood and
metadata_to_use. Also HMRF provides the opportunity for user to do
clustering by any customized spatial enrichment matrix
(existing_spatial_enrichm_to_use).
There are 3 clustering algorithm: K-means, Leiden, and Louvain to determine
initial centroids of HMRF. The initialization is
then finished. This function returns a list containing y (expression),
nei (neighborhood structure), numnei (number of neighbors),
blocks (graph colors), damp (dampened factor), mu (mean),
sigma (covariance), k, genes, edgelist, init.cl (initial clusters),
spat_unit, feat_type. This information is needed for the second step, doHMRF.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">GiottoData</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/loadGiottoMini.html" class="external-link">loadGiottoMini</a></span><span class="op">(</span><span class="st">"visium"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1. read Giotto object</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2. read Giotto feature information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3. read Giotto spatial information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3.1 read Giotto spatial shape information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3.2 read Giotto spatial centroid information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3.3 read Giotto spatial overlap information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4. read Giotto image information</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> checking default envname 'giotto_env'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> a system default python environment was found</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using python path:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  "/usr/bin/python3"</span>
<span class="r-in"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">g</span>, return_gobject <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This is the single parameter version of binSpect</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1. matrix binarization complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2. spatial enrichment test completed</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3. (optional) average expression of high</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  expressing cells calculated</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4. (optional) number of high expressing cells</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  calculated</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">initHMRF_V2</span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">g</span>, cl.method <span class="op">=</span> <span class="st">"km"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> If used in published research, please cite:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Q Zhu, S Shah, R Dries, L Cai, GC Yuan.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  'Identification of spatially associated subpopulations by combining</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  scRNAseq and sequential fluorescence in situ hybridization data'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Nature biotechnology 36 (12), 1183-1190. 2018</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> packages 'tidygraph', 'ggraph', 'smfishHmrf', 'graphcoloring' are not yet installed</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> </span>
<span class="r-err co"><span class="r-pr">#&gt;</span>  To install:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> install.packages(c("tidygraph", "ggraph"))</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> devtools::install_bitbucket("qzhudfci/smfishHmrf-r")</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> devtools::install_bitbucket("qzhudfci/graphcoloring")</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

