<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integrate Bento Analysis • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Integrate Bento Analysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Process Giotto</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactive selection</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integrate Bento Analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/bento.Rmd" class="external-link"><code>vignettes/bento.Rmd</code></a></small>
      <div class="hidden name"><code>bento.Rmd</code></div>

    </div>

    
    
<div class="section level2" number="1">
<h2 number="1" id="introduction-to-bento">
<span class="header-section-number">1</span> Introduction to Bento<a class="anchor" aria-label="anchor" href="#introduction-to-bento"></a>
</h2>
<p><a href="https://bento-tools.readthedocs.io/en/latest/index.html" class="external-link">Bento</a> is a Python toolkit for performing subcellular analysis of spatial transcriptomics data.</p>
</div>
<div class="section level2" number="2">
<h2 number="2" id="giotto-installation">
<span class="header-section-number">2</span> Giotto installation<a class="anchor" aria-label="anchor" href="#giotto-installation"></a>
</h2>
<p>You can install Giotto and dependent python envrionment using conda</p>
<p><a href="https://giottosuite.readthedocs.io/en/latest/subsections/datasets/installation_using_conda_231212.html" class="external-link">Giotto Installation Using Conda</a> or in R.</p>
<p><a href="https://drieslab.github.io/Giotto_website/articles/installation.html">Giotto Installation</a></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto Has been installed.</span></span>
<span></span>
<span><span class="co"># Set python path to your preferred python path. The python packages that Giotto depends on will be installed here if not installed before.</span></span>
<span><span class="va">python_path</span> <span class="op">&lt;-</span> <span class="st">"path/to/conda-env/envs/giotto_suite/bin/python"</span></span>
<span></span>
<span><span class="co"># Set python path to NULL if you want to automatically install (only the 1st time) and use the giotto miniconda environment</span></span>
<span><span class="va">python_path</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="va">genv_exists</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">python_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>    <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 number="3" id="set-up-giotto-environment">
<span class="header-section-number">3</span> Set up Giotto environment<a class="anchor" aria-label="anchor" href="#set-up-giotto-environment"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. ** SET WORKING DIRECTORY WHERE PROJECT OUPUTS WILL SAVE TO **</span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"/path/to/results/"</span></span>
<span></span>
<span><span class="co"># 2. Create Giotto instructions</span></span>
<span><span class="co"># Directly saving plots to the working directory without rendering them in the editor saves time.</span></span>
<span><span class="va">instructions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_instructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span></span>
<span>    python_path <span class="op">=</span> <span class="va">python_path</span>,</span>
<span>    save_dir <span class="op">=</span> <span class="va">results_folder</span>,</span>
<span>    save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    show_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    return_plot <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 number="4" id="dataset-explanation">
<span class="header-section-number">4</span> Dataset explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p>This vignette covers Giotto object creation and simple exploratory analysis with 10x Genomics’ subcellular Xenium In Situ platform data using their <a href="https://www.10xgenomics.com/products/xenium-in-situ/preview-dataset-human-breast" class="external-link">Human Breast Cancer Dataset</a> provided with their recent <a href="https://www.biorxiv.org/content/10.1101/2022.10.06.510405v1" class="external-link">bioRxiv pre-print</a>. The data from the first tissue replicate will be worked with.</p>
<p><img src="images%2Fbento%2Flarge_preview.png" width="608"></p>
</div>
<div class="section level2" number="5">
<h2 number="5" id="project-data-paths">
<span class="header-section-number">5</span> Project data paths<a class="anchor" aria-label="anchor" href="#project-data-paths"></a>
</h2>
<p>Xenium’s exact output folder structure and which file/formats will be available in the final release are still subject to change. This tutorial will be updated for any differences at release. Note: The cell_feature_matrix.tar.gz must be unpacked into a sub directory within the Xenium working directory before starting</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ** SET PATH TO FOLDER CONTAINING XENIUM DATA **</span></span>
<span><span class="va">xenium_folder</span> <span class="op">&lt;-</span> <span class="st">"path/to/Xenium/"</span></span>
<span></span>
<span><span class="co"># General files (some are supplemental files)</span></span>
<span><span class="va">settings_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"experiment.xenium"</span><span class="op">)</span></span>
<span><span class="va">he_img_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"Xenium_FFPE_Human_Breast_Cancer_Rep1_he_image.tif"</span><span class="op">)</span></span>
<span><span class="va">if_img_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"Xenium_FFPE_Human_Breast_Cancer_Rep1_if_image.tif"</span><span class="op">)</span></span>
<span><span class="va">panel_meta_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"Xenium_FFPE_Human_Breast_Cancer_Rep1_panel.tsv"</span><span class="op">)</span> <span class="co"># (optional)</span></span>
<span></span>
<span><span class="co"># Files (SUBCELLULAR): (tutorial focuses on working with these files)</span></span>
<span><span class="va">cell_bound_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"outs/cell_boundaries.csv.gz"</span><span class="op">)</span></span>
<span><span class="va">nuc_bound_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"outs/nucleus_boundaries.csv.gz"</span><span class="op">)</span></span>
<span><span class="va">tx_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"outs/transcripts.csv.gz"</span><span class="op">)</span></span>
<span><span class="va">feat_meta_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"outs/cell_feature_matrix/features.tsv.gz"</span><span class="op">)</span> <span class="co"># (also used in aggregate)</span></span>
<span></span>
<span><span class="co"># Files (AGGREGATE):</span></span>
<span><span class="va">expr_mat_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"outs/cell_feature_matrix"</span><span class="op">)</span></span>
<span><span class="va">cell_meta_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">xenium_folder</span>, <span class="st">"outs/cells.csv.gz"</span><span class="op">)</span> <span class="co"># contains spatlocs</span></span></code></pre></div>
</div>
<div class="section level2" number="6">
<h2 number="6" id="xenium-feature-types-exploration">
<span class="header-section-number">6</span> Xenium feature types exploration<a class="anchor" aria-label="anchor" href="#xenium-feature-types-exploration"></a>
</h2>
<p>features.tsv.gz within cell_feature_matrix.tar.gz provides information on the different feature types available within Xenium’s two types of expression outputs:</p>
<ol style="list-style-type: decimal">
<li>aggregated cell by feature matrix</li>
<li>transcript detections in transcripts.csv.gz There are four types of probes used, each of which is represented in both exports</li>
</ol>
<ul>
<li>gene expression: Gene expression detection</li>
<li>blank codeword: Unused codeword - there are no probes that will generate the codeword</li>
<li>negative control codeword: Valid codewords that do not have any probes that should yield that code, so they can be used to assess the specificity of the decoding algorithm</li>
<li>negative control probe: Probes that exist in the panel, but target ERCC or other non-biological sequences, which can be used to assess the specificity of the assay.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load features metadata</span></span>
<span><span class="co"># (Make sure cell_feature_matrix folder is unpacked)</span></span>
<span><span class="va">feature_dt</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">feat_meta_path</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">feature_dt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl_ID"</span>, <span class="st">"feat_name"</span>, <span class="st">"feat_type"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find the feature IDs that belong to each feature type</span></span>
<span><span class="va">feature_dt</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">feat_type</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">feat_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">feature_dt</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">feat_type</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">feat_types_IDs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>    <span class="va">feat_types</span>, <span class="kw">function</span><span class="op">(</span><span class="va">type</span><span class="op">)</span> <span class="va">feature_dt</span><span class="op">[</span><span class="va">feat_type</span> <span class="op">==</span> <span class="va">type</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">feat_name</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">feat_types_IDs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">feat_types</span></span></code></pre></div>
<p>This dataset has 313 probes that are dedicated for gene expression transcript detection.</p>
<ul>
<li>gene expression IDs</li>
<li>blank codeword IDs</li>
<li>negative control codeword IDs</li>
<li>negative control probe IDs</li>
</ul>
</div>
<div class="section level2" number="7">
<h2 number="7" id="loading-xenium-data">
<span class="header-section-number">7</span> Loading Xenium data<a class="anchor" aria-label="anchor" href="#loading-xenium-data"></a>
</h2>
<div class="section level3" number="7.1">
<h3 number="7.1" id="manual-method">
<span class="header-section-number">7.1</span> Manual Method<a class="anchor" aria-label="anchor" href="#manual-method"></a>
</h3>
<p>Giotto objects can be manually assembled feeding data and subobjects into a creation function.</p>
<div class="section level4" number="7.1.1">
<h4 number="7.1.1" id="load-transcript-level-data">
<span class="header-section-number">7.1.1</span> Load transcript-level data<a class="anchor" aria-label="anchor" href="#load-transcript-level-data"></a>
</h4>
<p>transcripts.csv.gz is a file containing x, y, z coordinates for individual transcript molecules detected during the Xenium run. It also contains a QC Phred score for which this tutorial will set a cutoff at 20, the same as what 10x uses.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tx_dt</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">tx_path</span><span class="op">)</span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setattr.html" class="external-link">setnames</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">tx_dt</span>,</span>
<span>    old <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature_name"</span>, <span class="st">"x_location"</span>, <span class="st">"y_location"</span><span class="op">)</span>,</span>
<span>    new <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feat_ID"</span>, <span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span></span>
<span>    <span class="st">"Transcripts info available:\n "</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'"'</span>, <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tx_dt</span><span class="op">)</span>, <span class="st">'"'</span><span class="op">)</span>, <span class="st">"\n"</span>,</span>
<span>    <span class="st">"with"</span>, <span class="va">tx_dt</span><span class="op">[</span>, <span class="va">.N</span><span class="op">]</span>, <span class="st">"unfiltered detections\n"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter by qv (Phred score)</span></span>
<span><span class="va">tx_dt_filtered</span> <span class="op">&lt;-</span> <span class="va">tx_dt</span><span class="op">[</span><span class="va">qv</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"and"</span>, <span class="va">tx_dt_filtered</span><span class="op">[</span>, <span class="va">.N</span><span class="op">]</span>, <span class="st">"filtered detections\n\n"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># separate detections by feature type</span></span>
<span><span class="va">tx_dt_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>    <span class="va">feat_types_IDs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">types</span><span class="op">)</span> <span class="va">tx_dt_filtered</span><span class="op">[</span><span class="va">feat_ID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">types</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">tx_dt_types</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tx_dt_types</span><span class="op">)</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>, <span class="st">"detections: "</span>, <span class="va">tx_dt_types</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="va">.N</span><span class="op">]</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Giotto loads these filtered subcellular detections in as a giottoPoints object and determines the correct columns by looking for columns named ‘feat_ID’, ‘x’, and ‘y’.</p>
<p>Here, we use the list of data.tables generated in the previous step to create a list of giottoPoints objects.</p>
<p>When previewing these objects using plot(), the default behavior is to plot ALL points within the object. For objects that contain many feature points, it is highly recommended to specify a subset of features to plot using the feats param.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gpoints_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>    <span class="va">tx_dt_types</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPoints.html" class="external-link">createGiottoPoints</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="co"># 208.499 sec elapsed</span></span>
<span></span>
<span><span class="co"># Preview QC probe detections</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Blank Codeword`</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"Blank Codeword"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Negative Control Codeword`</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"Negative Control Codeword"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Negative Control Probe`</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"Negative Control Probe"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Preview two genes (slower)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Gene Expression`</span>, <span class="co"># 77.843 sec elapsed</span></span>
<span>    feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KRT8"</span>, <span class="st">"MS4A1"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">tx_dt_types</span><span class="op">$</span><span class="va">`Gene Expression`</span><span class="op">[</span><span class="va">feat_ID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KRT8"</span>, <span class="st">"MS4A1"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">feat_ID</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p><img src="images%2Fbento%2Fgpoints_expr.png" width="982"></p>
</div>
<div class="section level4" number="7.1.2">
<h4 number="7.1.2" id="load-polygon-data">
<span class="header-section-number">7.1.2</span> Load polygon data<a class="anchor" aria-label="anchor" href="#load-polygon-data"></a>
</h4>
<p>Xenium output provides segmentation/cell boundary information in .csv.gz files. These are represented within Giotto as giottoPolygon objects and can also be directly plotted. This function also determines the correct columns to use by looking for columns named ‘poly_ID’, ‘x’, and ‘y’.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellPoly_dt</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">cell_bound_path</span><span class="op">)</span></span>
<span><span class="va">nucPoly_dt</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">nuc_bound_path</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setattr.html" class="external-link">setnames</a></span><span class="op">(</span><span class="va">cellPoly_dt</span>,</span>
<span>    old <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cell_id"</span>, <span class="st">"vertex_x"</span>, <span class="st">"vertex_y"</span><span class="op">)</span>,</span>
<span>    new <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"poly_ID"</span>, <span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setattr.html" class="external-link">setnames</a></span><span class="op">(</span><span class="va">nucPoly_dt</span>,</span>
<span>    old <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cell_id"</span>, <span class="st">"vertex_x"</span>, <span class="st">"vertex_y"</span><span class="op">)</span>,</span>
<span>    new <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"poly_ID"</span>, <span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">gpoly_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html" class="external-link">createGiottoPolygonsFromDfr</a></span><span class="op">(</span></span>
<span>    segmdfr <span class="op">=</span> <span class="va">cellPoly_dt</span>,</span>
<span>    name <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>    calc_centroids <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">gpoly_nucs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html" class="external-link">createGiottoPolygonsFromDfr</a></span><span class="op">(</span></span>
<span>    segmdfr <span class="op">=</span> <span class="va">nucPoly_dt</span>,</span>
<span>    name <span class="op">=</span> <span class="st">"nucleus"</span>,</span>
<span>    calc_centroids <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>giottoPolygon objects can be directly plotted with plot(), but the field of view here is so large that it would take a long time and the details would be lost. Here, we will only plot the polygon centroids for the cell nucleus polygons by accessing the calculated results within the giottoPolygon’s spatVectorCentroids slot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">gpoly_nucs</span>, point_size <span class="op">=</span> <span class="fl">0.1</span>, type <span class="op">=</span> <span class="st">"centroid"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fbento%2Fgpolys_centroids.png" width="713"></p>
</div>
<div class="section level4" number="7.1.3">
<h4 number="7.1.3" id="create-giotto-object">
<span class="header-section-number">7.1.3</span> Create Giotto Object<a class="anchor" aria-label="anchor" href="#create-giotto-object"></a>
</h4>
<p>Now that both the feature data and the boundaries are loaded in, a subcellular Giotto object can be created.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xenium_gobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObjectSubcellular.html" class="external-link">createGiottoObjectSubcellular</a></span><span class="op">(</span></span>
<span>    gpoints <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        rna <span class="op">=</span> <span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Gene Expression`</span>,</span>
<span>        blank_code <span class="op">=</span> <span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Blank Codeword`</span>,</span>
<span>        neg_code <span class="op">=</span> <span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Negative Control Codeword`</span>,</span>
<span>        neg_probe <span class="op">=</span> <span class="va">gpoints_list</span><span class="op">$</span><span class="va">`Negative Control Probe`</span></span>
<span>    <span class="op">)</span>,</span>
<span>    gpolygons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        cell <span class="op">=</span> <span class="va">gpoly_cells</span>,</span>
<span>        nucleus <span class="op">=</span> <span class="va">gpoly_nucs</span></span>
<span>    <span class="op">)</span>,</span>
<span>    instructions <span class="op">=</span> <span class="va">instructions</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2" number="8">
<h2 number="8" id="perform-bento-analysis">
<span class="header-section-number">8</span> Perform Bento Analysis<a class="anchor" aria-label="anchor" href="#perform-bento-analysis"></a>
</h2>
<div class="section level3" number="8.1">
<h3 number="8.1" id="create-bento-anndata-object">
<span class="header-section-number">8.1</span> Create Bento AnnData Object<a class="anchor" aria-label="anchor" href="#create-bento-anndata-object"></a>
</h3>
<div class="section level4" number="8.1.1">
<h4 number="8.1.1" id="subset-giotto-object-first">
<span class="header-section-number">8.1.1</span> Subset Giotto Object First<a class="anchor" aria-label="anchor" href="#subset-giotto-object-first"></a>
</h4>
<p>Large dataset may cause prolonged processing time for Bento.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subset_xenium_gobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiottoLocs.html" class="external-link">subsetGiottoLocs</a></span><span class="op">(</span><span class="va">xenium_gobj</span>,</span>
<span>                                       spat_unit <span class="op">=</span> <span class="st">"cell"</span>, </span>
<span>                                       feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                                       x_max <span class="op">=</span> <span class="fl">200</span>, </span>
<span>                                       x_min <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                                       y_max <span class="op">=</span> <span class="fl">200</span>, </span>
<span>                                       y_min <span class="op">=</span> <span class="fl">0</span></span>
<span>                                       <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4" number="8.1.2">
<h4 number="8.1.2" id="create-anndata-object">
<span class="header-section-number">8.1.2</span> Create AnnData Object<a class="anchor" aria-label="anchor" href="#create-anndata-object"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bento_adata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createBentoAdata.html" class="external-link">createBentoAdata</a></span><span class="op">(</span><span class="va">subset_xenium_gobj</span>,</span>
<span>                                env_to_use <span class="op">=</span> <span class="st">"path/to/conda-env/envs/giotto"</span> <span class="co"># use the default value "giotto_env" when you installed python dependencies automatically</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3" number="8.2">
<h3 number="8.2" id="bento-analysis">
<span class="header-section-number">8.2</span> Bento Analysis<a class="anchor" aria-label="anchor" href="#bento-analysis"></a>
</h3>
<div class="section level4" number="8.2.1">
<h4 number="8.2.1" id="load-python-modules">
<span class="header-section-number">8.2.1</span> Load Python Modules<a class="anchor" aria-label="anchor" href="#load-python-modules"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bento_analysis_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"python"</span>, <span class="st">"python_bento_analysis.py"</span>,</span>
<span>                                   package <span class="op">=</span> <span class="st">"Giotto"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">reticulate</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/source_python.html" class="external-link">source_python</a></span><span class="op">(</span><span class="va">bento_analysis_path</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4" number="8.2.2">
<h4 number="8.2.2" id="rna-forest-analysis">
<span class="header-section-number">8.2.2</span> RNA Forest Analysis<a class="anchor" aria-label="anchor" href="#rna-forest-analysis"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">analysis_rna_forest</span><span class="op">(</span>adata <span class="op">=</span> <span class="va">bento_adata</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_rna_forest_analysis_results</span><span class="op">(</span></span>
<span>    adata <span class="op">=</span> <span class="va">bento_adata</span>,</span>
<span>    fname1 <span class="op">=</span> <span class="st">"Bento_rna_forest_radvis.png"</span>,</span>
<span>    fname2 <span class="op">=</span> <span class="st">"Bento_rna_forest_upset.png"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fbento%2FBento_rna_forest_radvis1.png" width="1220"></p>
</div>
<div class="section level4" number="8.2.3">
<h4 number="8.2.3" id="colocalization-analysis">
<span class="header-section-number">8.2.3</span> Colocalization Analysis<a class="anchor" aria-label="anchor" href="#colocalization-analysis"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">analysis_colocalization</span><span class="op">(</span></span>
<span>    adata <span class="op">=</span> <span class="va">bento_adata</span>,</span>
<span>    fname <span class="op">=</span> <span class="st">"Bento_colocalization_knee_pos.png"</span>,</span>
<span>    ranks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set the rank according output hint.</span></span>
<span><span class="fu">plot_colocalization_analysis_results</span><span class="op">(</span></span>
<span>    adata <span class="op">=</span> <span class="va">bento_adata</span>,</span>
<span>    rank <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    fname <span class="op">=</span> <span class="st">"Bento_colocalization.png"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fbento%2FBento_colocalization_knee_pos.png" width="300"></p>
</div>
</div>
</div>
<div class="section level2" number="9">
<h2 number="9" id="session-info">
<span class="header-section-number">9</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co"># Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co"># Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># Matrix products: default</span></span>
<span><span class="co"># BLAS/LAPACK: /sc/arion/work/wangw32/conda-env/envs/giotto/lib/libopenblasp-r0.3.24.so</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># locale:</span></span>
<span><span class="co">#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C</span></span>
<span><span class="co">#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8</span></span>
<span><span class="co">#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8</span></span>
<span><span class="co">#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C</span></span>
<span><span class="co">#  [9] LC_ADDRESS=C               LC_TELEPHONE=C</span></span>
<span><span class="co"># [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># attached base packages:</span></span>
<span><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># other attached packages:</span></span>
<span><span class="co"># [1] Giotto_4.0.0        GiottoVisuals_0.1.0 GiottoClass_0.1.0</span></span>
<span><span class="co"># [4] GiottoUtils_0.1.0</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># loaded via a namespace (and not attached):</span></span>
<span><span class="co">#  [1] reticulate_1.34.0  tidyselect_1.2.0   terra_1.7-55       xfun_0.41</span></span>
<span><span class="co">#  [5] sf_1.0-14          lattice_0.22-5     colorspace_2.1-0   vctrs_0.6.5</span></span>
<span><span class="co">#  [9] generics_0.1.3     htmltools_0.5.7    yaml_2.3.7         utf8_1.2.4</span></span>
<span><span class="co"># [13] rlang_1.1.2        e1071_1.7-13       R.oo_1.25.0        pillar_1.9.0</span></span>
<span><span class="co"># [17] glue_1.6.2         withr_2.5.2        DBI_1.1.3          R.utils_2.12.3</span></span>
<span><span class="co"># [21] rappdirs_0.3.3     bit64_4.0.5        lifecycle_1.0.4    stringr_1.5.1</span></span>
<span><span class="co"># [25] munsell_0.5.0      gtable_0.3.4       R.methodsS3_1.8.2  codetools_0.2-19</span></span>
<span><span class="co"># [29] evaluate_0.23      knitr_1.45         fastmap_1.1.1      class_7.3-22</span></span>
<span><span class="co"># [33] parallel_4.2.3     fansi_1.0.5        Rcpp_1.0.11        KernSmooth_2.23-22</span></span>
<span><span class="co"># [37] scales_1.3.0       backports_1.4.1    classInt_0.4-10    checkmate_2.3.1</span></span>
<span><span class="co"># [41] jsonlite_1.8.7     bit_4.0.5          ggplot2_3.4.4      png_0.1-8</span></span>
<span><span class="co"># [45] digest_0.6.33      stringi_1.8.2      dplyr_1.1.4        grid_4.2.3</span></span>
<span><span class="co"># [49] scattermore_1.2    cli_3.6.1          tools_4.2.3        magrittr_2.0.3</span></span>
<span><span class="co"># [53] proxy_0.4-27       tibble_3.2.1       colorRamp2_0.1.0   pkgconfig_2.0.3</span></span>
<span><span class="co"># [57] Matrix_1.6-4       data.table_1.14.8  rmarkdown_2.25     rstudioapi_0.15.0</span></span>
<span><span class="co"># [61] R6_2.5.1           units_0.8-5        compiler_4.2.3</span></span></code></pre></div>
</div>
<div class="section level2" number="10">
<h2 number="10" id="python-session-info">
<span class="header-section-number">10</span> Python Session Info<a class="anchor" aria-label="anchor" href="#python-session-info"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">python_session_info</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># -----</span></span>
<span><span class="co"># anndata     0.9.2</span></span>
<span><span class="co"># bento       2.0.1</span></span>
<span><span class="co"># emoji       1.7.0</span></span>
<span><span class="co"># geopandas   0.10.2</span></span>
<span><span class="co"># kneed       0.8.5</span></span>
<span><span class="co"># log         NA</span></span>
<span><span class="co"># matplotlib  3.8.2</span></span>
<span><span class="co"># minisom     NA</span></span>
<span><span class="co"># numpy       1.26.2</span></span>
<span><span class="co"># pandas      1.5.3</span></span>
<span><span class="co"># rasterio    1.3.9</span></span>
<span><span class="co"># scipy       1.11.4</span></span>
<span><span class="co"># seaborn     0.12.2</span></span>
<span><span class="co"># shapely     1.8.5.post1</span></span>
<span><span class="co"># sklearn     1.3.2</span></span>
<span><span class="co"># tqdm        4.66.1</span></span>
<span><span class="co"># -----</span></span>
<span><span class="co"># IPython             8.18.1</span></span>
<span><span class="co"># PIL                 10.1.0</span></span>
<span><span class="co"># adjustText          NA</span></span>
<span><span class="co"># affine              2.4.0</span></span>
<span><span class="co"># astropy             5.3.4</span></span>
<span><span class="co"># asttokens           NA</span></span>
<span><span class="co"># attr                23.1.0</span></span>
<span><span class="co"># certifi             2023.11.17</span></span>
<span><span class="co"># cffi                1.16.0</span></span>
<span><span class="co"># click               8.1.7</span></span>
<span><span class="co"># comm                0.1.4</span></span>
<span><span class="co"># community           0.16</span></span>
<span><span class="co"># contourpy           1.2.0</span></span>
<span><span class="co"># cycler              0.12.1</span></span>
<span><span class="co"># cython_runtime      NA</span></span>
<span><span class="co"># dateutil            2.8.2</span></span>
<span><span class="co"># decorator           5.1.1</span></span>
<span><span class="co"># decoupler           1.5.0</span></span>
<span><span class="co"># defusedxml          0.7.1</span></span>
<span><span class="co"># erfa                2.0.1.1</span></span>
<span><span class="co"># exceptiongroup      1.2.0</span></span>
<span><span class="co"># executing           2.0.1</span></span>
<span><span class="co"># fiona               1.9.5</span></span>
<span><span class="co"># h5py                3.10.0</span></span>
<span><span class="co"># igraph              0.11.3</span></span>
<span><span class="co"># ipywidgets          8.1.1</span></span>
<span><span class="co"># jedi                0.19.1</span></span>
<span><span class="co"># joblib              1.3.2</span></span>
<span><span class="co"># kiwisolver          1.4.5</span></span>
<span><span class="co"># leidenalg           0.8.8</span></span>
<span><span class="co"># llvmlite            0.41.1</span></span>
<span><span class="co"># matplotlib_scalebar 0.8.1</span></span>
<span><span class="co"># mpl_toolkits        NA</span></span>
<span><span class="co"># natsort             8.4.0</span></span>
<span><span class="co"># networkx            3.2.1</span></span>
<span><span class="co"># numba               0.58.1</span></span>
<span><span class="co"># packaging           23.2</span></span>
<span><span class="co"># parso               0.8.3</span></span>
<span><span class="co"># patsy               0.5.4</span></span>
<span><span class="co"># pexpect             4.8.0</span></span>
<span><span class="co"># pickleshare         0.7.5</span></span>
<span><span class="co"># pkg_resources       NA</span></span>
<span><span class="co"># prompt_toolkit      3.0.41</span></span>
<span><span class="co"># psutil              5.9.5</span></span>
<span><span class="co"># ptyprocess          0.7.0</span></span>
<span><span class="co"># pure_eval           0.2.2</span></span>
<span><span class="co"># pycparser           2.21</span></span>
<span><span class="co"># pygeos              0.12.0</span></span>
<span><span class="co"># pygments            2.17.2</span></span>
<span><span class="co"># pyparsing           3.1.1</span></span>
<span><span class="co"># pyproj              3.6.1</span></span>
<span><span class="co"># pytz                2023.3.post1</span></span>
<span><span class="co"># rpycall             NA</span></span>
<span><span class="co"># rpytools            NA</span></span>
<span><span class="co"># session_info        1.0.0</span></span>
<span><span class="co"># setuptools          68.2.2</span></span>
<span><span class="co"># six                 1.16.0</span></span>
<span><span class="co"># sparse              0.13.0</span></span>
<span><span class="co"># stack_data          0.6.2</span></span>
<span><span class="co"># statsmodels         0.13.5</span></span>
<span><span class="co"># tensorly            0.7.0</span></span>
<span><span class="co"># texttable           1.7.0</span></span>
<span><span class="co"># threadpoolctl       3.2.0</span></span>
<span><span class="co"># traitlets           5.14.0</span></span>
<span><span class="co"># typing_extensions   NA</span></span>
<span><span class="co"># upsetplot           0.7.0</span></span>
<span><span class="co"># wcwidth             0.2.12</span></span>
<span><span class="co"># xgboost             1.4.2</span></span>
<span><span class="co"># yaml                6.0.1</span></span>
<span><span class="co"># zoneinfo            NA</span></span>
<span><span class="co"># -----</span></span>
<span><span class="co"># Python 3.10.13 | packaged by conda-forge | (main, Oct 26 2023, 18:20:51) [GCC 12.3.0]</span></span>
<span><span class="co"># Linux-3.10.0-1160.el7.x86_64-x86_64-with-glibc2.35</span></span>
<span><span class="co"># -----</span></span>
<span><span class="co"># Session information updated at 2023-12-19 14:29</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
