<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interoperability: Seurat • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Interoperability: Seurat">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/interoperability_seurat.html">Seurat</a>
    </li>
    <li>
      <a href="../articles/bento.html">Integrate Bento Analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
    <li>
      <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Interoperability: Seurat</h1>
                        <h4 data-toc-skip class="author">Iqra Amin</h4>
            
            <h4 data-toc-skip class="date">2024-03-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/interoperability_seurat.Rmd" class="external-link"><code>vignettes/interoperability_seurat.Rmd</code></a></small>
      <div class="hidden name"><code>interoperability_seurat.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Giotto facilitates seamless interoperability with various tools, including Seurat. The conversion between Giotto and Seurat relies on four primary functions. ‘giottoToSeurat_v4’ and ‘SeuratToGiotto_v4’ cater to Seurat version 4 , while ‘giottoToSeurat_v5’ and ‘SeuratToGiotto_v5’ are specifically for Seurat version 5. These functions play a vital role in ensuring effortless data transitions between Giotto and Seurat across different versions, ensuring compatibility and ease of use.</p>
<div class="section level3">
<h3 id="conversion-of-seurat-v5-to-giotto">Conversion of Seurat V5 to Giotto<a class="anchor" aria-label="anchor" href="#conversion-of-seurat-v5-to-giotto"></a>
</h3>
<p>In this section, we outline the steps to convert a Seurat V5 object to Giotto, offering a seamless transition between these two.</p>
<div class="section level4">
<h4 id="load-required-libraries">Load required Libraries<a class="anchor" aria-label="anchor" href="#load-required-libraries"></a>
</h4>
<p>To start, we load the required libraries.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/GiottoData/" class="external-link">GiottoData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratData</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-seurat-object">Load Seurat object<a class="anchor" aria-label="anchor" href="#load-seurat-object"></a>
</h4>
<p>We begin by loading a sample Seurat object named “stxBrain” with the type specified as “anterior1”.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu">LoadData</span><span class="op">(</span><span class="st">"stxBrain"</span>, type <span class="op">=</span> <span class="st">"anterior1"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="convert-seurat-object-to-giotto">Convert Seurat object to Giotto<a class="anchor" aria-label="anchor" href="#convert-seurat-object-to-giotto"></a>
</h4>
<p>To convert Seurat object to Giotto object, we use the function seuratToGiottoV5().</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/seuratToGiottoV5.html" class="external-link">seuratToGiottoV5</a></span><span class="op">(</span>sobject <span class="op">=</span> <span class="va">brain</span>, spatial_assay <span class="op">=</span> <span class="st">"Spatial"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="print-giotto-object">Print Giotto object<a class="anchor" aria-label="anchor" href="#print-giotto-object"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="downstream-analysis-using-giotto-object">Downstream Analysis using Giotto Object<a class="anchor" aria-label="anchor" href="#downstream-analysis-using-giotto-object"></a>
</h4>
<p>The downstream analysis of the giotto object being created by seurat object is given below.</p>
<div class="section level5">
<h5 id="high-efficiency-data-processing">High Efficiency Data Processing<a class="anchor" aria-label="anchor" href="#high-efficiency-data-processing"></a>
</h5>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/processGiotto.html">processGiotto</a></span><span class="op">(</span></span>
<span>    <span class="va">brainG</span>,</span>
<span>    filter_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                         feat_det_in_min_cells <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                         min_det_feats_per_cell <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    norm_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>norm_methods <span class="op">=</span> <span class="st">'standard'</span>,</span>
<span>                       scale_feats <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                       scalefactor <span class="op">=</span> <span class="fl">6000</span><span class="op">)</span>,</span>
<span>    stat_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_values <span class="op">=</span> <span class="st">'normalized'</span><span class="op">)</span>,</span>
<span>    adjust_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'normalized'</span><span class="op">)</span>,</span>
<span>                         covariate_columns <span class="op">=</span> <span class="st">'nr_feats'</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="dimension-reduction-and-pca">Dimension Reduction and PCA<a class="anchor" aria-label="anchor" href="#dimension-reduction-and-pca"></a>
</h5>
<p>First, we find HVF (highly varialble features) using the loess regression prediction model.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">brainG</span>, method <span class="op">=</span> <span class="st">'cov_loess'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/0-HVF.png"> then use HVF to compute the reduced dimensions using PCA.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Select genes highly variable genes that fit specified statistics</span></span>
<span><span class="co"># These are both found within feature metadata</span></span>
<span><span class="va">feature_metadata</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/getFeatureMetadata.html" class="external-link">getFeatureMetadata</a></span><span class="op">(</span><span class="va">brainG</span><span class="op">)</span><span class="op">[</span><span class="op">]</span></span>
<span><span class="va">featgenes</span> <span class="op">=</span> <span class="va">feature_metadata</span><span class="op">[</span><span class="va">hvf</span> <span class="op">==</span> <span class="st">'yes'</span> <span class="op">&amp;</span> <span class="va">perc_cells</span> <span class="op">&gt;</span> <span class="fl">4</span> <span class="op">&amp;</span> <span class="va">mean_expr_det</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">]</span><span class="op">$</span><span class="va">feat_ID</span></span>
<span></span>
<span><span class="co">## run PCA on expression values (default)</span></span>
<span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">brainG</span>, feats_to_use <span class="op">=</span> <span class="va">featgenes</span>, scale_unit <span class="op">=</span> <span class="cn">F</span>, center <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">brainG</span>,</span>
<span>                save_plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/1-PCA.png"></p>
<p>Here, we’ll proceed with running UMAP based on PCA dimension reduction and visualize the pre-clustering UMAP.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">brainG</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">)</span></span>
<span><span class="va">graph1</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">brainG</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/2-UMAP.png"></p>
</div>
<div class="section level5">
<h5 id="clustering">Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h5>
<p>Before proceeding with clustering cells based on their expression profiles, ensure that the Giotto Object has undergone PCA dimension reduction and includes either t-SNE or UMAP dimension reduction, as well as a defined neighbor network.</p>
<p>To initialize the clustering process, create a shared nearest neighbor network (sNN) with a specified number of nearest neighbors (‘k’):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">brainG</span>, type <span class="op">=</span> <span class="st">"sNN"</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span>, k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre></div>
<p>In Giotto, various clustering algorithms are available, such as k-means, Leiden, or Louvain. These algorithms automatically store cluster information within cell metadata, typically following default naming conventions. However, you can easily customize the cluster name by specifying the ‘name’ argument.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## k-means clustering</span></span>
<span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doKmeans.html">doKmeans</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">brainG</span>, dim_reduction_to_use <span class="op">=</span> <span class="st">'pca'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Plot UMAP post-clustering to visualize kmeans</span></span>
<span><span class="va">graph2</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span></span>
<span>    gobject <span class="op">=</span> <span class="va">brainG</span>,</span>
<span>    cell_color <span class="op">=</span> <span class="st">'kmeans'</span>,</span>
<span>    show_NN_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">2.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/3-UMAP.png"> #### Spatial co-expression</p>
<p>Using the spatial network previously established, we can employ the binSpect method to identify spatially variable genes.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialDelaunayNetwork.html" class="external-link">createSpatialDelaunayNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">brainG</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoSpatNetworks.html" class="external-link">showGiottoSpatNetworks</a></span><span class="op">(</span><span class="va">brainG</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ranktest</span> <span class="op">=</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span></span>
<span>    <span class="va">brainG</span>, bin_method <span class="op">=</span> <span class="st">'rank'</span>,</span>
<span>    calc_hub <span class="op">=</span> <span class="cn">T</span>, hub_min_int <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Here, we’ll narrow down our analysis by selecting the top 300 spatial genes identified through binSpect. Additionally, we’ll illustrate how to identify the genes that exhibit the highest spatial correlation.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3.1 cluster the top 500 spatial genes into 20 clusters</span></span>
<span><span class="va">ext_spatial_genes</span> <span class="op">=</span> <span class="va">ranktest</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">300</span>,<span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span></span>
<span><span class="co"># here we use existing detectSpatialCorGenes function to calculate pairwise distances between genes (but set network_smoothing=0 to use default clustering)</span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu"><a href="../reference/detectSpatialCorFeats.html">detectSpatialCorFeats</a></span><span class="op">(</span></span>
<span>    <span class="va">brainG</span>,</span>
<span>    method <span class="op">=</span> <span class="st">'network'</span>,</span>
<span>    spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>    subset_feats <span class="op">=</span> <span class="va">ext_spatial_genes</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># 3.2 identify most similar spatially correlated genes for one gene</span></span>
<span><span class="va">top10_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/showSpatialCorFeats.html">showSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, feats <span class="op">=</span> <span class="st">'Dsp'</span>, show_top_feats <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>Employing the pheatmap function, we can visualize spatial co-expression modules and iteratively adjust the number of clusters (k) for optimal representation.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3.3 identify potenial spatial co-expression</span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu"><a href="../reference/clusterSpatialCorFeats.html">clusterSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>, k <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize clusters</span></span>
<span><span class="va">graph3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/heatmSpatialCorFeats.html">heatmSpatialCorFeats</a></span><span class="op">(</span></span>
<span>    <span class="va">brainG</span>,</span>
<span>    spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>,</span>
<span>    use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>,</span>
<span>    heatmap_legend_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>title <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>,</span>
<span>    save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>base_height <span class="op">=</span> <span class="fl">6</span>, base_width <span class="op">=</span> <span class="fl">8</span>, units <span class="op">=</span> <span class="st">'cm'</span><span class="op">)</span>, </span>
<span>    show_plot <span class="op">=</span>  <span class="cn">TRUE</span>,</span>
<span>    return_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    save_plot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/4-heatmSpatialCorFeats.png"> We’ll proceed by isolating genes from individual co-expression modules and then consolidate them into meta-genes through aggregation.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3.4 create metagenes / co-expression modules</span></span>
<span><span class="va">cluster_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/getBalancedSpatCoexpressionFeats.html">getBalancedSpatCoexpressionFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, maximum <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainG</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createMetafeats.html" class="external-link">createMetafeats</a></span><span class="op">(</span><span class="va">brainG</span>, feat_clusters <span class="op">=</span> <span class="va">cluster_genes</span>, name <span class="op">=</span> <span class="st">'cluster_metagene'</span><span class="op">)</span>                     </span>
<span>    </span>
<span>    </span>
<span><span class="va">graph4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html" class="external-link">spatCellPlot</a></span><span class="op">(</span></span>
<span>    <span class="va">brainG</span>,</span>
<span>    spat_enr_names <span class="op">=</span> <span class="st">'cluster_metagene'</span>,</span>
<span>    cell_annotation_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">1</span>, cow_n_col <span class="op">=</span> <span class="fl">3</span>, gradient_style <span class="op">=</span> <span class="st">'sequential'</span></span>
<span><span class="op">)</span>   </span></code></pre></div>
<p><img src="images/interoperability_seurat/5-spatCellPlot2D.png"></p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="conversion-of-giotto-to-seurat-v5">Conversion of Giotto to Seurat V5<a class="anchor" aria-label="anchor" href="#conversion-of-giotto-to-seurat-v5"></a>
</h3>
<p>With Seurat’s recent upgrade to version 5, ensuring compatibility is essential. The ‘giottoToSeuratV5()’ function simplifies the process by seamlessly converting Giotto objects to the latest Seurat object.</p>
<div class="section level4">
<h4 id="reverting-giottoobj-to-seurat">Reverting GiottoObj to Seurat<a class="anchor" aria-label="anchor" href="#reverting-giottoobj-to-seurat"></a>
</h4>
<p>In this step, we revert the Giotto object, previously converted from Seurat, back to its original Seurat format:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giottoToSeuratV5.html" class="external-link">giottoToSeuratV5</a></span><span class="op">(</span><span class="va">brainG</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="print-seurat-object">Print Seurat Object<a class="anchor" aria-label="anchor" href="#print-seurat-object"></a>
</h4>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain2</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="downstream-analysis-of-giotto-to-seurat-v5">Downstream Analysis of Giotto to Seurat V5:<a class="anchor" aria-label="anchor" href="#downstream-analysis-of-giotto-to-seurat-v5"></a>
</h4>
<p>Following the conversion of Giotto to Seurat V5, downstream analysis can be conducted as outlined below:</p>
<div class="section level5">
<h5 id="load-libraries">Load libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h5>
<p>Load Required Libraries</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="visualization-of-cell-counts-distribution">Visualization of Cell Counts Distribution<a class="anchor" aria-label="anchor" href="#visualization-of-cell-counts-distribution"></a>
</h5>
<p>Visualize the distribution of cell counts using violin plots and spatial feature plots:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu">VlnPlot</span><span class="op">(</span><span class="va">brain2</span>, features <span class="op">=</span> <span class="st">"nCount_rna"</span>, pt.size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span> <span class="fu">NoLegend</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu">SpatialFeaturePlot</span><span class="op">(</span><span class="va">brain2</span>, features <span class="op">=</span> <span class="st">"nCount_rna"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span>
<span><span class="fu">wrap_plots</span><span class="op">(</span><span class="va">plot1</span>, <span class="va">plot2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/6-cellCountdist.png"> #### Data Transformation</p>
<p>To enhance the analysis, we apply SCTransform to perform data transformation on the RNA assay:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain2</span> <span class="op">&lt;-</span> <span class="fu">SCTransform</span><span class="op">(</span><span class="va">brain2</span>, assay <span class="op">=</span> <span class="st">"rna"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="dimensionality-reduction-and-clustering">Dimensionality Reduction and Clustering<a class="anchor" aria-label="anchor" href="#dimensionality-reduction-and-clustering"></a>
</h5>
<p>Following data transformation, we proceed with dimensionality reduction and clustering. This involves conducting PCA, identifying neighbors, performing clustering, and running UMAP for dimensionality reduction:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain2</span> <span class="op">&lt;-</span> <span class="fu">RunPCA</span><span class="op">(</span><span class="va">brain2</span>, assay <span class="op">=</span> <span class="st">"SCT"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">brain2</span> <span class="op">&lt;-</span> <span class="fu">FindNeighbors</span><span class="op">(</span><span class="va">brain2</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span><span class="va">brain2</span> <span class="op">&lt;-</span> <span class="fu">FindClusters</span><span class="op">(</span><span class="va">brain2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">brain2</span> <span class="op">&lt;-</span> <span class="fu">RunUMAP</span><span class="op">(</span><span class="va">brain2</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="visualization-of-clusters-on-umap">Visualization of Clusters on UMAP<a class="anchor" aria-label="anchor" href="#visualization-of-clusters-on-umap"></a>
</h5>
<p>To visualize clusters on UMAP, we generate dimensional plots with cluster labels for better interpretation:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">DimPlot</span><span class="op">(</span><span class="va">brain2</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">brain2</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, label.size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/7-seuratUMAP.png"> Additionally, we can highlight specific cell populations within the spatial dimensional plot:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">brain2</span>, cells.highlight <span class="op">=</span> <span class="fu">CellsByIdentities</span><span class="op">(</span>object <span class="op">=</span> <span class="va">brain2</span>, idents <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">3</span>,</span>
<span>    <span class="fl">5</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span>, facet.highlight <span class="op">=</span> <span class="cn">TRUE</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/8-spatialDim.png"> #### Identification of Spatially Variable Features</p>
<p>To identify spatially variable features between clusters, the FindMarkers function is employed. Subsequently, we visualize the top spatially variable features using SpatialFeaturePlot:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_markers</span> <span class="op">&lt;-</span> <span class="fu">FindMarkers</span><span class="op">(</span><span class="va">brain2</span>, ident.1 <span class="op">=</span> <span class="fl">5</span>, ident.2 <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span><span class="fu">SpatialFeaturePlot</span><span class="op">(</span>object <span class="op">=</span> <span class="va">brain2</span>, features <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">de_markers</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">1</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/9-spatVarFeat.png"></p>
<p>we identify spatially variable features within the spatial data. This is achieved through the <code>FindSpatiallyVariableFeatures</code> function, which calculates spatial statistics to determine the variability of features across spatial locations. In the following code chunk, we focus on the top 1000 variable features selected from the SCT assay using the Moransi selection method.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain2</span> <span class="op">&lt;-</span> <span class="fu">FindSpatiallyVariableFeatures</span><span class="op">(</span><span class="va">brain2</span>, assay <span class="op">=</span> <span class="st">"SCT"</span>, features <span class="op">=</span> <span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">brain2</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">]</span>,</span>
<span>    selection.method <span class="op">=</span> <span class="st">"moransi"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="subset-out-anatomical-regions">Subset out anatomical regions<a class="anchor" aria-label="anchor" href="#subset-out-anatomical-regions"></a>
</h5>
<p>This section focuses on subsetting anatomical regions from the dataset and visualizing them using spatial dimensional plots.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cortex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">brain2</span>, idents <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">6</span>, <span class="fl">7</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cortex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cortex</span>, <span class="va">anterior1_imagerow</span> <span class="op">&gt;</span> <span class="fl">400</span> <span class="op">|</span> <span class="va">anterior1_imagecol</span> <span class="op">&lt;</span> <span class="fl">150</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">cortex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cortex</span>, <span class="va">anterior1_imagerow</span> <span class="op">&gt;</span> <span class="fl">275</span> <span class="op">&amp;</span> <span class="va">anterior1_imagecol</span> <span class="op">&gt;</span> <span class="fl">370</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">cortex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cortex</span>, <span class="va">anterior1_imagerow</span> <span class="op">&gt;</span> <span class="fl">250</span> <span class="op">&amp;</span> <span class="va">anterior1_imagecol</span> <span class="op">&gt;</span> <span class="fl">440</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">cortex</span>, crop <span class="op">=</span> <span class="cn">TRUE</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">cortex</span>, crop <span class="op">=</span> <span class="cn">FALSE</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, pt.size.factor <span class="op">=</span> <span class="fl">1</span>, label.size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/10-anatomicalRegionSubset.png"></p>
</div>
<div class="section level5">
<h5 id="working-with-multiple-slices-in-seurat">Working with multiple slices in Seurat<a class="anchor" aria-label="anchor" href="#working-with-multiple-slices-in-seurat"></a>
</h5>
<p>To incorporate data from multiple slices in our analysis, we first load and transform the posterior slice:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brainP</span> <span class="op">&lt;-</span> <span class="fu">LoadData</span><span class="op">(</span><span class="st">"stxBrain"</span>, type <span class="op">=</span> <span class="st">"posterior1"</span><span class="op">)</span></span>
<span><span class="va">brainP</span> <span class="op">&lt;-</span> <span class="fu">SCTransform</span><span class="op">(</span><span class="va">brainP</span>, assay <span class="op">=</span> <span class="st">"Spatial"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Next, we merge the datasets obtained from the anterior and posterior slices:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain.merge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">brain2</span>, <span class="va">brainP</span><span class="op">)</span></span></code></pre></div>
<p>We set the default assay to “SCT” and combine variable features from both datasets:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DefaultAssay</span><span class="op">(</span><span class="va">brain.merge</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"SCT"</span></span>
<span><span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">brain.merge</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">brain</span><span class="op">)</span>, <span class="fu">VariableFeatures</span><span class="op">(</span><span class="va">brain2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">brain.merge</span> <span class="op">&lt;-</span> <span class="fu">RunPCA</span><span class="op">(</span><span class="va">brain.merge</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">brain.merge</span> <span class="op">&lt;-</span> <span class="fu">FindNeighbors</span><span class="op">(</span><span class="va">brain.merge</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span><span class="va">brain.merge</span> <span class="op">&lt;-</span> <span class="fu">FindClusters</span><span class="op">(</span><span class="va">brain.merge</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">brain.merge</span> <span class="op">&lt;-</span> <span class="fu">RunUMAP</span><span class="op">(</span><span class="va">brain.merge</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p>After performing PCA, identifying neighbors, clustering, and running UMAP, we visualize the results:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">brain.merge</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ident"</span>, <span class="st">"orig.ident"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/11-dimPlot.png"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialDimPlot</span><span class="op">(</span><span class="va">brain.merge</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/12-brainMerge.png"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SpatialFeaturePlot</span><span class="op">(</span><span class="va">brain.merge</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hpca"</span>, <span class="st">"Plp1"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/interoperability_seurat/13-brainMerge2.png"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true"></a>R version <span class="dv">4</span>.<span class="fl">3.1</span> (<span class="dv">2023-06-16</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true"></a>Platform<span class="op">:</span><span class="st"> </span>x86_<span class="dv">64</span><span class="op">-</span>pc<span class="op">-</span>linux<span class="op">-</span><span class="kw">gnu</span> (<span class="dv">64</span><span class="op">-</span>bit)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true"></a>Running under<span class="op">:</span><span class="st"> </span>AlmaLinux <span class="fl">8.9</span> (Midnight Oncilla)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true"></a> </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true"></a>Matrix products<span class="op">:</span><span class="st"> </span>default</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true"></a>BLAS<span class="op">:</span><span class="st">   </span><span class="er">/</span>share<span class="op">/</span>pkg<span class="fl">.8</span><span class="op">/</span>r<span class="op">/</span><span class="dv">4</span>.<span class="fl">3.1</span><span class="op">/</span>install<span class="op">/</span>lib64<span class="op">/</span>R<span class="op">/</span>lib<span class="op">/</span>libRblas.so </span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true"></a>LAPACK<span class="op">:</span><span class="st"> </span><span class="er">/</span>share<span class="op">/</span>pkg<span class="fl">.8</span><span class="op">/</span>r<span class="op">/</span><span class="dv">4</span>.<span class="fl">3.1</span><span class="op">/</span>install<span class="op">/</span>lib64<span class="op">/</span>R<span class="op">/</span>lib<span class="op">/</span>libRlapack.so;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true"></a> </span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true"></a>locale<span class="op">:</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true"></a><span class="st">  </span>[<span class="dv">1</span>] LC_CTYPE=en_US.UTF<span class="dv">-8</span>       LC_NUMERIC=C              </span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true"></a>  [<span class="dv">3</span>] LC_TIME=en_US.UTF<span class="dv">-8</span>        LC_COLLATE=en_US.UTF<span class="dv">-8</span>    </span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true"></a>  [<span class="dv">5</span>] LC_MONETARY=en_US.UTF<span class="dv">-8</span>    LC_MESSAGES=en_US.UTF<span class="dv">-8</span>   </span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true"></a>  [<span class="dv">7</span>] LC_PAPER=en_US.UTF<span class="dv">-8</span>       LC_NAME=C                 </span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true"></a>  [<span class="dv">9</span>] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true"></a> [<span class="dv">11</span>] LC_MEASUREMENT=en_US.UTF<span class="dv">-8</span> LC_IDENTIFICATION=C       </span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true"></a> </span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true"></a> time zone<span class="op">:</span><span class="st"> </span>America<span class="op">/</span>New_York</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true"></a> tzcode source<span class="op">:</span><span class="st"> </span><span class="kw">system</span> (glibc)</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true"></a> </span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true"></a> attached base packages<span class="op">:</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true"></a><span class="st"> </span>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true"></a></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true"></a> other attached packages<span class="op">:</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true"></a><span class="st"> </span>[<span class="dv">1</span>] dplyr_<span class="dv">1</span>.<span class="fl">1.4</span>               patchwork_<span class="dv">1</span>.<span class="dv">2</span>.<span class="fl">0.9000</span>     </span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true"></a> [<span class="dv">3</span>] ggplot2_<span class="dv">3</span>.<span class="dv">5</span>.<span class="fl">0.9000</span>        stxBrain.SeuratData_<span class="dv">0</span>.<span class="fl">1.1</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true"></a> [<span class="dv">5</span>] SeuratData_<span class="dv">0</span>.<span class="dv">2</span>.<span class="fl">2.9001</span>     Seurat_<span class="dv">5</span>.<span class="fl">0.1</span>             </span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true"></a> [<span class="dv">7</span>] SeuratObject_<span class="dv">5</span>.<span class="fl">0.1</span>        sp_<span class="fl">2.1</span><span class="dv">-3</span>                 </span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true"></a> [<span class="dv">9</span>] GiottoData_<span class="dv">0</span>.<span class="dv">2</span>.<span class="fl">9.0</span>        GiottoUtils_<span class="dv">0</span>.<span class="fl">1.6</span>        </span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true"></a> [<span class="dv">11</span>] Giotto_<span class="dv">4</span>.<span class="fl">0.6</span>              GiottoClass_<span class="dv">0</span>.<span class="fl">2.4</span>        </span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true"></a> [<span class="dv">13</span>] data.table_<span class="dv">1</span>.<span class="fl">15.4</span>        </span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true"></a></span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true"></a> loaded via a <span class="kw">namespace</span> (and not attached)<span class="op">:</span></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true"></a><span class="st"> </span>[<span class="dv">1</span>] matrixStats_<span class="dv">1</span>.<span class="fl">3.0</span>           spatstat.sparse_<span class="fl">3.0</span><span class="dv">-1</span>      </span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true"></a> [<span class="dv">3</span>] bitops_<span class="fl">1.0</span><span class="dv">-7</span>                httr_<span class="dv">1</span>.<span class="fl">4.7</span>                 </span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true"></a> [<span class="dv">5</span>] RColorBrewer_<span class="fl">1.1</span><span class="dv">-3</span>          doParallel_<span class="dv">1</span>.<span class="fl">0.17</span>          </span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true"></a> [<span class="dv">7</span>] tools_<span class="dv">4</span>.<span class="fl">3.1</span>                 sctransform_<span class="dv">0</span>.<span class="fl">4.1</span>          </span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true"></a> [<span class="dv">9</span>] backports_<span class="dv">1</span>.<span class="fl">4.1</span>             utf8_<span class="dv">1</span>.<span class="fl">2.4</span>                 </span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true"></a> [<span class="dv">11</span>] R6_<span class="dv">2</span>.<span class="fl">5.1</span>                    HDF5Array_<span class="dv">1</span>.<span class="fl">28.1</span>           </span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true"></a> [<span class="dv">13</span>] lazyeval_<span class="dv">0</span>.<span class="fl">2.2</span>              uwot_<span class="dv">0</span>.<span class="fl">1.14</span>                </span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true"></a> [<span class="dv">15</span>] rhdf5filters_<span class="dv">1</span>.<span class="fl">12.1</span>         GetoptLong_<span class="dv">1</span>.<span class="fl">0.5</span>           </span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true"></a> [<span class="dv">17</span>] withr_<span class="dv">3</span>.<span class="fl">0.0</span>                 gridExtra_<span class="fl">2.3</span>              </span>
<span id="cb34-42"><a href="#cb34-42" aria-hidden="true"></a> [<span class="dv">19</span>] progressr_<span class="dv">0</span>.<span class="fl">13.0</span>            cli_<span class="dv">3</span>.<span class="fl">6.2</span>                  </span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true"></a> [<span class="dv">21</span>] Biobase_<span class="dv">2</span>.<span class="fl">60.0</span>              Cairo_<span class="fl">1.6</span><span class="dv">-0</span>                </span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true"></a> [<span class="dv">23</span>] spatstat.explore_<span class="fl">3.2</span><span class="dv">-1</span>      fastDummies_<span class="dv">1</span>.<span class="fl">7.3</span>          </span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true"></a> [<span class="dv">25</span>] labeling_<span class="dv">0</span>.<span class="fl">4.3</span>              sass_<span class="dv">0</span>.<span class="fl">4.9</span>                 </span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true"></a> [<span class="dv">27</span>] spatstat.data_<span class="fl">3.0</span><span class="dv">-1</span>         ggridges_<span class="dv">0</span>.<span class="fl">5.4</span>             </span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true"></a> [<span class="dv">29</span>] pbapply_<span class="fl">1.7</span><span class="dv">-0</span>               dbscan_<span class="fl">1.1</span><span class="dv">-12</span>              </span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true"></a> [<span class="dv">31</span>] R.utils_<span class="dv">2</span>.<span class="fl">12.2</span>              parallelly_<span class="dv">1</span>.<span class="fl">36.0</span>          </span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true"></a> [<span class="dv">33</span>] limma_<span class="dv">3</span>.<span class="fl">56.2</span>                rstudioapi_<span class="fl">0.14</span>            </span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true"></a> [<span class="dv">35</span>] FNN_<span class="dv">1</span>.<span class="dv">1</span>.<span class="fl">3.2</span>                 generics_<span class="dv">0</span>.<span class="fl">1.3</span>             </span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true"></a> [<span class="dv">37</span>] shape_<span class="dv">1</span>.<span class="fl">4.6</span>                 gtools_<span class="dv">3</span>.<span class="fl">9.5</span>               </span>
<span id="cb34-52"><a href="#cb34-52" aria-hidden="true"></a> [<span class="dv">39</span>] ica_<span class="fl">1.0</span><span class="dv">-3</span>                   spatstat.random_<span class="fl">3.1</span><span class="dv">-5</span>      </span>
<span id="cb34-53"><a href="#cb34-53" aria-hidden="true"></a> [<span class="dv">41</span>] Matrix_<span class="fl">1.6</span><span class="dv">-4</span>                ggbeeswarm_<span class="dv">0</span>.<span class="fl">7.2</span>           </span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true"></a> [<span class="dv">43</span>] fansi_<span class="dv">1</span>.<span class="fl">0.6</span>                 S4Vectors_<span class="dv">0</span>.<span class="fl">38.1</span>           </span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true"></a> [<span class="dv">45</span>] abind_<span class="fl">1.4</span><span class="dv">-7</span>                 R.methodsS3_<span class="dv">1</span>.<span class="fl">8.2</span>          </span>
<span id="cb34-56"><a href="#cb34-56" aria-hidden="true"></a> [<span class="dv">47</span>] terra_<span class="fl">1.7</span><span class="dv">-71</span>                lifecycle_<span class="dv">1</span>.<span class="fl">0.4</span>            </span>
<span id="cb34-57"><a href="#cb34-57" aria-hidden="true"></a> [<span class="dv">49</span>] yaml_<span class="dv">2</span>.<span class="fl">3.8</span>                  edgeR_<span class="dv">3</span>.<span class="fl">42.4</span>               </span>
<span id="cb34-58"><a href="#cb34-58" aria-hidden="true"></a> [<span class="dv">51</span>] SummarizedExperiment_<span class="dv">1</span>.<span class="fl">30.2</span> rhdf5_<span class="dv">2</span>.<span class="fl">44.0</span>               </span>
<span id="cb34-59"><a href="#cb34-59" aria-hidden="true"></a> [<span class="dv">53</span>] Rtsne_<span class="fl">0.16</span>                  grid_<span class="dv">4</span>.<span class="fl">3.1</span>                 </span>
<span id="cb34-60"><a href="#cb34-60" aria-hidden="true"></a> [<span class="dv">55</span>] promises_<span class="dv">1</span>.<span class="fl">3.0</span>              dqrng_<span class="dv">0</span>.<span class="fl">3.0</span>                </span>
<span id="cb34-61"><a href="#cb34-61" aria-hidden="true"></a> [<span class="dv">57</span>] crayon_<span class="dv">1</span>.<span class="fl">5.2</span>                miniUI_<span class="dv">0</span>.<span class="dv">1</span>.<span class="fl">1.1</span>             </span>
<span id="cb34-62"><a href="#cb34-62" aria-hidden="true"></a> [<span class="dv">59</span>] lattice_<span class="fl">0.21</span><span class="dv">-8</span>              beachmat_<span class="dv">2</span>.<span class="fl">16.0</span>            </span>
<span id="cb34-63"><a href="#cb34-63" aria-hidden="true"></a> [<span class="dv">61</span>] cowplot_<span class="dv">1</span>.<span class="fl">1.3</span>               magick_<span class="dv">2</span>.<span class="fl">8.3</span>               </span>
<span id="cb34-64"><a href="#cb34-64" aria-hidden="true"></a> [<span class="dv">63</span>] pillar_<span class="dv">1</span>.<span class="fl">9.0</span>                knitr_<span class="fl">1.46</span>                 </span>
<span id="cb34-65"><a href="#cb34-65" aria-hidden="true"></a> [<span class="dv">65</span>] ComplexHeatmap_<span class="dv">2</span>.<span class="fl">16.0</span>       GenomicRanges_<span class="dv">1</span>.<span class="fl">52.0</span>       </span>
<span id="cb34-66"><a href="#cb34-66" aria-hidden="true"></a> [<span class="dv">67</span>] rjson_<span class="dv">0</span>.<span class="fl">2.21</span>                future.apply_<span class="dv">1</span>.<span class="fl">11.0</span>        </span>
<span id="cb34-67"><a href="#cb34-67" aria-hidden="true"></a> [<span class="dv">69</span>] codetools_<span class="fl">0.2</span><span class="dv">-19</span>            leiden_<span class="dv">0</span>.<span class="fl">4.3</span>               </span>
<span id="cb34-68"><a href="#cb34-68" aria-hidden="true"></a> [<span class="dv">71</span>] glue_<span class="dv">1</span>.<span class="fl">7.0</span>                  vctrs_<span class="dv">0</span>.<span class="fl">6.4</span>                </span>
<span id="cb34-69"><a href="#cb34-69" aria-hidden="true"></a> [<span class="dv">73</span>] png_<span class="fl">0.1</span><span class="dv">-8</span>                   spam_<span class="fl">2.9</span><span class="dv">-1</span>                 </span>
<span id="cb34-70"><a href="#cb34-70" aria-hidden="true"></a> [<span class="dv">75</span>] gtable_<span class="dv">0</span>.<span class="fl">3.4</span>                cachem_<span class="dv">1</span>.<span class="fl">0.8</span>               </span>
<span id="cb34-71"><a href="#cb34-71" aria-hidden="true"></a> [<span class="dv">77</span>] xfun_<span class="fl">0.43</span>                   S4Arrays_<span class="dv">1</span>.<span class="fl">0.4</span>             </span>
<span id="cb34-72"><a href="#cb34-72" aria-hidden="true"></a> [<span class="dv">79</span>] mime_<span class="fl">0.12</span>                   DropletUtils_<span class="dv">1</span>.<span class="fl">20.0</span>        </span>
<span id="cb34-73"><a href="#cb34-73" aria-hidden="true"></a> [<span class="dv">81</span>] survival_<span class="fl">3.5</span><span class="dv">-5</span>              SingleCellExperiment_<span class="dv">1</span>.<span class="fl">22.0</span></span>
<span id="cb34-74"><a href="#cb34-74" aria-hidden="true"></a> [<span class="dv">83</span>] iterators_<span class="dv">1</span>.<span class="fl">0.14</span>            ellipsis_<span class="dv">0</span>.<span class="fl">3.2</span>             </span>
<span id="cb34-75"><a href="#cb34-75" aria-hidden="true"></a> [<span class="dv">85</span>] fitdistrplus_<span class="fl">1.1</span><span class="dv">-11</span>         ROCR_<span class="fl">1.0</span><span class="dv">-11</span>                </span>
<span id="cb34-76"><a href="#cb34-76" aria-hidden="true"></a> [<span class="dv">87</span>] nlme_<span class="fl">3.1</span><span class="dv">-162</span>                RcppAnnoy_<span class="dv">0</span>.<span class="fl">0.20</span>           </span>
<span id="cb34-77"><a href="#cb34-77" aria-hidden="true"></a> [<span class="dv">89</span>] GenomeInfoDb_<span class="dv">1</span>.<span class="fl">36.1</span>         bslib_<span class="dv">0</span>.<span class="fl">7.0</span>                </span>
<span id="cb34-78"><a href="#cb34-78" aria-hidden="true"></a> [<span class="dv">91</span>] irlba_<span class="dv">2</span>.<span class="dv">3</span>.<span class="fl">5.1</span>               vipor_<span class="dv">0</span>.<span class="fl">4.5</span>                </span>
<span id="cb34-79"><a href="#cb34-79" aria-hidden="true"></a> [<span class="dv">93</span>] KernSmooth_<span class="fl">2.23</span><span class="dv">-21</span>          colorspace_<span class="fl">2.1</span><span class="dv">-0</span>           </span>
<span id="cb34-80"><a href="#cb34-80" aria-hidden="true"></a> [<span class="dv">95</span>] BiocGenerics_<span class="dv">0</span>.<span class="fl">46.0</span>         ggrastr_<span class="dv">1</span>.<span class="fl">0.2</span>              </span>
<span id="cb34-81"><a href="#cb34-81" aria-hidden="true"></a> [<span class="dv">97</span>] tidyselect_<span class="dv">1</span>.<span class="fl">2.1</span>            compiler_<span class="dv">4</span>.<span class="fl">3.1</span>             </span>
<span id="cb34-82"><a href="#cb34-82" aria-hidden="true"></a> [<span class="dv">99</span>] DelayedArray_<span class="dv">0</span>.<span class="fl">26.3</span>         plotly_<span class="dv">4</span>.<span class="fl">10.4</span>              </span>
<span id="cb34-83"><a href="#cb34-83" aria-hidden="true"></a>[<span class="dv">101</span>] checkmate_<span class="dv">2</span>.<span class="fl">3.1</span>             scales_<span class="dv">1</span>.<span class="fl">3.0</span>               </span>
<span id="cb34-84"><a href="#cb34-84" aria-hidden="true"></a>[<span class="dv">103</span>] lmtest_<span class="fl">0.9</span><span class="dv">-40</span>               rappdirs_<span class="dv">0</span>.<span class="fl">3.3</span>             </span>
<span id="cb34-85"><a href="#cb34-85" aria-hidden="true"></a>[<span class="dv">105</span>] stringr_<span class="dv">1</span>.<span class="fl">5.1</span>               SpatialExperiment_<span class="dv">1</span>.<span class="fl">10.0</span>   </span>
<span id="cb34-86"><a href="#cb34-86" aria-hidden="true"></a>[<span class="dv">107</span>] digest_<span class="dv">0</span>.<span class="fl">6.35</span>               goftest_<span class="fl">1.2</span><span class="dv">-3</span>              </span>
<span id="cb34-87"><a href="#cb34-87" aria-hidden="true"></a>[<span class="dv">109</span>] spatstat.utils_<span class="fl">3.0</span><span class="dv">-3</span>        rmarkdown_<span class="fl">2.26</span>             </span>
<span id="cb34-88"><a href="#cb34-88" aria-hidden="true"></a>[<span class="dv">111</span>] XVector_<span class="dv">0</span>.<span class="fl">40.0</span>              htmltools_<span class="dv">0</span>.<span class="dv">5</span>.<span class="fl">8.1</span>          </span>
<span id="cb34-89"><a href="#cb34-89" aria-hidden="true"></a>[<span class="dv">113</span>] GiottoVisuals_<span class="dv">0</span>.<span class="fl">1.7</span>         pkgconfig_<span class="dv">2</span>.<span class="fl">0.3</span>            </span>
<span id="cb34-90"><a href="#cb34-90" aria-hidden="true"></a>[<span class="dv">115</span>] sparseMatrixStats_<span class="dv">1</span>.<span class="fl">12.1</span>    MatrixGenerics_<span class="dv">1</span>.<span class="fl">12.2</span>      </span>
<span id="cb34-91"><a href="#cb34-91" aria-hidden="true"></a>[<span class="dv">117</span>] highr_<span class="fl">0.10</span>                  fastmap_<span class="dv">1</span>.<span class="fl">1.1</span>              </span>
<span id="cb34-92"><a href="#cb34-92" aria-hidden="true"></a>[<span class="dv">119</span>] rlang_<span class="dv">1</span>.<span class="fl">1.3</span>                 GlobalOptions_<span class="dv">0</span>.<span class="fl">1.2</span>        </span>
<span id="cb34-93"><a href="#cb34-93" aria-hidden="true"></a>[<span class="dv">121</span>] htmlwidgets_<span class="dv">1</span>.<span class="fl">6.4</span>           shiny_<span class="dv">1</span>.<span class="fl">7.4</span>                </span>
<span id="cb34-94"><a href="#cb34-94" aria-hidden="true"></a>[<span class="dv">123</span>] DelayedMatrixStats_<span class="dv">1</span>.<span class="fl">22.1</span>   farver_<span class="dv">2</span>.<span class="fl">1.1</span>               </span>
<span id="cb34-95"><a href="#cb34-95" aria-hidden="true"></a>[<span class="dv">125</span>] jquerylib_<span class="dv">0</span>.<span class="fl">1.4</span>             zoo_<span class="fl">1.8</span><span class="dv">-13</span>                 </span>
<span id="cb34-96"><a href="#cb34-96" aria-hidden="true"></a>[<span class="dv">127</span>] jsonlite_<span class="dv">1</span>.<span class="fl">8.8</span>              BiocParallel_<span class="dv">1</span>.<span class="fl">34.2</span>        </span>
<span id="cb34-97"><a href="#cb34-97" aria-hidden="true"></a>[<span class="dv">129</span>] R.oo_<span class="dv">1</span>.<span class="fl">25.0</span>                 BiocSingular_<span class="dv">1</span>.<span class="fl">16.0</span>        </span>
<span id="cb34-98"><a href="#cb34-98" aria-hidden="true"></a>[<span class="dv">131</span>] RCurl_<span class="fl">1.98-1.12</span>             magrittr_<span class="dv">2</span>.<span class="fl">0.3</span>             </span>
<span id="cb34-99"><a href="#cb34-99" aria-hidden="true"></a>[<span class="dv">133</span>] scuttle_<span class="dv">1</span>.<span class="fl">10.1</span>              GenomeInfoDbData_<span class="dv">1</span>.<span class="fl">2.10</span>    </span>
<span id="cb34-100"><a href="#cb34-100" aria-hidden="true"></a>[<span class="dv">135</span>] dotCall64_<span class="fl">1.0</span><span class="dv">-2</span>             Rhdf5lib_<span class="dv">1</span>.<span class="fl">22.0</span>            </span>
<span id="cb34-101"><a href="#cb34-101" aria-hidden="true"></a>[<span class="dv">137</span>] munsell_<span class="dv">0</span>.<span class="fl">5.1</span>               Rcpp_<span class="dv">1</span>.<span class="fl">0.12</span>                </span>
<span id="cb34-102"><a href="#cb34-102" aria-hidden="true"></a>[<span class="dv">139</span>] ape_<span class="fl">5.7</span><span class="dv">-1</span>                   reticulate_<span class="dv">1</span>.<span class="fl">36.0</span>          </span>
<span id="cb34-103"><a href="#cb34-103" aria-hidden="true"></a>[<span class="dv">141</span>] stringi_<span class="dv">1</span>.<span class="fl">8.3</span>               zlibbioc_<span class="dv">1</span>.<span class="fl">46.0</span>            </span>
<span id="cb34-104"><a href="#cb34-104" aria-hidden="true"></a>[<span class="dv">143</span>] MASS_<span class="fl">7.3</span><span class="dv">-60</span>                 plyr_<span class="dv">1</span>.<span class="fl">8.9</span>                 </span>
<span id="cb34-105"><a href="#cb34-105" aria-hidden="true"></a>[<span class="dv">145</span>] parallel_<span class="dv">4</span>.<span class="fl">3.1</span>              listenv_<span class="dv">0</span>.<span class="fl">9.0</span>              </span>
<span id="cb34-106"><a href="#cb34-106" aria-hidden="true"></a>[<span class="dv">147</span>] ggrepel_<span class="dv">0</span>.<span class="fl">9.5</span>               deldir_<span class="fl">2.0</span><span class="dv">-4</span>               </span>
<span id="cb34-107"><a href="#cb34-107" aria-hidden="true"></a>[<span class="dv">149</span>] splines_<span class="dv">4</span>.<span class="fl">3.1</span>               tensor_<span class="fl">1.5</span>                 </span>
<span id="cb34-108"><a href="#cb34-108" aria-hidden="true"></a>[<span class="dv">151</span>] circlize_<span class="dv">0</span>.<span class="fl">4.15</span>             locfit_<span class="fl">1.5-9.8</span>             </span>
<span id="cb34-109"><a href="#cb34-109" aria-hidden="true"></a>[<span class="dv">153</span>] colorRamp2_<span class="dv">0</span>.<span class="fl">1.0</span>            igraph_<span class="dv">2</span>.<span class="fl">0.3</span>               </span>
<span id="cb34-110"><a href="#cb34-110" aria-hidden="true"></a>[<span class="dv">155</span>] spatstat.geom_<span class="fl">3.2</span><span class="dv">-1</span>         RcppHNSW_<span class="dv">0</span>.<span class="fl">4.1</span>             </span>
<span id="cb34-111"><a href="#cb34-111" aria-hidden="true"></a>[<span class="dv">157</span>] reshape2_<span class="dv">1</span>.<span class="fl">4.4</span>              stats4_<span class="dv">4</span>.<span class="fl">3.1</span>               </span>
<span id="cb34-112"><a href="#cb34-112" aria-hidden="true"></a>[<span class="dv">159</span>] ScaledMatrix_<span class="dv">1</span>.<span class="fl">8.1</span>          evaluate_<span class="fl">0.23</span>              </span>
<span id="cb34-113"><a href="#cb34-113" aria-hidden="true"></a>[<span class="dv">161</span>] foreach_<span class="dv">1</span>.<span class="fl">5.2</span>               httpuv_<span class="dv">1</span>.<span class="fl">6.11</span>              </span>
<span id="cb34-114"><a href="#cb34-114" aria-hidden="true"></a>[<span class="dv">163</span>] RANN_<span class="dv">2</span>.<span class="fl">6.1</span>                  tidyr_<span class="dv">1</span>.<span class="fl">3.1</span>                </span>
<span id="cb34-115"><a href="#cb34-115" aria-hidden="true"></a>[<span class="dv">165</span>] purrr_<span class="dv">1</span>.<span class="fl">0.2</span>                 polyclip_<span class="fl">1.10</span><span class="dv">-4</span>            </span>
<span id="cb34-116"><a href="#cb34-116" aria-hidden="true"></a>[<span class="dv">167</span>] future_<span class="dv">1</span>.<span class="fl">32.0</span>               clue_<span class="fl">0.3</span><span class="dv">-64</span>                </span>
<span id="cb34-117"><a href="#cb34-117" aria-hidden="true"></a>[<span class="dv">169</span>] scattermore_<span class="fl">1.2</span>             rsvd_<span class="dv">1</span>.<span class="fl">0.5</span>                 </span>
<span id="cb34-118"><a href="#cb34-118" aria-hidden="true"></a>[<span class="dv">171</span>] xtable_<span class="fl">1.8</span><span class="dv">-6</span>                RSpectra_<span class="fl">0.16</span><span class="dv">-1</span>            </span>
<span id="cb34-119"><a href="#cb34-119" aria-hidden="true"></a>[<span class="dv">173</span>] later_<span class="dv">1</span>.<span class="fl">3.2</span>                 viridisLite_<span class="dv">0</span>.<span class="fl">4.2</span>          </span>
<span id="cb34-120"><a href="#cb34-120" aria-hidden="true"></a>[<span class="dv">175</span>] tibble_<span class="dv">3</span>.<span class="fl">2.1</span>                beeswarm_<span class="dv">0</span>.<span class="fl">4.0</span>             </span>
<span id="cb34-121"><a href="#cb34-121" aria-hidden="true"></a>[<span class="dv">177</span>] IRanges_<span class="dv">2</span>.<span class="fl">34.1</span>              cluster_<span class="dv">2</span>.<span class="fl">1.4</span>              </span>
<span id="cb34-122"><a href="#cb34-122" aria-hidden="true"></a>[<span class="dv">179</span>] globals_<span class="dv">0</span>.<span class="fl">16.2</span></span></code></pre></div>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
