<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MERFISH Mouse Hypothalamic Preoptic Region â€¢ Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MERFISH Mouse Hypothalamic Preoptic Region">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Process Giotto</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactivity</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
        <li>
          <a href="../articles/interactive_selection_cosmx.html">Interactive region selection on large datasets</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MERFISH Mouse Hypothalamic Preoptic Region</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/merfish_mouse_hypothalamic.Rmd" class="external-link"><code>vignettes/merfish_mouse_hypothalamic.Rmd</code></a></small>
      <div class="hidden name"><code>merfish_mouse_hypothalamic.Rmd</code></div>

    </div>

    
    
<div class="section level2" number="1">
<h2 number="1" id="dataset-explanation">
<span class="header-section-number">1</span> Dataset Explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p><a href="https://doi.org/10.1126/science.aau5324" class="external-link">Moffitt et al.</a> created a 3D spatial expression dataset consisting of 155 genes from ~1 million single cells acquired from mouse hypothalamic preoptic regions. Please ensure that <a href="https://www.gnu.org/software/wget/?" class="external-link">wget</a> is installed locally to streamline the download.</p>
<p>Clustering, 3D visualization, and cell type identification of clusters using marker genes will be explored in this tutorial.</p>
</div>
<div class="section level2" number="2">
<h2 number="2" id="start-giotto">
<span class="header-section-number">2</span> Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure GiottoData, a small, helper module for tutorials, is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"GiottoData"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/GiottoData"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 number="3" id="download-dataset">
<span class="header-section-number">3</span> Download Dataset<a class="anchor" aria-label="anchor" href="#download-dataset"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/GiottoData/" class="external-link">GiottoData</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Specify path from which data may be retrieved/stored</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/data/"</span></span>
<span></span>
<span><span class="co"># Specify path to which results may be saved</span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"/path/to/results/"</span></span>
<span></span>
<span><span class="co"># Optional: Specify a path to a Python executable within a conda or miniconda </span></span>
<span><span class="co"># environment. If set to NULL (default), the Python executable within the previously</span></span>
<span><span class="co"># installed Giotto environment will be used.</span></span>
<span><span class="va">python_path</span> <span class="op">&lt;-</span> <span class="cn">NULL</span> <span class="co"># alternatively, "/local/python/path/python" if desired.</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># In the event of authentication issues with wget, </span></span>
<span><span class="co"># add ", extra = "--no-check-certificate" " after the method argument.</span></span>
<span><span class="co"># Get the dataset:</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/getSpatialDataset.html" class="external-link">getSpatialDataset</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"merfish_preoptic"</span>, </span>
<span>                  directory <span class="op">=</span> <span class="va">data_path</span>, </span>
<span>                  method <span class="op">=</span> <span class="st">"wget"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 number="4" id="create-giotto-instructions-prepare-data">
<span class="header-section-number">4</span> Create Giotto Instructions &amp; Prepare Data<a class="anchor" aria-label="anchor" href="#create-giotto-instructions-prepare-data"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Optional, but encouraged: Set Giotto instructions</span></span>
<span><span class="va">instructions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_instructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span>save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                         show_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                         return_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                         save_dir <span class="op">=</span> <span class="va">results_folder</span>, </span>
<span>                                         python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create file paths to feed data into Giotto Object</span></span>
<span><span class="va">expr_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"merFISH_3D_data_expression.txt.gz"</span><span class="op">)</span></span>
<span><span class="va">loc_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"merFISH_3D_data_cell_locations.txt"</span><span class="op">)</span></span>
<span><span class="va">meta_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"merFISH_3D_metadata.txt"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create Giotto object</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expr_path</span>,</span>
<span>                                      spatial_locs <span class="op">=</span> <span class="va">loc_path</span>,</span>
<span>                                      instructions <span class="op">=</span> <span class="va">instructions</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add additional metadata</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">meta_path</span><span class="op">)</span></span>
<span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                                   new_metadata <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">layer_ID</span>, </span>
<span>                                   vector_name <span class="op">=</span> <span class="st">"layer_ID"</span><span class="op">)</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>                                   new_metadata <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">orig_cell_types</span>,</span>
<span>                                   vector_name <span class="op">=</span> <span class="st">"orig_cell_types"</span><span class="op">)</span></span></code></pre></div>
<p>First pre-test filter parameters for both features and cells.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                    detection <span class="op">=</span> <span class="st">"feats"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F0-filterDistributions.png"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                    detection <span class="op">=</span> <span class="st">"cells"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F1-filterDistributions.png"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterCombinations.html">filterCombinations</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>                   expression_thresholds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1e-6</span>,<span class="fl">1e-5</span><span class="op">)</span>,</span>
<span>                   feat_det_in_min_cells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">1000</span>, <span class="fl">1500</span><span class="op">)</span>,</span>
<span>                   min_det_feats_per_cell <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F2-filterCombinations.png"></p>
</div>
<div class="section level2" number="5">
<h2 number="5" id="data-processing">
<span class="header-section-number">5</span> Data Processing<a class="anchor" aria-label="anchor" href="#data-processing"></a>
</h2>
<p>Use the previously generated plots to inform filter decisions.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter data</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>,</span>
<span>                                feat_det_in_min_cells <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                                min_det_feats_per_cell <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Normalize data</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                   scalefactor <span class="op">=</span> <span class="fl">10000</span>, </span>
<span>                                   verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add statistics to Giotto Object</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                 expression_values <span class="op">=</span> <span class="st">"normalized"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Adjust for covariates</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjustGiottoMatrix.html">adjustGiottoMatrix</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                      expression_values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                                      batch_columns <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                                      covariate_columns <span class="op">=</span> <span class="st">"layer_ID"</span>,</span>
<span>                                      return_gobject <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                      update_slot <span class="op">=</span> <span class="st">"custom"</span><span class="op">)</span></span></code></pre></div>
<p>Now, take a glance at the data in both 2D and 3D.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 2D</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F3-spatPlot2D.png"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3D</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.25</span>, </span>
<span>           axis_scale <span class="op">=</span> <span class="st">"real"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F4-spat3D.png"></p>
</div>
<div class="section level2" number="6">
<h2 number="6" id="dimension-reduction">
<span class="header-section-number">6</span> Dimension Reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<p>There are only 155 genes within this dataset. Use them all (default) within the dimension reduction.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                          feats_to_use <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                          scale_unit <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                          center <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View details about the principal components</span></span>
<span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">merFISH_gobject</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F5-screePlot.png"></p>
<p>After the PCA, a UMAP may be run. Run the UMAP so clusters may be visualized upon it.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                           dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, </span>
<span>                           n_components <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                           n_threads <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP_3D.html" class="external-link">plotUMAP_3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>            point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> </span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F6-UMAP_3D.png"></p>
</div>
<div class="section level2" number="7">
<h2 number="7" id="clustering">
<span class="header-section-number">7</span> Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<p>Create a nearest network, then perform Leiden clustering. The clusters may be visualized on a UMAP.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a sNN network (default)</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                        dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, </span>
<span>                                        k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Leiden cluster</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                   resolution <span class="op">=</span> <span class="fl">0.2</span>, </span>
<span>                                   n_iterations <span class="op">=</span> <span class="fl">200</span>,</span>
<span>                                   name <span class="op">=</span> <span class="st">"leiden_0.2_200"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the clusters upon the UMAP</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP_3D.html" class="external-link">plotUMAP_3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">"leiden_0.2_200"</span>, </span>
<span>            point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>            show_center_label <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F7-UMAP_3D.png"></p>
</div>
<div class="section level2" number="8">
<h2 number="8" id="co-visualize">
<span class="header-section-number">8</span> Co-Visualize<a class="anchor" aria-label="anchor" href="#co-visualize"></a>
</h2>
<p>View the clusters in-tissue on each layer.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"leiden_0.2_200"</span>, </span>
<span>           group_by <span class="op">=</span> <span class="st">"layer_ID"</span>, </span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">260</span>, <span class="fl">160</span>, <span class="fl">60</span>, <span class="op">-</span><span class="fl">40</span>, <span class="op">-</span><span class="fl">140</span>, <span class="op">-</span><span class="fl">240</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F8-spatPlot2D.png"></p>
</div>
<div class="section level2" number="9">
<h2 number="9" id="cell-type-marker-gene-detection">
<span class="header-section-number">9</span> Cell Type Marker Gene Detection<a class="anchor" aria-label="anchor" href="#cell-type-marker-gene-detection"></a>
</h2>
<p>Call <strong>findMarkers_one_vs_all</strong> to identify marker features. Click the function to see alternate methods, or look findGiniMarkers section for details on the gini method. Once marker features have been determined, observe the differential expression across clusters within the violin plot.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers_gini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>,</span>
<span>                                  method <span class="op">=</span> <span class="st">"gini"</span>,</span>
<span>                                  expression_values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                                  cluster_column <span class="op">=</span> <span class="st">"leiden_0.2_200"</span>,</span>
<span>                                  min_feats <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                  rank_score <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display details about the marker genes in-console</span></span>
<span><span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># violinplot</span></span>
<span><span class="va">topgenes_gini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/violinPlot.html" class="external-link">violinPlot</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>           feats <span class="op">=</span> <span class="va">topgenes_gini</span>, </span>
<span>           cluster_column <span class="op">=</span> <span class="st">"leiden_0.2_200"</span>, </span>
<span>           strip_position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F9-violinPlot.png"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">topgenes_gini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">6</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotMetaDataHeatmap.html" class="external-link">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                    expression_values <span class="op">=</span> <span class="st">"scaled"</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="st">"leiden_0.2_200"</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">topgenes_gini</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F10-plotMetaDataHeatmap.png"></p>
</div>
<div class="section level2" number="10">
<h2 number="10" id="cell-type-annotation">
<span class="header-section-number">10</span> Cell Type Annotation<a class="anchor" aria-label="anchor" href="#cell-type-annotation"></a>
</h2>
<p>Use known marker and differentially expressed genes (DEGs) to identify cell type for each cluster.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Known markers and DEGs</span></span>
<span><span class="va">selected_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Myh11"</span>, <span class="st">"Klf4"</span>, <span class="st">"Fn1"</span>, <span class="st">"Cd24a"</span>, <span class="st">"Cyr61"</span>, <span class="st">"Nnat"</span>, <span class="st">"Trh"</span>, </span>
<span>                    <span class="st">"Selplg"</span>, <span class="st">"Pou3f2"</span>, <span class="st">"Aqp4"</span>, <span class="st">"Traf4"</span>, <span class="st">"Pdgfra"</span>, <span class="st">"Opalin"</span>, </span>
<span>                    <span class="st">"Mbp"</span>, <span class="st">"Ttyh2"</span>, <span class="st">"Fezf1"</span>, <span class="st">"Cbln1"</span>, <span class="st">"Slc17a6"</span>, <span class="st">"Scg2"</span>, <span class="st">"Isl1"</span>, <span class="st">"Gad1"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">merFISH_gobject</span><span class="op">)</span></span>
<span><span class="va">cluster_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">cell_metadata</span><span class="op">$</span><span class="va">leiden_0.2_200</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotMetaDataHeatmap.html" class="external-link">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                    expression_values <span class="op">=</span> <span class="st">"scaled"</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="st">"leiden_0.2_200"</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">selected_genes</span>,</span>
<span>                    custom_feat_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">selected_genes</span><span class="op">)</span>,</span>
<span>                    custom_cluster_order <span class="op">=</span> <span class="va">cluster_order</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F11-plotMetaDataHeatmap.png"></p>
<p>Since there are more Leiden clusters than there are types of cells in this dataset, the same cell type may be assigned to different cluster numbers. This may be done only after verifying that particular clusters highly express marker genes corresponding to the same cell type. The above heatmap is used to streamline this process. Call <strong>annotateGiotto</strong> to map cell types to Leiden clusters; these will appear in <em>cell_metadata</em> within the <strong>giottoObject</strong>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Name clusters</span></span>
<span><span class="va">clusters_cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibitory"</span>, <span class="st">"Inhibitory"</span>, <span class="st">"Excitatory"</span>, </span>
<span>                         <span class="st">"Astrocyte"</span>, <span class="st">"OD Mature"</span>, <span class="st">"Endothelial"</span>,</span>
<span>                         <span class="st">"OD Mature"</span>, <span class="st">"OD Immature"</span>,  <span class="st">"Ambiguous"</span>,</span>
<span>                         <span class="st">"Ependymal"</span>, <span class="st">"Endothelial"</span>, <span class="st">"Microglia"</span>, </span>
<span>                         <span class="st">"OD Mature"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusters_cell_types</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">cluster_order</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/annotateGiotto.html" class="external-link">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                  annotation_vector <span class="op">=</span> <span class="va">clusters_cell_types</span>,</span>
<span>                                  cluster_column <span class="op">=</span> <span class="st">"leiden_0.2_200"</span>, </span>
<span>                                  name <span class="op">=</span> <span class="st">"cell_types"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## show heatmap</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotMetaDataHeatmap.html" class="external-link">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                    expression_values <span class="op">=</span> <span class="st">"scaled"</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">selected_genes</span>,</span>
<span>                    custom_feat_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">selected_genes</span><span class="op">)</span>,</span>
<span>                    custom_cluster_order <span class="op">=</span> <span class="va">clusters_cell_types</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F12-plotMetaDataHeatmap.png"></p>
</div>
<div class="section level2" number="11">
<h2 number="11" id="visualize">
<span class="header-section-number">11</span> Visualize<a class="anchor" aria-label="anchor" href="#visualize"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assign colors to each cell type </span></span>
<span><span class="va">mycolorcode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"lightblue"</span>, <span class="st">"yellowgreen"</span>,<span class="st">"purple"</span>, <span class="st">"darkred"</span>,</span>
<span>                <span class="st">"magenta"</span>, <span class="st">"mediumblue"</span>, <span class="st">"yellow"</span>, <span class="st">"gray"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mycolorcode</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibitory"</span>, <span class="st">"Excitatory"</span>, <span class="st">"OD Mature"</span>, <span class="st">"OD Immature"</span>, </span>
<span>                        <span class="st">"Astrocyte"</span>, <span class="st">"Microglia"</span>, <span class="st">"Ependymal"</span>, <span class="st">"Endothelial"</span>, </span>
<span>                        <span class="st">"Ambiguous"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP_3D.html" class="external-link">plotUMAP_3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>            point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>            cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F13-UMAP_3D.png"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           axis_scale <span class="op">=</span> <span class="st">"real"</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">"sdimx"</span>, </span>
<span>           sdimy <span class="op">=</span> <span class="st">"sdimy"</span>, </span>
<span>           sdimz <span class="op">=</span> <span class="st">"sdimz"</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F14-spat3D.png"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"layer_ID"</span>, </span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F15-spatPlot2D.png"></p>
<div class="section level3" number="11.1">
<h3 number="11.1" id="excitatory-cells-only">
<span class="header-section-number">11.1</span> Excitatory Cells Only<a class="anchor" aria-label="anchor" href="#excitatory-cells-only"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           axis_scale <span class="op">=</span> <span class="st">"real"</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">"sdimx"</span>, </span>
<span>           sdimy <span class="op">=</span> <span class="st">"sdimy"</span>, </span>
<span>           sdimz <span class="op">=</span> <span class="st">"sdimz"</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">"Excitatory"</span>,</span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F16-spat3D.png"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">"Excitatory"</span>, </span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"layer_ID"</span>, </span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F17-spatPlot2D.png"></p>
</div>
<div class="section level3" number="11.2">
<h3 number="11.2" id="inhibitory-cells-only">
<span class="header-section-number">11.2</span> Inhibitory Cells Only<a class="anchor" aria-label="anchor" href="#inhibitory-cells-only"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           axis_scale <span class="op">=</span> <span class="st">"real"</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">"sdimx"</span>, </span>
<span>           sdimy <span class="op">=</span> <span class="st">"sdimy"</span>, </span>
<span>           sdimz <span class="op">=</span> <span class="st">"sdimz"</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">"Inhibitory"</span>, </span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F18-spat3D.png"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">"Inhibitory"</span>,</span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"layer_ID"</span>, </span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F19-spatPlot2D.png"></p>
</div>
<div class="section level3" number="11.3">
<h3 number="11.3" id="od-and-astrocytes-only">
<span class="header-section-number">11.3</span> OD and Astrocytes Only<a class="anchor" aria-label="anchor" href="#od-and-astrocytes-only"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           axis_scale <span class="op">=</span> <span class="st">"real"</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">"sdimx"</span>, </span>
<span>           sdimy <span class="op">=</span> <span class="st">"sdimy"</span>, </span>
<span>           sdimz <span class="op">=</span> <span class="st">"sdimz"</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Astrocyte"</span>, <span class="st">"OD Mature"</span>, <span class="st">"OD Immature"</span><span class="op">)</span>, </span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F20-spat3D.png"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Astrocyte"</span>, <span class="st">"OD Mature"</span>, <span class="st">"OD Immature"</span><span class="op">)</span>, </span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"layer_ID"</span>, </span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F21-spatPlot2D.png"></p>
</div>
<div class="section level3" number="11.4">
<h3 number="11.4" id="other-cells-only">
<span class="header-section-number">11.4</span> Other Cells Only<a class="anchor" aria-label="anchor" href="#other-cells-only"></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           axis_scale <span class="op">=</span> <span class="st">"real"</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">"sdimx"</span>, </span>
<span>           sdimy <span class="op">=</span> <span class="st">"sdimy"</span>, </span>
<span>           sdimz <span class="op">=</span> <span class="st">"sdimz"</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Microglia"</span>, <span class="st">"Ependymal"</span>, <span class="st">"Endothelial"</span><span class="op">)</span>, </span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F22-spat3D.png"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Microglia"</span>, <span class="st">"Ependymal"</span>, <span class="st">"Endothelial"</span><span class="op">)</span>, </span>
<span>           show_other_cells <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"layer_ID"</span>, </span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fmerfish_mouse_hypothalamic%2F23-spatPlot2D.png"></p>
</div>
</div>
<div class="section level2" number="12">
<h2 number="12" id="session-info">
<span class="header-section-number">12</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true"></a>R version <span class="dv">4</span>.<span class="fl">3.2</span> (<span class="dv">2023-10-31</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true"></a>Platform<span class="op">:</span><span class="st"> </span>x86_<span class="dv">64</span><span class="op">-</span>apple<span class="op">-</span><span class="kw">darwin20</span> (<span class="dv">64</span><span class="op">-</span>bit)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true"></a>Running under<span class="op">:</span><span class="st"> </span>macOS Sonoma <span class="dv">14</span>.<span class="fl">3.1</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true"></a>Matrix products<span class="op">:</span><span class="st"> </span>default</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true"></a>BLAS<span class="op">:</span><span class="st">   </span><span class="er">/</span>System<span class="op">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>Accelerate.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>Frameworks<span class="op">/</span>vecLib.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>libBLAS.dylib </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true"></a>LAPACK<span class="op">:</span><span class="st"> </span><span class="er">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>R.framework<span class="op">/</span>Versions<span class="op">/</span><span class="fl">4.3</span><span class="op">-</span>x86_<span class="dv">64</span><span class="op">/</span>Resources<span class="op">/</span>lib<span class="op">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true"></a></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true"></a>locale<span class="op">:</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>C<span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true"></a>time zone<span class="op">:</span><span class="st"> </span>America<span class="op">/</span>New_York</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true"></a>tzcode source<span class="op">:</span><span class="st"> </span>internal</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true"></a></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true"></a>attached base packages<span class="op">:</span></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true"></a></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true"></a>other attached packages<span class="op">:</span></span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true"></a>[<span class="dv">1</span>] GiottoData_<span class="dv">0</span>.<span class="dv">2</span>.<span class="fl">7.0</span> GiottoUtils_<span class="dv">0</span>.<span class="fl">1.5</span>  Giotto_<span class="dv">4</span>.<span class="fl">0.2</span>       GiottoClass_<span class="dv">0</span>.<span class="fl">1.3</span> </span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true"></a></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true"></a>loaded via a <span class="kw">namespace</span> (and not attached)<span class="op">:</span></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true"></a><span class="st">  </span>[<span class="dv">1</span>] colorRamp2_<span class="dv">0</span>.<span class="fl">1.0</span>            bitops_<span class="fl">1.0</span><span class="dv">-7</span>                rlang_<span class="dv">1</span>.<span class="fl">1.3</span>                </span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true"></a>  [<span class="dv">4</span>] magrittr_<span class="dv">2</span>.<span class="fl">0.3</span>              RcppAnnoy_<span class="dv">0</span>.<span class="fl">0.22</span>            matrixStats_<span class="dv">1</span>.<span class="fl">2.0</span>          </span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true"></a>  [<span class="dv">7</span>] compiler_<span class="dv">4</span>.<span class="fl">3.2</span>              png_<span class="fl">0.1</span><span class="dv">-8</span>                   systemfonts_<span class="dv">1</span>.<span class="fl">0.5</span>          </span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true"></a> [<span class="dv">10</span>] vctrs_<span class="dv">0</span>.<span class="fl">6.5</span>                 pkgconfig_<span class="dv">2</span>.<span class="fl">0.3</span>             SpatialExperiment_<span class="dv">1</span>.<span class="fl">12.0</span>   </span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true"></a> [<span class="dv">13</span>] crayon_<span class="dv">1</span>.<span class="fl">5.2</span>                fastmap_<span class="dv">1</span>.<span class="fl">1.1</span>               backports_<span class="dv">1</span>.<span class="fl">4.1</span>            </span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true"></a> [<span class="dv">16</span>] magick_<span class="dv">2</span>.<span class="fl">8.2</span>                XVector_<span class="dv">0</span>.<span class="fl">42.0</span>              ellipsis_<span class="dv">0</span>.<span class="fl">3.2</span>             </span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true"></a> [<span class="dv">19</span>] labeling_<span class="dv">0</span>.<span class="fl">4.3</span>              utf8_<span class="dv">1</span>.<span class="fl">2.4</span>                  rmarkdown_<span class="fl">2.25</span>             </span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true"></a> [<span class="dv">22</span>] ragg_<span class="dv">1</span>.<span class="fl">2.7</span>                  purrr_<span class="dv">1</span>.<span class="fl">0.2</span>                 xfun_<span class="fl">0.42</span>                  </span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true"></a> [<span class="dv">25</span>] zlibbioc_<span class="dv">1</span>.<span class="fl">48.0</span>             beachmat_<span class="dv">2</span>.<span class="fl">18.0</span>             GenomeInfoDb_<span class="dv">1</span>.<span class="fl">38.6</span>        </span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true"></a> [<span class="dv">28</span>] jsonlite_<span class="dv">1</span>.<span class="fl">8.8</span>              DelayedArray_<span class="dv">0</span>.<span class="fl">28.0</span>         BiocParallel_<span class="dv">1</span>.<span class="fl">36.0</span>        </span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true"></a> [<span class="dv">31</span>] terra_<span class="fl">1.7</span><span class="dv">-71</span>                irlba_<span class="dv">2</span>.<span class="dv">3</span>.<span class="fl">5.1</span>               parallel_<span class="dv">4</span>.<span class="fl">3.2</span>             </span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true"></a> [<span class="dv">34</span>] R6_<span class="dv">2</span>.<span class="fl">5.1</span>                    RColorBrewer_<span class="fl">1.1</span><span class="dv">-3</span>          limma_<span class="dv">3</span>.<span class="fl">58.1</span>               </span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true"></a> [<span class="dv">37</span>] reticulate_<span class="dv">1</span>.<span class="fl">35.0</span>           GenomicRanges_<span class="dv">1</span>.<span class="fl">54.1</span>        Rcpp_<span class="dv">1</span>.<span class="fl">0.12</span>                </span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true"></a> [<span class="dv">40</span>] SummarizedExperiment_<span class="dv">1</span>.<span class="fl">32.0</span> knitr_<span class="fl">1.45</span>                  R.utils_<span class="dv">2</span>.<span class="fl">12.3</span>             </span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true"></a> [<span class="dv">43</span>] IRanges_<span class="dv">2</span>.<span class="fl">36.0</span>              igraph_<span class="dv">2</span>.<span class="dv">0</span>.<span class="fl">1.1</span>              Matrix_<span class="fl">1.6</span><span class="dv">-5</span>               </span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true"></a> [<span class="dv">46</span>] tidyselect_<span class="dv">1</span>.<span class="fl">2.0</span>            rstudioapi_<span class="dv">0</span>.<span class="fl">15.0</span>           abind_<span class="fl">1.4</span><span class="dv">-5</span>                </span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true"></a> [<span class="dv">49</span>] yaml_<span class="dv">2</span>.<span class="fl">3.8</span>                  codetools_<span class="fl">0.2</span><span class="dv">-19</span>            lattice_<span class="fl">0.22</span><span class="dv">-5</span>             </span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true"></a> [<span class="dv">52</span>] tibble_<span class="dv">3</span>.<span class="fl">2.1</span>                Biobase_<span class="dv">2</span>.<span class="fl">62.0</span>              withr_<span class="dv">3</span>.<span class="fl">0.0</span>                </span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true"></a> [<span class="dv">55</span>] evaluate_<span class="fl">0.23</span>               pillar_<span class="dv">1</span>.<span class="fl">9.0</span>                MatrixGenerics_<span class="dv">1</span>.<span class="fl">14.0</span>      </span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true"></a> [<span class="dv">58</span>] checkmate_<span class="dv">2</span>.<span class="fl">3.1</span>             stats4_<span class="dv">4</span>.<span class="fl">3.2</span>                dbscan_<span class="fl">1.1</span><span class="dv">-12</span>              </span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true"></a> [<span class="dv">61</span>] plotly_<span class="dv">4</span>.<span class="fl">10.4</span>               generics_<span class="dv">0</span>.<span class="fl">1.3</span>              RCurl_<span class="fl">1.98-1.14</span>            </span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true"></a> [<span class="dv">64</span>] S4Vectors_<span class="dv">0</span>.<span class="fl">40.2</span>            ggplot2_<span class="dv">3</span>.<span class="fl">4.4</span>               munsell_<span class="dv">0</span>.<span class="fl">5.0</span>              </span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true"></a> [<span class="dv">67</span>] scales_<span class="dv">1</span>.<span class="fl">3.0</span>                gtools_<span class="dv">3</span>.<span class="fl">9.5</span>                glue_<span class="dv">1</span>.<span class="fl">7.0</span>                 </span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true"></a> [<span class="dv">70</span>] lazyeval_<span class="dv">0</span>.<span class="fl">2.2</span>              tools_<span class="dv">4</span>.<span class="fl">3.2</span>                 GiottoVisuals_<span class="dv">0</span>.<span class="fl">1.4</span>        </span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true"></a> [<span class="dv">73</span>] data.table_<span class="dv">1</span>.<span class="fl">15.0</span>           ScaledMatrix_<span class="dv">1</span>.<span class="fl">10.0</span>         cowplot_<span class="dv">1</span>.<span class="fl">1.3</span>              </span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true"></a> [<span class="dv">76</span>] grid_<span class="dv">4</span>.<span class="fl">3.2</span>                  tidyr_<span class="dv">1</span>.<span class="fl">3.1</span>                 crosstalk_<span class="dv">1</span>.<span class="fl">2.1</span>            </span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true"></a> [<span class="dv">79</span>] colorspace_<span class="fl">2.1</span><span class="dv">-0</span>            SingleCellExperiment_<span class="dv">1</span>.<span class="fl">24.0</span> GenomeInfoDbData_<span class="dv">1</span>.<span class="fl">2.11</span>    </span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true"></a> [<span class="dv">82</span>] BiocSingular_<span class="dv">1</span>.<span class="fl">18.0</span>         cli_<span class="dv">3</span>.<span class="fl">6.2</span>                   rsvd_<span class="dv">1</span>.<span class="fl">0.5</span>                 </span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true"></a> [<span class="dv">85</span>] textshaping_<span class="dv">0</span>.<span class="fl">3.7</span>           fansi_<span class="dv">1</span>.<span class="fl">0.6</span>                 S4Arrays_<span class="dv">1</span>.<span class="fl">2.0</span>             </span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true"></a> [<span class="dv">88</span>] viridisLite_<span class="dv">0</span>.<span class="fl">4.2</span>           dplyr_<span class="dv">1</span>.<span class="fl">1.4</span>                 uwot_<span class="dv">0</span>.<span class="fl">1.16</span>                </span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true"></a> [<span class="dv">91</span>] gtable_<span class="dv">0</span>.<span class="fl">3.4</span>                R.methodsS3_<span class="dv">1</span>.<span class="fl">8.2</span>           digest_<span class="dv">0</span>.<span class="fl">6.34</span>              </span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true"></a> [<span class="dv">94</span>] progressr_<span class="dv">0</span>.<span class="fl">14.0</span>            BiocGenerics_<span class="dv">0</span>.<span class="fl">48.1</span>         SparseArray_<span class="dv">1</span>.<span class="fl">2.4</span>          </span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true"></a> [<span class="dv">97</span>] ggrepel_<span class="dv">0</span>.<span class="fl">9.5</span>               rjson_<span class="dv">0</span>.<span class="fl">2.21</span>                htmlwidgets_<span class="dv">1</span>.<span class="fl">6.4</span>          </span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true"></a>[<span class="dv">100</span>] farver_<span class="dv">2</span>.<span class="fl">1.1</span>                htmltools_<span class="dv">0</span>.<span class="fl">5.7</span>             R.oo_<span class="dv">1</span>.<span class="fl">26.0</span>                </span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true"></a>[<span class="dv">103</span>] lifecycle_<span class="dv">1</span>.<span class="fl">0.4</span>             httr_<span class="dv">1</span>.<span class="fl">4.7</span>                  statmod_<span class="dv">1</span>.<span class="fl">5.0</span>   </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. ChÃ¡vez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew Oâ€™Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
