<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Multi-omics Mouse Embryo DBiT-Seq • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Multi-omics Mouse Embryo DBiT-Seq">
<meta property="og:description" content="Giotto">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/interoperability_seurat.html">Seurat</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Multi-omics Mouse Embryo DBiT-Seq</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/vignettes/mouse_embryo_dbitseq.Rmd" class="external-link"><code>vignettes/mouse_embryo_dbitseq.Rmd</code></a></small>
      <div class="hidden name"><code>mouse_embryo_dbitseq.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="dataset-explanation">1. Dataset explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p>The dataset was created by <a href="https://www.sciencedirect.com/science/article/pii/S0092867420313908?via%3Dihub" class="external-link">Liu,
et al 2020</a> and downloaded from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE137986" class="external-link uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE137986</a>.
For running this tutorial, we will use the dataset E10 Whole (50 μm)
2.</p>
</div>
<div class="section level2">
<h2 id="start-giotto">2. Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed</span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/checkGiottoEnvironment.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/installGiottoEnvironment.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-giotto-object">3. Create Giotto object<a class="anchor" aria-label="anchor" href="#create-giotto-object"></a>
</h2>
<p>Read expression matrix 22846 genes, 22 proteins, 901 cells</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_dir</span> <span class="op">=</span> <span class="st">"/path/to/your/data/"</span></span>
<span></span>
<span><span class="co">## RNA</span></span>
<span><span class="va">rna_expression</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>,<span class="st">"GSE137986_RAW/GSM4189613_0702cL.tsv.gz"</span><span class="op">)</span>, </span>
<span>                            sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>                            header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rna_expression</span><span class="op">)</span> <span class="op">=</span> <span class="va">rna_expression</span><span class="op">$</span><span class="va">X</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Protein</span></span>
<span><span class="va">protein_expression</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>,<span class="st">"GSE137986_RAW/GSM4202307_0702aL.tsv.gz"</span><span class="op">)</span>, </span>
<span>                                sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>                                header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">protein_expression</span><span class="op">)</span> <span class="op">=</span> <span class="va">protein_expression</span><span class="op">$</span><span class="va">X</span></span></code></pre></div>
<p>Transpose matrix</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rna_expression</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">rna_expression</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">protein_expression</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">protein_expression</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>Get the spatial coordinates</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spatial_coords</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cell_ID <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rna_expression</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">spatial_coords</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">spatial_coords</span>, </span>
<span>                       <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">spatial_coords</span>, <span class="va">cell_ID</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x"</span>,<span class="st">"y"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"x"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spatial_coords</span><span class="op">$</span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">spatial_coords</span><span class="op">$</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="va">spatial_coords</span><span class="op">$</span><span class="va">y</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">spatial_coords</span><span class="op">$</span><span class="va">y</span><span class="op">)</span><span class="op">*</span><span class="op">-</span><span class="fl">1</span></span></code></pre></div>
<p>Create the Giotto object</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">save_dir</span> <span class="op">=</span> <span class="st">'results'</span></span>
<span><span class="va">instrs</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">save_dir</span>,</span>
<span>                                  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  show_plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>raw <span class="op">=</span> <span class="va">rna_expression</span>,</span>
<span>                                                     raw <span class="op">=</span> <span class="va">protein_expression</span><span class="op">)</span>,</span>
<span>                                   expression_feat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'rna'</span>, <span class="st">'protein'</span><span class="op">)</span>,</span>
<span>                                   spatial_locs <span class="op">=</span> <span class="va">spatial_coords</span>,</span>
<span>                                   instructions <span class="op">=</span> <span class="va">instrs</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="processing">4. Processing<a class="anchor" aria-label="anchor" href="#processing"></a>
</h2>
<div class="section level3">
<h3 id="filtering">4.1 Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## RNA</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>                             expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                             feat_det_in_min_cells <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                             min_det_feats_per_cell <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                             expression_values <span class="op">=</span> <span class="st">'raw'</span>,</span>
<span>                             verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Protein</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>                             spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                             feat_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>                             expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                             feat_det_in_min_cells <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                             min_det_feats_per_cell <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                             expression_values <span class="op">=</span> <span class="st">'raw'</span>,</span>
<span>                             verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="normalization">4.2 Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## RNA</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                scalefactor <span class="op">=</span> <span class="fl">6000</span>, </span>
<span>                                verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Protein</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                                feat_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>                                scalefactor <span class="op">=</span> <span class="fl">6000</span>, </span>
<span>                                verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="statistics">4.3 Statistics<a class="anchor" aria-label="anchor" href="#statistics"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## RNA</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">giottoObject</span>,</span>
<span>           spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>           feat_type <span class="op">=</span> <span class="st">'rna'</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"nr_feats"</span>,</span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">giottoObject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"total_expr"</span>,</span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/0-spatPlot2D.png"><img src="images/mouse_embryo_dbitseq/1-spatPlot2D.png"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Protein</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>                              spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                              feat_type <span class="op">=</span> <span class="st">'protein'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">giottoObject</span>,</span>
<span>           spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>           feat_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"total_expr"</span>,</span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/2-spatPlot2D.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="dimension-reduction">5. Dimension Reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<div class="section level3">
<h3 id="identify-highly-variable-features-hvf">5.1 Identify highly variable features (HVF)<a class="anchor" aria-label="anchor" href="#identify-highly-variable-features-hvf"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">giottoObject</span> <span class="op">=</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/3-HVFplot.png"></p>
</div>
<div class="section level3">
<h3 id="pca">5.2 PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">giottoObject</span>, </span>
<span>          ncp <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/4-screePlot.png"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/5-PCA.png"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Protein</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>                       spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                       feat_type <span class="op">=</span> <span class="st">'protein'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">giottoObject</span>, </span>
<span>          spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>          feat_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>          ncp <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/6-screePlot.png"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>        spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>        feat_type <span class="op">=</span> <span class="st">'protein'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/7-PCA.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="clustering">6. Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<div class="section level3">
<h3 id="umap">6.1 UMAP<a class="anchor" aria-label="anchor" href="#umap"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">giottoObject</span>, </span>
<span>                        dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/8-UMAP.png"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Protein</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">giottoObject</span>, </span>
<span>                        spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>                        feat_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>                        dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>         spat_unit <span class="op">=</span> <span class="st">'cell'</span>,</span>
<span>         feat_type <span class="op">=</span> <span class="st">'protein'</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/9-UMAP.png"></p>
</div>
<div class="section level3">
<h3 id="create-shared-nearest-network-snn-and-perform-leiden-clustering">6.2 Create shared nearest network (sNN) and perform leiden
clustering<a class="anchor" aria-label="anchor" href="#create-shared-nearest-network-snn-and-perform-leiden-clustering"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                     dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>                                     k <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                resolution <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Protein</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                     spat_unit <span class="op">=</span> <span class="st">'cell'</span>, </span>
<span>                                     feat_type <span class="op">=</span> <span class="st">'protein'</span>, </span>
<span>                                     dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>                                     k <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                spat_unit <span class="op">=</span> <span class="st">'cell'</span>, </span>
<span>                                feat_type <span class="op">=</span> <span class="st">'protein'</span>, </span>
<span>                                resolution <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-umap-cluster-results">6.3 Visualize UMAP cluster results<a class="anchor" aria-label="anchor" href="#visualize-umap-cluster-results"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, </span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         title <span class="op">=</span> <span class="st">""</span>,</span>
<span>         axis_text <span class="op">=</span> <span class="fl">14</span>,</span>
<span>         axis_title <span class="op">=</span> <span class="fl">18</span>,</span>
<span>         legend_text <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/10-UMAP.png"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Protein</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>         spat_unit <span class="op">=</span> <span class="st">'cell'</span>, </span>
<span>         feat_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, </span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         title <span class="op">=</span> <span class="st">""</span>,</span>
<span>         axis_text <span class="op">=</span> <span class="fl">14</span>,</span>
<span>         axis_title <span class="op">=</span> <span class="fl">18</span>,</span>
<span>         legend_text <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/11-UMAP.png"></p>
</div>
<div class="section level3">
<h3 id="visualize-spatial-results">6.4 Visualize spatial results<a class="anchor" aria-label="anchor" href="#visualize-spatial-results"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>           show_image <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>           title <span class="op">=</span> <span class="st">""</span>,</span>
<span>           axis_text <span class="op">=</span> <span class="fl">14</span>,</span>
<span>           axis_title <span class="op">=</span> <span class="fl">18</span>,</span>
<span>           legend_text <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/12-spatPlot2D.png"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Protein</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>           spat_unit <span class="op">=</span> <span class="st">'cell'</span>, </span>
<span>           feat_type <span class="op">=</span> <span class="st">'protein'</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>           title <span class="op">=</span> <span class="st">""</span>,</span>
<span>           axis_text <span class="op">=</span> <span class="fl">14</span>,</span>
<span>           axis_title <span class="op">=</span> <span class="fl">18</span>,</span>
<span>           legend_text <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/13-spatPlot2D.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="multi-omics-integration">7. Multi-omics integration<a class="anchor" aria-label="anchor" href="#multi-omics-integration"></a>
</h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                     type <span class="op">=</span> <span class="st">'kNN'</span>,</span>
<span>                                     dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>                                     k <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Protein</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>, </span>
<span>                                     spat_unit <span class="op">=</span> <span class="st">'cell'</span>, </span>
<span>                                     feat_type <span class="op">=</span> <span class="st">'protein'</span>, </span>
<span>                                     type <span class="op">=</span> <span class="st">'kNN'</span>,</span>
<span>                                     dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>                                     k <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runWNN.html">runWNN</a></span><span class="op">(</span><span class="va">giottoObject</span>,</span>
<span>                       spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                       modality_1 <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                       modality_2 <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                       pca_name_modality_1 <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>                       pca_name_modality_2 <span class="op">=</span> <span class="st">"protein.pca"</span>,</span>
<span>                       k <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                       verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runIntegratedUMAP.html">runIntegratedUMAP</a></span><span class="op">(</span><span class="va">giottoObject</span>,</span>
<span>                                  modality1 <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                                  modality2 <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                                  spread <span class="op">=</span> <span class="fl">7</span>,</span>
<span>                                  min_dist <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                  force <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>                                spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                                feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                                nn_network_to_use <span class="op">=</span> <span class="st">"kNN"</span>,</span>
<span>                                network_name <span class="op">=</span> <span class="st">"integrated_kNN"</span>,</span>
<span>                                name <span class="op">=</span> <span class="st">"integrated_leiden_clus"</span>,</span>
<span>                                resolution <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>         spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>         feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">'integrated_leiden_clus'</span>,</span>
<span>         dim_reduction_name <span class="op">=</span> <span class="st">"integrated.umap"</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         title <span class="op">=</span> <span class="st">""</span>,</span>
<span>         axis_text <span class="op">=</span> <span class="fl">14</span>,</span>
<span>         axis_title <span class="op">=</span> <span class="fl">18</span>,</span>
<span>         legend_text <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/14-UMAP.png"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">giottoObject</span>,</span>
<span>           spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>           feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"integrated_leiden_clus"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           title <span class="op">=</span> <span class="st">""</span>,</span>
<span>           axis_text <span class="op">=</span> <span class="fl">14</span>,</span>
<span>           axis_title <span class="op">=</span> <span class="fl">18</span>,</span>
<span>           legend_text <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/15-spatPlot2D.png"></p>
</div>
<div class="section level2">
<h2 id="deconvolution">8. Deconvolution<a class="anchor" aria-label="anchor" href="#deconvolution"></a>
</h2>
<p>We used the scRNAseq from <a href="https://www.nature.com/articles/s41586-019-0969-x" class="external-link">Cao et al.,
2019</a> as reference. J. Cao, M. Spielmann, X. Qiu, X. Huang, D.M.
Ibrahim, A.J. Hill, F. Zhang, S. Mundlos, L. Christiansen, F.J.
Steemers, et al. The single-cell transcriptional landscape of mammalian
organogenesis. You can download the gene_count_cleaned_sampled_100k.RDS
file from <a href="https://oncoscape.v3.sttrcancer.org/atlas.gs.washington.edu.mouse.rna/downloads" class="external-link">here.</a></p>
<div class="section level3">
<h3 id="preparation-of-the-scrnaseq">8.1 Preparation of the scRNAseq<a class="anchor" aria-label="anchor" href="#preparation-of-the-scrnaseq"></a>
</h3>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read single cell procesed object</span></span>
<span><span class="va">gene_count_cleaned_sampled_100k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>,<span class="st">"gene_count_cleaned_sampled_100k.RDS"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># download cell annotations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://shendure-web.gs.washington.edu/content/members/cao1025/public/mouse_embryo_atlas/cell_annotate.csv"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>,<span class="st">"cell_annotate.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># or run in your terminal: wget https://shendure-web.gs.washington.edu/content/members/cao1025/public/mouse_embryo_atlas/cell_annotate.csv</span></span>
<span></span>
<span><span class="va">cell_annotation</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>,<span class="st">"cell_annotate.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cell_annotation</span> <span class="op">=</span> <span class="va">cell_annotation</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"Total_mRNAs"</span>, <span class="st">"num_genes_expressed"</span>, <span class="st">"Main_cell_type"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cell_annotation</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">=</span> <span class="st">"cell_ID"</span></span>
<span></span>
<span><span class="va">cell_annotation</span> <span class="op">=</span> <span class="va">cell_annotation</span><span class="op">[</span><span class="va">cell_annotation</span><span class="op">$</span><span class="va">cell_ID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">gene_count_cleaned_sampled_100k</span><span class="op">)</span>,<span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc_giotto</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">gene_count_cleaned_sampled_100k</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sc_giotto</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html" class="external-link">subsetGiotto</a></span><span class="op">(</span><span class="va">sc_giotto</span>,</span>
<span>                          cell_ids <span class="op">=</span> <span class="va">cell_annotation</span><span class="op">$</span><span class="va">cell_ID</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sc_giotto</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span><span class="va">sc_giotto</span>,</span>
<span>                             new_metadata <span class="op">=</span> <span class="va">cell_annotation</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc_giotto</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span><span class="va">sc_giotto</span>, </span>
<span>                             log_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                             scale_feats <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                             scale_cells <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Find markergenes</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers_scran</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">sc_giotto</span>,</span>
<span>                                        method <span class="op">=</span> <span class="st">"scran"</span>,</span>
<span>                                        expression_values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                                        cluster_column <span class="op">=</span> <span class="st">'Main_cell_type'</span>,</span>
<span>                                        min_feats <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">markergenes_scran</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers_scran</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">30</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span><span class="op">[[</span><span class="st">"feats"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>Create DWLS matrix</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DWLS_matrix_direct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignMatrixDWLSfromMatrix.html">makeSignMatrixDWLSfromMatrix</a></span><span class="op">(</span></span>
<span>  matrix <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/getExpression.html" class="external-link">getExpression</a></span><span class="op">(</span><span class="va">sc_giotto</span>,</span>
<span>                         values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                         output <span class="op">=</span> <span class="st">"matrix"</span><span class="op">)</span>,</span>
<span>  cell_type <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">sc_giotto</span><span class="op">)</span><span class="op">$</span><span class="va">Main_cell_type</span>,</span>
<span>  sign_gene <span class="op">=</span> <span class="va">markergenes_scran</span><span class="op">)</span></span></code></pre></div>
<p>Fix gene names</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download gene annotations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://shendure-web.gs.washington.edu/content/members/cao1025/public/mouse_embryo_atlas/gene_annotate.csv"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>,<span class="st">"gene_annotate.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sc_gene_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>,<span class="st">"gene_annotate.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ENSMUS_names</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">DWLS_matrix_direct</span><span class="op">)</span></span>
<span><span class="va">sc_gene_names</span> <span class="op">=</span> <span class="va">sc_gene_names</span><span class="op">[</span><span class="va">sc_gene_names</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">ENSMUS_names</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">DWLS_matrix_direct</span><span class="op">)</span> <span class="op">=</span> <span class="va">sc_gene_names</span><span class="op">$</span><span class="va">gene_short_name</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-deconvolution">8.2 Run deconvolution<a class="anchor" aria-label="anchor" href="#run-deconvolution"></a>
</h3>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run DWLS using integrated leiden clusters</span></span>
<span><span class="va">giottoObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runDWLSDeconv.html">runDWLSDeconv</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giottoObject</span>,</span>
<span>                              sign_matrix <span class="op">=</span> <span class="va">DWLS_matrix_direct</span>,</span>
<span>                              cluster_column <span class="op">=</span> <span class="st">"integrated_leiden_clus"</span><span class="op">)</span></span></code></pre></div>
<p>Plot DWLS deconvolution result</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot DWLS deconvolution result with Pie plots dataset 1</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDeconvPlot.html" class="external-link">spatDeconvPlot</a></span><span class="op">(</span><span class="va">giottoObject</span>,</span>
<span>               show_image <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>               radius <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>               return_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"integrated_deconvolution"</span><span class="op">)</span>,</span>
<span>               title <span class="op">=</span> <span class="st">""</span>,</span>
<span>               axis_text <span class="op">=</span> <span class="fl">14</span>,</span>
<span>               axis_title <span class="op">=</span> <span class="fl">18</span>,</span>
<span>               legend_text <span class="op">=</span> <span class="fl">0</span>,</span>
<span>               background_color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/mouse_embryo_dbitseq/integrated_deconvolution.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">9. Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">3.2</span> (<span class="dv">2023-10-31</span>)</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin20</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="dv">14</span>.<span class="fl">3.1</span></span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.3</span><span class="sc">-</span>x86_64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb42-8"><a href="#cb42-8" tabindex="-1"></a></span>
<span id="cb42-9"><a href="#cb42-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb42-10"><a href="#cb42-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb42-11"><a href="#cb42-11" tabindex="-1"></a></span>
<span id="cb42-12"><a href="#cb42-12" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb42-13"><a href="#cb42-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb42-14"><a href="#cb42-14" tabindex="-1"></a></span>
<span id="cb42-15"><a href="#cb42-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb42-16"><a href="#cb42-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb42-17"><a href="#cb42-17" tabindex="-1"></a></span>
<span id="cb42-18"><a href="#cb42-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb42-19"><a href="#cb42-19" tabindex="-1"></a>[<span class="dv">1</span>] Giotto_4.<span class="fl">0.2</span>      GiottoClass_0.<span class="fl">1.3</span></span>
<span id="cb42-20"><a href="#cb42-20" tabindex="-1"></a></span>
<span id="cb42-21"><a href="#cb42-21" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb42-22"><a href="#cb42-22" tabindex="-1"></a>  [<span class="dv">1</span>] IRanges_2.<span class="fl">36.0</span>                SpatialExperiment_1.<span class="fl">12.0</span>      urlchecker_1.<span class="fl">0.1</span>             </span>
<span id="cb42-23"><a href="#cb42-23" tabindex="-1"></a>  [<span class="dv">4</span>] DT_0<span class="fl">.31</span>                       Biostrings_2.<span class="fl">70.2</span>             vctrs_0.<span class="fl">6.5</span>                  </span>
<span id="cb42-24"><a href="#cb42-24" tabindex="-1"></a>  [<span class="dv">7</span>] digest_0.<span class="fl">6.34</span>                 png_0<span class="fl">.1</span><span class="dv">-8</span>                     shape_1.<span class="fl">4.6</span>                  </span>
<span id="cb42-25"><a href="#cb42-25" tabindex="-1"></a> [<span class="dv">10</span>] ggrepel_0.<span class="fl">9.5</span>                 parallelly_1.<span class="fl">36.0</span>             magick_2.<span class="fl">8.2</span>                 </span>
<span id="cb42-26"><a href="#cb42-26" tabindex="-1"></a> [<span class="dv">13</span>] MASS_7<span class="fl">.3</span><span class="dv">-60</span>.<span class="fl">0.1</span>               pkgdown_2.<span class="fl">0.7</span>                 tictoc_1<span class="fl">.2</span>                   </span>
<span id="cb42-27"><a href="#cb42-27" tabindex="-1"></a> [<span class="dv">16</span>] reshape2_1.<span class="fl">4.4</span>                MAST_1.<span class="fl">28.0</span>                   foreach_1.<span class="fl">5.2</span>                </span>
<span id="cb42-28"><a href="#cb42-28" tabindex="-1"></a> [<span class="dv">19</span>] httpuv_1.<span class="fl">6.14</span>                 BiocGenerics_0.<span class="fl">48.1</span>           withr_3.<span class="fl">0.0</span>                  </span>
<span id="cb42-29"><a href="#cb42-29" tabindex="-1"></a> [<span class="dv">22</span>] ggfun_0.<span class="fl">1.4</span>                   xfun_0<span class="fl">.42</span>                     ellipsis_0.<span class="fl">3.2</span>               </span>
<span id="cb42-30"><a href="#cb42-30" tabindex="-1"></a> [<span class="dv">25</span>] memoise_2.<span class="fl">0.1</span>                 ggbeeswarm_0.<span class="fl">7.2</span>              clustree_0.<span class="fl">5.1</span>               </span>
<span id="cb42-31"><a href="#cb42-31" tabindex="-1"></a> [<span class="dv">28</span>] emmeans_1.<span class="fl">10.0</span>                profvis_0.<span class="fl">3.8</span>                 gmp_0<span class="fl">.7</span><span class="dv">-4</span>                    </span>
<span id="cb42-32"><a href="#cb42-32" tabindex="-1"></a> [<span class="dv">31</span>] systemfonts_1.<span class="fl">0.5</span>             ragg_1.<span class="fl">2.7</span>                    GlobalOptions_0.<span class="fl">1.2</span>          </span>
<span id="cb42-33"><a href="#cb42-33" tabindex="-1"></a> [<span class="dv">34</span>] gtools_3.<span class="fl">9.5</span>                  sys_3.<span class="fl">4.2</span>                     KEGGREST_1.<span class="fl">42.0</span>              </span>
<span id="cb42-34"><a href="#cb42-34" tabindex="-1"></a> [<span class="dv">37</span>] promises_1.<span class="fl">2.1</span>                scatterplot3d_0<span class="fl">.3</span><span class="dv">-44</span>          httr_1.<span class="fl">4.7</span>                   </span>
<span id="cb42-35"><a href="#cb42-35" tabindex="-1"></a> [<span class="dv">40</span>] restfulr_0.<span class="fl">0.15</span>               globals_0.<span class="fl">16.2</span>                pak_0.<span class="fl">7.1</span>                    </span>
<span id="cb42-36"><a href="#cb42-36" tabindex="-1"></a> [<span class="dv">43</span>] rhdf5filters_1.<span class="fl">14.1</span>           ps_1.<span class="fl">7.6</span>                      rhdf5_2.<span class="fl">46.1</span>                 </span>
<span id="cb42-37"><a href="#cb42-37" tabindex="-1"></a> [<span class="dv">46</span>] rstudioapi_0.<span class="fl">15.0</span>             miniUI_0.<span class="dv">1</span>.<span class="fl">1.1</span>                generics_0.<span class="fl">1.3</span>               </span>
<span id="cb42-38"><a href="#cb42-38" tabindex="-1"></a> [<span class="dv">49</span>] processx_3.<span class="fl">8.3</span>                curl_5.<span class="fl">2.0</span>                    S4Vectors_0.<span class="fl">40.2</span>             </span>
<span id="cb42-39"><a href="#cb42-39" tabindex="-1"></a> [<span class="dv">52</span>] zlibbioc_1.<span class="fl">48.0</span>               ScaledMatrix_1.<span class="fl">10.0</span>           ggraph_2.<span class="fl">1.0</span>                 </span>
<span id="cb42-40"><a href="#cb42-40" tabindex="-1"></a> [<span class="dv">55</span>] polyclip_1<span class="fl">.10</span><span class="dv">-6</span>               RcppZiggurat_0.<span class="fl">1.6</span>            quadprog_1<span class="fl">.5</span><span class="dv">-8</span>               </span>
<span id="cb42-41"><a href="#cb42-41" tabindex="-1"></a> [<span class="dv">58</span>] GenomeInfoDbData_1.<span class="fl">2.11</span>       ExperimentHub_2.<span class="fl">10.0</span>          SparseArray_1.<span class="fl">2.4</span>            </span>
<span id="cb42-42"><a href="#cb42-42" tabindex="-1"></a> [<span class="dv">61</span>] FactoMineR_2<span class="fl">.9</span>                interactiveDisplayBase_1.<span class="fl">40.0</span> xtable_1<span class="fl">.8</span><span class="dv">-4</span>                 </span>
<span id="cb42-43"><a href="#cb42-43" tabindex="-1"></a> [<span class="dv">64</span>] stringr_1.<span class="fl">5.1</span>                 desc_1.<span class="fl">4.3</span>                    trendsceek_1.<span class="fl">0.0</span>             </span>
<span id="cb42-44"><a href="#cb42-44" tabindex="-1"></a> [<span class="dv">67</span>] pracma_2.<span class="fl">4.4</span>                  doParallel_1.<span class="fl">0.17</span>             evaluate_0<span class="fl">.23</span>                </span>
<span id="cb42-45"><a href="#cb42-45" tabindex="-1"></a> [<span class="dv">70</span>] S4Arrays_1.<span class="fl">2.0</span>                Rfast_2.<span class="fl">1.0</span>                   gitcreds_0.<span class="fl">1.2</span>               </span>
<span id="cb42-46"><a href="#cb42-46" tabindex="-1"></a> [<span class="dv">73</span>] BiocFileCache_2.<span class="fl">10.1</span>          GenomicRanges_1.<span class="fl">54.1</span>          irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>                </span>
<span id="cb42-47"><a href="#cb42-47" tabindex="-1"></a> [<span class="dv">76</span>] colorspace_2<span class="fl">.1</span><span class="dv">-0</span>              filelock_1.<span class="fl">0.3</span>                harmony_1.<span class="fl">2.0</span>                </span>
<span id="cb42-48"><a href="#cb42-48" tabindex="-1"></a> [<span class="dv">79</span>] reticulate_1.<span class="fl">35.0</span>             magrittr_2.<span class="fl">0.3</span>                later_1.<span class="fl">3.2</span>                  </span>
<span id="cb42-49"><a href="#cb42-49" tabindex="-1"></a> [<span class="dv">82</span>] viridis_0.<span class="fl">6.5</span>                 lattice_0<span class="fl">.22</span><span class="dv">-5</span>                future.apply_1.<span class="fl">11.1</span>          </span>
<span id="cb42-50"><a href="#cb42-50" tabindex="-1"></a> [<span class="dv">85</span>] GiottoUtils_0.<span class="fl">1.5</span>             XML_3<span class="fl">.99</span><span class="dv">-0</span>.<span class="fl">16.1</span>               scuttle_1.<span class="fl">12.0</span>               </span>
<span id="cb42-51"><a href="#cb42-51" tabindex="-1"></a> [<span class="dv">88</span>] cowplot_1.<span class="fl">1.3</span>                 matrixStats_1.<span class="fl">2.0</span>             pillar_1.<span class="fl">9.0</span>                 </span>
<span id="cb42-52"><a href="#cb42-52" tabindex="-1"></a> [<span class="dv">91</span>] iterators_1.<span class="fl">0.14</span>              STexampleData_1.<span class="fl">10.0</span>          compiler_4.<span class="fl">3.2</span>               </span>
<span id="cb42-53"><a href="#cb42-53" tabindex="-1"></a> [<span class="dv">94</span>] beachmat_2.<span class="fl">18.0</span>               stringi_1.<span class="fl">8.3</span>                 SummarizedExperiment_1.<span class="fl">32.0</span>  </span>
<span id="cb42-54"><a href="#cb42-54" tabindex="-1"></a> [<span class="dv">97</span>] devtools_2.<span class="fl">4.5</span>                GenomicAlignments_1.<span class="fl">38.2</span>      jackstraw_1.<span class="fl">3.9</span>              </span>
<span id="cb42-55"><a href="#cb42-55" tabindex="-1"></a>[<span class="dv">100</span>] plyr_1.<span class="fl">8.9</span>                    BiocIO_1.<span class="fl">12.0</span>                 crayon_1.<span class="fl">5.2</span>                 </span>
<span id="cb42-56"><a href="#cb42-56" tabindex="-1"></a>[<span class="dv">103</span>] abind_1<span class="fl">.4</span><span class="dv">-5</span>                   scater_1.<span class="fl">30.1</span>                 ggdendro_0.<span class="fl">1.23</span>              </span>
<span id="cb42-57"><a href="#cb42-57" tabindex="-1"></a>[<span class="dv">106</span>] locfit_1<span class="fl">.5-9.8</span>                graphlayouts_1.<span class="fl">1.0</span>            bit_4.<span class="fl">0.5</span>                    </span>
<span id="cb42-58"><a href="#cb42-58" tabindex="-1"></a>[<span class="dv">109</span>] terra_1<span class="fl">.7</span><span class="dv">-71</span>                  dplyr_1.<span class="fl">1.4</span>                   whisker_0.<span class="fl">4.1</span>                </span>
<span id="cb42-59"><a href="#cb42-59" tabindex="-1"></a>[<span class="dv">112</span>] codetools_0<span class="fl">.2</span><span class="dv">-19</span>              textshaping_0.<span class="fl">3.7</span>             GiottoVisuals_0.<span class="fl">1.4</span>          </span>
<span id="cb42-60"><a href="#cb42-60" tabindex="-1"></a>[<span class="dv">115</span>] BiocSingular_1.<span class="fl">18.0</span>           openssl_2.<span class="fl">1.1</span>                 bslib_0.<span class="fl">6.1</span>                  </span>
<span id="cb42-61"><a href="#cb42-61" tabindex="-1"></a>[<span class="dv">118</span>] GetoptLong_1.<span class="fl">0.5</span>              mime_0<span class="fl">.12</span>                     multinet_4.<span class="fl">1.2</span>               </span>
<span id="cb42-62"><a href="#cb42-62" tabindex="-1"></a>[<span class="dv">121</span>] circlize_0.<span class="fl">4.15</span>               Rcpp_1.<span class="fl">0.12</span>                   dbplyr_2.<span class="fl">4.0</span>                 </span>
<span id="cb42-63"><a href="#cb42-63" tabindex="-1"></a>[<span class="dv">124</span>] sparseMatrixStats_1.<span class="fl">14.0</span>      leaps_3<span class="fl">.1</span>                     knitr_1<span class="fl">.45</span>                   </span>
<span id="cb42-64"><a href="#cb42-64" tabindex="-1"></a>[<span class="dv">127</span>] blob_1.<span class="fl">2.4</span>                    utf8_1.<span class="fl">2.4</span>                    clue_0<span class="fl">.3</span><span class="dv">-65</span>                  </span>
<span id="cb42-65"><a href="#cb42-65" tabindex="-1"></a>[<span class="dv">130</span>] BiocVersion_3.<span class="fl">18.1</span>            fs_1.<span class="fl">6.3</span>                      listenv_0.<span class="fl">9.1</span>                </span>
<span id="cb42-66"><a href="#cb42-66" tabindex="-1"></a>[<span class="dv">133</span>] checkmate_2.<span class="fl">3.1</span>               DelayedMatrixStats_1.<span class="fl">24.0</span>     Rdpack_2<span class="fl">.6</span>                   </span>
<span id="cb42-67"><a href="#cb42-67" tabindex="-1"></a>[<span class="dv">136</span>] pkgbuild_1.<span class="fl">4.3</span>                gh_1.<span class="fl">4.0</span>                      estimability_1.<span class="fl">4.1</span>           </span>
<span id="cb42-68"><a href="#cb42-68" tabindex="-1"></a>[<span class="dv">139</span>] tibble_3.<span class="fl">2.1</span>                  Matrix_1<span class="fl">.6</span><span class="dv">-5</span>                  callr_3.<span class="fl">7.3</span>                  </span>
<span id="cb42-69"><a href="#cb42-69" tabindex="-1"></a>[<span class="dv">142</span>] statmod_1.<span class="fl">5.0</span>                 tweenr_2.<span class="fl">0.2</span>                  pkgconfig_2.<span class="fl">0.3</span>              </span>
<span id="cb42-70"><a href="#cb42-70" tabindex="-1"></a>[<span class="dv">145</span>] pheatmap_1.<span class="fl">0.12</span>               tools_4.<span class="fl">3.2</span>                   cachem_1.<span class="fl">0.8</span>                 </span>
<span id="cb42-71"><a href="#cb42-71" tabindex="-1"></a>[<span class="dv">148</span>] rbibutils_2.<span class="fl">2.16</span>              RSQLite_2.<span class="fl">3.5</span>                 viridisLite_0.<span class="fl">4.2</span>            </span>
<span id="cb42-72"><a href="#cb42-72" tabindex="-1"></a>[<span class="dv">151</span>] DBI_1.<span class="fl">2.1</span>                     fastmap_1.<span class="fl">1.1</span>                 rmarkdown_2<span class="fl">.25</span>               </span>
<span id="cb42-73"><a href="#cb42-73" tabindex="-1"></a>[<span class="dv">154</span>] scales_1.<span class="fl">3.0</span>                  grid_4.<span class="fl">3.2</span>                    credentials_2.<span class="fl">0.1</span>            </span>
<span id="cb42-74"><a href="#cb42-74" tabindex="-1"></a>[<span class="dv">157</span>] ggspavis_1.<span class="fl">8.0</span>                usethis_2.<span class="fl">2.2</span>                 Rsamtools_2.<span class="fl">18.0</span>             </span>
<span id="cb42-75"><a href="#cb42-75" tabindex="-1"></a>[<span class="dv">160</span>] AnnotationHub_3.<span class="fl">10.0</span>          sass_0.<span class="fl">4.8</span>                    FNN_1.<span class="fl">1.4</span>                    </span>
<span id="cb42-76"><a href="#cb42-76" tabindex="-1"></a>[<span class="dv">163</span>] BiocManager_1.<span class="fl">30.22</span>           farver_2.<span class="fl">1.1</span>                  scatterpie_0.<span class="fl">2.1</span>             </span>
<span id="cb42-77"><a href="#cb42-77" tabindex="-1"></a>[<span class="dv">166</span>] tidygraph_1.<span class="fl">3.1</span>               yaml_2.<span class="fl">3.8</span>                    MatrixGenerics_1.<span class="fl">14.0</span>        </span>
<span id="cb42-78"><a href="#cb42-78" tabindex="-1"></a>[<span class="dv">169</span>] rtracklayer_1.<span class="fl">62.0</span>            cli_3.<span class="fl">6.2</span>                     purrr_1.<span class="fl">0.2</span>                  </span>
<span id="cb42-79"><a href="#cb42-79" tabindex="-1"></a>[<span class="dv">172</span>] stats4_4.<span class="fl">3.2</span>                  lifecycle_1.<span class="fl">0.4</span>               dbscan_1<span class="fl">.1</span><span class="dv">-12</span>                </span>
<span id="cb42-80"><a href="#cb42-80" tabindex="-1"></a>[<span class="dv">175</span>] askpass_1.<span class="fl">2.0</span>                 uwot_0.<span class="fl">1.16</span>                   Biobase_2.<span class="fl">62.0</span>               </span>
<span id="cb42-81"><a href="#cb42-81" tabindex="-1"></a>[<span class="dv">178</span>] mvtnorm_1<span class="fl">.2</span><span class="dv">-4</span>                 bluster_1.<span class="fl">12.0</span>                sessioninfo_1.<span class="fl">2.2</span>            </span>
<span id="cb42-82"><a href="#cb42-82" tabindex="-1"></a>[<span class="dv">181</span>] backports_1.<span class="fl">4.1</span>               BiocParallel_1.<span class="fl">36.0</span>           gtable_0.<span class="fl">3.4</span>                 </span>
<span id="cb42-83"><a href="#cb42-83" tabindex="-1"></a>[<span class="dv">184</span>] rjson_0.<span class="fl">2.21</span>                  progressr_0.<span class="fl">14.0</span>              colorRamp2_0.<span class="fl">1.0</span>             </span>
<span id="cb42-84"><a href="#cb42-84" tabindex="-1"></a>[<span class="dv">187</span>] parallel_4.<span class="fl">3.2</span>                limma_3.<span class="fl">58.1</span>                  jsonlite_1.<span class="fl">8.8</span>               </span>
<span id="cb42-85"><a href="#cb42-85" tabindex="-1"></a>[<span class="dv">190</span>] edgeR_4.<span class="fl">0.15</span>                  bitops_1<span class="fl">.0</span><span class="dv">-7</span>                  ggplot2_3.<span class="fl">4.4</span>                </span>
<span id="cb42-86"><a href="#cb42-86" tabindex="-1"></a>[<span class="dv">193</span>] multcompView_0<span class="fl">.1</span><span class="dv">-9</span>            bit64_4.<span class="fl">0.5</span>                   Rtsne_0<span class="fl">.17</span>                   </span>
<span id="cb42-87"><a href="#cb42-87" tabindex="-1"></a>[<span class="dv">196</span>] BiocNeighbors_1.<span class="fl">20.2</span>          RcppParallel_5.<span class="fl">1.7</span>            ggside_0.<span class="fl">2.3</span>                 </span>
<span id="cb42-88"><a href="#cb42-88" tabindex="-1"></a>[<span class="dv">199</span>] jquerylib_0.<span class="fl">1.4</span>               metapod_1.<span class="fl">10.1</span>                ClusterR_1.<span class="fl">3.2</span>               </span>
<span id="cb42-89"><a href="#cb42-89" tabindex="-1"></a>[<span class="dv">202</span>] dqrng_0.<span class="fl">3.2</span>                   downlit_0.<span class="fl">4.3</span>                 shiny_1.<span class="fl">8.0</span>                  </span>
<span id="cb42-90"><a href="#cb42-90" tabindex="-1"></a>[<span class="dv">205</span>] htmltools_0.<span class="fl">5.7</span>               rappdirs_0.<span class="fl">3.3</span>                glue_1.<span class="fl">7.0</span>                   </span>
<span id="cb42-91"><a href="#cb42-91" tabindex="-1"></a>[<span class="dv">208</span>] httr2_1.<span class="fl">0.0</span>                   GiottoData_0.<span class="dv">2</span>.<span class="fl">7.0</span>            XVector_0.<span class="fl">42.0</span>               </span>
<span id="cb42-92"><a href="#cb42-92" tabindex="-1"></a>[<span class="dv">211</span>] RCurl_1<span class="fl">.98-1.14</span>               rprojroot_2.<span class="fl">0.4</span>               scran_1.<span class="fl">30.2</span>                 </span>
<span id="cb42-93"><a href="#cb42-93" tabindex="-1"></a>[<span class="dv">214</span>] gridExtra_2<span class="fl">.3</span>                 flashClust_1<span class="fl">.01</span><span class="dv">-2</span>             igraph_2.<span class="dv">0</span>.<span class="fl">1.1</span>               </span>
<span id="cb42-94"><a href="#cb42-94" tabindex="-1"></a>[<span class="dv">217</span>] R6_2.<span class="fl">5.1</span>                      smfishHmrf_0<span class="fl">.1</span>                tidyr_1.<span class="fl">3.1</span>                  </span>
<span id="cb42-95"><a href="#cb42-95" tabindex="-1"></a>[<span class="dv">220</span>] SingleCellExperiment_1.<span class="fl">24.0</span>   labeling_0.<span class="fl">4.3</span>                cluster_2.<span class="fl">1.6</span>                </span>
<span id="cb42-96"><a href="#cb42-96" tabindex="-1"></a>[<span class="dv">223</span>] pkgload_1.<span class="fl">3.4</span>                 Rhdf5lib_1.<span class="fl">24.2</span>               ArchR_1.<span class="fl">0.3</span>                  </span>
<span id="cb42-97"><a href="#cb42-97" tabindex="-1"></a>[<span class="dv">226</span>] GenomeInfoDb_1.<span class="fl">38.6</span>           DelayedArray_0.<span class="fl">28.0</span>           tidyselect_1.<span class="fl">2.0</span>             </span>
<span id="cb42-98"><a href="#cb42-98" tabindex="-1"></a>[<span class="dv">229</span>] vipor_0.<span class="fl">4.7</span>                   ggforce_0.<span class="fl">4.1</span>                 xml2_1.<span class="fl">3.6</span>                   </span>
<span id="cb42-99"><a href="#cb42-99" tabindex="-1"></a>[<span class="dv">232</span>] AnnotationDbi_1.<span class="fl">64.1</span>          future_1.<span class="fl">33.1</span>                 rsvd_1.<span class="fl">0.5</span>                   </span>
<span id="cb42-100"><a href="#cb42-100" tabindex="-1"></a>[<span class="dv">235</span>] munsell_0.<span class="fl">5.0</span>                 data.table_1.<span class="fl">15.0</span>             htmlwidgets_1.<span class="fl">6.4</span>            </span>
<span id="cb42-101"><a href="#cb42-101" tabindex="-1"></a>[<span class="dv">238</span>] ComplexHeatmap_2.<span class="fl">18.0</span>         RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>            rlang_1.<span class="fl">1.3</span>                  </span>
<span id="cb42-102"><a href="#cb42-102" tabindex="-1"></a>[<span class="dv">241</span>] gert_2.<span class="fl">0.1</span>                    remotes_2.<span class="dv">4</span>.<span class="fl">2.1</span>               fansi_1.<span class="fl">0.6</span>                  </span>
<span id="cb42-103"><a href="#cb42-103" tabindex="-1"></a>[<span class="dv">244</span>] beeswarm_0.<span class="fl">4.0</span> </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
