<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Giotto Object Creation • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Giotto Object Creation">
<meta property="og:description" content="Giotto">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/interoperability_seurat.html">Seurat</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Giotto Object Creation</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/vignettes/object_creation.Rmd" class="external-link"><code>vignettes/object_creation.Rmd</code></a></small>
      <div class="hidden name"><code>object_creation.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="how-to-create-a-giotto-object">1. How to create a Giotto Object<a class="anchor" aria-label="anchor" href="#how-to-create-a-giotto-object"></a>
</h2>
<p>In this tutorial, the methodology and syntax to create a
<code>giotto</code> object is shown and osmFISH data is used throughout
the tutorial.</p>
<div class="section level3">
<h3 id="import-giotto-and-download-the-data">1.1 Import Giotto and Download the Data<a class="anchor" aria-label="anchor" href="#import-giotto-and-download-the-data"></a>
</h3>
<p>To download this data, please ensure that <a href="https://www.gnu.org/software/wget/?" class="external-link">wget</a> is installed
locally.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/GiottoData/" class="external-link">GiottoData</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/checkGiottoEnvironment.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/installGiottoEnvironment.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Specify path from which data may be retrieved/stored</span></span>
<span><span class="va">data_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">'/gobject_data/'</span><span class="op">)</span></span>
<span><span class="co"># alternatively, "/path/to/where/the/data/lives/"</span></span>
<span></span>
<span><span class="co"># Specify path to which results may be saved</span></span>
<span><span class="va">results_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">'/gobject_results/'</span><span class="op">)</span> </span>
<span><span class="co"># alternatively, "/path/to/store/the/results/"</span></span>
<span></span>
<span><span class="co"># Download osmFISH dataset to data_directory</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/getSpatialDataset.html" class="external-link">getSpatialDataset</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">'osmfish_SS_cortex'</span>, </span>
<span>                  directory <span class="op">=</span> <span class="va">data_directory</span>, </span>
<span>                  method <span class="op">=</span> <span class="st">'wget'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="minimum-requirements-for-a-giotto-object">1.2 Minimum requirements for a Giotto Object<a class="anchor" aria-label="anchor" href="#minimum-requirements-for-a-giotto-object"></a>
</h3>
<ul>
<li>Expression matrix</li>
<li>Spatial locations (<em>unnecessary for scRNAseq analysis</em>)</li>
</ul>
<p>Here, creating a <code>giotto</code> object with the minimum
requirements is shown in two examples. Data formatting guidelines are
shown below this code block.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example 1.</span></span>
<span><span class="co"># Create a Giotto object using data directly from file paths </span></span>
<span><span class="va">osm_exprs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"osmFISH_prep_expression.txt"</span><span class="op">)</span></span>
<span><span class="va">osm_locs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"osmFISH_prep_cell_coordinates.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">minimum_gobject1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">osm_exprs</span>,</span>
<span>                                       spatial_locs <span class="op">=</span> <span class="va">osm_locs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Example 2.</span></span>
<span><span class="co"># Create a Giotto object using objects already loaded into workspace</span></span>
<span><span class="va">expression_matrix</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/readExprMatrix.html" class="external-link">readExprMatrix</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">osm_exprs</span><span class="op">)</span> <span class="co"># fast method to read expression matrix</span></span>
<span><span class="va">cell_locations</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">osm_locs</span><span class="op">)</span></span>
<span></span>
<span><span class="va">minimum_gobject2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expression_matrix</span>,</span>
<span>                                       spatial_locs <span class="op">=</span> <span class="va">cell_locations</span><span class="op">)</span></span></code></pre></div>
<p><strong>Expression file formatting</strong></p>
<p>Expression count matrices accepted by <em>Giotto</em> should have
columns correlating to individual cells (<strong>cell IDs</strong>) and
rows to individual features (<strong>feature IDs</strong>).</p>
<p><strong>data.table</strong> objects and text-based files ie:
(<strong>.csv, .tsv, .txt</strong>) to be read in should have column 1
be <strong>feature IDs</strong>.</p>
<p><strong>matrix</strong> and <strong>DelayedMatrix</strong> objects
should have <strong>feature IDs</strong> already incorporated as the
rownames.</p>
<p><strong>Locations file formatting</strong></p>
<p>Numerical columns will be interpreted in order of x, y, and
(optionally) z coordinate. The first non-numerical column will be taken
as <strong>cell IDs</strong></p>
<p><strong>*Note:</strong> multiple expression files during can be given
at once during Giotto object creation by using a named list.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Arbitrary modifications</span></span>
<span><span class="va">scaled_matrix</span> <span class="op">=</span> <span class="va">expression_matrix</span> <span class="op">*</span> <span class="fl">1.2</span></span>
<span><span class="va">custom_matrix</span> <span class="op">=</span> <span class="va">expression_matrix</span> <span class="op">*</span> <span class="fl">0.5</span></span>
<span></span>
<span><span class="co"># Provide multiple expression matrices at once to the Giotto Object</span></span>
<span><span class="co"># If these matrices are stored in files rather than in the workspace,</span></span>
<span><span class="co"># file paths may be provided instead of variables</span></span>
<span><span class="va">multi_expr_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>raw <span class="op">=</span> <span class="va">expression_matrix</span>,</span>
<span>                                                           scaled <span class="op">=</span> <span class="va">scaled_matrix</span>,</span>
<span>                                                           custom <span class="op">=</span> <span class="va">custom_matrix</span><span class="op">)</span>,</span>
<span>                                         spatial_locs <span class="op">=</span> <span class="va">cell_locations</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="customizing-the-giotto-object">1.3 Customizing the Giotto Object<a class="anchor" aria-label="anchor" href="#customizing-the-giotto-object"></a>
</h3>
<p>By providing values to other <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject()</a></code>
parameters, it is possible to add:</p>
<ul>
<li>
<strong>Cell</strong> or <strong>feature (gene) metadata</strong>:
see <a href="../reference/addCellMetadata.html">addCellMetadata</a> and
<a href="../reference/addFeatMetadata.html">addFeatMetadata</a>
</li>
<li>
<strong>Spatial networks</strong> or <strong>grids</strong>: see <a href="./Visualizations.html">Visualizations</a>
</li>
<li>
<strong>Dimension reduction</strong>: see <a href="./dimension_reduction.html">Clustering</a>
</li>
<li>
<strong>Images</strong>: see <a href="./getting_started_images.html">Imaging</a>
</li>
<li>
<strong>giottoInstructions</strong>: see <a href="../docs/reference/createGiottoInstructions.html">createGiottoInstructions</a>
and below</li>
</ul>
<p>Providing <code>giottoInstructions</code> allows the specification
of:</p>
<ul>
<li>An alternative python path if using the Giotto Environment (default)
is not desired</li>
<li>A directory to which resulting plots will save</li>
<li>Plot formatting</li>
</ul>
<p><code>createGiottoInstruction()</code> is used to create the
instructions that are provided to <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject()</a></code>. The
<code><a href="https://drieslab.github.io/GiottoClass/reference/instructions-generic.html" class="external-link">instructions()</a></code> function can then be used to view, set, or
modify one or more of these instructions after they have been added to a
<code>giotto</code> object.</p>
<p>Here is an example of a more customized Giotto object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Specify data with file paths </span></span>
<span><span class="va">osm_exprs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"osmFISH_prep_expression.txt"</span><span class="op">)</span></span>
<span><span class="va">osm_locs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"osmFISH_prep_cell_coordinates.txt"</span><span class="op">)</span></span>
<span><span class="va">meta_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"osmFISH_prep_cell_metadata.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create instructions</span></span>
<span></span>
<span><span class="co"># Optional: Specify a path to a Python executable within a conda or miniconda </span></span>
<span><span class="co"># environment. If set to NULL (default), the Python executable within the previously</span></span>
<span><span class="co"># installed Giotto environment will be used.</span></span>
<span><span class="va">my_python_path</span> <span class="op">=</span> <span class="cn">NULL</span> <span class="co"># alternatively, "/local/python/path/python" if desired.</span></span>
<span></span>
<span><span class="va">instrs</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span>python_path <span class="op">=</span> <span class="va">my_python_path</span>,</span>
<span>                                  save_dir <span class="op">=</span> <span class="va">results_directory</span>,</span>
<span>                                  plot_format <span class="op">=</span> <span class="st">'png'</span>,</span>
<span>                                  dpi <span class="op">=</span> <span class="fl">200</span>,</span>
<span>                                  height <span class="op">=</span> <span class="fl">9</span>,</span>
<span>                                  width <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create Giotto object</span></span>
<span><span class="va">custom_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">osm_exprs</span>,</span>
<span>                                     spatial_locs <span class="op">=</span> <span class="va">osm_locs</span>,</span>
<span>                                     instructions <span class="op">=</span> <span class="va">instrs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add field annotations as cell metadata</span></span>
<span><span class="va">metadata</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">meta_path</span><span class="op">)</span></span>
<span></span>
<span><span class="va">custom_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span><span class="va">custom_gobject</span>, </span>
<span>                                  new_metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>                                  by_column <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                                  column_cell_ID <span class="op">=</span> <span class="st">'CellID'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show the Giotto instructions associated with the Giotto object</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoInstructions.html" class="external-link">showGiottoInstructions</a></span><span class="op">(</span><span class="va">custom_gobject</span><span class="op">)</span></span></code></pre></div>
<p>Note that although parameters <code>show_plot</code>,
<code>return_plot</code>, and <code>save_plot</code> were not specified
within the call to <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html" class="external-link">createGiottoInstructions()</a></code>, default
values were provided to these instruction parameters. All instruction
parameters have default values, such that
<code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html" class="external-link">createGiottoInstructions()</a></code> may be called with some or no
arguments yet all instruction parameters will have a value after its
execution.</p>
<p>Alternatively, a named list may also be provided to the
<code>instructions</code> argument of <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject()</a></code>.
However, ensure that all arguments to
<code></code>createGiottoInstructions()
<code>are defined when providing instructions as a named list, since default values are only applied to instructions when made with createGiottoInstructions() \*Note that</code>python_path`
must be specified when providing instructions as a named list, and may
not be provided as NULL.</p>
<p>The <code>giottoInstructions</code> may be changed, or completely
replaced:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Change a specific previously set parameter, e.g. change dpi = 200 to dpi = 300</span></span>
<span><span class="va">custom_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/changeGiottoInstructions.html" class="external-link">changeGiottoInstructions</a></span><span class="op">(</span><span class="va">custom_gobject</span>, </span>
<span>                                           params <span class="op">=</span> <span class="st">'dpi'</span>,</span>
<span>                                           new_values <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Observe that the instructions have changed</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/instructions-generic.html" class="external-link">instructions</a></span><span class="op">(</span><span class="va">custom_gobject</span>, <span class="st">'dpi'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create new instructions using a named list</span></span>
<span><span class="va">sub_results_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">results_directory</span>, <span class="st">'specific_results/'</span><span class="op">)</span></span>
<span><span class="va">my_python_path</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/instructions-generic.html" class="external-link">instructions</a></span><span class="op">(</span><span class="va">custom_gobject</span>, <span class="st">'python_path'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">new_instrs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>python_path <span class="op">=</span> <span class="va">my_python_path</span>,</span>
<span>                  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  return_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  save_dir <span class="op">=</span> <span class="va">sub_results_directory</span>,</span>
<span>                  plot_format <span class="op">=</span> <span class="st">'jpg'</span>,</span>
<span>                  dpi <span class="op">=</span> <span class="fl">250</span>,</span>
<span>                  units <span class="op">=</span> <span class="st">'in'</span>,</span>
<span>                  height <span class="op">=</span> <span class="fl">12</span>,</span>
<span>                  width <span class="op">=</span> <span class="fl">12</span>,</span>
<span>                  is_docker <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Change all instructions</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/instructions-generic.html" class="external-link">instructions</a></span><span class="op">(</span><span class="va">custom_gobject</span><span class="op">)</span> <span class="op">=</span> <span class="va">new_instrs</span></span>
<span></span>
<span><span class="co"># Observe that the instructions have changed</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoInstructions.html" class="external-link">showGiottoInstructions</a></span><span class="op">(</span><span class="va">custom_gobject</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="active-spatial-unit-and-feature-type">1.3.1 Active spatial unit and feature type<a class="anchor" aria-label="anchor" href="#active-spatial-unit-and-feature-type"></a>
</h4>
<p>Many of <em>Giotto</em>’s functions have <code>spat_unit</code> and
<code>feat_type</code> parameters that govern which set of data to use.
The active spatial unit and feature type is visible when directly
returning the <code>giotto</code> object and decides what defaults are
used when those parameters are not supplied.</p>
<p>This setting is also stored within <code>giottoInstructions</code>
and there are convenient accessors specific for those two settings:</p>
<ul>
<li><code><a href="https://drieslab.github.io/GiottoClass/reference/activeSpatUnit-generic.html" class="external-link">activeSpatUnit()</a></code></li>
<li><code><a href="https://drieslab.github.io/GiottoClass/reference/activeFeatType-generic.html" class="external-link">activeFeatType()</a></code></li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="plotting-data-from-a-giotto-object">1.4 Plotting Data from a Giotto Object<a class="anchor" aria-label="anchor" href="#plotting-data-from-a-giotto-object"></a>
</h3>
<p>Each plotting function in Giotto has three important binary
parameters:</p>
<ul>
<li>
<code>show_plot</code>: print the plot to the console, default is
TRUE</li>
<li>
<code>return_plot</code>: return the plot as an object, default is
TRUE</li>
<li>
<code>save_plot</code>: automatically save the plot, default is
FALSE</li>
</ul>
<p>These parameters are stored within a <code>giotto</code> object that
was provided instructions from <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html" class="external-link">createGiottoInstructions()</a></code>
and are provided to plotting functions accordingly. To change these
parameters from the default values, the instructions may be changed or
replaced, or these parameters may be <em>manually overwritten</em>
within plotting functions.</p>
<p>See <code><a href="https://drieslab.github.io/GiottoVisuals/reference/showSaveParameters.html" class="external-link">showSaveParameters()</a></code> and the
<code>Saving Options</code> tutorial for alternative methods to save
plots.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot according to Giotto Instructions (default)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span><span class="va">custom_gobject</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/object_creation/0-spatPlot2D.jpg"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot clusters, create, and save to a new subdirectory, all while overwriting formatting</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span><span class="va">custom_gobject</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">'ClusterName'</span>, </span>
<span>         save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         return_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_folder <span class="op">=</span> <span class="st">'plots/'</span>, <span class="co"># Create subdirectory</span></span>
<span>                           save_name <span class="op">=</span> <span class="st">'cell_clusters'</span>, </span>
<span>                           save_format <span class="op">=</span> <span class="st">'png'</span>, </span>
<span>                           units <span class="op">=</span> <span class="st">'in'</span>,</span>
<span>                           base_height <span class="op">=</span> <span class="fl">9</span>,</span>
<span>                           base_width <span class="op">=</span> <span class="fl">9</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/object_creation/cell_clusters.png"></p>
<p>For a more in-depth look at the <code>giotto</code> object structure,
take a look at the <a href="./classes_intro.html">introduction to giotto
classes</a></p>
</div>
<div class="section level3">
<h3 id="session-info">2. Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">3.2</span> (<span class="dv">2023-10-31</span>)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin20</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="dv">14</span>.<span class="fl">3.1</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.3</span><span class="sc">-</span>x86_64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a>[<span class="dv">1</span>] GiottoData_0.<span class="dv">2</span>.<span class="fl">7.0</span> GiottoUtils_0.<span class="fl">1.5</span>  Giotto_4.<span class="fl">0.2</span>       GiottoClass_0.<span class="fl">1.3</span> </span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a> [<span class="dv">1</span>] tidyselect_1.<span class="fl">2.0</span>            farver_2.<span class="fl">1.1</span>                dplyr_1.<span class="fl">1.4</span>                </span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a> [<span class="dv">4</span>] GiottoVisuals_0.<span class="fl">1.4</span>         bitops_1<span class="fl">.0</span><span class="dv">-7</span>                fastmap_1.<span class="fl">1.1</span>              </span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a> [<span class="dv">7</span>] SingleCellExperiment_1.<span class="fl">24.0</span> RCurl_1<span class="fl">.98-1.14</span>             digest_0.<span class="fl">6.34</span>              </span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>[<span class="dv">10</span>] lifecycle_1.<span class="fl">0.4</span>             processx_3.<span class="fl">8.3</span>              terra_1<span class="fl">.7</span><span class="dv">-71</span>               </span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>[<span class="dv">13</span>] magrittr_2.<span class="fl">0.3</span>              compiler_4.<span class="fl">3.2</span>              rlang_1.<span class="fl">1.3</span>                </span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a>[<span class="dv">16</span>] tools_4.<span class="fl">3.2</span>                 utf8_1.<span class="fl">2.4</span>                  yaml_2.<span class="fl">3.8</span>                 </span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a>[<span class="dv">19</span>] data.table_1.<span class="fl">15.0</span>           knitr_1<span class="fl">.45</span>                  S4Arrays_1.<span class="fl">2.0</span>             </span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>[<span class="dv">22</span>] labeling_0.<span class="fl">4.3</span>              reticulate_1.<span class="fl">35.0</span>           DelayedArray_0.<span class="fl">28.0</span>        </span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a>[<span class="dv">25</span>] RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>          abind_1<span class="fl">.4</span><span class="dv">-5</span>                 withr_3.<span class="fl">0.0</span>                </span>
<span id="cb9-31"><a href="#cb9-31" tabindex="-1"></a>[<span class="dv">28</span>] BiocGenerics_0.<span class="fl">48.1</span>         grid_4.<span class="fl">3.2</span>                  stats4_4.<span class="fl">3.2</span>               </span>
<span id="cb9-32"><a href="#cb9-32" tabindex="-1"></a>[<span class="dv">31</span>] fansi_1.<span class="fl">0.6</span>                 colorspace_2<span class="fl">.1</span><span class="dv">-0</span>            ggplot2_3.<span class="fl">4.4</span>              </span>
<span id="cb9-33"><a href="#cb9-33" tabindex="-1"></a>[<span class="dv">34</span>] scales_1.<span class="fl">3.0</span>                gtools_3.<span class="fl">9.5</span>                SummarizedExperiment_1.<span class="fl">32.0</span></span>
<span id="cb9-34"><a href="#cb9-34" tabindex="-1"></a>[<span class="dv">37</span>] cli_3.<span class="fl">6.2</span>                   rmarkdown_2<span class="fl">.25</span>              crayon_1.<span class="fl">5.2</span>               </span>
<span id="cb9-35"><a href="#cb9-35" tabindex="-1"></a>[<span class="dv">40</span>] ragg_1.<span class="fl">2.7</span>                  generics_0.<span class="fl">1.3</span>              remotes_2.<span class="dv">4</span>.<span class="fl">2.1</span>            </span>
<span id="cb9-36"><a href="#cb9-36" tabindex="-1"></a>[<span class="dv">43</span>] rstudioapi_0.<span class="fl">15.0</span>           rjson_0.<span class="fl">2.21</span>                zlibbioc_1.<span class="fl">48.0</span>            </span>
<span id="cb9-37"><a href="#cb9-37" tabindex="-1"></a>[<span class="dv">46</span>] parallel_4.<span class="fl">3.2</span>              XVector_0.<span class="fl">42.0</span>              matrixStats_1.<span class="fl">2.0</span>          </span>
<span id="cb9-38"><a href="#cb9-38" tabindex="-1"></a>[<span class="dv">49</span>] vctrs_0.<span class="fl">6.5</span>                 Matrix_1<span class="fl">.6</span><span class="dv">-5</span>                jsonlite_1.<span class="fl">8.8</span>             </span>
<span id="cb9-39"><a href="#cb9-39" tabindex="-1"></a>[<span class="dv">52</span>] callr_3.<span class="fl">7.3</span>                 IRanges_2.<span class="fl">36.0</span>              S4Vectors_0.<span class="fl">40.2</span>           </span>
<span id="cb9-40"><a href="#cb9-40" tabindex="-1"></a>[<span class="dv">55</span>] systemfonts_1.<span class="fl">0.5</span>           magick_2.<span class="fl">8.2</span>                pak_0.<span class="fl">7.1</span>                  </span>
<span id="cb9-41"><a href="#cb9-41" tabindex="-1"></a>[<span class="dv">58</span>] glue_1.<span class="fl">7.0</span>                  codetools_0<span class="fl">.2</span><span class="dv">-19</span>            ps_1.<span class="fl">7.6</span>                   </span>
<span id="cb9-42"><a href="#cb9-42" tabindex="-1"></a>[<span class="dv">61</span>] cowplot_1.<span class="fl">1.3</span>               gtable_0.<span class="fl">3.4</span>                GenomeInfoDb_1.<span class="fl">38.6</span>        </span>
<span id="cb9-43"><a href="#cb9-43" tabindex="-1"></a>[<span class="dv">64</span>] GenomicRanges_1.<span class="fl">54.1</span>        munsell_0.<span class="fl">5.0</span>               tibble_3.<span class="fl">2.1</span>               </span>
<span id="cb9-44"><a href="#cb9-44" tabindex="-1"></a>[<span class="dv">67</span>] pillar_1.<span class="fl">9.0</span>                htmltools_0.<span class="fl">5.7</span>             GenomeInfoDbData_1.<span class="fl">2.11</span>    </span>
<span id="cb9-45"><a href="#cb9-45" tabindex="-1"></a>[<span class="dv">70</span>] R6_2.<span class="fl">5.1</span>                    textshaping_0.<span class="fl">3.7</span>           evaluate_0<span class="fl">.23</span>              </span>
<span id="cb9-46"><a href="#cb9-46" tabindex="-1"></a>[<span class="dv">73</span>] lattice_0<span class="fl">.22</span><span class="dv">-5</span>              Biobase_2.<span class="fl">62.0</span>              png_0<span class="fl">.1</span><span class="dv">-8</span>                  </span>
<span id="cb9-47"><a href="#cb9-47" tabindex="-1"></a>[<span class="dv">76</span>] backports_1.<span class="fl">4.1</span>             SpatialExperiment_1.<span class="fl">12.0</span>    Rcpp_1.<span class="fl">0.12</span>                </span>
<span id="cb9-48"><a href="#cb9-48" tabindex="-1"></a>[<span class="dv">79</span>] SparseArray_1.<span class="fl">2.4</span>           checkmate_2.<span class="fl">3.1</span>             colorRamp2_0.<span class="fl">1.0</span>           </span>
<span id="cb9-49"><a href="#cb9-49" tabindex="-1"></a>[<span class="dv">82</span>] xfun_0<span class="fl">.42</span>                   MatrixGenerics_1.<span class="fl">14.0</span>       pkgconfig_2.<span class="fl">0.3</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
