<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integrate ONTraC Analysis â€¢ Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Integrate ONTraC Analysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
        <li>
          <a href="../articles/create_simulated_object.html">Create a simulated datastet object</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Data Processing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
        <li>
          <a href="../articles/dimension_reduction.html">Dimension Reduction</a>
        </li>
        <li>
          <a href="../articles/networks.html">Spatial and Feature Networks</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactivity</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
        <li>
          <a href="../articles/interactive_selection_cosmx.html">Interactive region selection on large datasets</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/ontrac.html">Integrate ONTraC Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integrate ONTraC Analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/ontrac.Rmd" class="external-link"><code>vignettes/ontrac.Rmd</code></a></small>
      <div class="hidden name"><code>ontrac.Rmd</code></div>

    </div>

    
    
<div class="section level2" number="1">
<h2 number="1" id="overview-of-ontrac">
<span class="header-section-number">1</span> Overview of ONTraC<a class="anchor" aria-label="anchor" href="#overview-of-ontrac"></a>
</h2>
<p>ONTraC (Ordered Niche Trajectory Construction) is a niche-centered, machine learning method for constructing spatially continuous trajectories. ONTraC differs from existing tools in that it treats a niche, rather than an individual cell, as the basic unit for spatial trajectory analysis. In this context, we define niche as a multicellular, spatially localized region where different cell types may coexist and interact with each other. ONTraC seamlessly integrates cell-type composition and spatial information by using the graph neural network modeling framework. Its output, which is called the niche trajectory, can be viewed as a one dimensional representation of the tissue microenvironment continuum. By disentangling cell-level and niche-level properties, niche trajectory analysis provides a coherent framework to study coordinated responses from all the cells in association with continuous tissue microenvironment variations.</p>
<p><img src="images%2Fontrac%2F0-ONTraCWorkFlow.png" width="80%" style="display: block; margin: auto;"></p>
<p><a href="https://www.biorxiv.org/content/10.1101/2024.04.23.590827v1" class="external-link">ONTraC paper</a></p>
<p><a href="https://github.com/gyuanlab/ONTraC" class="external-link">ONTraC GitHub repository</a></p>
</div>
<div class="section level2" number="2">
<h2 number="2" id="dataset-description">
<span class="header-section-number">2</span> Dataset Description<a class="anchor" aria-label="anchor" href="#dataset-description"></a>
</h2>
<p>In this tutorial, we ues a MERFISH mouse motor cortex dataset comprising 61 tissue sections and containing approximately 280,000 cells characterised by a 258-gene panel. The study identified 3 classes of cells, glutamatergic, GABAergic and non-neuronal cell groups, and further clustered into 23 annotated plus 1 other subclass-level cell types.</p>
<p><img src="images%2Fontrac%2F0-Dataset.png" width="100%" style="display: block; margin: auto;"></p>
<p>MERFISH is a massively multiplexed single-molecule imaging technology for spatially resolved transcriptomics capable of simultaneously measuring the copy number and spatial distribution of hundreds to tens of thousands of RNA species in individual cells. For further information, please visit the official <a href="https://vizgen.com/technology/#merfish" class="external-link">website</a>.</p>
<p><img src="images%2Fontrac%2F0-MERFISH.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2" number="3">
<h2 number="3" id="pre-settings">
<span class="header-section-number">3</span> Pre-settings<a class="anchor" aria-label="anchor" href="#pre-settings"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">python_path</span> <span class="op">&lt;-</span> <span class="st">"giotto_ontrac_env"</span></span>
<span><span class="va">env_name</span> <span class="op">&lt;-</span> <span class="st">"giotto_ontrac_env"</span></span></code></pre></div>
<div class="section level3" number="3.1">
<h3 number="3.1" id="giotto-installation">
<span class="header-section-number">3.1</span> Giotto Installation<a class="anchor" aria-label="anchor" href="#giotto-installation"></a>
</h3>
<p>We need <code>R&gt;=4.4</code> for this tutorial.</p>
<ul>
<li>Install Giotto</li>
</ul>
<p>Please following the <a href="installation.html">Giotto Installation Tutorial</a>.</p>
</div>
<div class="section level3" number="3.2">
<h3 number="3.2" id="ontrac-installation">
<span class="header-section-number">3.2</span> ONTraC Installation<a class="anchor" aria-label="anchor" href="#ontrac-installation"></a>
</h3>
<p>Since ONTraC is a GNN algorithm that relies on <a href="https://pytorch-geometric.readthedocs.io/en/latest/" class="external-link">pytorch-geometric</a>, we create a new environment here in order not to interfere with other software using <code>installGiottoONTraCEnvironment()</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span><span class="fu">installGiottoONTraCEnvironment</span><span class="op">(</span>envname <span class="op">=</span> <span class="va">env_name</span><span class="op">)</span></span></code></pre></div>
<p>You can also update/re-install ONTraC in a specific env.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">reticulate</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/conda-tools.html" class="external-link">conda_install</a></span><span class="op">(</span>envname <span class="op">=</span> <span class="va">env_name</span>,</span>
<span>                          packages <span class="op">=</span> <span class="st">'ontrac'</span>,</span>
<span>                          pip <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="3.3">
<h3 number="3.3" id="configuring-data-and-results-paths">
<span class="header-section-number">3.3</span> Configuring Data and Results Paths<a class="anchor" aria-label="anchor" href="#configuring-data-and-results-paths"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span> <span class="co"># In case of network interrupt</span></span>
<span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/data"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">data_path</span>, recursive<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"/path/to/results"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">results_folder</span>, recursive<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="4">
<h2 number="4" id="downloading-dataset">
<span class="header-section-number">4</span> Downloading Dataset<a class="anchor" aria-label="anchor" href="#downloading-dataset"></a>
</h2>
<p>The MERFISH mouse motor cortex data to run this tutorial can be found <a href="https://doi.brainimagelibrary.org/doi/10.35077/g.21" class="external-link">here</a></p>
<p>You need to download the processed expression, metadata, and cell segmentation information by running these commands:</p>
<p>Note 1: there are 61 slices here, we run on two of them to save the time. Note 2: due to the instability of network, download processing may be interrupt.</p>
<p>We recommend to download these data in advance or download the processing giotto object from <a href="https://zenodo.org/records/13713948" class="external-link">Zenodo</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/counts.h5ad"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>,<span class="st">"counts.h5ad"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/cell_labels.csv"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>,<span class="st">"cell_labels.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/segmented_cells_mouse2sample1.csv"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>,<span class="st">"segmented_cells_mouse2sample1.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/segmented_cells_mouse2sample6.csv"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>,<span class="st">"segmented_cells_mouse2sample6.csv"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="5">
<h2 number="5" id="data-analysis">
<span class="header-section-number">5</span> Data Analysis<a class="anchor" aria-label="anchor" href="#data-analysis"></a>
</h2>
<div class="section level3" number="5.1">
<h3 number="5.1" id="creating-the-giotto-object">
<span class="header-section-number">5.1</span> Creating the Giotto Object<a class="anchor" aria-label="anchor" href="#creating-the-giotto-object"></a>
</h3>
<p>There are two options for creating the giotto object. Option 1 is create the giotto object step by step. Option 2 is load the pre-constructed giotto object from <a href="https://zenodo.org/records/13713948" class="external-link">Zenodo</a>.</p>
<div class="section level4" number="5.1.1">
<h4 number="5.1.1" id="option-1">
<span class="header-section-number">5.1.1</span> Option 1<a class="anchor" aria-label="anchor" href="#option-1"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">set_giotto_python_path</a></span><span class="op">(</span>python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Install the anndata and scanpy package since the counts matrix was stored in h5ad format. We need it to convert an anndata object to giotto object.</span></span>
<span><span class="fu">reticulate</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/conda-tools.html" class="external-link">conda_install</a></span><span class="op">(</span>envname <span class="op">=</span> <span class="va">env_name</span>, packages <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'scanpy'</span><span class="op">)</span>, pip <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create Giotto object from expression counts. This file contains 61 slices here.</span></span>
<span><span class="va">giotto_all_slices_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/anndataToGiotto.html" class="external-link">anndataToGiotto</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"counts.h5ad"</span><span class="op">)</span>, python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load meta_data</span></span>
<span><span class="va">meta_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"cell_labels.csv"</span><span class="op">)</span>,</span>
<span>                    colClasses <span class="op">=</span> <span class="st">"character"</span><span class="op">)</span> <span class="co"># as the cell IDs are 30 digit numbers, set the type as character to avoid the limitation of R in handling larger integers</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">meta_df</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"cell_ID"</span></span>
<span></span>
<span><span class="co">### We use two slices here to speed up</span></span>
<span><span class="va">slice1_cells</span> <span class="op">&lt;-</span> <span class="va">meta_df</span><span class="op">[</span><span class="va">meta_df</span><span class="op">$</span><span class="va">slice_id</span> <span class="op">==</span> <span class="st">"mouse2_slice229"</span>,<span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span></span>
<span><span class="va">slice2_cells</span> <span class="op">&lt;-</span> <span class="va">meta_df</span><span class="op">[</span><span class="va">meta_df</span><span class="op">$</span><span class="va">slice_id</span> <span class="op">==</span> <span class="st">"mouse2_slice300"</span>,<span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span></span>
<span><span class="va">selected_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">slice1_cells</span>, <span class="va">slice2_cells</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Subset giotto obj by cell ID</span></span>
<span><span class="va">giotto_slice1_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html" class="external-link">subsetGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_all_slices_obj</span>,</span>
<span>                                  cell_ids <span class="op">=</span> <span class="va">slice1_cells</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_slice2_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html" class="external-link">subsetGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_all_slices_obj</span>,</span>
<span>                                  cell_ids <span class="op">=</span> <span class="va">slice2_cells</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Add cell metadata</span></span>
<span><span class="va">giotto_slice1_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_slice1_obj</span>,</span>
<span>                                     new_metadata <span class="op">=</span> <span class="va">meta_df</span>,</span>
<span>                                     by_column <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     column_cell_ID <span class="op">=</span> <span class="st">'cell_ID'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_slice2_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_slice2_obj</span>,</span>
<span>                                     new_metadata <span class="op">=</span> <span class="va">meta_df</span>,</span>
<span>                                     by_column <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     column_cell_ID <span class="op">=</span> <span class="st">'cell_ID'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Cell segmentation. Calculate center (median of vertices) of each cell.</span></span>
<span><span class="va">segments_1_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"segmented_cells_mouse2sample1.csv"</span><span class="op">)</span>, </span>
<span>                          row.names<span class="op">=</span><span class="fl">1</span>, colClasses <span class="op">=</span> <span class="st">"character"</span><span class="op">)</span> <span class="co"># as the cell IDs are 30 digit numbers, set the type as character to avoid the limitation of R in handling larger integers</span></span>
<span></span>
<span><span class="va">segments_2_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"segmented_cells_mouse2sample6.csv"</span><span class="op">)</span>, </span>
<span>                          row.names<span class="op">=</span><span class="fl">1</span>, colClasses <span class="op">=</span> <span class="st">"character"</span><span class="op">)</span> <span class="co"># as the cell IDs are 30 digit numbers, set the type as character to avoid the limitation of R in handling larger integers</span></span>
<span></span>
<span><span class="va">segments_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">segments_1_df</span>, <span class="va">segments_2_df</span><span class="op">)</span></span>
<span></span>
<span><span class="va">loc.use</span> <span class="op">&lt;-</span> <span class="va">segments_df</span><span class="op">[</span><span class="va">selected_cells</span>,<span class="op">]</span></span>
<span><span class="va">loc.x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"boundaryX_"</span>,<span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">loc.use</span><span class="op">)</span>,value <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">loc.y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"boundaryY_"</span>,<span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">loc.use</span><span class="op">)</span>,value <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">centr.x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">loc.use</span><span class="op">[</span>,<span class="va">loc.x</span><span class="op">]</span>,<span class="fl">1</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">x</span>,<span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">y</span>,<span class="st">", "</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">centr.y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">loc.use</span><span class="op">[</span>,<span class="va">loc.y</span><span class="op">]</span>,<span class="fl">1</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">x</span>,<span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">y</span>,<span class="st">", "</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## create spatial locations object</span></span>
<span><span class="va">spatial_locs_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cell_ID <span class="op">=</span> <span class="va">selected_cells</span>,</span>
<span>                              sdimx <span class="op">=</span> <span class="va">centr.x</span>,</span>
<span>                              sdimy <span class="op">=</span> <span class="va">centr.y</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spatial_locs_slice1_df</span> <span class="op">&lt;-</span> <span class="va">spatial_locs_df</span><span class="op">[</span><span class="va">slice1_cells</span>,<span class="op">]</span></span>
<span><span class="va">spatial_locs_slice2_df</span> <span class="op">&lt;-</span> <span class="va">spatial_locs_df</span><span class="op">[</span><span class="va">slice2_cells</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">spat_locs_slice1_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/readSpatLocsData.html" class="external-link">readSpatLocsData</a></span><span class="op">(</span>data_list <span class="op">=</span> <span class="va">spatial_locs_slice1_df</span><span class="op">)</span></span>
<span><span class="va">spat_locs_slice2_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/readSpatLocsData.html" class="external-link">readSpatLocsData</a></span><span class="op">(</span>data_list <span class="op">=</span> <span class="va">spatial_locs_slice2_df</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## add spatial location info</span></span>
<span><span class="va">giotto_slice1_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/setSpatialLocations.html" class="external-link">setSpatialLocations</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_slice1_obj</span>,</span>
<span>                                         x <span class="op">=</span> <span class="va">spat_locs_slice1_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_slice2_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/setSpatialLocations.html" class="external-link">setSpatialLocations</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_slice2_obj</span>,</span>
<span>                                         x <span class="op">=</span> <span class="va">spat_locs_slice2_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## merge two giotto objects together</span></span>
<span><span class="va">giotto_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects</a></span><span class="op">(</span>gobject_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">giotto_slice1_obj</span>,</span>
<span>                                                    <span class="va">giotto_slice2_obj</span><span class="op">)</span>,</span>
<span>                               gobject_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mouse2_slice229"</span>,</span>
<span>                                                 <span class="st">"mouse2_slice300"</span><span class="op">)</span>,  <span class="co"># name for each samples</span></span>
<span>                               join_method <span class="op">=</span> <span class="st">"z_stack"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## save giotto obj</span></span>
<span><span class="co"># saveGiotto</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/saveGiotto.html" class="external-link">saveGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>, foldername <span class="op">=</span> <span class="st">"gobject"</span>, dir<span class="op">=</span><span class="va">results_folder</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4" number="5.1.2">
<h4 number="5.1.2" id="option-2">
<span class="header-section-number">5.1.2</span> Option 2<a class="anchor" aria-label="anchor" href="#option-2"></a>
</h4>
<p>If you facing network issue when downloading the raw dataset. Please download the processing giotto obj from <a href="https://zenodo.org/records/13713948" class="external-link">Zenodo</a>, unzip and move it to <code>results</code> folder.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">set_giotto_python_path</a></span><span class="op">(</span>python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span>
<span><span class="va">giotto_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/loadGiotto.html" class="external-link">loadGiotto</a></span><span class="op">(</span>path_to_folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_folder</span>, <span class="st">"gobject"</span><span class="op">)</span>, python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3" number="5.2">
<h3 number="5.2" id="spatial-distribution-of-cell-type">
<span class="header-section-number">5.2</span> Spatial distribution of cell type<a class="anchor" aria-label="anchor" href="#spatial-distribution-of-cell-type"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">giotto_obj</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"slice_id"</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"subclass"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>           point_border_stroke <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>           legend_text <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F1-spatialCellTypeDis.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="5.3">
<h3 number="5.3" id="generate-input-file-for-ontrac">
<span class="header-section-number">5.3</span> Generate input file for ONTraC<a class="anchor" aria-label="anchor" href="#generate-input-file-for-ontrac"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We skip the processing process here to save time and use the given cell type</span></span>
<span><span class="co"># annotation directly</span></span>
<span><span class="va">ONTraC_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getONTraCv1Input.html">getONTraCv1Input</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                                 cell_type <span class="op">=</span> <span class="st">"subclass"</span>,</span>
<span>                                 output_path <span class="op">=</span> <span class="va">results_folder</span>,</span>
<span>                                 spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                                 feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                                 verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ONTraC_input</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#   Cell_ID                                                     Sample          x           y           Cell_Type          </span></span>
<span><span class="co">#   &lt;chr&gt;                                                       &lt;chr&gt;           &lt;dbl&gt;       &lt;dbl&gt;       &lt;chr&gt;</span></span>
<span><span class="co">#   mouse2_slice229-100101435705986292663283283043431511315     mouse2_slice229 -4828.728   -2203.4502  L6 CT</span></span>
<span><span class="co">#   mouse2_slice229-100104370212612969023746137269354247741     mouse2_slice229 -5405.400   -995.6467   OPC</span></span>
<span><span class="co">#   mouse2_slice229-100128078183217482733448056590230529739     mouse2_slice229 -5731.403   -1071.1735  L2/3 IT</span></span>
<span><span class="co">#   mouse2_slice229-100209662400867003194056898065587980841     mouse2_slice229 -5468.113   -1286.2465  Oligo</span></span>
<span><span class="co">#   mouse2_slice229-100218038012295593766653119076639444055     mouse2_slice229 -6399.986   -959.7440   L2/3 IT</span></span>
<span><span class="co">#   mouse2_slice229-100252992997994275968450436343196667192     mouse2_slice229 -6637.847   -1659.6237  Astro</span></span></code></pre></div>
</div>
<div class="section level3" number="5.4">
<h3 number="5.4" id="perform-spatial-niche-trajectory-analysis-using-ontrac">
<span class="header-section-number">5.4</span> Perform spatial niche trajectory analysis using ONTraC<a class="anchor" aria-label="anchor" href="#perform-spatial-niche-trajectory-analysis-using-ontrac"></a>
</h3>
<div class="section level4" number="5.4.1">
<h4 number="5.4.1" id="option-1-run-ontrac-through-giotto">
<span class="header-section-number">5.4.1</span> Option 1: run ONTraC through Giotto<a class="anchor" aria-label="anchor" href="#option-1-run-ontrac-through-giotto"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">runONTraCV1</span><span class="op">(</span>ONTraC_input <span class="op">=</span> <span class="va">ONTraC_input</span>,</span>
<span>       dataset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_folder</span>, <span class="st">"ONTraC_dataset_input.csv"</span><span class="op">)</span>,</span>
<span>       preprocessing_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_folder</span>, <span class="st">"preprocessing_dir"</span><span class="op">)</span>,</span>
<span>       GNN_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_folder</span>, <span class="st">"GNN_dir"</span><span class="op">)</span>,</span>
<span>       NTScore_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_folder</span>, <span class="st">"NTScore_dir"</span><span class="op">)</span>,</span>
<span>       n_neighbors <span class="op">=</span> <span class="fl">50L</span>,</span>
<span>       n_local <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>       device <span class="op">=</span> <span class="st">"cpu"</span>,</span>
<span>       seed <span class="op">=</span> <span class="fl">42L</span>,</span>
<span>       k <span class="op">=</span> <span class="fl">6L</span>,</span>
<span>       modularity_loss_weight <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>       purity_loss_weight <span class="op">=</span> <span class="fl">300.0</span>,</span>
<span>       regularization_loss_weight <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>       beta <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>       python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4" number="5.4.2">
<h4 number="5.4.2" id="option-2-run-ontrac-in-a-separate-machine">
<span class="header-section-number">5.4.2</span> Option 2: run ONTraC in a separate machine<a class="anchor" aria-label="anchor" href="#option-2-run-ontrac-in-a-separate-machine"></a>
</h4>
<p>You can run ONTraC on an HPC with an NVIDIA GPU node, which would save a lot of time when applying it on larger datasets.</p>
<ul>
<li>Install ONTraC.</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true"></a><span class="bu">source</span> ~/.bash_profile</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true"></a><span class="ex">conda</span> create -y -n ONTraC python=3.11</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true"></a><span class="ex">conda</span> activate ONTraC</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true"></a><span class="ex">pip</span> install ONTraC</span></code></pre></div>
<ul>
<li>Copy the input file to that machine and running ONTraC.</li>
</ul>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true"></a><span class="bu">source</span> ~/.bash_profile</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true"></a><span class="ex">conda</span> activate ONTraC</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true"></a><span class="ex">ONTraC</span> -d /path/to/results/ONTraC_dataset_input.csv --preprocessing-dir /path/to/results/preprocessing_dir --GNN-dir /path/to/results/GNN_dir --NTScore-dir /path/to/results/NTScore_dir --device cuda --epochs 1000 -s 42 --patience 100 --min-delta 0.001 --min-epochs 50 --lr 0.03 --hidden-feats 4 -k 6 --modularity-loss-weight 0.3 --purity-loss-weight 300 --regularization-loss-weight 0.3 --beta 0.03 <span class="op">2&gt;&amp;1</span> <span class="kw">|</span> <span class="fu">tee</span> /path/to/results/merfish_subset.log</span></code></pre></div>
<ul>
<li><p>Copy the output directories (<code>/path/to/results/preprocessing_dir</code>, <code>/path/to/results/GNN_dir</code>, and <code>/path/to/results/NTScore_dir</code>) back to the machine running Giotto.</p></li>
<li><p>For details, please refer <a href="https://github.com/gyuanlab/ONTraC" class="external-link">ONTraCâ€™s website</a>.</p></li>
</ul>
</div>
</div>
</div>
<div class="section level2" number="6">
<h2 number="6" id="visualization">
<span class="header-section-number">6</span> Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="section level3" number="6.1">
<h3 number="6.1" id="load-ontrac-results">
<span class="header-section-number">6.1</span> Load ONTraC results<a class="anchor" aria-label="anchor" href="#load-ontrac-results"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">giotto_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadOntraCResults.html">loadOntraCResults</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                                ontrac_results_dir <span class="op">=</span> <span class="va">results_folder</span>,</span>
<span>                                NTScore_reverse <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># whether to reverse the NTScore (direction of the trajectory)</span></span>
<span>                                <span class="op">)</span></span></code></pre></div>
<p>This function will look for the <code>preprocessing_dir</code>, <code>GNN_dir</code>, and <code>NTScore_dir</code> under the <code>results_folder</code> automatically.</p>
<p>The NTScore and binarized niche cluster info were stored in cell metadata as below:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">giotto_obj</span>, spat_unit <span class="op">=</span> <span class="st">"cell"</span>, feat_type <span class="op">=</span> <span class="st">"rna"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#                                                       cell_ID      sample_id        slice_id    class_label subclass   label           list_ID  NTScore   niche cluster</span></span>
<span><span class="co">#                                                        &lt;char&gt;         &lt;char&gt;          &lt;char&gt;         &lt;char&gt;   &lt;char&gt;  &lt;char&gt;            &lt;char&gt;  &lt;dbl&gt;     &lt;fctr&gt;</span></span>
<span><span class="co">#   1:  mouse2_slice229-100101435705986292663283283043431511315 mouse2_sample6  mouse2_slice229 Glutamatergic   L6 CT    L6_CT_5   mouse2_slice229  0.2002081 NicheCluster_3</span></span>
<span><span class="co">#   2:  mouse2_slice229-100104370212612969023746137269354247741 mouse2_sample6  mouse2_slice229 Other           OPC      OPC       mouse2_slice229  0.7999791 NicheCluster_1</span></span>
<span><span class="co">#   3:  mouse2_slice229-100128078183217482733448056590230529739 mouse2_sample6  mouse2_slice229 Glutamatergic   L2/3 IT  L23_IT_4  mouse2_slice229  0.7662198 NicheCluster_1</span></span>
<span><span class="co">#   4:  mouse2_slice229-100209662400867003194056898065587980841 mouse2_sample6  mouse2_slice229 Other           Oligo    Oligo_1   mouse2_slice229  0.6010420 NicheCluster_5</span></span>
<span><span class="co">#   5:  mouse2_slice229-100218038012295593766653119076639444055 mouse2_sample6  mouse2_slice229 Glutamatergic   L2/3 IT  L23_IT_4  mouse2_slice229  0.7132024 NicheCluster_1</span></span>
<span><span class="co">#   6:  mouse2_slice229-100252992997994275968450436343196667192 mouse2_sample6  mouse2_slice229 Other           Astro    Astro_2   mouse2_slice229  0.1980136 NicheCluster_3</span></span></code></pre></div>
<p>The 1) probability matrix of each cell assigned to each niche cluster and 2) connectivity between niche cluster were stored as additional matrix here:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">GiottoClass</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/list_expression.html" class="external-link">list_expression</a></span><span class="op">(</span><span class="va">giotto_obj</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#        spat_unit     feat_type       name</span></span>
<span><span class="co">#           &lt;char&gt;        &lt;char&gt;     &lt;char&gt;</span></span>
<span><span class="co"># 1:          cell           rna        raw</span></span>
<span><span class="co"># 2:          cell niche cluster       prob</span></span>
<span><span class="co"># 3: niche cluster  connectivity normalized</span></span></code></pre></div>
</div>
<div class="section level3" number="6.2">
<h3 number="6.2" id="niche-cluster-probability-distribution">
<span class="header-section-number">6.2</span> Niche cluster probability distribution<a class="anchor" aria-label="anchor" href="#niche-cluster-probability-distribution"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotSpatNicheClusterProb</span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                         group_by <span class="op">=</span> <span class="st">"list_ID"</span>,</span>
<span>                         gradient_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                         point_border_col <span class="op">=</span> <span class="st">"gray"</span>,</span>
<span>                         show_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F2-spatialNicheClusterProb.png" width="60%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="6.3">
<h3 number="6.3" id="binarized-niche-cluster-for-each-cell">
<span class="header-section-number">6.3</span> Binarized niche cluster for each cell<a class="anchor" aria-label="anchor" href="#binarized-niche-cluster-for-each-cell"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotSpatNicheClusterBin</span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                        group_by <span class="op">=</span> <span class="st">"list_ID"</span>,</span>
<span>                        point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                        point_border_stroke <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>                        show_plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F3-spatialBinNicheCluster.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="6.4">
<h3 number="6.4" id="niche-cluster-spatial-connectivity">
<span class="header-section-number">6.4</span> Niche cluster spatial connectivity<a class="anchor" aria-label="anchor" href="#niche-cluster-spatial-connectivity"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span>  <span class="co"># fix the node positions</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotNicheClusterConnectivity.html">plotNicheClusterConnectivity</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                             show_plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F4-nicheClusterConnectivity.png" width="60%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="6.5">
<h3 number="6.5" id="nt-niche-trajectory-score">
<span class="header-section-number">6.5</span> NT (niche trajectory) score<a class="anchor" aria-label="anchor" href="#nt-niche-trajectory-score"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>, </span>
<span>           spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>           feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"list_ID"</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"NTScore"</span>,</span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           cell_color_gradient <span class="op">=</span> <span class="st">"turbo"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>           point_border_stroke <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>           show_plot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F5-2-newSpatialNTScore.png" width="80%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/Giotto_website/reference/plotCellTypeNTScore.html">plotCellTypeNTScore</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                    cell_type <span class="op">=</span> <span class="st">"subclass"</span>,</span>
<span>                    values <span class="op">=</span> <span class="st">"NTScore"</span>,</span>
<span>                    spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                    feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                    show_plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F6-violinCellTypeOverNTScore.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="6.6">
<h3 number="6.6" id="cell-type-composition-within-niche-cluster">
<span class="header-section-number">6.6</span> Cell type composition within niche cluster<a class="anchor" aria-label="anchor" href="#cell-type-composition-within-niche-cluster"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCTCompositionInNicheCluster.html">plotCTCompositionInNicheCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                                cell_type <span class="op">=</span> <span class="st">"subclass"</span>,</span>
<span>                                normalization <span class="op">=</span> <span class="st">"by_niche_cluster"</span>,</span>
<span>                                show_plot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>                                <span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F7-heatmapCellTypeCompositionInNicheClusterNormByNicheCluster.png" width="80%" style="display: block; margin: auto;"></p>
<p>We can also normalized by cell type.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCTCompositionInNicheCluster.html">plotCTCompositionInNicheCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_obj</span>,</span>
<span>                                cell_type <span class="op">=</span> <span class="st">"subclass"</span>,</span>
<span>                                normalization <span class="op">=</span> <span class="st">"by_cell_type"</span>,</span>
<span>                                show_plot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>                                <span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fontrac%2F7-heatmapCellTypeCompositionInNicheClusterNormByNicheCluster.png" width="80%" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2" number="7">
<h2 number="7" id="session-info">
<span class="header-section-number">7</span> Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co"># Platform: aarch64-apple-darwin20</span></span>
<span><span class="co"># Running under: macOS Ventura 13.6.6</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># Matrix products: default</span></span>
<span><span class="co"># BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib </span></span>
<span><span class="co"># LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># locale:</span></span>
<span><span class="co"># [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># time zone: America/New_York</span></span>
<span><span class="co"># tzcode source: internal</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># attached base packages:</span></span>
<span><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># other attached packages:</span></span>
<span><span class="co"># [1] ggraph_2.2.1.9000 ggplot2_3.5.1     Giotto_4.1.3      GiottoClass_0.4.0</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># loaded via a namespace (and not attached):</span></span>
<span><span class="co">#  [1] gtable_0.3.5        xfun_0.44           htmlwidgets_1.6.4   ggrepel_0.9.5      </span></span>
<span><span class="co">#  [5] lattice_0.22-6      vctrs_0.6.5         tools_4.4.0         generics_0.1.3     </span></span>
<span><span class="co">#  [9] parallel_4.4.0      tibble_3.2.1        fansi_1.0.6         colorRamp2_0.1.0   </span></span>
<span><span class="co"># [13] pkgconfig_2.0.3     R.oo_1.26.0         Matrix_1.7-0        data.table_1.15.4  </span></span>
<span><span class="co"># [17] checkmate_2.3.2     RColorBrewer_1.1-3  lifecycle_1.0.4     compiler_4.4.0     </span></span>
<span><span class="co"># [21] farver_2.1.2        deldir_2.0-4        GiottoUtils_0.1.12  munsell_0.5.1      </span></span>
<span><span class="co"># [25] ggforce_0.4.2       terra_1.7-78        graphlayouts_1.1.1  codetools_0.2-20   </span></span>
<span><span class="co"># [29] scattermore_1.2     htmltools_0.5.8.1   yaml_2.3.8          lazyeval_0.2.2     </span></span>
<span><span class="co"># [33] plotly_4.10.4       pillar_1.9.0        tidyr_1.3.1         MASS_7.3-60.2      </span></span>
<span><span class="co"># [37] R.utils_2.12.3      cachem_1.1.0        dbscan_1.2-0        viridis_0.6.5      </span></span>
<span><span class="co"># [41] magick_2.8.4        gtools_3.9.5        tidyselect_1.2.1    digest_0.6.35      </span></span>
<span><span class="co"># [45] dplyr_1.1.4         purrr_1.0.2         labeling_0.4.3      polyclip_1.10-6    </span></span>
<span><span class="co"># [49] cowplot_1.1.3       fastmap_1.2.0       grid_4.4.0          colorspace_2.1-0   </span></span>
<span><span class="co"># [53] cli_3.6.3           magrittr_2.0.3      tidygraph_1.3.1     utf8_1.2.4         </span></span>
<span><span class="co"># [57] withr_3.0.1         scales_1.3.0        backports_1.5.0     sp_2.1-4           </span></span>
<span><span class="co"># [61] rmarkdown_2.27      httr_1.4.7          matrixStats_1.3.0   igraph_2.0.3       </span></span>
<span><span class="co"># [65] gridExtra_2.3       reticulate_1.38.0   GiottoVisuals_0.2.5 png_0.1-8          </span></span>
<span><span class="co"># [69] R.methodsS3_1.8.2   memoise_2.0.1       evaluate_0.24.0     knitr_1.47         </span></span>
<span><span class="co"># [73] viridisLite_0.4.2   rlang_1.1.4         Rcpp_1.0.13         glue_1.7.0         </span></span>
<span><span class="co"># [77] tweenr_2.0.3        rstudioapi_0.16.0   jsonlite_1.8.8      R6_2.5.1    </span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. ChÃ¡vez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew Oâ€™Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
