<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SeqFish+ Mouse Cortex Example • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SeqFish+ Mouse Cortex Example">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/core_functions.html">Giotto Core Functions</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
        <li>
          <a href="../articles/create_simulated_object.html">Create a simulated datastet object</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Data Processing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
        <li>
          <a href="../articles/dimension_reduction.html">Dimension Reduction</a>
        </li>
        <li>
          <a href="../articles/networks.html">Spatial and Feature Networks</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactivity</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
        <li>
          <a href="../articles/interactive_selection_cosmx.html">Interactive region selection on large datasets</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/ontrac.html">Integrate ONTraC Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SeqFish+ Mouse Cortex Example</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/seqfish_cortex.Rmd" class="external-link"><code>vignettes/seqfish_cortex.Rmd</code></a></small>
      <div class="hidden name"><code>seqfish_cortex.Rmd</code></div>

    </div>

    
    
<div class="section level2" number="1">
<h2 number="1" id="start-giotto">
<span class="header-section-number">1</span> Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure GiottoData, a small, helper module for tutorials, is installed.</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"GiottoData"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/GiottoData"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>    <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 number="2" id="dataset-explanation">
<span class="header-section-number">2</span> Dataset Explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p>Several fields - containing 100’s of cells - in the mouse cortex and subventricular zone were imaged for seqFISH+. The coordinates of the cells within each field are independent of each other, so in order to visualize and process all cells together imaging fields will be stitched together by providing x and y-offset values specific to each field.</p>
<p>These offset values are known or estimates based on the original raw image:</p>
<p><img src="images%2Fseqfish_cortex%2Fcortex_svz_location_fields.png"></p>
</div>
<div class="section level2" number="3">
<h2 number="3" id="download-data">
<span class="header-section-number">3</span> Download Data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/data/"</span></span>
<span></span>
<span><span class="fu">GiottoData</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/getSpatialDataset.html" class="external-link">getSpatialDataset</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"seqfish_SS_cortex"</span>, </span>
<span>                              directory <span class="op">=</span> <span class="va">data_path</span>, </span>
<span>                              method <span class="op">=</span> <span class="st">"wget"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 number="4" id="giotto-instructions-and-preparation">
<span class="header-section-number">4</span> Giotto Instructions and Preparation<a class="anchor" aria-label="anchor" href="#giotto-instructions-and-preparation"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#  set Giotto instructions</span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"/path/to/results/"</span></span>
<span></span>
<span><span class="va">instructions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_instructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span></span>
<span>    save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    show_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    return_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    save_dir <span class="op">=</span> <span class="va">results_folder</span>,</span>
<span>    python_path <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create giotto object from provided paths ####</span></span>
<span><span class="va">expr_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"cortex_svz_expression.txt"</span><span class="op">)</span></span>
<span><span class="va">loc_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"cortex_svz_centroids_coord.txt"</span><span class="op">)</span></span>
<span><span class="va">meta_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"cortex_svz_centroids_annot.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># This dataset contains multiple field of views which need to be stitched together</span></span>
<span><span class="co"># first merge location and additional metadata</span></span>
<span><span class="va">SS_locations</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">loc_path</span><span class="op">)</span></span>
<span><span class="va">cortex_fields</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">meta_path</span><span class="op">)</span></span>
<span><span class="va">SS_loc_annot</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/merge.html" class="external-link">merge.data.table</a></span><span class="op">(</span><span class="va">SS_locations</span>, </span>
<span>                                             <span class="va">cortex_fields</span>, </span>
<span>                                             by <span class="op">=</span> <span class="st">"ID"</span><span class="op">)</span></span>
<span><span class="va">SS_loc_annot</span><span class="op">[</span>, <span class="va">ID</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ID</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cell_"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">913</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorder</a></span><span class="op">(</span><span class="va">SS_loc_annot</span>, <span class="va">ID</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create file with offset information</span></span>
<span><span class="va">my_offset_file</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span></span>
<span>    field <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">6</span><span class="op">)</span>,</span>
<span>    x_offset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1654.97</span>, <span class="fl">1750.75</span>, <span class="fl">1674.35</span>, <span class="fl">675.5</span>, <span class="fl">2048</span>, <span class="fl">675</span><span class="op">)</span>,</span>
<span>    y_offset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="op">-</span><span class="fl">1438.02</span>, <span class="op">-</span><span class="fl">1438.02</span>, <span class="fl">0</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create a stitch file</span></span>
<span><span class="va">stitch_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/stitchFieldCoordinates.html" class="external-link">stitchFieldCoordinates</a></span><span class="op">(</span></span>
<span>    location_file <span class="op">=</span> <span class="va">SS_loc_annot</span>,</span>
<span>    offset_file <span class="op">=</span> <span class="va">my_offset_file</span>,</span>
<span>    cumulate_offset_x <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    cumulate_offset_y <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    field_col <span class="op">=</span> <span class="st">"FOV"</span>,</span>
<span>    reverse_final_x <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    reverse_final_y <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">stitch_file</span> <span class="op">&lt;-</span> <span class="va">stitch_file</span><span class="op">[</span>, <span class="fu">.</span><span class="op">(</span><span class="va">ID</span>, <span class="va">X_final</span>, <span class="va">Y_final</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">stitch_file</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">stitch_file</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span></span>
<span><span class="va">my_offset_file</span> <span class="op">&lt;-</span> <span class="va">my_offset_file</span><span class="op">[</span>, <span class="fu">.</span><span class="op">(</span><span class="va">field</span>, <span class="va">x_offset_final</span>, <span class="va">y_offset_final</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2" number="5">
<h2 number="5" id="create-giotto-object-process-data">
<span class="header-section-number">5</span> Create Giotto object &amp; process data<a class="anchor" aria-label="anchor" href="#create-giotto-object-process-data"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create Giotto object</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expr_path</span>,</span>
<span>                                 spatial_locs <span class="op">=</span> <span class="va">stitch_file</span>,</span>
<span>                                 offset_file <span class="op">=</span> <span class="va">my_offset_file</span>,</span>
<span>                                 instructions <span class="op">=</span> <span class="va">instructions</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add additional annotation if wanted</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                              new_metadata <span class="op">=</span> <span class="va">cortex_fields</span>,</span>
<span>                              by_column <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                              column_cell_ID <span class="op">=</span> <span class="st">"ID"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># subset data to the cortex field of views</span></span>
<span><span class="va">cell_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">SS_seqfish</span><span class="op">)</span></span>
<span><span class="va">cortex_cell_ids</span> <span class="op">&lt;-</span> <span class="va">cell_metadata</span><span class="op">[</span><span class="va">FOV</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fl">0</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html" class="external-link">subsetGiotto</a></span><span class="op">(</span><span class="va">SS_seqfish</span>, cell_ids <span class="op">=</span> <span class="va">cortex_cell_ids</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                           expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                           feat_det_in_min_cells <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                           min_det_feats_per_cell <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                           expression_values <span class="op">=</span> <span class="st">"raw"</span>,</span>
<span>                           verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># normalize</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                              scalefactor <span class="op">=</span> <span class="fl">6000</span>, </span>
<span>                              verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add gene &amp; cell statistics</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># adjust expression matrix for technical or known variables</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjustGiottoMatrix.html">adjustGiottoMatrix</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                                 expression_values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                                 covariate_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"nr_feats"</span>, <span class="st">"total_expr"</span><span class="op">)</span>,</span>
<span>                                 return_gobject <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 update_slot <span class="op">=</span> <span class="st">"custom"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F0-spatPlot2D.png"></p>
</div>
<div class="section level2" number="6">
<h2 number="6" id="dimension-reduction">
<span class="header-section-number">6</span> Dimension Reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## highly variable features (HVF)</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                           save_plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F1-HVFplot.png"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## select genes based on highly variable features and gene statistics, both found in feature (gene) metadata</span></span>
<span><span class="va">gene_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/fDataDT.html" class="external-link">fDataDT</a></span><span class="op">(</span><span class="va">SS_seqfish</span><span class="op">)</span></span>
<span><span class="va">featgenes</span> <span class="op">&lt;-</span> <span class="va">gene_metadata</span><span class="op">[</span><span class="va">hvf</span> <span class="op">==</span> <span class="st">"yes"</span> <span class="op">&amp;</span> <span class="va">perc_cells</span> <span class="op">&gt;</span> <span class="fl">4</span> <span class="op">&amp;</span> <span class="va">mean_expr_det</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">]</span><span class="op">$</span><span class="va">feat_ID</span></span>
<span></span>
<span><span class="co">## run PCA on expression values (default)</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                     genes_to_use <span class="op">=</span> <span class="va">featgenes</span>, </span>
<span>                     scale_unit <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                     center <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">SS_seqfish</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F2-screePlot.png"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F3-PCA.png"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">SS_seqfish</span>, </span>
<span>                      dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span>, </span>
<span>                      n_threads <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F4-UMAP.png"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runtSNE.html">runtSNE</a></span><span class="op">(</span><span class="va">SS_seqfish</span>, </span>
<span>                      dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotTSNE.html" class="external-link">plotTSNE</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F5-tSNE.png"></p>
</div>
<div class="section level2" number="7">
<h2 number="7" id="cluster">
<span class="header-section-number">7</span> Cluster<a class="anchor" aria-label="anchor" href="#cluster"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## sNN network (default)</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                                   dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span>,</span>
<span>                                   k <span class="op">=</span> <span class="fl">15</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Leiden clustering</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                              resolution <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>                              n_iterations <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F6-UMAP.png"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Leiden subclustering for specified clusters</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subClusterCells.html">doLeidenSubCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                                 cluster_column <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>                                 resolution <span class="op">=</span> <span class="fl">0.2</span>, </span>
<span>                                 k_neighbors <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                                 pca_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_values <span class="op">=</span> <span class="st">"normalized"</span>, </span>
<span>                                                  scale_unit <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>                                 nn_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>,</span>
<span>                                 selected_clusters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">6</span>, <span class="fl">7</span><span class="op">)</span>,</span>
<span>                                 name <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## set colors for clusters</span></span>
<span><span class="va">subleiden_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="fl">1.1</span>, <span class="fl">2.1</span>, <span class="fl">3.1</span>, <span class="fl">4.1</span>, <span class="fl">5.1</span>, <span class="fl">5.2</span>,</span>
<span>    <span class="fl">6.1</span>, <span class="fl">6.2</span>, <span class="fl">7.1</span>, <span class="fl">7.2</span>, <span class="fl">8.1</span>, <span class="fl">9.1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoUtils/reference/getDistinctColors.html" class="external-link">getDistinctColors</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">subleiden_order</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">giotto_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">subleiden_order</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>, </span>
<span>         cell_color_code <span class="op">=</span> <span class="va">giotto_colors</span>,</span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>         show_center_label <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>         legend_text <span class="op">=</span> <span class="fl">12</span>, </span>
<span>         legend_symbol_size <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F7-UMAP.png"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show cluster relationships</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/showClusterHeatmap.html" class="external-link">showClusterHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                   cluster_column <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>,</span>
<span>                   row_names_gp <span class="op">=</span> <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>, </span>
<span>                   column_names_gp <span class="op">=</span> <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F8-showClusterHeatmap.png"></p>
<p>The following step requires the installation of {ggdendro}.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages('ggdendro')</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://andrie.github.io/ggdendro/" class="external-link">ggdendro</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/showClusterDendrogram.html" class="external-link">showClusterDendrogram</a></span><span class="op">(</span><span class="va">SS_seqfish</span>, </span>
<span>                      h <span class="op">=</span> <span class="fl">0.5</span>, </span>
<span>                      rotate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                      cluster_column <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F9-showClusterDendrogram.png"></p>
</div>
<div class="section level2" number="8">
<h2 number="8" id="visualize-spatial-and-expression-space">
<span class="header-section-number">8</span> Visualize Spatial and Expression Space<a class="anchor" aria-label="anchor" href="#visualize-spatial-and-expression-space"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># expression and spatial</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html" class="external-link">spatDimPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>,</span>
<span>            cell_color_code <span class="op">=</span> <span class="va">giotto_colors</span>,</span>
<span>            dim_point_size <span class="op">=</span> <span class="fl">2</span>, </span>
<span>            spat_point_size <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F10-spatDimPlot2D.png"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># selected groups and provide new colors</span></span>
<span><span class="va">groups_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6.1</span>, <span class="fl">6.2</span>, <span class="fl">7.1</span>, <span class="fl">7.2</span><span class="op">)</span></span>
<span><span class="va">group_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"green"</span>, <span class="st">"blue"</span>, <span class="st">"purple"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">group_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">groups_of_interest</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html" class="external-link">spatDimPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>,</span>
<span>            dim_point_size <span class="op">=</span> <span class="fl">2</span>, </span>
<span>            spat_point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>            select_cell_groups <span class="op">=</span> <span class="va">groups_of_interest</span>, </span>
<span>            cell_color_code <span class="op">=</span> <span class="va">group_colors</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F11-spatDimPlot2D.png"></p>
</div>
<div class="section level2" number="9">
<h2 number="9" id="cell-type-marker-gene-detection">
<span class="header-section-number">9</span> Cell Type Marker Gene Detection<a class="anchor" aria-label="anchor" href="#cell-type-marker-gene-detection"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## gini</span></span>
<span><span class="va">markers_gini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                                       method <span class="op">=</span> <span class="st">"gini"</span>,</span>
<span>                                       expression_values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                                       cluster_column <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>,</span>
<span>                                       min_feats <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                                       min_expr_gini_score <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                                       min_det_gini_score <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">topgenes_gini</span> <span class="op">&lt;-</span> <span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## violin plot</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/violinPlot.html" class="external-link">violinPlot</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>           feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_gini</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>, </span>
<span>           cluster_column <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>,</span>
<span>           strip_text <span class="op">=</span> <span class="fl">8</span>, </span>
<span>           strip_position <span class="op">=</span> <span class="st">"right"</span>, </span>
<span>           cluster_custom_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_gini</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F12-violinPlot.png"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cluster heatmap</span></span>
<span><span class="va">topgenes_gini</span> <span class="op">&lt;-</span> <span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">6</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotMetaDataHeatmap.html" class="external-link">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_gini</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>,</span>
<span>                    custom_feat_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_gini</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>,</span>
<span>                    custom_cluster_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_gini</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>, </span>
<span>                    x_text_size <span class="op">=</span> <span class="fl">10</span>, y_text_size <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F13-plotMetaDataHeatmap.png"></p>
</div>
<div class="section level2" number="10">
<h2 number="10" id="cell-type-annotation">
<span class="header-section-number">10</span> Cell Type Annotation<a class="anchor" aria-label="anchor" href="#cell-type-annotation"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## general cell types</span></span>
<span><span class="co">## create vector with names</span></span>
<span><span class="va">clusters_cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1.1"</span> <span class="op">=</span> <span class="st">"L4 eNeuron"</span>,</span>
<span>                         <span class="st">"2.1"</span> <span class="op">=</span> <span class="st">"L6 eNeuron"</span>,</span>
<span>                         <span class="st">"3.1"</span> <span class="op">=</span> <span class="st">"L5 eNeuron"</span>, </span>
<span>                         <span class="st">"4.1"</span> <span class="op">=</span> <span class="st">"endothelial"</span>,</span>
<span>                         <span class="st">"5.1"</span> <span class="op">=</span> <span class="st">"Lhx6 iNeuron"</span>, </span>
<span>                         <span class="st">"5.2"</span> <span class="op">=</span> <span class="st">"Adarb2 iNeuron"</span>,</span>
<span>                         <span class="st">"6.1"</span> <span class="op">=</span> <span class="st">"Olig"</span>,</span>
<span>                         <span class="st">"6.2"</span> <span class="op">=</span> <span class="st">"OPC"</span>,</span>
<span>                         <span class="st">"7.1"</span> <span class="op">=</span> <span class="st">"astrocytes"</span>,</span>
<span>                         <span class="st">"8.1"</span> <span class="op">=</span> <span class="st">"microglia"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/annotateGiotto.html" class="external-link">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                             annotation_vector <span class="op">=</span> <span class="va">clusters_cell_types</span>,</span>
<span>                             cluster_column <span class="op">=</span> <span class="st">"sub_leiden_clus_select"</span>,</span>
<span>                             name <span class="op">=</span> <span class="st">"cell_types"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># cell type order and colors</span></span>
<span><span class="va">cell_type_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"L4 eNeuron"</span>, <span class="st">"L5 eNeuron"</span>, <span class="st">"L6 eNeuron"</span>,</span>
<span>    <span class="st">"astrocytes"</span>, <span class="st">"Olig"</span>, <span class="st">"OPC"</span>, <span class="st">"Adarb2 iNeuron"</span>, <span class="st">"Lhx6 iNeuron"</span>,</span>
<span>    <span class="st">"endothelial"</span>, <span class="st">"microglia"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_type_colors</span> <span class="op">&lt;-</span> <span class="va">giotto_colors</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cell_type_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">clusters_cell_types</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">giotto_colors</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">cell_type_colors</span> <span class="op">&lt;-</span> <span class="va">cell_type_colors</span><span class="op">[</span><span class="va">cell_type_order</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## violin plot</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/violinPlot.html" class="external-link">violinPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>           feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_gini</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>,</span>
<span>           strip_text <span class="op">=</span> <span class="fl">7</span>, </span>
<span>           strip_position <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>           cluster_custom_order <span class="op">=</span> <span class="va">cell_type_order</span>,</span>
<span>           cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>, </span>
<span>           color_violin <span class="op">=</span> <span class="st">"cluster"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F14-violinPlot.png"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># co-visualization</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html" class="external-link">spatDimPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>            dim_point_size <span class="op">=</span> <span class="fl">2</span>, </span>
<span>            spat_point_size <span class="op">=</span> <span class="fl">2</span>, </span>
<span>            dim_show_cluster_center <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            dim_show_center_label <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F15-spatDimPlot2D.png"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## heatmap genes vs cells</span></span>
<span><span class="va">markers_gini</span><span class="op">[</span>, <span class="va">cell_types</span> <span class="op">:=</span> <span class="va">clusters_cell_types</span><span class="op">[</span><span class="va">cluster</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">markers_gini</span><span class="op">[</span>, <span class="va">cell_types</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cell_types</span>, <span class="va">cell_type_order</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorder</a></span><span class="op">(</span><span class="va">markers_gini</span>, <span class="va">cell_types</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotHeatmap.html" class="external-link">plotHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>            feats <span class="op">=</span> <span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">3</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cell_types"</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,</span>
<span>            feat_order <span class="op">=</span> <span class="st">"custom"</span>,</span>
<span>            feat_custom_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">3</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>,</span>
<span>            cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>, cluster_order <span class="op">=</span> <span class="st">"custom"</span>,</span>
<span>            cluster_custom_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">3</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cell_types"</span><span class="op">]</span><span class="op">$</span><span class="va">cell_types</span><span class="op">)</span>,</span>
<span>            legend_nrows <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F16-plotHeatmap.png"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotHeatmap.html" class="external-link">plotHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>            feats <span class="op">=</span> <span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">6</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cell_types"</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,</span>
<span>            feat_order <span class="op">=</span> <span class="st">"custom"</span>,</span>
<span>            feat_label_selection <span class="op">=</span> <span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,</span>
<span>            feat_custom_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">6</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>,</span>
<span>            cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>, cluster_order <span class="op">=</span> <span class="st">"custom"</span>,</span>
<span>            cluster_custom_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers_gini</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">3</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cell_types"</span><span class="op">]</span><span class="op">$</span><span class="va">cell_types</span><span class="op">)</span>,</span>
<span>            legend_nrows <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F17-plotHeatmap.png"></p>
</div>
<div class="section level2" number="11">
<h2 number="11" id="spatial-grid">
<span class="header-section-number">11</span> Spatial Grid<a class="anchor" aria-label="anchor" href="#spatial-grid"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialGrid.html" class="external-link">createSpatialGrid</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                                sdimx_stepsize <span class="op">=</span> <span class="fl">500</span>,</span>
<span>                                sdimy_stepsize <span class="op">=</span> <span class="fl">500</span>,</span>
<span>                                minimum_padding <span class="op">=</span> <span class="fl">50</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>         show_grid <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F18-spatPlot2D.png"></p>
</div>
<div class="section level2" number="12">
<h2 number="12" id="spatial-network">
<span class="header-section-number">12</span> Spatial Network<a class="anchor" aria-label="anchor" href="#spatial-network"></a>
</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## delaunay network: stats + creation</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotStatDelaunayNetwork.html" class="external-link">plotStatDelaunayNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                        maximum_distance <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F19-plotStatDelaunayNetwork.png"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html" class="external-link">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                                   minimum_k <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                                   maximum_distance_delaunay <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## create spatial networks based on k and/or distance from centroid</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html" class="external-link">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                                   method <span class="op">=</span> <span class="st">"kNN"</span>, </span>
<span>                                   k <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                                   name <span class="op">=</span> <span class="st">"spatial_network"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html" class="external-link">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                                   method <span class="op">=</span> <span class="st">"kNN"</span>, </span>
<span>                                   k <span class="op">=</span> <span class="fl">10</span>, </span>
<span>                                   name <span class="op">=</span> <span class="st">"large_network"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html" class="external-link">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                                   method <span class="op">=</span> <span class="st">"kNN"</span>, </span>
<span>                                   k <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                                   maximum_distance_knn <span class="op">=</span> <span class="fl">200</span>, </span>
<span>                                   minimum_k <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                                   name <span class="op">=</span> <span class="st">"distance_network"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## visualize different spatial networks on first field (~ layer 1)</span></span>
<span><span class="va">cell_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">SS_seqfish</span><span class="op">)</span></span>
<span><span class="va">field1_ids</span> <span class="op">&lt;-</span> <span class="va">cell_metadata</span><span class="op">[</span><span class="va">FOV</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span></span>
<span></span>
<span><span class="va">subSS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html" class="external-link">subsetGiotto</a></span><span class="op">(</span><span class="va">SS_seqfish</span>, </span>
<span>                              cell_ids <span class="op">=</span> <span class="va">field1_ids</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subSS_seqfish</span>, </span>
<span>         show_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">"blue"</span>, </span>
<span>         spatial_network_name <span class="op">=</span> <span class="st">"Delaunay_network"</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">"cell_types"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F20-spatPlot2D.png"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subSS_seqfish</span>, </span>
<span>         show_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">"blue"</span>, </span>
<span>         spatial_network_name <span class="op">=</span> <span class="st">"spatial_network"</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">"cell_types"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F21-spatPlot2D.png"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subSS_seqfish</span>, </span>
<span>         show_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">"blue"</span>, </span>
<span>         spatial_network_name <span class="op">=</span> <span class="st">"large_network"</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">"cell_types"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F22-spatPlot2D.png"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subSS_seqfish</span>, </span>
<span>         show_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">"blue"</span>, </span>
<span>         spatial_network_name <span class="op">=</span> <span class="st">"distance_network"</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">"cell_types"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F23-spatPlot2D.png"></p>
</div>
<div class="section level2" number="13">
<h2 number="13" id="spatial-genes">
<span class="header-section-number">13</span> Spatial Genes<a class="anchor" aria-label="anchor" href="#spatial-genes"></a>
</h2>
<div class="section level3" number="13.1">
<h3 number="13.1" id="individual-spatial-genes">
<span class="header-section-number">13.1</span> Individual spatial genes<a class="anchor" aria-label="anchor" href="#individual-spatial-genes"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## identify spatial genes</span></span>
<span><span class="va">km_spatialfeats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">SS_seqfish</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatFeatPlot2D.html" class="external-link">spatFeatPlot2D</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>               expression_values <span class="op">=</span> <span class="st">"scaled"</span>, </span>
<span>               feats <span class="op">=</span> <span class="va">km_spatialfeats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,</span>
<span>               point_shape <span class="op">=</span> <span class="st">"border"</span>, </span>
<span>               point_border_stroke <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>               show_network <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>               network_color <span class="op">=</span> <span class="st">"lightgrey"</span>, </span>
<span>               point_size <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F24-spatFeatPlot2D.png"></p>
</div>
<div class="section level3" number="13.2">
<h3 number="13.2" id="spatial-genes-co-expression-modules">
<span class="header-section-number">13.2</span> Spatial Genes Co-Expression Modules<a class="anchor" aria-label="anchor" href="#spatial-genes-co-expression-modules"></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## spatial co-expression patterns ##</span></span>
<span><span class="va">ext_spatial_genes</span> <span class="op">&lt;-</span> <span class="va">km_spatialfeats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span></span>
<span><span class="co">## 1. calculate gene spatial correlation and single-cell correlation</span></span>
<span><span class="co">## create spatial correlation object</span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/detectSpatialCorFeats.html">detectSpatialCorFeats</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                                          method <span class="op">=</span> <span class="st">"network"</span>,</span>
<span>                                          spatial_network_name <span class="op">=</span> <span class="st">"Delaunay_network"</span>,</span>
<span>                                          subset_feats <span class="op">=</span> <span class="va">ext_spatial_genes</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 2. cluster correlated genes &amp; visualize</span></span>
<span><span class="va">spat_cor_netw_DT</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clusterSpatialCorFeats.html">clusterSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>,</span>
<span>                                           name <span class="op">=</span> <span class="st">"spat_netw_clus"</span>,</span>
<span>                                           k <span class="op">=</span> <span class="fl">8</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/heatmSpatialCorFeats.html">heatmSpatialCorFeats</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                     spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>, </span>
<span>                     use_clus_name <span class="op">=</span> <span class="st">"spat_netw_clus"</span>,</span>
<span>                     heatmap_legend_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>title <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F25-heatmSpatialCorFeats.png"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3. rank spatial correlated clusters and show genes for selected clusters</span></span>
<span><span class="va">netw_ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rankSpatialCorGroups.html">rankSpatialCorGroups</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                                   spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>,</span>
<span>                                   use_clus_name <span class="op">=</span> <span class="st">"spat_netw_clus"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">top_netw_spat_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/showSpatialCorFeats.html">showSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>,</span>
<span>                                             use_clus_name <span class="op">=</span> <span class="st">"spat_netw_clus"</span>,</span>
<span>                                             selected_clusters <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                                             show_top_feats <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F26-rankSpatialCorGroups.png"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 4. create metagene enrichment score for clusters</span></span>
<span><span class="va">cluster_genes_DT</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/showSpatialCorFeats.html">showSpatialCorFeats</a></span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>,</span>
<span>                                        use_clus_name <span class="op">=</span> <span class="st">"spat_netw_clus"</span>,</span>
<span>                                        show_top_feats <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_genes</span> <span class="op">&lt;-</span> <span class="va">cluster_genes_DT</span><span class="op">$</span><span class="va">clus</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cluster_genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">cluster_genes_DT</span><span class="op">$</span><span class="va">feat_ID</span></span>
<span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createMetafeats.html" class="external-link">createMetafeats</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                              feat_clusters <span class="op">=</span> <span class="va">cluster_genes</span>,</span>
<span>                              name <span class="op">=</span> <span class="st">"cluster_metagene"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html" class="external-link">spatCellPlot</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>             spat_enr_names <span class="op">=</span> <span class="st">"cluster_metagene"</span>,</span>
<span>             cell_annotation_values <span class="op">=</span> <span class="va">netw_ranks</span><span class="op">$</span><span class="va">clusters</span>,</span>
<span>             point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>             cow_n_col <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F27-spatCellPlot2D.png"></p>
</div>
</div>
<div class="section level2" number="14">
<h2 number="14" id="hmrf-spatial-domains">
<span class="header-section-number">14</span> HMRF Spatial Domains<a class="anchor" aria-label="anchor" href="#hmrf-spatial-domains"></a>
</h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hmrf_folder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">results_folder</span>, <span class="st">"/11_HMRF/"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">hmrf_folder</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">hmrf_folder</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">my_spatial_genes</span> <span class="op">&lt;-</span> <span class="va">km_spatialfeats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span></span>
<span><span class="co"># do HMRF with different betas</span></span>
<span><span class="va">HMRF_spatial_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doHMRF.html">doHMRF</a></span><span class="op">(</span></span>
<span>    gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>    expression_values <span class="op">=</span> <span class="st">"scaled"</span>,</span>
<span>    spatial_genes <span class="op">=</span> <span class="va">my_spatial_genes</span>,</span>
<span>    spatial_network_name <span class="op">=</span> <span class="st">"Delaunay_network"</span>,</span>
<span>    k <span class="op">=</span> <span class="fl">9</span>,</span>
<span>    betas <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">28</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>    output_folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">hmrf_folder</span>, <span class="st">"/Spatial_genes/SG_top100_k9_scaled"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## view results of HMRF</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">28</span>, <span class="fl">32</span>, by <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/viewHMRFresults2D.html">viewHMRFresults2D</a></span><span class="op">(</span></span>
<span>        gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>        HMRFoutput <span class="op">=</span> <span class="va">HMRF_spatial_genes</span>,</span>
<span>        k <span class="op">=</span> <span class="fl">9</span>, </span>
<span>        betas_to_view <span class="op">=</span> <span class="va">i</span>,</span>
<span>        point_size <span class="op">=</span> <span class="fl">2</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## add HMRF of interest to giotto object</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addHMRF.html">addHMRF</a></span><span class="op">(</span></span>
<span>    gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>    HMRFoutput <span class="op">=</span> <span class="va">HMRF_spatial_genes</span>,</span>
<span>    k <span class="op">=</span> <span class="fl">9</span>, </span>
<span>    betas_to_add <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">28</span><span class="op">)</span>,</span>
<span>    hmrf_name <span class="op">=</span> <span class="st">"HMRF_2"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## visualize</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span></span>
<span>    gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>    cell_color <span class="op">=</span> <span class="st">"HMRF_2_k9_b.28"</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    coord_fix_ratio <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F28-spatPlot2D.png"></p>
</div>
<div class="section level2" number="15">
<h2 number="15" id="cell-neighborhood-cell-typecell-type-interactions">
<span class="header-section-number">15</span> Cell Neighborhood: Cell-Type/Cell-Type Interactions<a class="anchor" aria-label="anchor" href="#cell-neighborhood-cell-typecell-type-interactions"></a>
</h2>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_proximities</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cellProximityEnrichment.html">cellProximityEnrichment</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                                            cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>                                            spatial_network_name <span class="op">=</span> <span class="st">"Delaunay_network"</span>,</span>
<span>                                            adjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>                                            number_of_simulations <span class="op">=</span> <span class="fl">2000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## barplot</span></span>
<span><span class="fu"><a href="../reference/cellProximityBarplot.html">cellProximityBarplot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>,</span>
<span>                     min_orig_ints <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                     min_sim_ints <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F29-cellProximityBarplot.png"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## heatmap</span></span>
<span><span class="fu"><a href="../reference/cellProximityHeatmap.html">cellProximityHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>,</span>
<span>                     order_cell_types <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     color_breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1.5</span>, <span class="fl">0</span>, <span class="fl">1.5</span><span class="op">)</span>,</span>
<span>                     color_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F30-cellProximityHeatmap.png"></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## network</span></span>
<span><span class="fu"><a href="../reference/cellProximityNetwork.html">cellProximityNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>, </span>
<span>                     remove_self_edges <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     only_show_enrichment_edges <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F31-cellProximityNetwork.png"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## network with self-edges</span></span>
<span><span class="fu"><a href="../reference/cellProximityNetwork.html">cellProximityNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>, </span>
<span>                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>,</span>
<span>                     remove_self_edges <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                     self_loop_strength <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>                     only_show_enrichment_edges <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     rescale_edge_weights <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     node_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                     edge_weight_range_depletion <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>                     edge_weight_range_enrichment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">5</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F32-cellProximityNetwork.png"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## visualization of specific cell types</span></span>
<span><span class="co"># Option 1</span></span>
<span><span class="va">spec_interaction</span> <span class="op">&lt;-</span> <span class="st">"astrocytes--Olig"</span></span>
<span></span>
<span><span class="fu"><a href="../reference/cellProximitySpatPlot2D.html">cellProximitySpatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                        interaction_name <span class="op">=</span> <span class="va">spec_interaction</span>,</span>
<span>                        show_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                        cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>                        cell_color <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>                        cell_color_code <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>astrocytes <span class="op">=</span> <span class="st">"lightblue"</span>, Olig <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span>,</span>
<span>                        point_size_select <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                        point_size_other <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F33-cellProximitySpatPlot2D.png"></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 2: create additional metadata</span></span>
<span><span class="va">SS_seqfish</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCellIntMetadata.html">addCellIntMetadata</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                                 spatial_network <span class="op">=</span> <span class="st">"spatial_network"</span>,</span>
<span>                                 cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>                                 cell_interaction <span class="op">=</span> <span class="va">spec_interaction</span>,</span>
<span>                                 name <span class="op">=</span> <span class="st">"astro_olig_ints"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">"astro_olig_ints"</span>,</span>
<span>         select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"other_astrocytes"</span>, <span class="st">"other_Olig"</span>, <span class="st">"select_astrocytes"</span>, <span class="st">"select_Olig"</span><span class="op">)</span>,</span>
<span>         legend_symbol_size <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F34-spatPlot2D.png"></p>
</div>
<div class="section level2" number="16">
<h2 number="16" id="cell-neighborhood-interaction-changed-features">
<span class="header-section-number">16</span> Cell Neighborhood: Interaction Changed Features<a class="anchor" aria-label="anchor" href="#cell-neighborhood-interaction-changed-features"></a>
</h2>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true"></a><span class="kw">library</span>(future)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true"></a><span class="co">## select top 25th highest expressing genes</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true"></a>gene_metadata &lt;-<span class="st"> </span><span class="kw">fDataDT</span>(SS_seqfish)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true"></a><span class="kw">plot</span>(gene_metadata<span class="op">$</span>nr_cells, gene_metadata<span class="op">$</span>mean_expr)</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true"></a></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true"></a><span class="kw">plot</span>(gene_metadata<span class="op">$</span>nr_cells, gene_metadata<span class="op">$</span>mean_expr_det)</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true"></a></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true"></a><span class="kw">quantile</span>(gene_metadata<span class="op">$</span>mean_expr_det)</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true"></a></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true"></a>high_expressed_genes &lt;-<span class="st"> </span>gene_metadata[mean_expr_det <span class="op">&gt;</span><span class="st"> </span><span class="fl">3.5</span>]<span class="op">$</span>feat_ID</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true"></a></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true"></a><span class="co">## identify genes that are associated with proximity to other cell types</span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true"></a><span class="kw">plan</span>(<span class="st">"multisession"</span>, <span class="dt">workers =</span> <span class="dv">6</span>)</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true"></a></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true"></a>ICFsForesHighGenes &lt;-<span class="st"> </span><span class="kw">findInteractionChangedFeats</span>(</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true"></a>    <span class="dt">gobject =</span> SS_seqfish,</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true"></a>    <span class="dt">selected_feats =</span> high_expressed_genes,</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true"></a>    <span class="dt">spatial_network_name =</span> <span class="st">"Delaunay_network"</span>,</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true"></a>    <span class="dt">cluster_column =</span> <span class="st">"cell_types"</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true"></a>    <span class="dt">diff_test =</span> <span class="st">"permutation"</span>,</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true"></a>    <span class="dt">adjust_method =</span> <span class="st">"fdr"</span>,</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true"></a>    <span class="dt">nr_permutations =</span> <span class="dv">2000</span>,</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true"></a>    <span class="dt">do_parallel =</span> <span class="ot">TRUE</span></span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true"></a>)</span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true"></a></span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true"></a><span class="co">## visualize all genes</span></span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true"></a><span class="kw">plotCellProximityFeats</span>(SS_seqfish,</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true"></a>                       <span class="dt">icfObject =</span> ICFsForesHighGenes,</span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true"></a>                       <span class="dt">method =</span> <span class="st">"dotplot"</span>)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## filter genes</span></span>
<span><span class="va">ICFscoresFilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterInteractionChangedFeats.html">filterICF</a></span><span class="op">(</span><span class="va">ICFsForesHighGenes</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## visualize subset of interaction changed genes (ICFs)</span></span>
<span><span class="va">ICF_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jakmip1"</span>, <span class="st">"Golgb1"</span>, <span class="st">"Dact2"</span>, <span class="st">"Ddx27"</span>, <span class="st">"Abl1"</span>, <span class="st">"Zswim8"</span><span class="op">)</span></span>
<span><span class="va">ICF_genes_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lhx6 iNeuron"</span>, <span class="st">"Lhx6 iNeuron"</span>, <span class="st">"L4 eNeuron"</span>, <span class="st">"L4 eNeuron"</span>, <span class="st">"astrocytes"</span>, <span class="st">"astrocytes"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ICF_genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">ICF_genes_types</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotICF.html">plotICF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>        icfObject <span class="op">=</span> <span class="va">ICFsForesHighGenes</span>,</span>
<span>        source_type <span class="op">=</span> <span class="st">"endothelial"</span>,</span>
<span>        source_markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pltp"</span>, <span class="st">"Cldn5"</span>, <span class="st">"Apcdd1"</span><span class="op">)</span>,</span>
<span>        ICF_feats <span class="op">=</span> <span class="va">ICF_genes</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="17">
<h2 number="17" id="cell-neighborhood-ligand-receptor-cell-cell-communication">
<span class="header-section-number">17</span> Cell Neighborhood: Ligand-Receptor Cell-Cell Communication<a class="anchor" aria-label="anchor" href="#cell-neighborhood-ligand-receptor-cell-cell-communication"></a>
</h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## LR expression</span></span>
<span><span class="co">## LR activity changes</span></span>
<span><span class="va">LR_data</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mouse_ligand_receptors.txt"</span>, package <span class="op">=</span> <span class="st">"GiottoData"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LR_data</span><span class="op">[</span>, <span class="va">ligand_det</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">LR_data</span><span class="op">$</span><span class="va">mouseLigand</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">SS_seqfish</span><span class="op">@</span><span class="va">feat_ID</span><span class="op">$</span><span class="va">rna</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">LR_data</span><span class="op">[</span>, <span class="va">receptor_det</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">LR_data</span><span class="op">$</span><span class="va">mouseReceptor</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">SS_seqfish</span><span class="op">@</span><span class="va">feat_ID</span><span class="op">$</span><span class="va">rna</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">LR_data_det</span> <span class="op">&lt;-</span> <span class="va">LR_data</span><span class="op">[</span><span class="va">ligand_det</span> <span class="op">==</span> <span class="cn">TRUE</span> <span class="op">&amp;</span> <span class="va">receptor_det</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">]</span></span>
<span></span>
<span><span class="va">select_ligands</span> <span class="op">&lt;-</span> <span class="va">LR_data_det</span><span class="op">$</span><span class="va">mouseLigand</span></span>
<span></span>
<span><span class="va">select_receptors</span> <span class="op">&lt;-</span> <span class="va">LR_data_det</span><span class="op">$</span><span class="va">mouseReceptor</span></span>
<span></span>
<span><span class="co">## get statistical significance of gene pair expression changes based on expression</span></span>
<span><span class="va">expr_only_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exprCellCellcom.html">exprCellCellcom</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                                    cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>                                    random_iter <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>                                    feat_set_1 <span class="op">=</span> <span class="va">select_ligands</span>,</span>
<span>                                    feat_set_2 <span class="op">=</span> <span class="va">select_receptors</span>,</span>
<span>                                    verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## get statistical significance of gene pair expression changes upon cell-cell interaction</span></span>
<span><span class="va">spatial_all_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatCellCellcom.html">spatCellCellcom</a></span><span class="op">(</span><span class="va">SS_seqfish</span>,</span>
<span>                                      spatial_network_name <span class="op">=</span> <span class="st">"spatial_network"</span>,</span>
<span>                                      cluster_column <span class="op">=</span> <span class="st">"cell_types"</span>,</span>
<span>                                      random_iter <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>                                      feat_set_1 <span class="op">=</span> <span class="va">select_ligands</span>,</span>
<span>                                      feat_set_2 <span class="op">=</span> <span class="va">select_receptors</span>,</span>
<span>                                      adjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>                                      do_parallel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                      cores <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                                      verbose <span class="op">=</span> <span class="st">"a little"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## select top LR ##</span></span>
<span><span class="va">selected_spat</span> <span class="op">&lt;-</span> <span class="va">spatial_all_scores</span><span class="op">[</span><span class="va">p.adj</span> <span class="op">&lt;=</span> <span class="fl">0.01</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">log2fc</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.25</span> <span class="op">&amp;</span> <span class="va">lig_nr</span> <span class="op">&gt;=</span> <span class="fl">4</span> <span class="op">&amp;</span> <span class="va">rec_nr</span> <span class="op">&gt;=</span> <span class="fl">4</span><span class="op">]</span></span>
<span></span>
<span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorder</a></span><span class="op">(</span><span class="va">selected_spat</span>, <span class="op">-</span><span class="va">PI</span><span class="op">)</span></span>
<span></span>
<span><span class="va">top_LR_ints</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">selected_spat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">PI</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">$</span><span class="va">LR_comb</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">33</span><span class="op">]</span></span>
<span></span>
<span><span class="va">top_LR_cell_ints</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">selected_spat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">PI</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">$</span><span class="va">LR_cell_comb</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">33</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotCCcomDotplot.html">plotCCcomDotplot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                 comScores <span class="op">=</span> <span class="va">spatial_all_scores</span>,</span>
<span>                 selected_LR <span class="op">=</span> <span class="va">top_LR_ints</span>,</span>
<span>                 selected_cell_LR <span class="op">=</span> <span class="va">top_LR_cell_ints</span>,</span>
<span>                 cluster_on <span class="op">=</span> <span class="st">"PI"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F37-plotCCcomDotplot.png"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## spatial vs rank ####</span></span>
<span><span class="va">comb_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combCCcom.html">combCCcom</a></span><span class="op">(</span></span>
<span>    spatialCC <span class="op">=</span> <span class="va">spatial_all_scores</span>,</span>
<span>    exprCC <span class="op">=</span> <span class="va">expr_only_scores</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## highest levels of ligand and receptor prediction</span></span>
<span><span class="co">## top differential activity levels for ligand receptor pairs</span></span>
<span><span class="fu"><a href="../reference/plotRankSpatvsExpr.html">plotRankSpatvsExpr</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>                   <span class="va">comb_comm</span>,</span>
<span>                   expr_rnk_column <span class="op">=</span> <span class="st">"LR_expr_rnk"</span>,</span>
<span>                   spat_rnk_column <span class="op">=</span> <span class="st">"LR_spat_rnk"</span>,</span>
<span>                   gradient_midpoint <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F38-plotRankSpatvsExpr.png"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## recovery</span></span>
<span><span class="fu"><a href="../reference/plotRecovery.html">plotRecovery</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">SS_seqfish</span>,</span>
<span>             <span class="va">comb_comm</span>,</span>
<span>             expr_rnk_column <span class="op">=</span> <span class="st">"LR_expr_rnk"</span>,</span>
<span>             spat_rnk_column <span class="op">=</span> <span class="st">"LR_spat_rnk"</span>,</span>
<span>             ground_truth <span class="op">=</span> <span class="st">"spatial"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqfish_cortex%2F39-plotRecovery.png"></p>
</div>
<div class="section level2" number="18">
<h2 number="18" id="session-info">
<span class="header-section-number">18</span> Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true"></a>R version <span class="dv">4</span>.<span class="fl">4.1</span> (<span class="dv">2024-06-14</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true"></a>Platform<span class="op">:</span><span class="st"> </span>x86_<span class="dv">64</span><span class="op">-</span>apple<span class="op">-</span>darwin20</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true"></a>Running under<span class="op">:</span><span class="st"> </span>macOS Sonoma <span class="dv">14</span>.<span class="fl">6.1</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true"></a></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true"></a>Matrix products<span class="op">:</span><span class="st"> </span>default</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true"></a>BLAS<span class="op">:</span><span class="st">   </span><span class="er">/</span>System<span class="op">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>Accelerate.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>Frameworks<span class="op">/</span>vecLib.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>libBLAS.dylib </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true"></a>LAPACK<span class="op">:</span><span class="st"> </span><span class="er">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>R.framework<span class="op">/</span>Versions<span class="op">/</span><span class="fl">4.4</span><span class="op">-</span>x86_<span class="dv">64</span><span class="op">/</span>Resources<span class="op">/</span>lib<span class="op">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true"></a></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true"></a>locale<span class="op">:</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>C<span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true"></a></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true"></a>time zone<span class="op">:</span><span class="st"> </span>America<span class="op">/</span>New_York</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true"></a>tzcode source<span class="op">:</span><span class="st"> </span>internal</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true"></a></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true"></a>attached base packages<span class="op">:</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true"></a></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true"></a>other attached packages<span class="op">:</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true"></a>[<span class="dv">1</span>] future_<span class="dv">1</span>.<span class="fl">34.0</span>     ggdendro_<span class="dv">0</span>.<span class="fl">2.0</span>    Giotto_<span class="dv">4</span>.<span class="fl">1.3</span>      GiottoClass_<span class="dv">0</span>.<span class="fl">4.0</span></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true"></a></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true"></a>loaded via a <span class="kw">namespace</span> (and not attached)<span class="op">:</span></span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true"></a><span class="st">  </span>[<span class="dv">1</span>] RColorBrewer_<span class="fl">1.1</span><span class="dv">-3</span>          rstudioapi_<span class="dv">0</span>.<span class="fl">16.0</span>          </span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true"></a>  [<span class="dv">3</span>] jsonlite_<span class="dv">1</span>.<span class="fl">8.8</span>              shape_<span class="dv">1</span>.<span class="dv">4</span>.<span class="fl">6.1</span>              </span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true"></a>  [<span class="dv">5</span>] magrittr_<span class="dv">2</span>.<span class="fl">0.3</span>              magick_<span class="dv">2</span>.<span class="fl">8.4</span>               </span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true"></a>  [<span class="dv">7</span>] farver_<span class="dv">2</span>.<span class="fl">1.2</span>                rmarkdown_<span class="fl">2.28</span>             </span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true"></a>  [<span class="dv">9</span>] GlobalOptions_<span class="dv">0</span>.<span class="fl">1.2</span>         zlibbioc_<span class="dv">1</span>.<span class="fl">50.0</span>            </span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true"></a> [<span class="dv">11</span>] ragg_<span class="dv">1</span>.<span class="fl">3.2</span>                  vctrs_<span class="dv">0</span>.<span class="fl">6.5</span>                </span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true"></a> [<span class="dv">13</span>] memoise_<span class="dv">2</span>.<span class="fl">0.1</span>               Cairo_<span class="fl">1.6</span><span class="dv">-2</span>                </span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true"></a> [<span class="dv">15</span>] GiottoUtils_<span class="dv">0</span>.<span class="fl">1.12</span>          terra_<span class="fl">1.7</span><span class="dv">-78</span>               </span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true"></a> [<span class="dv">17</span>] htmltools_<span class="dv">0</span>.<span class="dv">5</span>.<span class="fl">8.1</span>           S4Arrays_<span class="dv">1</span>.<span class="fl">4.1</span>             </span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true"></a> [<span class="dv">19</span>] SparseArray_<span class="dv">1</span>.<span class="fl">4.8</span>           parallelly_<span class="dv">1</span>.<span class="fl">38.0</span>          </span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true"></a> [<span class="dv">21</span>] htmlwidgets_<span class="dv">1</span>.<span class="fl">6.4</span>           cachem_<span class="dv">1</span>.<span class="fl">1.0</span>               </span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true"></a> [<span class="dv">23</span>] plotly_<span class="dv">4</span>.<span class="fl">10.4</span>               igraph_<span class="dv">2</span>.<span class="fl">0.3</span>               </span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true"></a> [<span class="dv">25</span>] lifecycle_<span class="dv">1</span>.<span class="fl">0.4</span>             iterators_<span class="dv">1</span>.<span class="fl">0.14</span>           </span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true"></a> [<span class="dv">27</span>] pkgconfig_<span class="dv">2</span>.<span class="fl">0.3</span>             rsvd_<span class="dv">1</span>.<span class="fl">0.5</span>                 </span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true"></a> [<span class="dv">29</span>] Matrix_<span class="fl">1.7</span><span class="dv">-0</span>                R6_<span class="dv">2</span>.<span class="fl">5.1</span>                   </span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true"></a> [<span class="dv">31</span>] fastmap_<span class="dv">1</span>.<span class="fl">2.0</span>               GenomeInfoDbData_<span class="dv">1</span>.<span class="fl">2.12</span>    </span>
<span id="cb45-38"><a href="#cb45-38" aria-hidden="true"></a> [<span class="dv">33</span>] MatrixGenerics_<span class="dv">1</span>.<span class="fl">16.0</span>       clue_<span class="fl">0.3</span><span class="dv">-65</span>                </span>
<span id="cb45-39"><a href="#cb45-39" aria-hidden="true"></a> [<span class="dv">35</span>] digest_<span class="dv">0</span>.<span class="fl">6.37</span>               colorspace_<span class="fl">2.1</span><span class="dv">-1</span>           </span>
<span id="cb45-40"><a href="#cb45-40" aria-hidden="true"></a> [<span class="dv">37</span>] S4Vectors_<span class="dv">0</span>.<span class="fl">42.1</span>            irlba_<span class="dv">2</span>.<span class="dv">3</span>.<span class="fl">5.1</span>              </span>
<span id="cb45-41"><a href="#cb45-41" aria-hidden="true"></a> [<span class="dv">39</span>] textshaping_<span class="dv">0</span>.<span class="fl">4.0</span>           GenomicRanges_<span class="dv">1</span>.<span class="fl">56.1</span>       </span>
<span id="cb45-42"><a href="#cb45-42" aria-hidden="true"></a> [<span class="dv">41</span>] beachmat_<span class="dv">2</span>.<span class="fl">20.0</span>             labeling_<span class="dv">0</span>.<span class="fl">4.3</span>             </span>
<span id="cb45-43"><a href="#cb45-43" aria-hidden="true"></a> [<span class="dv">43</span>] progressr_<span class="dv">0</span>.<span class="fl">14.0</span>            fansi_<span class="dv">1</span>.<span class="fl">0.6</span>                </span>
<span id="cb45-44"><a href="#cb45-44" aria-hidden="true"></a> [<span class="dv">45</span>] polyclip_<span class="fl">1.10</span><span class="dv">-7</span>             httr_<span class="dv">1</span>.<span class="fl">4.7</span>                 </span>
<span id="cb45-45"><a href="#cb45-45" aria-hidden="true"></a> [<span class="dv">47</span>] abind_<span class="fl">1.4</span><span class="dv">-5</span>                 compiler_<span class="dv">4</span>.<span class="fl">4.1</span>             </span>
<span id="cb45-46"><a href="#cb45-46" aria-hidden="true"></a> [<span class="dv">49</span>] withr_<span class="dv">3</span>.<span class="fl">0.1</span>                 doParallel_<span class="dv">1</span>.<span class="fl">0.17</span>          </span>
<span id="cb45-47"><a href="#cb45-47" aria-hidden="true"></a> [<span class="dv">51</span>] backports_<span class="dv">1</span>.<span class="fl">5.0</span>             BiocParallel_<span class="dv">1</span>.<span class="fl">38.0</span>        </span>
<span id="cb45-48"><a href="#cb45-48" aria-hidden="true"></a> [<span class="dv">53</span>] viridis_<span class="dv">0</span>.<span class="fl">6.5</span>               ggforce_<span class="dv">0</span>.<span class="fl">4.2</span>              </span>
<span id="cb45-49"><a href="#cb45-49" aria-hidden="true"></a> [<span class="dv">55</span>] R.utils_<span class="dv">2</span>.<span class="fl">12.3</span>              MASS_<span class="fl">7.3-60.2</span>              </span>
<span id="cb45-50"><a href="#cb45-50" aria-hidden="true"></a> [<span class="dv">57</span>] DelayedArray_<span class="dv">0</span>.<span class="fl">30.1</span>         rjson_<span class="dv">0</span>.<span class="fl">2.22</span>               </span>
<span id="cb45-51"><a href="#cb45-51" aria-hidden="true"></a> [<span class="dv">59</span>] gtools_<span class="dv">3</span>.<span class="fl">9.5</span>                GiottoVisuals_<span class="dv">0</span>.<span class="fl">2.5</span>        </span>
<span id="cb45-52"><a href="#cb45-52" aria-hidden="true"></a> [<span class="dv">61</span>] tools_<span class="dv">4</span>.<span class="fl">4.1</span>                 future.apply_<span class="dv">1</span>.<span class="fl">11.2</span>        </span>
<span id="cb45-53"><a href="#cb45-53" aria-hidden="true"></a> [<span class="dv">63</span>] R.oo_<span class="dv">1</span>.<span class="fl">26.0</span>                 glue_<span class="dv">1</span>.<span class="fl">7.0</span>                 </span>
<span id="cb45-54"><a href="#cb45-54" aria-hidden="true"></a> [<span class="dv">65</span>] dbscan_<span class="fl">1.2</span><span class="dv">-0</span>                grid_<span class="dv">4</span>.<span class="fl">4.1</span>                 </span>
<span id="cb45-55"><a href="#cb45-55" aria-hidden="true"></a> [<span class="dv">67</span>] checkmate_<span class="dv">2</span>.<span class="fl">3.2</span>             Rtsne_<span class="fl">0.17</span>                 </span>
<span id="cb45-56"><a href="#cb45-56" aria-hidden="true"></a> [<span class="dv">69</span>] cluster_<span class="dv">2</span>.<span class="fl">1.6</span>               generics_<span class="dv">0</span>.<span class="fl">1.3</span>             </span>
<span id="cb45-57"><a href="#cb45-57" aria-hidden="true"></a> [<span class="dv">71</span>] gtable_<span class="dv">0</span>.<span class="fl">3.5</span>                R.methodsS3_<span class="dv">1</span>.<span class="fl">8.2</span>          </span>
<span id="cb45-58"><a href="#cb45-58" aria-hidden="true"></a> [<span class="dv">73</span>] tidyr_<span class="dv">1</span>.<span class="fl">3.1</span>                 data.table_<span class="dv">1</span>.<span class="fl">16.0</span>          </span>
<span id="cb45-59"><a href="#cb45-59" aria-hidden="true"></a> [<span class="dv">75</span>] tidygraph_<span class="dv">1</span>.<span class="fl">3.1</span>             BiocSingular_<span class="dv">1</span>.<span class="fl">20.0</span>        </span>
<span id="cb45-60"><a href="#cb45-60" aria-hidden="true"></a> [<span class="dv">77</span>] ScaledMatrix_<span class="dv">1</span>.<span class="fl">12.0</span>         sp_<span class="fl">2.1</span><span class="dv">-4</span>                   </span>
<span id="cb45-61"><a href="#cb45-61" aria-hidden="true"></a> [<span class="dv">79</span>] utf8_<span class="dv">1</span>.<span class="fl">2.4</span>                  XVector_<span class="dv">0</span>.<span class="fl">44.0</span>             </span>
<span id="cb45-62"><a href="#cb45-62" aria-hidden="true"></a> [<span class="dv">81</span>] BiocGenerics_<span class="dv">0</span>.<span class="fl">50.0</span>         ggrepel_<span class="dv">0</span>.<span class="fl">9.6</span>              </span>
<span id="cb45-63"><a href="#cb45-63" aria-hidden="true"></a> [<span class="dv">83</span>] foreach_<span class="dv">1</span>.<span class="fl">5.2</span>               pillar_<span class="dv">1</span>.<span class="fl">9.0</span>               </span>
<span id="cb45-64"><a href="#cb45-64" aria-hidden="true"></a> [<span class="dv">85</span>] limma_<span class="dv">3</span>.<span class="fl">60.4</span>                circlize_<span class="dv">0</span>.<span class="fl">4.16</span>            </span>
<span id="cb45-65"><a href="#cb45-65" aria-hidden="true"></a> [<span class="dv">87</span>] tweenr_<span class="dv">2</span>.<span class="fl">0.3</span>                dplyr_<span class="dv">1</span>.<span class="fl">1.4</span>                </span>
<span id="cb45-66"><a href="#cb45-66" aria-hidden="true"></a> [<span class="dv">89</span>] lattice_<span class="fl">0.22</span><span class="dv">-6</span>              FNN_<span class="dv">1</span>.<span class="fl">1.4</span>                  </span>
<span id="cb45-67"><a href="#cb45-67" aria-hidden="true"></a> [<span class="dv">91</span>] deldir_<span class="fl">2.0</span><span class="dv">-4</span>                tidyselect_<span class="dv">1</span>.<span class="fl">2.1</span>           </span>
<span id="cb45-68"><a href="#cb45-68" aria-hidden="true"></a> [<span class="dv">93</span>] ComplexHeatmap_<span class="dv">2</span>.<span class="fl">20.0</span>       SingleCellExperiment_<span class="dv">1</span>.<span class="fl">26.0</span></span>
<span id="cb45-69"><a href="#cb45-69" aria-hidden="true"></a> [<span class="dv">95</span>] knitr_<span class="fl">1.48</span>                  gridExtra_<span class="fl">2.3</span>              </span>
<span id="cb45-70"><a href="#cb45-70" aria-hidden="true"></a> [<span class="dv">97</span>] IRanges_<span class="dv">2</span>.<span class="fl">38.1</span>              SummarizedExperiment_<span class="dv">1</span>.<span class="fl">34.0</span></span>
<span id="cb45-71"><a href="#cb45-71" aria-hidden="true"></a> [<span class="dv">99</span>] scattermore_<span class="fl">1.2</span>             stats4_<span class="dv">4</span>.<span class="fl">4.1</span>               </span>
<span id="cb45-72"><a href="#cb45-72" aria-hidden="true"></a>[<span class="dv">101</span>] xfun_<span class="fl">0.47</span>                   graphlayouts_<span class="dv">1</span>.<span class="fl">1.1</span>         </span>
<span id="cb45-73"><a href="#cb45-73" aria-hidden="true"></a>[<span class="dv">103</span>] Biobase_<span class="dv">2</span>.<span class="fl">64.0</span>              statmod_<span class="dv">1</span>.<span class="fl">5.0</span>              </span>
<span id="cb45-74"><a href="#cb45-74" aria-hidden="true"></a>[<span class="dv">105</span>] matrixStats_<span class="dv">1</span>.<span class="fl">4.1</span>           UCSC.utils_<span class="dv">1</span>.<span class="fl">0.0</span>           </span>
<span id="cb45-75"><a href="#cb45-75" aria-hidden="true"></a>[<span class="dv">107</span>] lazyeval_<span class="dv">0</span>.<span class="fl">2.2</span>              yaml_<span class="dv">2</span>.<span class="fl">3.10</span>                </span>
<span id="cb45-76"><a href="#cb45-76" aria-hidden="true"></a>[<span class="dv">109</span>] evaluate_<span class="dv">0</span>.<span class="fl">24.0</span>             codetools_<span class="fl">0.2</span><span class="dv">-20</span>           </span>
<span id="cb45-77"><a href="#cb45-77" aria-hidden="true"></a>[<span class="dv">111</span>] ggraph_<span class="dv">2</span>.<span class="fl">2.1</span>                GiottoData_<span class="dv">0</span>.<span class="fl">2.13</span>          </span>
<span id="cb45-78"><a href="#cb45-78" aria-hidden="true"></a>[<span class="dv">113</span>] tibble_<span class="dv">3</span>.<span class="fl">2.1</span>                colorRamp2_<span class="dv">0</span>.<span class="fl">1.0</span>           </span>
<span id="cb45-79"><a href="#cb45-79" aria-hidden="true"></a>[<span class="dv">115</span>] cli_<span class="dv">3</span>.<span class="fl">6.3</span>                   uwot_<span class="dv">0</span>.<span class="fl">2.2</span>                 </span>
<span id="cb45-80"><a href="#cb45-80" aria-hidden="true"></a>[<span class="dv">117</span>] reticulate_<span class="dv">1</span>.<span class="fl">39.0</span>           systemfonts_<span class="dv">1</span>.<span class="fl">1.0</span>          </span>
<span id="cb45-81"><a href="#cb45-81" aria-hidden="true"></a>[<span class="dv">119</span>] munsell_<span class="dv">0</span>.<span class="fl">5.1</span>               Rcpp_<span class="dv">1</span>.<span class="fl">0.13</span>                </span>
<span id="cb45-82"><a href="#cb45-82" aria-hidden="true"></a>[<span class="dv">121</span>] GenomeInfoDb_<span class="dv">1</span>.<span class="fl">40.1</span>         globals_<span class="dv">0</span>.<span class="fl">16.3</span>             </span>
<span id="cb45-83"><a href="#cb45-83" aria-hidden="true"></a>[<span class="dv">123</span>] png_<span class="fl">0.1</span><span class="dv">-8</span>                   parallel_<span class="dv">4</span>.<span class="fl">4.1</span>             </span>
<span id="cb45-84"><a href="#cb45-84" aria-hidden="true"></a>[<span class="dv">125</span>] ggplot2_<span class="dv">3</span>.<span class="fl">5.1</span>               listenv_<span class="dv">0</span>.<span class="fl">9.1</span>              </span>
<span id="cb45-85"><a href="#cb45-85" aria-hidden="true"></a>[<span class="dv">127</span>] SpatialExperiment_<span class="dv">1</span>.<span class="fl">14.0</span>    viridisLite_<span class="dv">0</span>.<span class="fl">4.2</span>          </span>
<span id="cb45-86"><a href="#cb45-86" aria-hidden="true"></a>[<span class="dv">129</span>] scales_<span class="dv">1</span>.<span class="fl">3.0</span>                purrr_<span class="dv">1</span>.<span class="fl">0.2</span>                </span>
<span id="cb45-87"><a href="#cb45-87" aria-hidden="true"></a>[<span class="dv">131</span>] crayon_<span class="dv">1</span>.<span class="fl">5.3</span>                GetoptLong_<span class="dv">1</span>.<span class="fl">0.5</span>           </span>
<span id="cb45-88"><a href="#cb45-88" aria-hidden="true"></a>[<span class="dv">133</span>] rlang_<span class="dv">1</span>.<span class="fl">1.4</span>                 cowplot_<span class="dv">1</span>.<span class="fl">1.3</span> </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
