<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Seq-Scope Mouse liver â€¢ Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Seq-Scope Mouse liver">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/core_functions.html">Giotto Core Functions</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
        <li>
          <a href="../articles/create_simulated_object.html">Create a simulated datastet object</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Data Processing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
        <li>
          <a href="../articles/dimension_reduction.html">Dimension Reduction</a>
        </li>
        <li>
          <a href="../articles/networks.html">Spatial and Feature Networks</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactivity</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
        <li>
          <a href="../articles/interactive_selection_cosmx.html">Interactive region selection on large datasets</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/ontrac.html">Integrate ONTraC Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Seq-Scope Mouse liver</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/seqscope_mouse_liver.rmd" class="external-link"><code>vignettes/seqscope_mouse_liver.rmd</code></a></small>
      <div class="hidden name"><code>seqscope_mouse_liver.rmd</code></div>

    </div>

    
    
<div class="section level2" number="1">
<h2 number="1" id="dataset-explanation">
<span class="header-section-number">1</span> Dataset explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/34115981/" class="external-link">Seqscope</a> is a illumina sequencing based spatial sequencing platform developed by <a href="https://lee.lab.medicine.umich.edu/seq-scope" class="external-link">Jun Hee Lee Lab</a>. The basic strategy is to use illumina sequencing by synthesis to generate the spatial barcodes and use the barcodes to capture mRNAs in tissue.</p>
<div class="section level3" number="1.1">
<h3 number="1.1" id="example-raw-data-needed-for-seqscope">
<span class="header-section-number">1.1</span> Example Raw Data needed for seqscope:<a class="anchor" aria-label="anchor" href="#example-raw-data-needed-for-seqscope"></a>
</h3>
<p>1st-seq data (single-ended, for generating spatial barcodes)</p>
<ul>
<li>SeqScope_1st.fastq.gz 2nd-seq data (pair-ended, for generating count matrix)</li>
<li>SeqScope_2nd_R1.fastq.gz</li>
<li>SeqScope_2nd_R2.fastq.gz</li>
</ul>
<p>Image (for seqmentation)</p>
<ul>
<li>Tile_No_Segmented.png</li>
</ul>
</div>
<div class="section level3" number="1.2">
<h3 number="1.2" id="preprocessing-to-generate-count-matrix">
<span class="header-section-number">1.2</span> Preprocessing to generate count matrix<a class="anchor" aria-label="anchor" href="#preprocessing-to-generate-count-matrix"></a>
</h3>
<p>Seqscope has its own pipeline to generate the count matrix(Gene barcode). Please refer to their <a href="https://github.com/leeju-umich/Cho_Xi_Seqscope" class="external-link">github page</a> or use customized methods to proceed.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="co"># Note you need to install seqtk, STARsolo and clone their github page first. All bash scripts are stored in script directory of their github.</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="co">## First generate the whitelist of spatial barcodes:</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a>bash extractCoord.sh [SeqScope_1st.fastq.gz] [SeqScope_2nd_R1.fastq.gz] [HDMI_length]</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a><span class="co">## Second, generating the count matrix via STAR.Solo:</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a>bash align.sh [SeqScope_2nd_R1.fastq.gz] [abc_SeqScope_2nd_R2.fastq.gz] [HDMI_length] [whitelists.txt] [outprefix] [starpath] [seqtkpath] [geneIndex]</span></code></pre></div>
<p>You should now have the count matrix for Giotto Object. Note the Spatial Barcodes are typically at 0.6 um resolution. One way to analyze is to use the getSimpleGrid.R or getSlidingGrid.R provided by Lee Lab to bin the counts and simply follow the Giotto analysis steps of <a href="https://drieslab.github.io/Giotto_website/articles/visium_mouse_brain.html">Visium</a> and <a href="https://drieslab.github.io/Giotto_website/articles/visium_prostate_integration.html">Integration of Visium</a></p>
</div>
</div>
<div class="section level2" number="2">
<h2 number="2" id="start-giotto">
<span class="header-section-number">2</span> Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 number="3" id="set-up-giotto-environment">
<span class="header-section-number">3</span> Set up Giotto Environment<a class="anchor" aria-label="anchor" href="#set-up-giotto-environment"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 1. set working directory</span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"/path/to/results/"</span></span>
<span></span>
<span><span class="co"># 2. set giotto python path</span></span>
<span><span class="co"># set python path to your preferred python version path</span></span>
<span><span class="co"># set python path to conda env/bin/ directory if manually installed Giotto python dependencies by conda</span></span>
<span><span class="co"># python_path = "/path_to_conda/.conda/envs/giotto/bin/python"</span></span>
<span><span class="co"># set python path to NULL if you want to automatically install (only the 1st time) and use the giotto miniconda environment</span></span>
<span><span class="co"># python_path = NULL</span></span>
<span><span class="co">#if(is.null(python_path)) {</span></span>
<span><span class="co">#  installGiottoEnvironment()</span></span>
<span><span class="co">#}</span></span>
<span></span>
<span><span class="co"># 3. create giotto instructions</span></span>
<span><span class="va">instructions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_instructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">results_folder</span>,</span>
<span>                                         save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                         show_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                         return_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 number="4" id="create-giotto-subcellular-object">
<span class="header-section-number">4</span> Create Giotto Subcellular Object<a class="anchor" aria-label="anchor" href="#create-giotto-subcellular-object"></a>
</h2>
<p>In order to do single cell and subcellular analysis with Giotto, we suggest using the GiottoSubcellular Object. Giotto Subcellular objects take cell polygons and a Giotto Point file (a data.table contains Gene Name, Expression, x, y, etc). Therefore, first we need to do some transformation of the count matrix.</p>
<p><strong>Giotto Point file formatting</strong></p>
<p>Unlike normal cell by gene matrix, a Giotto Point file is usually a data.table contains Gene Name, Expression, x, y, where each row represent a subcellular point. For Seqscope data, one HDMI typically have more than one subcellular point.</p>
<p>Example Giotto Point file for <strong>Seqscope</strong></p>
<table class="table">
<thead><tr class="header">
<th align="left">HDMI</th>
<th align="left">Feat_ID</th>
<th align="left">Count</th>
<th align="left">sdimX</th>
<th align="left">sdimY</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">HDMI1</td>
<td align="left">GeneA</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">HDMI1</td>
<td align="left">GeneB</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">HDMI2</td>
<td align="left">GeneA</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">HDMI3</td>
<td align="left">GeneC</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
</tbody>
</table>
<div class="section level3" number="4.1">
<h3 number="4.1" id="process-giotto-point-file-per-tile">
<span class="header-section-number">4.1</span> Process Giotto Point file per tile<a class="anchor" aria-label="anchor" href="#process-giotto-point-file-per-tile"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##expression matrix</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/Solo.out/GeneFull/raw/"</span></span>
<span></span>
<span><span class="va">expr_matrix</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="../reference/get10Xmatrix.html">get10Xmatrix</a></span><span class="op">(</span>path_to_data <span class="op">=</span> <span class="va">data_path</span>, </span>
<span>                                    gene_column_index <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##Spatial coordinates</span></span>
<span><span class="va">spatial_coords_Dir</span> <span class="op">&lt;-</span> <span class="st">"/path/to/extractCoord.sh/results/spatialcoordinates.txt"</span></span>
<span><span class="va">spatial_coords</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"spatial_coords_Dir"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">spatial_coords</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HDMI"</span>,<span class="st">"Lane"</span>,<span class="st">"Tile"</span>,<span class="st">"X"</span>,<span class="st">"Y"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##Prepare Giotto Point</span></span>
<span><span class="co"># Subset expression and spatial info by tile</span></span>
<span><span class="va">spatial_coords_tile</span> <span class="op">&lt;-</span> <span class="va">spatial_coords</span><span class="op">[</span><span class="va">Tile</span> <span class="op">==</span> <span class="st">"2104"</span><span class="op">]</span></span>
<span><span class="va">expr_matrix_tile</span> <span class="op">&lt;-</span> <span class="va">expr_matrix</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expr_matrix</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">spatial_coords_tile</span><span class="op">$</span><span class="va">HDMI</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># convert expression matrix to minimal data.table object</span></span>
<span><span class="va">matrix_tile_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/as.data.table.html" class="external-link">as.data.table</a></span><span class="op">(</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu">summary</span><span class="op">(</span><span class="va">expr_matrix_tile</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">expr_matrix_tile</span><span class="op">@</span><span class="va">Dimnames</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">expr_matrix_tile</span><span class="op">@</span><span class="va">Dimnames</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">matrix_tile_dt</span><span class="op">[</span>, <span class="va">gene</span> <span class="op">:=</span> <span class="va">genes</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">matrix_tile_dt</span><span class="op">[</span>, <span class="va">hdmi</span> <span class="op">:=</span> <span class="va">samples</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># merge data.table matrix and spatial coordinates to create input for Giotto Polygons</span></span>
<span><span class="va">gpoints</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/merge.html" class="external-link">merge.data.table</a></span><span class="op">(</span><span class="va">matrix_tile_dt</span>, </span>
<span>                            <span class="va">spatial_coords_tile</span>, </span>
<span>                            by.x <span class="op">=</span> <span class="st">"hdmi"</span>, by.y <span class="op">=</span> <span class="st">"HDMI"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gpoints</span> <span class="op">&lt;-</span> <span class="va">gpoints</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">hdmi</span>, <span class="va">X</span>, <span class="va">Y</span>, <span class="va">gene</span>, <span class="va">x</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">gpoints</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hdmi"</span>, <span class="st">"x"</span>, <span class="st">"y"</span>, <span class="st">"gene"</span>, <span class="st">"counts"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="4.2">
<h3 number="4.2" id="prepare-the-polygon-mask-file">
<span class="header-section-number">4.2</span> Prepare the polygon mask file<a class="anchor" aria-label="anchor" href="#prepare-the-polygon-mask-file"></a>
</h3>
<p>Giotto can read in a variety of different mask files provided by common segmentation tools. But first we need to check if we need to filp the x and y axis.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># check total counts per hdmi</span></span>
<span><span class="va">gpoints_aggr</span> <span class="op">&lt;-</span> <span class="va">gpoints</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, by <span class="op">=</span> <span class="fu">.</span><span class="op">(</span><span class="va">hdmi</span>, <span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">gpoints_aggr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hdmi"</span>,<span class="st">"x"</span>,<span class="st">"y"</span>,<span class="st">"total_counts"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorder</a></span><span class="op">(</span><span class="va">gpoints_aggr</span>, <span class="op">-</span><span class="va">total_counts</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">pl</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">gpoints_aggr</span><span class="op">[</span><span class="va">total_counts</span> <span class="op">&lt;</span> <span class="fl">1000</span> <span class="op">&amp;</span> <span class="va">total_counts</span> <span class="op">&gt;</span> <span class="fl">8</span><span class="op">]</span>, </span>
<span>                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span>, color <span class="op">=</span> <span class="va">total_counts</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">pl</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradient2</a></span><span class="op">(</span>midpoint <span class="op">=</span> <span class="fl">15</span>, low <span class="op">=</span> <span class="st">"blue"</span>, mid <span class="op">=</span> <span class="st">"yellow"</span>, high <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">pl</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2FHDMI_scatter.png"></p>
<p>Read polygon mask file</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segmentation_mask</span> <span class="op">&lt;-</span> <span class="st">"/path/to/segmentation.tif"</span></span>
<span><span class="va">final_polygons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html" class="external-link">createGiottoPolygonsFromMask</a></span><span class="op">(</span><span class="va">segmentation_mask</span>,</span>
<span>                                               flip_vertical <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                               flip_horizontal <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">final_polygons</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2Fpolygon.png"></p>
<p><strong>Polygon mask file manual formatting</strong></p>
<p>However, sometimes Giotto does not read in the provided mask file and we will need a manual processing for the mask file and we will do that using terra and createGiottoPolygonsFromDfr.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raster</span> <span class="op">&lt;-</span> <span class="fu">terra</span><span class="fu">::</span><span class="fu"><a href="https://rspatial.github.io/terra/reference/rast.html" class="external-link">rast</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">segmentation_mask</span><span class="op">)</span></span>
<span><span class="va">terra_polygon</span> <span class="op">&lt;-</span> <span class="fu">terra</span><span class="fu">::</span><span class="fu"><a href="https://rspatial.github.io/terra/reference/as.polygons.html" class="external-link">as.polygons</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">raster</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># convert polygon to data.table and remove unwantedly detected polygons (e.g. canvas etc)</span></span>
<span><span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu">GiottospatVector_to_dt</span><span class="op">(</span><span class="va">terra_polygon</span><span class="op">)</span></span>
<span><span class="va">npolygons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">part</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span><span class="va">filter_dt</span> <span class="op">&lt;-</span> <span class="va">dt</span><span class="op">[</span><span class="va">geom</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">hole</span> <span class="op">==</span> <span class="fl">0</span> <span class="op">&amp;</span> <span class="va">part</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="va">npolygons</span><span class="op">)</span>, <span class="fu">.</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">part</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">filter_dt</span><span class="op">[</span>, <span class="va">part</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">part</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># create new polygons from filtered data.table</span></span>
<span><span class="va">final_polygons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html" class="external-link">createGiottoPolygonsFromDfr</a></span><span class="op">(</span>segmdfr <span class="op">=</span> <span class="va">filter_dt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create giotto points first to get the extent of the points (hdmi)</span></span>
<span><span class="va">original_points</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPoints.html" class="external-link">createGiottoPoints</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">gpoints</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">gene</span>, <span class="va">hdmi</span>, <span class="va">counts</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">original_feat_ext</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rspatial.github.io/terra/reference/ext.html" class="external-link">ext</a></span><span class="op">(</span><span class="va">original_points</span><span class="op">@</span><span class="va">spatVector</span><span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="co"># convert polygon to spatRaster to change extent to that of original points</span></span>
<span><span class="va">final_spatraster</span> <span class="op">&lt;-</span> <span class="fu">Giottopolygon_to_raster</span><span class="op">(</span>polygon <span class="op">=</span> <span class="va">final_polygons</span><span class="op">@</span><span class="va">spatVector</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rspatial.github.io/terra/reference/ext.html" class="external-link">ext</a></span><span class="op">(</span><span class="va">final_spatraster</span><span class="op">$</span><span class="va">raster</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">original_feat_ext</span></span>
<span><span class="va">final_polygons</span><span class="op">@</span><span class="va">spatVector</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rspatial.github.io/terra/reference/as.polygons.html" class="external-link">as.polygons</a></span><span class="op">(</span><span class="va">final_spatraster</span><span class="op">$</span><span class="va">raster</span><span class="op">)</span></span>
<span><span class="va">final_polygons</span><span class="op">@</span><span class="va">spatVector</span><span class="op">$</span><span class="va">poly_ID</span> <span class="op">&lt;-</span> <span class="va">final_spatraster</span><span class="op">$</span><span class="va">ID_vector</span><span class="op">[</span><span class="va">final_polygons</span><span class="op">@</span><span class="va">spatVector</span><span class="op">$</span><span class="va">poly_i</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># flip and shift, if needed</span></span>
<span><span class="co">#final_polygons@spatVector = flip(final_polygons@spatVector)</span></span>
<span><span class="co">#yshift = ymin(original_feat_ext) - ymax(original_feat_ext)</span></span>
<span><span class="co">#final_polygons@spatVector = terra::shift(final_polygons@spatVector, dy = -yshift)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">final_polygons</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="4.3">
<h3 number="4.3" id="create-giotto-object">
<span class="header-section-number">4.3</span> Create Giotto Object<a class="anchor" aria-label="anchor" href="#create-giotto-object"></a>
</h3>
<p>Add a random jitter to the HDMI location to make a pseudo-in situ transcript file.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add giotto points class</span></span>
<span><span class="va">gpoints_subset</span> <span class="op">&lt;-</span> <span class="va">gpoints</span><span class="op">[</span><span class="va">hdmi</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">gpoints_aggr</span><span class="op">[</span><span class="va">total_counts</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">]</span><span class="op">$</span><span class="va">hdmi</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># multiply rows with multiple counts and add jitter</span></span>
<span><span class="va">gpoints_extra</span> <span class="op">&lt;-</span> <span class="va">gpoints_subset</span><span class="op">[</span><span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">gpoints_extra</span> <span class="op">&lt;-</span> <span class="va">gpoints_extra</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">counts</span>, <span class="va">counts</span><span class="op">)</span>, by <span class="op">=</span> <span class="fu">.</span><span class="op">(</span><span class="va">hdmi</span>, <span class="va">gene</span>, <span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">gpoints_extra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">gpoints_extra</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">hdmi</span>, <span class="va">gene</span>, <span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">]</span>, <span class="va">gpoints_subset</span><span class="op">[</span><span class="va">counts</span> <span class="op">==</span> <span class="fl">1</span> ,<span class="fu">.</span><span class="op">(</span><span class="va">hdmi</span>, <span class="va">gene</span>, <span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">jitter_x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gpoints_extra</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">jitter_y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gpoints_extra</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">gpoints_extra</span><span class="op">[</span>, <span class="va">x</span> <span class="op">:=</span> <span class="va">x</span> <span class="op">+</span> <span class="va">jitter_x</span><span class="op">]</span></span>
<span><span class="va">gpoints_extra</span><span class="op">[</span>, <span class="va">y</span> <span class="op">:=</span> <span class="va">y</span> <span class="op">+</span> <span class="va">jitter_y</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add subcellular information</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObjectSubcellular.html" class="external-link">createGiottoObjectSubcellular</a></span><span class="op">(</span>gpoints <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gpoints_extra</span><span class="op">[</span>,<span class="fu">.</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">gene</span>, <span class="va">hdmi</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                                         gpolygons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">final_polygons</span><span class="op">)</span>,</span>
<span>                                         instructions <span class="op">=</span> <span class="va">instructions</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add centroids</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addSpatialCentroidLocations.html" class="external-link">addSpatialCentroidLocations</a></span><span class="op">(</span><span class="va">seqscope</span>,</span>
<span>                                       poly_info <span class="op">=</span> <span class="st">"cell"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Overlap to Polygon information</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/calculateOverlapRaster.html" class="external-link">calculateOverlapRaster</a></span><span class="op">(</span><span class="va">seqscope</span><span class="op">)</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/overlapToMatrix.html" class="external-link">overlapToMatrix</a></span><span class="op">(</span><span class="va">seqscope</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize top 200 expressed genes in situ</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html" class="external-link">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">seqscope</span>, </span>
<span>                     show_legend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     show_image <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     feats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"rna"</span> <span class="op">=</span> <span class="va">seqscope</span><span class="op">@</span><span class="va">feat_info</span><span class="op">$</span><span class="va">rna</span><span class="op">@</span><span class="va">spatVector</span><span class="op">$</span><span class="va">feat_ID</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                     spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                     point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     use_overlap <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     polygon_feat_type <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>                     polygon_bg_color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>                     coord_fix_ratio <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     background_color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2Ftop200.png"></p>
</div>
</div>
<div class="section level2" number="5">
<h2 number="5" id="process-giotto-and-quality-control">
<span class="header-section-number">5</span> Process Giotto and Quality Control<a class="anchor" aria-label="anchor" href="#process-giotto-and-quality-control"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>,</span>
<span>                         expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                         feat_det_in_min_cells <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                         min_det_feats_per_cell <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#normalize</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>, </span>
<span>                            scalefactor <span class="op">=</span> <span class="fl">5000</span>, </span>
<span>                            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># add statistics</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View cellular data</span></span>
<span><span class="co"># pDataDT(seqscope)</span></span>
<span><span class="co"># View rna data</span></span>
<span><span class="co"># fDataDT(seqscope)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"total_expr"</span>, </span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">2.5</span>, </span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>           coord_fix_ratio <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2Ftotal_expr.png"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">seqscope</span>, feat_type <span class="op">=</span> <span class="st">"rna"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">cell_metadata</span><span class="op">$</span><span class="va">nr_feats</span>, <span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2Ffeature_distribution.png"></p>
</div>
<div class="section level2" number="6">
<h2 number="6" id="dimension-reduction">
<span class="header-section-number">6</span> Dimension Reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cluster cells</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>, </span>
<span>                         HVFname <span class="op">=</span> <span class="st">"hvg_orig"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>,</span>
<span>                   expression_values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                   scale_unit <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                   center <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">seqscope</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="7">
<h2 number="7" id="cluster">
<span class="header-section-number">7</span> Cluster<a class="anchor" aria-label="anchor" href="#cluster"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>, </span>
<span>                                 dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, </span>
<span>                                 k <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>, </span>
<span>                            resolution <span class="op">=</span> <span class="fl">0.9</span>, </span>
<span>                            n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize UMAP cluster results</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2Fleiden_umap.png"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatInSituPlotPoints.html" class="external-link">spatInSituPlotPoints</a></span><span class="op">(</span><span class="va">seqscope</span>,</span>
<span>                     show_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     polygon_color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                     polygon_line_size <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                     polygon_fill <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>                     polygon_fill_as_factor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     coord_fix_ratio <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2Fleiden_spat.png"></p>
</div>
<div class="section level2" number="8">
<h2 number="8" id="find-spatial-genes">
<span class="header-section-number">8</span> Find spatial genes<a class="anchor" aria-label="anchor" href="#find-spatial-genes"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqscope</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html" class="external-link">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqscope</span>, </span>
<span>                                 minimum_k <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                                 maximum_distance_delaunay <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="va">km_spatialfeats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">seqscope</span>, </span>
<span>                            subset_feats <span class="op">=</span> <span class="va">seqscope</span><span class="op">@</span><span class="va">feat_ID</span><span class="op">$</span><span class="va">rna</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatFeatPlot2D.html" class="external-link">spatFeatPlot2D</a></span><span class="op">(</span><span class="va">seqscope</span>, </span>
<span>               expression_values <span class="op">=</span> <span class="st">"scaled"</span>,</span>
<span>               feats <span class="op">=</span> <span class="va">km_spatialfeats</span><span class="op">$</span><span class="va">feats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>               cell_color_gradient <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>               point_shape <span class="op">=</span> <span class="st">"border"</span>, </span>
<span>               point_border_stroke <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>               show_network <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>               network_color <span class="op">=</span> <span class="st">"lightgrey"</span>, </span>
<span>               point_size <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fseqscope_mouse_liver%2Fspatgenes.png"></p>
</div>
<div class="section level2" number="9">
<h2 number="9" id="session-info">
<span class="header-section-number">9</span> Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. ChÃ¡vez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew Oâ€™Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
