<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integration of single cell datasets â€¢ Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Integration of single cell datasets">
<meta property="og:description" content="Giotto">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
    <li>
      <a href="../articles/contributing.html">Contributing guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/object_creation.html">Giotto Object Creation</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/configuration.html">Giotto Configuration</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Spatial manipulation</li>
    <li>
      <a href="../articles/spatial_transformations.html">Spatial transformations</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Image alignment</li>
    <li>
      <a href="../articles/image_alignment.html">Image alignment</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interactive selection</li>
    <li>
      <a href="../articles/interactive_selection.html">Interactive selection</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualizations</li>
    <li>
      <a href="../articles/visualizations.html">Visualizations</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Saving and Loading</li>
    <li>
      <a href="../articles/saving_object.html">Saving a Giotto object</a>
    </li>
    <li>
      <a href="../articles/saving_plots.html">Saving plots</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interoperability</li>
    <li>
      <a href="../articles/interoperability_anndata.html">AnnData</a>
    </li>
    <li>
      <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
    </li>
    <li>
      <a href="../articles/interoperability_seurat.html">Seurat</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Try Giotto in the Cloud</li>
    <li>
      <a href="../articles/terra.html">Terra.bio</a>
    </li>
    <li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/singularity.html">Singularity</a>
    </li>
    <li>
      <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integration of single cell datasets</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/HEAD/vignettes/singlecell_prostate_integration.Rmd" class="external-link"><code>vignettes/singlecell_prostate_integration.Rmd</code></a></small>
      <div class="hidden name"><code>singlecell_prostate_integration.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="dataset-explanation">Dataset explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p>This is a tutorial for Harmony integration of different single cell
RNAseq datasets using two prostate cancer patient datasets. <a href="https:/pubmed.ncbi.nlm.nih.gov/33032611/">Ma et al.</a> Processed
10X Single Cell RNAseq from two prostate cancer patients. The raw
dataset can be found <a href="https:/www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE157703">here</a></p>
</div>
<div class="section level2">
<h2 id="set-up-giotto-environment">Set up Giotto Environment<a class="anchor" aria-label="anchor" href="#set-up-giotto-environment"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu">pkg_install</span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/checkGiottoEnvironment.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/installGiottoEnvironment.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 1. set working directory</span></span>
<span><span class="va">results_folder</span> <span class="op">=</span> <span class="st">'path/to/result'</span></span>
<span></span>
<span><span class="co"># Optional: Specify a path to a Python executable within a conda or miniconda </span></span>
<span><span class="co"># environment. If set to NULL (default), the Python executable within the previously</span></span>
<span><span class="co"># installed Giotto environment will be used.</span></span>
<span><span class="va">my_python_path</span> <span class="op">=</span> <span class="cn">NULL</span> <span class="co"># alternatively, "/local/python/path/python" if desired.</span></span>
<span></span>
<span><span class="co"># 2. create giotto instructions</span></span>
<span><span class="va">instrs</span> <span class="op">=</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">results_folder</span>,</span>
<span>                                  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  show_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                  python_path <span class="op">=</span> <span class="va">my_python_path</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-giotto-object-from-10x-dataset-and-join">1. Create Giotto object from 10X dataset and join<a class="anchor" aria-label="anchor" href="#create-giotto-object-from-10x-dataset-and-join"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_dir</span> <span class="op">=</span> <span class="st">"path/to/data/"</span></span>
<span></span>
<span><span class="va">expression1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"GSM4773521_PCa1_gene_counts_matrix.txt"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_P1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expression1</span>,</span>
<span>                                instructions <span class="op">=</span> <span class="va">instrs</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">expression2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"GSM4773522_PCa2_gene_counts_matrix.txt"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_P2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expression2</span>,</span>
<span>                                instructions <span class="op">=</span> <span class="va">instrs</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects</a></span><span class="op">(</span>gobject_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">giotto_P1</span>, <span class="va">giotto_P2</span><span class="op">)</span>,</span>
<span>                                    gobject_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'P1'</span>, <span class="st">'P2'</span><span class="op">)</span>,</span>
<span>                                    join_method <span class="op">=</span> <span class="st">"z_stack"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="process-joined-object">2. Process Joined object<a class="anchor" aria-label="anchor" href="#process-joined-object"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                               expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                               feat_det_in_min_cells <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                               min_det_feats_per_cell <span class="op">=</span> <span class="fl">500</span>,</span>
<span>                               expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'raw'</span><span class="op">)</span>,</span>
<span>                               verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## normalize</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                                  scalefactor <span class="op">=</span> <span class="fl">6000</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## add gene &amp; cell statistics</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                                expression_values <span class="op">=</span> <span class="st">'raw'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dimension-reduction-and-clustering">3. Dimension reduction and clustering<a class="anchor" aria-label="anchor" href="#dimension-reduction-and-clustering"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## PCA ##</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                         center <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                         scale_unit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check screeplot to select number of PCs for clustering</span></span>
<span><span class="co"># screePlot(giotto_SC_join, ncp = 30, save_param = list(save_name = '3_scree_plot'))</span></span>
<span></span>
<span><span class="co">## WITHOUT INTEGRATION ##</span></span>
<span><span class="co"># --------------------- #</span></span>
<span></span>
<span><span class="co">## cluster and run UMAP ##</span></span>
<span><span class="co"># sNN network (default)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoDimRed.html" class="external-link">showGiottoDimRed</a></span><span class="op">(</span><span class="va">giotto_SC_join</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                                       dim_reduction_to_use <span class="op">=</span> <span class="st">'pca'</span>,</span>
<span>                                       dim_reduction_name <span class="op">=</span> <span class="st">'pca'</span>,</span>
<span>                                       dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                                       k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Leiden clustering</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                                  resolution <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>                                  n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># UMAP</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">giotto_SC_join</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>,</span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4_cluster_without_integration"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/singlecell_prostate_integration/4_cluster_without_integration.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/dimPlot.html" class="external-link">dimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>          dim_reduction_name <span class="op">=</span> <span class="st">'umap'</span>,</span>
<span>          point_shape <span class="op">=</span> <span class="st">'no_border'</span>,</span>
<span>          cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>          group_by <span class="op">=</span> <span class="st">"list_ID"</span>,</span>
<span>          show_NN_network <span class="op">=</span> <span class="cn">F</span>,</span>
<span>          point_size <span class="op">=</span> <span class="fl">0.5</span>, </span>
<span>          show_center_label <span class="op">=</span> <span class="cn">F</span>,</span>
<span>          show_legend <span class="op">=</span> <span class="cn">F</span>,</span>
<span>          save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4_list_without_integration"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/singlecell_prostate_integration/4_list_without_integration.png" style="width:50.0%"></p>
<p>Harmony is a integration algorithm developed by <a href="https:/www.nature.com/articles/s41592-019-0619-0">Korsunsky, I. et
al.</a>. It was designed for integration of single cell data but also
work well on spatial datasets.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## WITH INTEGRATION ##</span></span>
<span><span class="co"># --------------------- #</span></span>
<span></span>
<span><span class="co">## data integration, cluster and run UMAP ##</span></span>
<span></span>
<span><span class="co"># harmony</span></span>
<span><span class="co">#library(devtools)</span></span>
<span><span class="co">#install_github("immunogenomics/harmony")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">harmony</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#pDataDT(giotto_SC_join)</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runGiottoHarmony.html">runGiottoHarmony</a></span><span class="op">(</span><span class="va">giotto_SC_join</span>,</span>
<span>                                   vars_use <span class="op">=</span> <span class="st">'list_ID'</span>,</span>
<span>                                   do_pca <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## sNN network (default)</span></span>
<span><span class="co">#showGiottoDimRed(giotto_SC_join)</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                                       dim_reduction_to_use <span class="op">=</span> <span class="st">'harmony'</span>,</span>
<span>                                       dim_reduction_name <span class="op">=</span> <span class="st">'harmony'</span>,</span>
<span>                                       name <span class="op">=</span> <span class="st">'NN.harmony'</span>,</span>
<span>                                       dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                                       k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Leiden clustering</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>                                  network_name <span class="op">=</span> <span class="st">'NN.harmony'</span>,</span>
<span>                                  resolution <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>                                  n_iterations <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>                                  name <span class="op">=</span> <span class="st">'leiden_harmony'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># UMAP dimension reduction</span></span>
<span><span class="co">#showGiottoDimRed(giotto_SC_join)</span></span>
<span><span class="va">giotto_SC_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">giotto_SC_join</span>,</span>
<span>                          dim_reduction_name <span class="op">=</span> <span class="st">'harmony'</span>,</span>
<span>                          dim_reduction_to_use <span class="op">=</span> <span class="st">'harmony'</span>,</span>
<span>                          name <span class="op">=</span> <span class="st">'umap_harmony'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>         dim_reduction_name <span class="op">=</span> <span class="st">'umap_harmony'</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">'leiden_harmony'</span>,</span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4_cluster_with_integration"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/singlecell_prostate_integration/4_cluster_with_integration.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/dimPlot.html" class="external-link">dimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC_join</span>,</span>
<span>          dim_reduction_name <span class="op">=</span> <span class="st">'umap_harmony'</span>,</span>
<span>          point_shape <span class="op">=</span> <span class="st">'no_border'</span>,</span>
<span>          cell_color <span class="op">=</span> <span class="st">"leiden_harmony"</span>,</span>
<span>          group_by <span class="op">=</span> <span class="st">"list_ID"</span>,</span>
<span>          show_NN_network <span class="op">=</span> <span class="cn">F</span>,</span>
<span>          point_size <span class="op">=</span> <span class="fl">0.5</span>, </span>
<span>          show_center_label <span class="op">=</span> <span class="cn">F</span>,</span>
<span>          show_legend <span class="op">=</span><span class="cn">F</span> ,</span>
<span>          save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">"4_list_with_integration"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/singlecell_prostate_integration/4_list_with_integration.png" style="width:50.0%"> # 4. Session Info</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">3.2</span> (<span class="dv">2023-10-31</span>)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>Platform<span class="sc">:</span> aarch64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin20</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="dv">14</span>.<span class="fl">2.1</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.3</span><span class="sc">-</span>arm64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a>[<span class="dv">1</span>] harmony_1.<span class="fl">2.0</span>     Rcpp_1.<span class="fl">0.12</span>       Giotto_4.<span class="fl">0.3</span>      GiottoClass_0.<span class="fl">1.3</span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a> [<span class="dv">1</span>] tidyselect_1.<span class="fl">2.0</span>            farver_2.<span class="fl">1.1</span>                dplyr_1.<span class="fl">1.4</span>                </span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a> [<span class="dv">4</span>] GiottoVisuals_0.<span class="fl">1.4</span>         bitops_1<span class="fl">.0</span><span class="dv">-7</span>                fastmap_1.<span class="fl">1.1</span>              </span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a> [<span class="dv">7</span>] SingleCellExperiment_1.<span class="fl">24.0</span> RCurl_1<span class="fl">.98-1.14</span>             rsvd_1.<span class="fl">0.5</span>                 </span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a>[<span class="dv">10</span>] digest_0.<span class="fl">6.34</span>               lifecycle_1.<span class="fl">0.4</span>             terra_1<span class="fl">.7</span><span class="dv">-71</span>               </span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a>[<span class="dv">13</span>] dbscan_1<span class="fl">.1</span><span class="dv">-12</span>               magrittr_2.<span class="fl">0.3</span>              compiler_4.<span class="fl">3.2</span>             </span>
<span id="cb10-27"><a href="#cb10-27" tabindex="-1"></a>[<span class="dv">16</span>] rlang_1.<span class="fl">1.3</span>                 tools_4.<span class="fl">3.2</span>                 igraph_2.<span class="fl">0.2</span>               </span>
<span id="cb10-28"><a href="#cb10-28" tabindex="-1"></a>[<span class="dv">19</span>] utf8_1.<span class="fl">2.4</span>                  yaml_2.<span class="fl">3.8</span>                  data.table_1.<span class="fl">15.0</span>          </span>
<span id="cb10-29"><a href="#cb10-29" tabindex="-1"></a>[<span class="dv">22</span>] knitr_1<span class="fl">.45</span>                  S4Arrays_1.<span class="fl">2.0</span>              labeling_0.<span class="fl">4.3</span>             </span>
<span id="cb10-30"><a href="#cb10-30" tabindex="-1"></a>[<span class="dv">25</span>] reticulate_1.<span class="fl">35.0</span>           DelayedArray_0.<span class="fl">28.0</span>         RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>         </span>
<span id="cb10-31"><a href="#cb10-31" tabindex="-1"></a>[<span class="dv">28</span>] BiocParallel_1.<span class="fl">36.0</span>         abind_1<span class="fl">.4</span><span class="dv">-5</span>                 withr_3.<span class="fl">0.0</span>                </span>
<span id="cb10-32"><a href="#cb10-32" tabindex="-1"></a>[<span class="dv">31</span>] BiocGenerics_0.<span class="fl">48.1</span>         grid_4.<span class="fl">3.2</span>                  stats4_4.<span class="fl">3.2</span>               </span>
<span id="cb10-33"><a href="#cb10-33" tabindex="-1"></a>[<span class="dv">34</span>] fansi_1.<span class="fl">0.6</span>                 beachmat_2.<span class="fl">18.1</span>             colorspace_2<span class="fl">.1</span><span class="dv">-0</span>           </span>
<span id="cb10-34"><a href="#cb10-34" tabindex="-1"></a>[<span class="dv">37</span>] future_1.<span class="fl">33.1</span>               ggplot2_3.<span class="fl">4.4</span>               globals_0.<span class="fl">16.2</span>             </span>
<span id="cb10-35"><a href="#cb10-35" tabindex="-1"></a>[<span class="dv">40</span>] scales_1.<span class="fl">3.0</span>                gtools_3.<span class="fl">9.5</span>                SummarizedExperiment_1.<span class="fl">32.0</span></span>
<span id="cb10-36"><a href="#cb10-36" tabindex="-1"></a>[<span class="dv">43</span>] cli_3.<span class="fl">6.2</span>                   rmarkdown_2<span class="fl">.25</span>              crayon_1.<span class="fl">5.2</span>               </span>
<span id="cb10-37"><a href="#cb10-37" tabindex="-1"></a>[<span class="dv">46</span>] ragg_1.<span class="fl">2.7</span>                  generics_0.<span class="fl">1.3</span>              rstudioapi_0.<span class="fl">15.0</span>          </span>
<span id="cb10-38"><a href="#cb10-38" tabindex="-1"></a>[<span class="dv">49</span>] future.apply_1.<span class="fl">11.1</span>         rjson_0.<span class="fl">2.21</span>                zlibbioc_1.<span class="fl">48.0</span>            </span>
<span id="cb10-39"><a href="#cb10-39" tabindex="-1"></a>[<span class="dv">52</span>] parallel_4.<span class="fl">3.2</span>              XVector_0.<span class="fl">42.0</span>              matrixStats_1.<span class="fl">2.0</span>          </span>
<span id="cb10-40"><a href="#cb10-40" tabindex="-1"></a>[<span class="dv">55</span>] vctrs_0.<span class="fl">6.5</span>                 Matrix_1<span class="fl">.6</span><span class="dv">-5</span>                jsonlite_1.<span class="fl">8.8</span>             </span>
<span id="cb10-41"><a href="#cb10-41" tabindex="-1"></a>[<span class="dv">58</span>] BiocSingular_1.<span class="fl">18.0</span>         IRanges_2.<span class="fl">36.0</span>              S4Vectors_0.<span class="fl">40.2</span>           </span>
<span id="cb10-42"><a href="#cb10-42" tabindex="-1"></a>[<span class="dv">61</span>] ggrepel_0.<span class="fl">9.5</span>               irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>               listenv_0.<span class="fl">9.1</span>              </span>
<span id="cb10-43"><a href="#cb10-43" tabindex="-1"></a>[<span class="dv">64</span>] systemfonts_1.<span class="fl">0.5</span>           magick_2.<span class="fl">8.3</span>                GiottoUtils_0.<span class="fl">1.5</span>          </span>
<span id="cb10-44"><a href="#cb10-44" tabindex="-1"></a>[<span class="dv">67</span>] glue_1.<span class="fl">7.0</span>                  parallelly_1.<span class="fl">37.0</span>           codetools_0<span class="fl">.2</span><span class="dv">-19</span>           </span>
<span id="cb10-45"><a href="#cb10-45" tabindex="-1"></a>[<span class="dv">70</span>] uwot_0.<span class="fl">1.16</span>                 cowplot_1.<span class="fl">1.3</span>               RcppAnnoy_0.<span class="fl">0.22</span>           </span>
<span id="cb10-46"><a href="#cb10-46" tabindex="-1"></a>[<span class="dv">73</span>] gtable_0.<span class="fl">3.4</span>                GenomeInfoDb_1.<span class="fl">38.6</span>         GenomicRanges_1.<span class="fl">54.1</span>       </span>
<span id="cb10-47"><a href="#cb10-47" tabindex="-1"></a>[<span class="dv">76</span>] ScaledMatrix_1.<span class="fl">10.0</span>         munsell_0.<span class="fl">5.0</span>               tibble_3.<span class="fl">2.1</span>               </span>
<span id="cb10-48"><a href="#cb10-48" tabindex="-1"></a>[<span class="dv">79</span>] pillar_1.<span class="fl">9.0</span>                htmltools_0.<span class="fl">5.7</span>             GenomeInfoDbData_1.<span class="fl">2.11</span>    </span>
<span id="cb10-49"><a href="#cb10-49" tabindex="-1"></a>[<span class="dv">82</span>] R6_2.<span class="fl">5.1</span>                    textshaping_0.<span class="fl">3.7</span>           evaluate_0<span class="fl">.23</span>              </span>
<span id="cb10-50"><a href="#cb10-50" tabindex="-1"></a>[<span class="dv">85</span>] lattice_0<span class="fl">.22</span><span class="dv">-5</span>              Biobase_2.<span class="fl">62.0</span>              png_0<span class="fl">.1</span><span class="dv">-8</span>                  </span>
<span id="cb10-51"><a href="#cb10-51" tabindex="-1"></a>[<span class="dv">88</span>] backports_1.<span class="fl">4.1</span>             RhpcBLASctl_0<span class="fl">.23</span><span class="dv">-42</span>         SpatialExperiment_1.<span class="fl">12.0</span>   </span>
<span id="cb10-52"><a href="#cb10-52" tabindex="-1"></a>[<span class="dv">91</span>] SparseArray_1.<span class="fl">2.4</span>           checkmate_2.<span class="fl">3.1</span>             colorRamp2_0.<span class="fl">1.0</span>           </span>
<span id="cb10-53"><a href="#cb10-53" tabindex="-1"></a>[<span class="dv">94</span>] xfun_0<span class="fl">.42</span>                   MatrixGenerics_1.<span class="fl">14.0</span>       pkgconfig_2.<span class="fl">0.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. ChÃ¡vez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew Oâ€™Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
