<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Splitting and Joining • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Splitting and Joining">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/core_functions.html">Giotto Core Functions</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Object creation</a>
        </li>
        <li>
          <a href="../articles/split_join.html">Object splitting and joining</a>
        </li>
        <li>
          <a href="../articles/create_simulated_object.html">Create a simulated dataset object</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Data Processing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
        <li>
          <a href="../articles/dimension_reduction.html">Dimension reduction</a>
        </li>
        <li>
          <a href="../articles/networks.html">Spatial and feature networks</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactivity</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
        <li>
          <a href="../articles/interactive_selection_cosmx.html">Interactive region selection on large datasets</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/ontrac.html">Integrate ONTraC Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Splitting and Joining</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/split_join.Rmd" class="external-link"><code>vignettes/split_join.Rmd</code></a></small>
      <div class="hidden name"><code>split_join.Rmd</code></div>

    </div>

    
    
<p><code>splitGiotto()</code> and <code><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects()</a></code> are how Giotto works with multiple samples.</p>
<ul>
<li>
<code>splitGiotto()</code> - separate a single <code>giotto</code> object into a list of several based on a cell metadata column</li>
<li>
<code><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects()</a></code> - combine a list of multiple <code>giotto</code> objects into a single <code>giotto</code> object.</li>
</ul>
<p>This example shows working with a 10x Visium <a href="https://www.10xgenomics.com/datasets/mouse-tissue-microarray-in-3x3-layout-with-2-mm-edge-to-edge-spacing-ffpe-2-standard" class="external-link">mouse TMA</a> dataset.</p>
<div class="section level2" number="1">
<h2 number="1" id="setup">
<span class="header-section-number">1</span> Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed.</span></span>
<span><span class="va">genv_exists</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 number="2" id="download-the-data">
<span class="header-section-number">2</span> Download the Data<a class="anchor" aria-label="anchor" href="#download-the-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## provide path to a folder to save the example data to</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/data/"</span></span>
<span></span>
<span><span class="va">mat_link</span> <span class="op">&lt;-</span> <span class="st">"https://cf.10xgenomics.com/samples/spatial-exp/2.0.1/CytAssist_11mm_FFPE_Mouse_TMA_3x3_2mm/CytAssist_11mm_FFPE_Mouse_TMA_3x3_2mm_filtered_feature_bc_matrix.h5"</span></span>
<span><span class="va">spat_link</span> <span class="op">&lt;-</span> <span class="st">"https://cf.10xgenomics.com/samples/spatial-exp/2.0.1/CytAssist_11mm_FFPE_Mouse_TMA_3x3_2mm/CytAssist_11mm_FFPE_Mouse_TMA_3x3_2mm_spatial.tar.gz"</span></span>
<span></span>
<span><span class="va">mat_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"filtered_matrix.h5"</span><span class="op">)</span></span>
<span><span class="va">spat_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"spatial.tar.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">mat_link</span>, destfile <span class="op">=</span> <span class="va">mat_path</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">spat_link</span>, destfile <span class="op">=</span> <span class="va">spat_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># untar the spatial folder</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="va">spat_path</span>, exdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">spat_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"spatial"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 number="3" id="create-giotto-object">
<span class="header-section-number">3</span> Create Giotto Object<a class="anchor" aria-label="anchor" href="#create-giotto-object"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 1. set working directory</span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"path/to/results"</span></span>
<span></span>
<span><span class="co"># Optional: Specify a path to a Python executable within a conda or miniconda </span></span>
<span><span class="co"># environment. If set to NULL (default), the Python executable within the previously</span></span>
<span><span class="co"># installed Giotto environment will be used.</span></span>
<span><span class="va">python_path</span> <span class="op">&lt;-</span> <span class="cn">NULL</span> <span class="co"># alternatively, "/local/python/path/python" if desired.</span></span>
<span></span>
<span><span class="va">instrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_instructions.html" class="external-link">instructions</a></span><span class="op">(</span>python_path <span class="op">=</span> <span class="va">python_path</span>, save_dir <span class="op">=</span> <span class="va">results_folder</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createGiottoVisiumObject.html">createGiottoVisiumObject</a></span><span class="op">(</span></span>
<span>    h5_visium_path <span class="op">=</span> <span class="va">mat_path</span>, </span>
<span>    h5_tissue_positions_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">spat_dir</span>, <span class="st">"tissue_positions.csv"</span><span class="op">)</span>,</span>
<span>    h5_image_png_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">spat_dir</span>, <span class="st">"tissue_hires_image.png"</span><span class="op">)</span>,</span>
<span>    h5_json_scalefactors_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">spat_dir</span>, <span class="st">"scalefactors_json.json"</span><span class="op">)</span>,</span>
<span>    instructions <span class="op">=</span> <span class="va">instrs</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the data</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">tma</span>, show_image <span class="op">=</span> <span class="cn">TRUE</span>, point_size <span class="op">=</span> <span class="fl">1.5</span>, point_alpha <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fsplit_join%2F1_spatplot.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2" number="4">
<h2 number="4" id="detect-individual-cores">
<span class="header-section-number">4</span> Detect individual cores<a class="anchor" aria-label="anchor" href="#detect-individual-cores"></a>
</h2>
<p>By building a spatial network, we can find out which data points are spatially contiguous by looking checking spatial distance, breaking ones that are too far away, and then checking for graph membership.</p>
<p>This functionality is implemented as <code><a href="../reference/identifyTMAcores.html">identifyTMAcores()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create a default delaunay spatial network</span></span>
<span><span class="va">tma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialNetwork.html" class="external-link">createSpatialNetwork</a></span><span class="op">(</span><span class="va">tma</span><span class="op">)</span></span>
<span><span class="va">tma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/identifyTMAcores.html">identifyTMAcores</a></span><span class="op">(</span><span class="va">tma</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">tma</span>,</span>
<span>    show_image <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>    cell_color <span class="op">=</span> <span class="st">"core_id"</span>, </span>
<span>    point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>    point_alpha <span class="op">=</span> <span class="fl">0.7</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fsplit_join%2F2_identify_tma.png" width="100%" style="display: block; margin: auto;"></p>
<p>The cores have been assigned numerical IDs. There is also a “not_connected” group which are too small to be considered a spatial region and not connected to a larger group of data points.</p>
<p>The identification also split one cores it did not need to, so let’s manually repair it</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># fix the core id annotation</span></span>
<span><span class="va">tma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/annotateGiotto.html" class="external-link">annotateGiotto</a></span><span class="op">(</span><span class="va">tma</span>, </span>
<span>    name <span class="op">=</span> <span class="st">"core_id_fixed"</span>, </span>
<span>    cluster_column <span class="op">=</span> <span class="st">"core_id"</span>, </span>
<span>    annotation_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"1"</span> <span class="op">=</span> <span class="st">"1"</span>,</span>
<span>        <span class="st">"2"</span> <span class="op">=</span> <span class="st">"2"</span>,</span>
<span>        <span class="st">"3"</span> <span class="op">=</span> <span class="st">"3"</span>,</span>
<span>        <span class="st">"4"</span> <span class="op">=</span> <span class="st">"4"</span>,</span>
<span>        <span class="st">"5"</span> <span class="op">=</span> <span class="st">"5"</span>,</span>
<span>        <span class="st">"6"</span> <span class="op">=</span> <span class="st">"6"</span>,</span>
<span>        <span class="st">"7"</span> <span class="op">=</span> <span class="st">"7"</span>,</span>
<span>        <span class="st">"8"</span> <span class="op">=</span> <span class="st">"8"</span>,</span>
<span>        <span class="st">"9"</span> <span class="op">=</span> <span class="st">"9"</span>,</span>
<span>        <span class="st">"10"</span> <span class="op">=</span> <span class="st">"9"</span>,</span>
<span>        <span class="st">"11"</span> <span class="op">=</span> <span class="st">"not_connected"</span>,</span>
<span>        <span class="st">"12"</span> <span class="op">=</span> <span class="st">"not_connected"</span>,</span>
<span>        <span class="st">"not_connected"</span> <span class="op">=</span> <span class="st">"not_connected"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># also add the core annotations from the 10x dataset description</span></span>
<span><span class="va">tma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/annotateGiotto.html" class="external-link">annotateGiotto</a></span><span class="op">(</span><span class="va">tma</span>, </span>
<span>    name <span class="op">=</span> <span class="st">"sample"</span>, </span>
<span>    cluster_column <span class="op">=</span> <span class="st">"core_id_fixed"</span>, </span>
<span>    annotation_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"1"</span> <span class="op">=</span> <span class="st">"lung1"</span>,</span>
<span>        <span class="st">"2"</span> <span class="op">=</span> <span class="st">"brain2"</span>,</span>
<span>        <span class="st">"3"</span> <span class="op">=</span> <span class="st">"brain3"</span>,</span>
<span>        <span class="st">"4"</span> <span class="op">=</span> <span class="st">"lung5"</span>,</span>
<span>        <span class="st">"5"</span> <span class="op">=</span> <span class="st">"brain1"</span>,</span>
<span>        <span class="st">"6"</span> <span class="op">=</span> <span class="st">"brain4"</span>,</span>
<span>        <span class="st">"7"</span> <span class="op">=</span> <span class="st">"lung4"</span>,</span>
<span>        <span class="st">"8"</span> <span class="op">=</span> <span class="st">"lung3"</span>,</span>
<span>        <span class="st">"9"</span> <span class="op">=</span> <span class="st">"lung2"</span>,</span>
<span>        <span class="st">"not_connected"</span> <span class="op">=</span> <span class="st">"not_connected"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">tma</span>,</span>
<span>    show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    cell_color <span class="op">=</span> <span class="st">"sample"</span>, </span>
<span>    point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>    point_alpha <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fsplit_join%2F3_annotated.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2" number="5">
<h2 number="5" id="split-giotto-object">
<span class="header-section-number">5</span> Split Giotto Object<a class="anchor" aria-label="anchor" href="#split-giotto-object"></a>
</h2>
<p>A split operation will split a single <code>giotto</code> object into a list of several based on a cell metadata column defined by the <code>by</code> param.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object_list</span> <span class="op">&lt;-</span> <span class="fu">splitGiotto</span><span class="op">(</span><span class="va">tma</span>, by <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">object_list</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 10</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">object_list</span><span class="op">)</span></span></code></pre></div>
<pre><code> [1] "lung1"         "brain2"        "brain3"        "lung5"         "brain1"       
 [6] "brain4"        "lung4"         "lung3"         "not_connected" "lung2"   </code></pre>
</div>
<div class="section level2" number="6">
<h2 number="6" id="join-giotto-objects">
<span class="header-section-number">6</span> Join Giotto Objects<a class="anchor" aria-label="anchor" href="#join-giotto-objects"></a>
</h2>
<p><code>giotto</code> objects can be joined together into a single one. This requires both the list of <code>giotto</code> objects and a list of names to assign those objects in the joined object. This involves updating the cell IDs used throughout the object so that they are disambiguated from possibly similar cell IDs in other objects. A new column (called “list_ID” by default) is also added to the cell metadata to help differentiate between samples. Additionally, a spatial x shift is performed by default, so that spatial data does not accidentally overlap each other and become hard to tell apart between samples. In this case, we do not need any additional shifting of values to avoid overlaps, and we can use <code>join_method = "no_change</code>.</p>
<p><code><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects()</a></code> is used to join together any sets of <code>giotto</code> objects for joint analysis, and it is not needed for the objects to have been from the same source object or to have first been split with <code>splitGiotto()</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lung_reps</span> <span class="op">&lt;-</span> <span class="va">object_list</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lung1"</span>, <span class="st">"lung2"</span>, <span class="st">"lung3"</span>, <span class="st">"lung4"</span>, <span class="st">"lung5"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">brain_reps</span> <span class="op">&lt;-</span> <span class="va">object_list</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"brain1"</span>, <span class="st">"brain2"</span>, <span class="st">"brain3"</span>, <span class="st">"brain4"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">lung</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects</a></span><span class="op">(</span></span>
<span>    gobject_list <span class="op">=</span> <span class="va">lung_reps</span>, </span>
<span>    gobject_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lung_reps</span><span class="op">)</span>, </span>
<span>    join_method <span class="op">=</span> <span class="st">"no_change"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">lung</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span><span class="va">lung</span>, expression_values <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">lung</span>,</span>
<span>    cell_color <span class="op">=</span> <span class="st">"total_expr"</span>, </span>
<span>    color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>    gradient_style <span class="op">=</span> <span class="st">"sequential"</span>,</span>
<span>    group_by <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    point_border_stroke <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    point_alpha <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>    show_image <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fsplit_join%2F4_lung_joined.png" width="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects</a></span><span class="op">(</span></span>
<span>    gobject_list <span class="op">=</span> <span class="va">brain_reps</span>, </span>
<span>    gobject_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">brain_reps</span><span class="op">)</span>, </span>
<span>    join_method <span class="op">=</span> <span class="st">"no_change"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span><span class="va">brain</span>, expression_values <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">brain</span>,</span>
<span>    cell_color <span class="op">=</span> <span class="st">"total_expr"</span>, </span>
<span>    color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>    gradient_style <span class="op">=</span> <span class="st">"sequential"</span>,</span>
<span>    group_by <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    point_border_stroke <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    point_alpha <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>    show_image <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fsplit_join%2F5_brain_joined.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2" number="7">
<h2 number="7" id="session-info">
<span class="header-section-number">7</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.4

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Giotto_4.1.3      GiottoClass_0.4.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1            viridisLite_0.4.2           farver_2.1.2               
 [4] dplyr_1.1.4                 GiottoVisuals_0.2.5         R.utils_2.12.3             
 [7] fastmap_1.2.0               SingleCellExperiment_1.26.0 lazyeval_0.2.2             
[10] digest_0.6.37               lifecycle_1.0.4             terra_1.7-78               
[13] magrittr_2.0.3              compiler_4.4.1              rlang_1.1.4                
[16] tools_4.4.1                 igraph_2.0.3                utf8_1.2.4                 
[19] data.table_1.16.0           knitr_1.48                  labeling_0.4.3             
[22] S4Arrays_1.4.0              htmlwidgets_1.6.4           bit_4.5.0                  
[25] sp_2.1-4                    reticulate_1.39.0           DelayedArray_0.30.0        
[28] RColorBrewer_1.1-3          abind_1.4-8                 withr_3.0.1                
[31] purrr_1.0.2                 BiocGenerics_0.50.0         R.oo_1.26.0                
[34] grid_4.4.1                  stats4_4.4.1                fansi_1.0.6                
[37] colorspace_2.1-1            ggplot2_3.5.1               scales_1.3.0               
[40] gtools_3.9.5                SummarizedExperiment_1.34.0 cli_3.6.3                  
[43] rmarkdown_2.28              crayon_1.5.3                generics_0.1.3             
[46] rstudioapi_0.16.0           httr_1.4.7                  rjson_0.2.21               
[49] zlibbioc_1.50.0             parallel_4.4.1              XVector_0.44.0             
[52] matrixStats_1.4.1           vctrs_0.6.5                 Matrix_1.7-0               
[55] jsonlite_1.8.9              IRanges_2.38.0              S4Vectors_0.42.0           
[58] bit64_4.5.2                 ggrepel_0.9.6               scattermore_1.2            
[61] hdf5r_1.3.10                magick_2.8.4                GiottoUtils_0.2.0          
[64] plotly_4.10.4               tidyr_1.3.1                 glue_1.8.0                 
[67] codetools_0.2-20            cowplot_1.1.3               gtable_0.3.5               
[70] deldir_2.0-4                GenomeInfoDb_1.40.0         GenomicRanges_1.56.0       
[73] UCSC.utils_1.0.0            munsell_0.5.1               tibble_3.2.1               
[76] pillar_1.9.0                htmltools_0.5.8.1           GenomeInfoDbData_1.2.12    
[79] R6_2.5.1                    evaluate_1.0.0              lattice_0.22-6             
[82] Biobase_2.64.0              png_0.1-8                   R.methodsS3_1.8.2          
[85] backports_1.5.0             SpatialExperiment_1.14.0    Rcpp_1.0.13                
[88] SparseArray_1.4.1           checkmate_2.3.2             colorRamp2_0.1.0           
[91] xfun_0.47                   MatrixGenerics_1.16.0       pkgconfig_2.0.3  </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
