<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to the Giotto Suite classes • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the Giotto Suite classes">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Process Giotto</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactive selection</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to the Giotto Suite classes</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/structure.Rmd" class="external-link"><code>vignettes/structure.Rmd</code></a></small>
      <div class="hidden name"><code>structure.Rmd</code></div>

    </div>

    
    
<p><em>Giotto</em> is a technique-agnostic framework and toolbox for spatial-omic analysis. Its structure and classes are designed to be flexible, intuitive, and readable. The framework supports working with both aggregate (cell x count) and un-aggregated spatial data where the polygon annotations are separate from the spatial expression data.</p>
<div class="section level2" number="1">
<h2 number="1" id="giotto-object-structure">
<span class="header-section-number">1</span> Giotto Object Structure<a class="anchor" aria-label="anchor" href="#giotto-object-structure"></a>
</h2>
<p>Usage of the <em>Giotto</em> package revolves around the <code>giotto</code> object. This is an S4 class that holds spatial expression data and facilitates its manipulation and visualization with the <em>Giotto</em> package"s functions. Additional metadata and other outputs generated from certain functions, which may be used in downstream analyses, are also be stored within the <code>giotto</code> object. Its self-contained nature provides a convenient representation of the entire spatial experiment and is why most <em>Giotto</em> functions take a given <code>giotto</code> object as input and return a <code>giotto</code> object as output.</p>
<p>Data is organized within the <code>giotto</code> object in defined <code>slots</code> as described in the diagram below.</p>
<p><img src="images%2Fstructure%2FGiotto_suite_object-01.svg"><img src="images%2Fstructure%2FGiotto_suite_object-02.svg"><img src="images%2Fstructure%2FGiotto_suite_object-03.svg"></p>
</div>
<div class="section level2" number="2">
<h2 number="2" id="nested-organization-of-the-giotto-object">
<span class="header-section-number">2</span> Nested Organization of the Giotto Object<a class="anchor" aria-label="anchor" href="#nested-organization-of-the-giotto-object"></a>
</h2>
<p>Biology happens across multiple scales of size and types of modalities. While it is possible to simply generate a new object for each combination of the two, the fact that data from most spatial methods are both high resolution and spatially contiguous, requires a more flexible approach that permits the coexistence of multiple spatial units within the same object. This allows the user to define the spatial unit(s) of biology that are most relevant to the analysis and re-aggregate the feature information to those units.</p>
<p>With this organization it is convenient to compare expression across different spatial units. Additionally, by determining spatial overlaps between these spatial units, it becomes possible to represent the hierarchical organization of biological subunits and make queries using it.</p>
<div class="section level3" number="2.1">
<h3 number="2.1" id="spatial-unit-and-feature-type">
<span class="header-section-number">2.1</span> Spatial unit and feature type<a class="anchor" aria-label="anchor" href="#spatial-unit-and-feature-type"></a>
</h3>
<p>To accommodate this complexity, information is subnested within many of the <code>giotto</code> object“s slots first by <code>spat_unit</code> (spatial unit) and then by <code>feat_type</code> (feature type). This structurally separates each set of information within <em>Giotto</em>”s framework so that there is minimal ambiguity.</p>
<p>A summary of what information the object contains can be viewed by directly returning it.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">vizmini</span> <span class="op">&lt;-</span> <span class="fu">GiottoData</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/loadGiottoMini.html" class="external-link">loadGiottoMini</a></span><span class="op">(</span><span class="st">"vizgen"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vizmini</span></span></code></pre></div>
<p>Included below is a description of the <code>giotto</code> object subnesting for each data slot and also the accessor functions for setting and getting information from them.</p>
<table class="table">
<colgroup>
<col width="24%">
<col width="21%">
<col width="30%">
<col width="22%">
</colgroup>
<thead><tr class="header">
<th>Slot</th>
<th>Nested</th>
<th>Example</th>
<th>Internal Accessors</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>@expression</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td></td>
<td>getExpression()</td>
</tr>
<tr class="even">
<td></td>
<td>
<span style="color: pink;">feat_type</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: pink;">rna</span> - <span style="color: darkred;">raw</span>
</td>
<td>setExpression()</td>
</tr>
<tr class="odd">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@cell_metadata</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: pink;">rna</span>
</td>
<td>getCellMetadata()</td>
</tr>
<tr class="even">
<td></td>
<td><span style="color: pink;">feat_type</span></td>
<td></td>
<td>setCellMetadata()</td>
</tr>
<tr class="odd">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="even">
<td><strong>@feat_metadata</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: pink;">rna</span>
</td>
<td>getFeatMetadata()</td>
</tr>
<tr class="odd">
<td></td>
<td><span style="color: pink;">feat_type</span></td>
<td></td>
<td>setFeatMetadata()</td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@spatial_grid</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td>
<span style="color: darkturquoise;">grid</span>- <span style="color: darkred;">grid</span>
</td>
<td>getSpatialGrid()</td>
</tr>
<tr class="even">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td>setSpatialGrid()</td>
</tr>
<tr class="odd">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="even">
<td><strong>@dimension_reduction</strong></td>
<td>
<span style="color: purple;">approach</span> -</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td></td>
<td>getDimReduction()</td>
</tr>
<tr class="even">
<td></td>
<td>
<span style="color: pink;">feat_type</span> -</td>
<td>
<span style="color: purple;">cells</span> - <span style="color: darkturquoise;">cell</span> - <span style="color: pink;">rna</span> - <span style="color: orange;">pca</span> - <span style="color: darkred;">pca</span>
</td>
<td>setDimReduction()</td>
</tr>
<tr class="odd">
<td></td>
<td>
<span style="color: orange;">method</span> -</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="even">
<td><strong>@multiomics</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>
<span style="color: pink;">feat_type</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: pink;">rna-protein</span> - <span style="color: orange;">WNN</span> - <span style="color: darkred;">theta_weighted_matrix</span>
</td>
<td>getMultiomics()</td>
</tr>
<tr class="even">
<td></td>
<td>
<span style="color: orange;">method</span> -</td>
<td></td>
<td>setMultiomics()</td>
</tr>
<tr class="odd">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@nn_network</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span>-</td>
<td></td>
<td>getNearestNetwork()</td>
</tr>
<tr class="even">
<td></td>
<td>
<span style="color: orange;">method</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: orange;">sNN</span> - <span style="color: darkred;">sNN_results1</span>
</td>
<td>setNearestNetwork()</td>
</tr>
<tr class="odd">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@spatial_enrichment</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td></td>
<td>getSpatialEnrichment()</td>
</tr>
<tr class="even">
<td></td>
<td>
<span style="color: pink;">feat_type</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: pink;">rna</span> - <span style="color: darkred;">results1</span>
</td>
<td>setSpatialEnrichment()</td>
</tr>
<tr class="odd">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@spatial_info</strong></td>
<td><span style="color: darkturquoise;">spat_unit</span></td>
<td><span style="color: darkturquoise;">cell</span></td>
<td>getPolygonInfo()</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td>setPolygonInfo()</td>
</tr>
<tr class="odd">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="even">
<td><strong>@spatial_locs</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: darkred;">raw</span>
</td>
<td>getSpatialLocations()</td>
</tr>
<tr class="odd">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td>setSpatialLocations()</td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@spatial_network</strong></td>
<td>
<span style="color: darkturquoise;">spat_unit</span> -</td>
<td>
<span style="color: darkturquoise;">cell</span> - <span style="color: darkred;">Delaunay_network1</span>
</td>
<td>getSpatialNetwork()</td>
</tr>
<tr class="even">
<td></td>
<td><span style="color: darkred;">name</span></td>
<td></td>
<td>setSpatialNetwork()</td>
</tr>
<tr class="odd">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="even">
<td><strong>@feat_info</strong></td>
<td><span style="color: pink;">feat_type</span></td>
<td><span style="color: pink;">rna</span></td>
<td>getFeatureInfo()</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td>setFeatureInfo()</td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@images</strong></td>
<td><span style="color: darkred;">name</span></td>
<td><span style="color: darkred;">image</span></td>
<td>getGiottoImage()</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td>setGiottoImage()</td>
</tr>
<tr class="odd">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="even">
<td><strong>@largeImages</strong></td>
<td><span style="color: darkred;">name</span></td>
<td><span style="color: darkred;">image</span></td>
<td>getGiottoImage()</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td>setGiottoImage()</td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
<tr class="odd">
<td><strong>@instructions</strong></td>
<td></td>
<td></td>
<td>instructions()</td>
</tr>
<tr class="even">
<td>—————————</td>
<td>————————</td>
<td>———————————-</td>
<td>————————-</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3" number="2.2">
<h3 number="2.2" id="show-and-list-functions">
<span class="header-section-number">2.2</span> Show and list functions<a class="anchor" aria-label="anchor" href="#show-and-list-functions"></a>
</h3>
<p>Show and list functions are also provided for determining what information exists within each of these slots and its nesting.</p>
<ul>
<li>
<code>show</code> functions print a preview of all the data within the slot, but do not return information</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoSpatLocs.html" class="external-link">showGiottoSpatLocs</a></span><span class="op">(</span><span class="va">vizmini</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/list_expression.html" class="external-link">list_expression</a></span><span class="op">(</span><span class="va">vizmini</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find specific spat_unit objects #</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/list_expression.html" class="external-link">list_expression</a></span><span class="op">(</span><span class="va">vizmini</span>, </span>
<span>                spat_unit <span class="op">=</span> <span class="st">"z0"</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>
<code>list names</code> (internal) functions return a <code>vector</code> of object names at the specified nesting</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/list_expression_names.html" class="external-link">list_expression_names</a></span><span class="op">(</span><span class="va">vizmini</span>, </span>
<span>                      spat_unit <span class="op">=</span> <span class="st">"z1"</span>, </span>
<span>                      feat_type <span class="op">=</span> <span class="st">"rna"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="2.3">
<h3 number="2.3" id="provenance">
<span class="header-section-number">2.3</span> Provenance<a class="anchor" aria-label="anchor" href="#provenance"></a>
</h3>
<p>Going further, sometimes different sources of information can be used when aggregating to a particular spatial unit. This is most easily shown with the subcellular datasets from the Vizgen MERSCOPE platform which provide both feature polygon information for multiple confocal planes within a tissue. The aggregated information produced then could be drawn from different z-planes or combinations thereof. Giotto tracks this provenance information for each set of aggregated data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/getExpression.html" class="external-link">getExpression</a></span><span class="op">(</span><span class="va">vizmini</span>, </span>
<span>                          spat_unit <span class="op">=</span> <span class="st">"aggregate"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/prov-generic.html" class="external-link">prov</a></span><span class="op">(</span><span class="va">expr_mat</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="3">
<h2 number="3" id="giotto-subobjects">
<span class="header-section-number">3</span> Giotto subobjects<a class="anchor" aria-label="anchor" href="#giotto-subobjects"></a>
</h2>
<p><em>Giotto</em> 3.0 update introduced S4 subobjects that are used within the <code>giotto</code> object and its processing. These subobjects provide more formalized definitions for what information and formatting is needed in each of the <code>giotto</code> object slots in order for it to be functional. These objects are standalone and extensible and commonly used spatial manipulation and plotting methods are being implemented for them.</p>
<p>In addition, these subobjects carry several pieces of metadata in additional slots alongside the main information (e.g. also slots for <code>spat_unit</code> and <code>feat_type</code> alongside the <code>exprDT</code> slot for the <code>exprObj</code> S4). This makes it so that nesting information is retained when they are taken out of the <code>giotto</code> object and that nesting information does not need to be supplied anymore when interacting with the <code>setter</code> functions.</p>
<p><code>getter</code> functions now have an <code>output</code> param that <strong>defaults</strong> to extracting the information from the <code>giotto</code> object as the S4 subobject. When extracting information that will be modified and then returned to the <code>giotto</code> object, it is preferred that the information is extracted as the S4 both so that tagged information is not lost, and because it is convenient to work with the S4"s main data slot through the <code>[</code> and <code>[&lt;-</code> generics (see Section 3.5).</p>
<div class="section level3" number="3.1">
<h3 number="3.1" id="creating-an-s4-subobject">
<span class="header-section-number">3.1</span> Creating an S4 subobject<a class="anchor" aria-label="anchor" href="#creating-an-s4-subobject"></a>
</h3>
<div class="section level4" number="3.1.1">
<h4 number="3.1.1" id="constructors">
<span class="header-section-number">3.1.1</span> Constructors<a class="anchor" aria-label="anchor" href="#constructors"></a>
</h4>
<p>For directly creating a subobject, constructor functions can be used.</p>
<p>constructors</p>
<p><code><a href="https://drieslab.github.io/GiottoClass/reference/createExprObj.html" class="external-link">createExprObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createCellMetaObj.html" class="external-link">createCellMetaObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createFeatMetaObj.html" class="external-link">createFeatMetaObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createDimObj.html" class="external-link">createDimObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetObj.html" class="external-link">createNearestNetObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createSpatLocsObj.html" class="external-link">createSpatLocsObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createSpatNetObj.html" class="external-link">createSpatNetObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createSpatEnrObj.html" class="external-link">createSpatEnrObj()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createSpatialGrid.html" class="external-link">createSpatialGrid()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPoints.html" class="external-link">createGiottoPoints()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html" class="external-link">createGiottoPolygonsFromDfr()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoPolygon.html" class="external-link">createGiottoPolygonsFromMask()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoImage.html" class="external-link">createGiottoImage()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoLargeImage.html" class="external-link">createGiottoLargeImage()</a></code></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coords</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span></span>
<span>  sdimx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>,</span>
<span>  sdimy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>,</span>
<span>  cell_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"C"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">st</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createSpatLocsObj.html" class="external-link">createSpatLocsObj</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"test"</span>,</span>
<span>                        spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                        coordinates <span class="op">=</span> <span class="va">coords</span>,</span>
<span>                        provenance <span class="op">=</span> <span class="st">"cell"</span><span class="op">)</span></span></code></pre></div>
<p>There are non numeric or integer columns for the spatial location input at column position(s): 3 The first non-numeric column will be considered as a cell ID to test for consistency with the expression matrix. Other non numeric columns will be removed</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">st</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4" number="3.1.2">
<h4 number="3.1.2" id="readers">
<span class="header-section-number">3.1.2</span> Readers<a class="anchor" aria-label="anchor" href="#readers"></a>
</h4>
<p>Alternatively, read functions can be used to take named nested lists of raw data input and convert them to lists of subobjects which are directly usable by the setter functions.</p>
<p>readers</p>
<p><code><a href="https://drieslab.github.io/GiottoClass/reference/readPolygonData.html" class="external-link">readPolygonData()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readFeatData.html" class="external-link">readFeatData()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readExprData.html" class="external-link">readExprData()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readCellMetadata.html" class="external-link">readCellMetadata()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readFeatMetadata.html" class="external-link">readFeatMetadata()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readSpatLocsData.html" class="external-link">readSpatLocsData()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readSpatNetData.html" class="external-link">readSpatNetData()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readSpatEnrichData.html" class="external-link">readSpatEnrichData()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readDimReducData.html" class="external-link">readDimReducData()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/readNearestNetData.html" class="external-link">readNearestNetData()</a></code></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/readSpatLocsData.html" class="external-link">readSpatLocsData</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>cell2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>test1 <span class="op">=</span> <span class="va">coords</span>,</span>
<span>                                         test2 <span class="op">=</span> <span class="va">coords</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>There are non numeric or integer columns for the spatial location input at column position(s): 3. The first non-numeric column will be considered as a cell ID to test for consistency with the expression matrix. Other non numeric columns will be removed</p>
<p>There are non numeric or integer columns for the spatial location input at column position(s): 3. The first non-numeric column will be considered as a cell ID to test for consistency with the expression matrix. Other non numeric columns will be removed</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st2</span></span></code></pre></div>
</div>
</div>
<div class="section level3" number="3.2">
<h3 number="3.2" id="giotto-accessors">
<span class="header-section-number">3.2</span> Giotto Accessors<a class="anchor" aria-label="anchor" href="#giotto-accessors"></a>
</h3>
<p><em>Giotto</em> provides <code>getter</code> and <code>setter</code> functions for manually accessing the information contained within the <code>giotto</code> object. Note that the <code>setters</code> require that the data be provided as compatible S4 subobjects or lists thereof. External data can read into the appropriate formats using the above <code>reader</code> functions. The <code>getter</code> functions return S4 subobjects by default.</p>
<p>getters</p>
<p><code><a href="https://drieslab.github.io/GiottoClass/reference/getExpression.html" class="external-link">getExpression()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getCellMetadata.html" class="external-link">getCellMetadata()</a></code> <code>getFeatMetadata()</code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getSpatialLocations.html" class="external-link">getSpatialLocations()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getDimReduction.html" class="external-link">getDimReduction()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getNearestNetwork.html" class="external-link">getNearestNetwork()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getSpatialNetwork.html" class="external-link">getSpatialNetwork()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getPolygonInfo.html" class="external-link">getPolygonInfo()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getFeatureInfo.html" class="external-link">getFeatureInfo()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getSpatialEnrichment.html" class="external-link">getSpatialEnrichment()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/getGiottoImage.html" class="external-link">getGiottoImage()</a></code></p>
<p>setters</p>
<p><code><a href="https://drieslab.github.io/GiottoClass/reference/setExpression.html" class="external-link">setExpression()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setCellMetadata.html" class="external-link">setCellMetadata()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setFeatureMetadata.html" class="external-link">setFeatureMetadata()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setSpatialLocations.html" class="external-link">setSpatialLocations()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setDimReduction.html" class="external-link">setDimReduction()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setNearestNetwork.html" class="external-link">setNearestNetwork()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setSpatialNetwork.html" class="external-link">setSpatialNetwork()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setPolygonInfo.html" class="external-link">setPolygonInfo()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setFeatureInfo.html" class="external-link">setFeatureInfo()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setSpatialEnrichment.html" class="external-link">setSpatialEnrichment()</a></code> <code><a href="https://drieslab.github.io/GiottoClass/reference/setGiottoImage.html" class="external-link">setGiottoImage()</a></code></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expval</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/getExpression.html" class="external-link">getExpression</a></span><span class="op">(</span><span class="va">vizmini</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expval</span></span></code></pre></div>
</div>
<div class="section level3" number="3.3">
<h3 number="3.3" id="get-and-set-s4-spat_unit-feat_type-provenance">
<span class="header-section-number">3.3</span> Get and set S4 spat_unit, feat_type, provenance<a class="anchor" aria-label="anchor" href="#get-and-set-s4-spat_unit-feat_type-provenance"></a>
</h3>
<p><code><a href="https://drieslab.github.io/GiottoClass/reference/spatUnit-generic.html" class="external-link">spatUnit()</a></code>, <code><a href="https://drieslab.github.io/GiottoClass/reference/featType-generic.html" class="external-link">featType()</a></code>, and <code><a href="https://drieslab.github.io/GiottoClass/reference/prov-generic.html" class="external-link">prov()</a></code> are replacement functions for tagged spatial unit, feature type, and provenance information respectively.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># spat_unit</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/spatUnit-generic.html" class="external-link">spatUnit</a></span><span class="op">(</span><span class="va">expval</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"new_spat"</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/spatUnit-generic.html" class="external-link">spatUnit</a></span><span class="op">(</span><span class="va">expval</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># feat_type</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/featType-generic.html" class="external-link">featType</a></span><span class="op">(</span><span class="va">expval</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"new_feat"</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/featType-generic.html" class="external-link">featType</a></span><span class="op">(</span><span class="va">expval</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># provenance</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/prov-generic.html" class="external-link">prov</a></span><span class="op">(</span><span class="va">expval</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"cell"</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/prov-generic.html" class="external-link">prov</a></span><span class="op">(</span><span class="va">expval</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="3.4">
<h3 number="3.4" id="setting-an-s4-subobject">
<span class="header-section-number">3.4</span> Setting an S4 subobject<a class="anchor" aria-label="anchor" href="#setting-an-s4-subobject"></a>
</h3>
<p>The <code>spat_unit</code>, <code>feat_type</code>, and <code>name</code> params no longer need to be given when setting an S4 subobject with tagged information into a <code>giottoObject</code>. However, if input is given to the <code>set</code> function parameters then it is prioritized over the tagged information and the tagged information is updated.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set exprObj to tagged nesting location</span></span>
<span><span class="va">vizmini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/setExpression.html" class="external-link">setExpression</a></span><span class="op">(</span><span class="va">vizmini</span>, </span>
<span>                         <span class="va">expval</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/list_expression.html" class="external-link">list_expression</a></span><span class="op">(</span><span class="va">vizmini</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="3.5">
<h3 number="3.5" id="working-with-s4-subobjects">
<span class="header-section-number">3.5</span> Working with S4 subobjects<a class="anchor" aria-label="anchor" href="#working-with-s4-subobjects"></a>
</h3>
<p>Giotto"s S4 subobjects each wrap one main data object. The empty <code>[]</code> and <code>[]&lt;-</code> operators are defined as shorthand for directly accessing this slot that contains the data. For example, with a <code>spatLocsObj</code>:</p>
<p><code>class(spatLocsObj[])</code> is equivalent to <code>class(spatLocsObj@coordinates)</code></p>
<p>In this way, the S4 subobjects can be used in contexts that the wrapped objects could be.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/getSpatialLocations.html" class="external-link">getSpatialLocations</a></span><span class="op">(</span><span class="va">vizmini</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">st</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># With empty brackets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">st</span><span class="op">[</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><strong>Setting information</strong></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st</span><span class="op">[</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">coords</span></span>
<span></span>
<span><span class="va">st</span></span></code></pre></div>
</div>
<div class="section level3" number="3.6">
<h3 number="3.6" id="session-info">
<span class="header-section-number">3.6</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true"></a>R version <span class="dv">4</span>.<span class="fl">4.0</span> (<span class="dv">2024-04-24</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true"></a>Platform<span class="op">:</span><span class="st"> </span>x86_<span class="dv">64</span><span class="op">-</span>apple<span class="op">-</span>darwin20</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true"></a>Running under<span class="op">:</span><span class="st"> </span>macOS Sonoma <span class="fl">14.5</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true"></a>Matrix products<span class="op">:</span><span class="st"> </span>default</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true"></a>BLAS<span class="op">:</span><span class="st">   </span><span class="er">/</span>System<span class="op">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>Accelerate.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>Frameworks<span class="op">/</span>vecLib.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>libBLAS.dylib </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true"></a>LAPACK<span class="op">:</span><span class="st"> </span><span class="er">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>R.framework<span class="op">/</span>Versions<span class="op">/</span><span class="fl">4.4</span><span class="op">-</span>x86_<span class="dv">64</span><span class="op">/</span>Resources<span class="op">/</span>lib<span class="op">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true"></a>locale<span class="op">:</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>C<span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true"></a></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true"></a>time zone<span class="op">:</span><span class="st"> </span>America<span class="op">/</span>New_York</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true"></a>tzcode source<span class="op">:</span><span class="st"> </span>internal</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true"></a>attached base packages<span class="op">:</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true"></a></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true"></a>other attached packages<span class="op">:</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true"></a>[<span class="dv">1</span>] data.table_<span class="dv">1</span>.<span class="fl">15.4</span> Giotto_<span class="dv">4</span>.<span class="fl">1.0</span>      GiottoClass_<span class="dv">0</span>.<span class="fl">3.4</span></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true"></a></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true"></a>loaded via a <span class="kw">namespace</span> (and not attached)<span class="op">:</span></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true"></a><span class="st"> </span>[<span class="dv">1</span>] tidyselect_<span class="dv">1</span>.<span class="fl">2.1</span>            viridisLite_<span class="dv">0</span>.<span class="fl">4.2</span>           dplyr_<span class="dv">1</span>.<span class="fl">1.4</span>                </span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true"></a> [<span class="dv">4</span>] GiottoVisuals_<span class="dv">0</span>.<span class="fl">2.4</span>         lazyeval_<span class="dv">0</span>.<span class="fl">2.2</span>              fastmap_<span class="dv">1</span>.<span class="fl">2.0</span>              </span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true"></a> [<span class="dv">7</span>] SingleCellExperiment_<span class="dv">1</span>.<span class="fl">26.0</span> digest_<span class="dv">0</span>.<span class="fl">6.36</span>               lifecycle_<span class="dv">1</span>.<span class="fl">0.4</span>            </span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true"></a>[<span class="dv">10</span>] terra_<span class="fl">1.7</span><span class="dv">-78</span>                magrittr_<span class="dv">2</span>.<span class="fl">0.3</span>              dbscan_<span class="fl">1.2</span><span class="dv">-0</span>               </span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true"></a>[<span class="dv">13</span>] compiler_<span class="dv">4</span>.<span class="fl">4.0</span>              rlang_<span class="dv">1</span>.<span class="fl">1.4</span>                 tools_<span class="dv">4</span>.<span class="fl">4.0</span>                </span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true"></a>[<span class="dv">16</span>] igraph_<span class="dv">2</span>.<span class="fl">0.3</span>                utf8_<span class="dv">1</span>.<span class="fl">2.4</span>                  yaml_<span class="dv">2</span>.<span class="fl">3.10</span>                </span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true"></a>[<span class="dv">19</span>] knitr_<span class="fl">1.48</span>                  htmlwidgets_<span class="dv">1</span>.<span class="fl">6.4</span>           S4Arrays_<span class="dv">1</span>.<span class="fl">4.1</span>             </span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true"></a>[<span class="dv">22</span>] sp_<span class="fl">2.1</span><span class="dv">-4</span>                    reticulate_<span class="dv">1</span>.<span class="fl">38.0</span>           DelayedArray_<span class="dv">0</span>.<span class="fl">30.1</span>        </span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true"></a>[<span class="dv">25</span>] plyr_<span class="dv">1</span>.<span class="fl">8.9</span>                  xml2_<span class="dv">1</span>.<span class="fl">3.6</span>                  RColorBrewer_<span class="fl">1.1</span><span class="dv">-3</span>         </span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true"></a>[<span class="dv">28</span>] abind_<span class="fl">1.4</span><span class="dv">-5</span>                 withr_<span class="dv">3</span>.<span class="fl">0.0</span>                 purrr_<span class="dv">1</span>.<span class="fl">0.2</span>                </span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true"></a>[<span class="dv">31</span>] BiocGenerics_<span class="dv">0</span>.<span class="fl">50.0</span>         grid_<span class="dv">4</span>.<span class="fl">4.0</span>                  stats4_<span class="dv">4</span>.<span class="fl">4.0</span>               </span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true"></a>[<span class="dv">34</span>] fansi_<span class="dv">1</span>.<span class="fl">0.6</span>                 colorspace_<span class="fl">2.1</span><span class="dv">-1</span>            ggplot2_<span class="dv">3</span>.<span class="fl">5.1</span>              </span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true"></a>[<span class="dv">37</span>] scales_<span class="dv">1</span>.<span class="fl">3.0</span>                gtools_<span class="dv">3</span>.<span class="fl">9.5</span>                SummarizedExperiment_<span class="dv">1</span>.<span class="fl">34.0</span></span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true"></a>[<span class="dv">40</span>] cli_<span class="dv">3</span>.<span class="fl">6.3</span>                   rmarkdown_<span class="fl">2.27</span>              crayon_<span class="dv">1</span>.<span class="fl">5.3</span>               </span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true"></a>[<span class="dv">43</span>] generics_<span class="dv">0</span>.<span class="fl">1.3</span>              rstudioapi_<span class="dv">0</span>.<span class="fl">16.0</span>           reshape2_<span class="dv">1</span>.<span class="fl">4.4</span>             </span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true"></a>[<span class="dv">46</span>] httr_<span class="dv">1</span>.<span class="fl">4.7</span>                  rjson_<span class="dv">0</span>.<span class="fl">2.21</span>                stringr_<span class="dv">1</span>.<span class="fl">5.1</span>              </span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true"></a>[<span class="dv">49</span>] zlibbioc_<span class="dv">1</span>.<span class="fl">50.0</span>             parallel_<span class="dv">4</span>.<span class="fl">4.0</span>              XVector_<span class="dv">0</span>.<span class="fl">44.0</span>             </span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true"></a>[<span class="dv">52</span>] matrixStats_<span class="dv">1</span>.<span class="fl">3.0</span>           vctrs_<span class="dv">0</span>.<span class="fl">6.5</span>                 Matrix_<span class="fl">1.7</span><span class="dv">-0</span>               </span>
<span id="cb23-40"><a href="#cb23-40" aria-hidden="true"></a>[<span class="dv">55</span>] jsonlite_<span class="dv">1</span>.<span class="fl">8.8</span>              GiottoData_<span class="dv">0</span>.<span class="fl">2.13</span>           IRanges_<span class="dv">2</span>.<span class="fl">38.1</span>             </span>
<span id="cb23-41"><a href="#cb23-41" aria-hidden="true"></a>[<span class="dv">58</span>] S4Vectors_<span class="dv">0</span>.<span class="fl">42.1</span>            ggrepel_<span class="dv">0</span>.<span class="fl">9.5</span>               scattermore_<span class="fl">1.2</span>            </span>
<span id="cb23-42"><a href="#cb23-42" aria-hidden="true"></a>[<span class="dv">61</span>] systemfonts_<span class="dv">1</span>.<span class="fl">1.0</span>           magick_<span class="dv">2</span>.<span class="fl">8.4</span>                GiottoUtils_<span class="dv">0</span>.<span class="fl">1.10</span>         </span>
<span id="cb23-43"><a href="#cb23-43" aria-hidden="true"></a>[<span class="dv">64</span>] plotly_<span class="dv">4</span>.<span class="fl">10.4</span>               tidyr_<span class="dv">1</span>.<span class="fl">3.1</span>                 glue_<span class="dv">1</span>.<span class="fl">7.0</span>                 </span>
<span id="cb23-44"><a href="#cb23-44" aria-hidden="true"></a>[<span class="dv">67</span>] codetools_<span class="fl">0.2</span><span class="dv">-20</span>            cowplot_<span class="dv">1</span>.<span class="fl">1.3</span>               stringi_<span class="dv">1</span>.<span class="fl">8.4</span>              </span>
<span id="cb23-45"><a href="#cb23-45" aria-hidden="true"></a>[<span class="dv">70</span>] gtable_<span class="dv">0</span>.<span class="fl">3.5</span>                GenomeInfoDb_<span class="dv">1</span>.<span class="fl">40.1</span>         deldir_<span class="fl">2.0</span><span class="dv">-4</span>               </span>
<span id="cb23-46"><a href="#cb23-46" aria-hidden="true"></a>[<span class="dv">73</span>] GenomicRanges_<span class="dv">1</span>.<span class="fl">56.1</span>        UCSC.utils_<span class="dv">1</span>.<span class="fl">0.0</span>            munsell_<span class="dv">0</span>.<span class="fl">5.1</span>              </span>
<span id="cb23-47"><a href="#cb23-47" aria-hidden="true"></a>[<span class="dv">76</span>] tibble_<span class="dv">3</span>.<span class="fl">2.1</span>                pillar_<span class="dv">1</span>.<span class="fl">9.0</span>                htmltools_<span class="dv">0</span>.<span class="dv">5</span>.<span class="fl">8.1</span>          </span>
<span id="cb23-48"><a href="#cb23-48" aria-hidden="true"></a>[<span class="dv">79</span>] GenomeInfoDbData_<span class="dv">1</span>.<span class="fl">2.12</span>     R6_<span class="dv">2</span>.<span class="fl">5.1</span>                    evaluate_<span class="dv">0</span>.<span class="fl">24.0</span>            </span>
<span id="cb23-49"><a href="#cb23-49" aria-hidden="true"></a>[<span class="dv">82</span>] kableExtra_<span class="dv">1</span>.<span class="fl">4.0</span>            lattice_<span class="fl">0.22</span><span class="dv">-6</span>              Biobase_<span class="dv">2</span>.<span class="fl">64.0</span>             </span>
<span id="cb23-50"><a href="#cb23-50" aria-hidden="true"></a>[<span class="dv">85</span>] png_<span class="fl">0.1</span><span class="dv">-8</span>                   backports_<span class="dv">1</span>.<span class="fl">5.0</span>             SpatialExperiment_<span class="dv">1</span>.<span class="fl">14.0</span>   </span>
<span id="cb23-51"><a href="#cb23-51" aria-hidden="true"></a>[<span class="dv">88</span>] Rcpp_<span class="dv">1</span>.<span class="fl">0.13</span>                 svglite_<span class="dv">2</span>.<span class="fl">1.3</span>               SparseArray_<span class="dv">1</span>.<span class="fl">4.8</span>          </span>
<span id="cb23-52"><a href="#cb23-52" aria-hidden="true"></a>[<span class="dv">91</span>] checkmate_<span class="dv">2</span>.<span class="fl">3.2</span>             colorRamp2_<span class="dv">0</span>.<span class="fl">1.0</span>            xfun_<span class="fl">0.46</span>                  </span>
<span id="cb23-53"><a href="#cb23-53" aria-hidden="true"></a>[<span class="dv">94</span>] MatrixGenerics_<span class="dv">1</span>.<span class="fl">16.0</span>       pkgconfig_<span class="dv">2</span>.<span class="fl">0.3</span> </span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
