<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visium Prostate Integration â€¢ Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visium Prostate Integration">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/general_workflow.html">General Giotto Workflow</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
        <li>
          <a href="../articles/VisiumHD_Human_Colorectal_Cancer.html">VisiumHD Human Colorectal Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Open-ST</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/openst_mouse_hippocampus.html">Open-ST Mouse Hippocampus</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Process Giotto</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/process_giotto.html">Process Giotto convenience function</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactivity</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
        <li>
          <a href="../articles/interactive_giotto_vitessceR.html">Interactive visualization with vitessceR</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialdata.html">SpatialData</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Visium Prostate Integration</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/visium_prostate_integration.Rmd" class="external-link"><code>vignettes/visium_prostate_integration.Rmd</code></a></small>
      <div class="hidden name"><code>visium_prostate_integration.Rmd</code></div>

    </div>

    
    
<div class="section level2" number="1">
<h2 number="1" id="dataset-explanation">
<span class="header-section-number">1</span> Dataset explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p><a href="https://www.10xgenomics.com/spatial-transcriptomics/" class="external-link">10X genomics</a> recently launched a new platform to obtain spatial expression data using a Visium Spatial Gene Expression slide.</p>
<p>The Visium Cancer Prostate data to run this tutorial can be found <a href="https://www.10xgenomics.com/resources/datasets/human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe-1-standard-1-3-0" class="external-link">here</a> The Visium Normal Prostate data to run this tutorial can be found <a href="https://www.10xgenomics.com/resources/datasets/normal-human-prostate-ffpe-1-standard-1-3-0" class="external-link">here</a></p>
<p>Visium technology:</p>
<p><img src="images%2Fgeneral_figs%2Fvisium_technology.png" width="100%"></p>
<p>High resolution png from original tissue:</p>
<p><img src="images%2Fgeneral_figs%2FVisium_FFPE_Human_Normal_Prostate_image.png" width="50%" style="display: block; margin: auto;"></p>
<p><img src="images%2Fgeneral_figs%2FVisium_FFPE_Human_Prostate_Cancer_image.png" width="50%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2" number="2">
<h2 number="2" id="start-giotto">
<span class="header-section-number">2</span> Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed</span></span>
<span><span class="va">genv_exists</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_python.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 1. set working directory</span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"/path/to/results/"</span></span>
<span></span>
<span><span class="co"># 2. set giotto python path</span></span>
<span><span class="co"># set python path to your preferred python version path</span></span>
<span><span class="co"># set python path to NULL if you want to automatically install (only the 1st time) and use the giotto miniconda environment</span></span>
<span><span class="va">python_path</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3. create giotto instructions</span></span>
<span><span class="va">instructions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/giotto_instructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">results_folder</span>, </span>
<span>                                         save_plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                         show_plot <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                         return_plot <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                         python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 number="3" id="create-giotto-objects-and-join">
<span class="header-section-number">3</span> Create Giotto objects and join<a class="anchor" aria-label="anchor" href="#create-giotto-objects-and-join"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This dataset must be downlaoded manually; please do so and change the path below as appropriate</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/data/"</span></span>
<span></span>
<span><span class="co">## obese upper</span></span>
<span><span class="va">N_pros</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createGiottoVisiumObject.html">createGiottoVisiumObject</a></span><span class="op">(</span></span>
<span>    visium_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"Visium_FFPE_Human_Normal_Prostate"</span><span class="op">)</span>,</span>
<span>    expr_data <span class="op">=</span> <span class="st">"raw"</span>,</span>
<span>    png_name <span class="op">=</span> <span class="st">"tissue_lowres_image.png"</span>,</span>
<span>    gene_column_index <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    instructions <span class="op">=</span> <span class="va">instructions</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## obese lower</span></span>
<span><span class="va">C_pros</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createGiottoVisiumObject.html">createGiottoVisiumObject</a></span><span class="op">(</span></span>
<span>    visium_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"Visium_FFPE_Human_Prostate_Cancer/"</span><span class="op">)</span>,</span>
<span>    expr_data <span class="op">=</span> <span class="st">"raw"</span>,</span>
<span>    png_name <span class="op">=</span> <span class="st">"tissue_lowres_image.png"</span>,</span>
<span>    gene_column_index <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    instructions <span class="op">=</span> <span class="va">instructions</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># join giotto objects</span></span>
<span><span class="co"># joining with x_shift has the advantage that you can join both 2D and 3D data</span></span>
<span><span class="co"># x_padding determines how much distance is between each dataset</span></span>
<span><span class="co"># if x_shift = NULL, then the total shift will be guessed from the giotto image</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/joinGiottoObjects.html" class="external-link">joinGiottoObjects</a></span><span class="op">(</span>gobject_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">N_pros</span>, <span class="va">C_pros</span><span class="op">)</span>,</span>
<span>                               gobject_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NP"</span>, <span class="st">"CP"</span><span class="op">)</span>,</span>
<span>                               join_method <span class="op">=</span> <span class="st">"shift"</span>, </span>
<span>                               x_padding <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># join info is stored in this slot</span></span>
<span><span class="co"># simple list for now</span></span>
<span><span class="va">testcombo</span><span class="op">@</span><span class="va">join_info</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># check joined Giotto object</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/fDataDT.html" class="external-link">fDataDT</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoImageNames.html" class="external-link">showGiottoImageNames</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoSpatLocs.html" class="external-link">showGiottoSpatLocs</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoExpression.html" class="external-link">showGiottoExpression</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># this plots all the images by list_ID</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"in_tissue"</span>,</span>
<span>           image_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NP-image"</span>, <span class="st">"CP-image"</span><span class="op">)</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">"list_ID"</span>, </span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F0-spatPlot2D.png" width="100%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># this plots one selected image</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"in_tissue"</span>,</span>
<span>           image_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NP-image"</span><span class="op">)</span>, </span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F1-spatPlot2D.png" width="100%"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># this plots two selected images</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"in_tissue"</span>,</span>
<span>           image_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"NP-image"</span>, <span class="st">"CP-image"</span><span class="op">)</span>,</span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F2-spatPlot2D.png" width="100%"></p>
</div>
<div class="section level2" number="4">
<h2 number="4" id="process-giotto-objects">
<span class="header-section-number">4</span> Process Giotto Objects<a class="anchor" aria-label="anchor" href="#process-giotto-objects"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># subset on in-tissue spots</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span>
<span><span class="va">in_tissue_barcodes</span> <span class="op">&lt;-</span> <span class="va">metadata</span><span class="op">[</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span></span>
<span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html" class="external-link">subsetGiotto</a></span><span class="op">(</span><span class="va">testcombo</span>, </span>
<span>                          cell_ids <span class="op">=</span> <span class="va">in_tissue_barcodes</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## filter</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>                          expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                          feat_det_in_min_cells <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                          min_det_feats_per_cell <span class="op">=</span> <span class="fl">500</span>,</span>
<span>                          expression_values <span class="op">=</span> <span class="st">"raw"</span>,</span>
<span>                          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## normalize</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>                             scalefactor <span class="op">=</span> <span class="fl">6000</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## add gene &amp; cell statistics</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>                           expression_values <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/fDataDT.html" class="external-link">fDataDT</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testfeats</span> <span class="op">&lt;-</span> <span class="va">gene_metadata</span><span class="op">[</span><span class="va">perc_cells</span> <span class="op">&gt;</span> <span class="fl">20</span> <span class="op">&amp;</span> <span class="va">perc_cells</span> <span class="op">&lt;</span> <span class="fl">50</span><span class="op">]</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">110</span><span class="op">]</span><span class="op">$</span><span class="va">feat_ID</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/violinPlot.html" class="external-link">violinPlot</a></span><span class="op">(</span><span class="va">testcombo</span>, </span>
<span>           feats <span class="op">=</span> <span class="va">testfeats</span>, </span>
<span>           cluster_column <span class="op">=</span> <span class="st">"list_ID"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F3-violinPlot.png" width="100%"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotMetaDataHeatmap.html" class="external-link">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">testcombo</span>, </span>
<span>                    selected_feats <span class="op">=</span> <span class="va">testfeats</span>, </span>
<span>                    metadata_cols <span class="op">=</span> <span class="st">"list_ID"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F4-plotMetaDataHeatmap.png" width="60%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## visualize</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>           group_by <span class="op">=</span> <span class="st">"list_ID"</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"nr_feats"</span>, </span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F5-spatPlot2D.png" width="100%"></p>
</div>
<div class="section level2" number="5">
<h2 number="5" id="dimension-reduction">
<span class="header-section-number">5</span> Dimension Reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## PCA ##</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>                    center <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                    scale_unit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">testcombo</span>, </span>
<span>          ncp <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F6-screePlot.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2" number="6">
<h2 number="6" id="clustering">
<span class="header-section-number">6</span> Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<div class="section level3" number="6.1">
<h3 number="6.1" id="without-integration">
<span class="header-section-number">6.1</span> Without Integration<a class="anchor" aria-label="anchor" href="#without-integration"></a>
</h3>
<p>Integration is usually needed for dataset of different conditions to minimize batch effects. Without integration means without using any integration methods.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## cluster and run UMAP ##</span></span>
<span><span class="co"># sNN network (default)</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>                                  dim_reduction_to_use <span class="op">=</span> <span class="st">"pca"</span>, </span>
<span>                                  dim_reduction_name <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>                                  dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>                                  k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Leiden clustering</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>                             resolution <span class="op">=</span> <span class="fl">0.2</span>, </span>
<span>                             n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># UMAP</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">testcombo</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>, </span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F7-UMAP.png" width="80%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>           group_by <span class="op">=</span> <span class="st">"list_ID"</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F8-spatPlot2D.png" width="100%"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html" class="external-link">spatDimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>              cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F9-spatDimPlot2D.png" width="100%"></p>
</div>
<div class="section level3" number="6.2">
<h3 number="6.2" id="with-harmony-integration">
<span class="header-section-number">6.2</span> With Harmony integration<a class="anchor" aria-label="anchor" href="#with-harmony-integration"></a>
</h3>
<p>Harmony is a integration algorithm developed by <a href="https://www.nature.com/articles/s41592-019-0619-0" class="external-link">Korsunsky, I. et al.</a>. It was designed for integration of single cell data but also work well on spatial datasets.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## run harmony integration</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runGiottoHarmony.html">runGiottoHarmony</a></span><span class="op">(</span><span class="va">testcombo</span>, </span>
<span>                              vars_use <span class="op">=</span> <span class="st">"list_ID"</span>, </span>
<span>                              do_pca <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## sNN network (default)</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>                                  dim_reduction_to_use <span class="op">=</span> <span class="st">"harmony"</span>, </span>
<span>                                  dim_reduction_name <span class="op">=</span> <span class="st">"harmony"</span>, </span>
<span>                                  name <span class="op">=</span> <span class="st">"NN.harmony"</span>,</span>
<span>                                  dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>                                  k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Leiden clustering</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>                             network_name <span class="op">=</span> <span class="st">"NN.harmony"</span>, </span>
<span>                             resolution <span class="op">=</span> <span class="fl">0.2</span>, </span>
<span>                             n_iterations <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>                             name <span class="op">=</span> <span class="st">"leiden_harmony"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># UMAP dimension reduction</span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">testcombo</span>, </span>
<span>                     dim_reduction_name <span class="op">=</span> <span class="st">"harmony"</span>, </span>
<span>                     dim_reduction_to_use <span class="op">=</span> <span class="st">"harmony"</span>, </span>
<span>                     name <span class="op">=</span> <span class="st">"umap_harmony"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>         dim_reduction_name <span class="op">=</span> <span class="st">"umap_harmony"</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">"leiden_harmony"</span>, </span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F10-UMAP.png" width="80%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># If you want to show NN network information, you will need to specify these arguments in the plotUMAP function</span></span>
<span><span class="co"># show_NN_network = TRUE, nn_network_to_use = "sNN" , network_name = "NN.harmony"</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>           group_by <span class="op">=</span> <span class="st">"list_ID"</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"leiden_harmony"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F11-spatPlot2D.png" width="100%"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatDimPlot.html" class="external-link">spatDimPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>              dim_reduction_to_use <span class="op">=</span> <span class="st">"umap"</span>, </span>
<span>              dim_reduction_name <span class="op">=</span> <span class="st">"umap_harmony"</span>,</span>
<span>              cell_color <span class="op">=</span> <span class="st">"leiden_harmony"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F12-spatDimPlot2D.png" width="100%"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># compare to previous results</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F13-spatPlot2D.png" width="100%"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"leiden_harmony"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F14-spatPlot2D.png" width="100%"></p>
</div>
</div>
<div class="section level2" number="7">
<h2 number="7" id="cell-type-annotation">
<span class="header-section-number">7</span> Cell type annotation<a class="anchor" aria-label="anchor" href="#cell-type-annotation"></a>
</h2>
<p>Visium spatial transcriptomics does not provide single-cell resolution, making cell type annotation a harder problem. Giotto provides several ways to calculate enrichment of specific cell-type signature gene list:</p>
<ul>
<li>PAGE</li>
<li>hypergeometric test</li>
<li>Rank</li>
<li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02362-7" class="external-link">DWLS Deconvolution</a></li>
</ul>
<p>This is also the easiest way to integrate Visium datasets with single cell data. Example shown here is from <a href="https://pubmed.ncbi.nlm.nih.gov/33032611/" class="external-link">Ma et al.</a> from two prostate cancer patients. The raw dataset can be found <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE157703" class="external-link">here</a> Giotto_SC is processed variable in the <a href="https://giottosuite.com/articles/singlecell_prostate_standard_Sep29_21.html" class="external-link">single cell RNAseq tutorial</a>. You can also get access to the processed files of this dataset using getSpatialDataset</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download data to results directory ####</span></span>
<span><span class="co"># if wget is installed, set method = "wget"</span></span>
<span><span class="co"># if you run into authentication issues with wget, then add " extra = "--no-check-certificate" "</span></span>
<span><span class="fu">GiottoData</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoData/reference/getSpatialDataset.html" class="external-link">getSpatialDataset</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"scRNA_prostate"</span>, </span>
<span>                              directory <span class="op">=</span> <span class="va">data_path</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"prostate_sc_expression_matrix.csv.gz"</span><span class="op">)</span></span>
<span><span class="va">sc_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_path</span>, <span class="st">"prostate_sc_metadata.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_SC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoObject.html" class="external-link">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">sc_expression</span>,</span>
<span>                                instructions <span class="op">=</span> <span class="va">instructions</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">giotto_SC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/addCellMetadata.html" class="external-link">addCellMetadata</a></span><span class="op">(</span><span class="va">giotto_SC</span>, </span>
<span>                             new_metadata <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">sc_metadata</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">giotto_SC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span><span class="va">giotto_SC</span><span class="op">)</span></span></code></pre></div>
<div class="section level3" number="7.1">
<h3 number="7.1" id="page-enrichment">
<span class="header-section-number">7.1</span> PAGE enrichment<a class="anchor" aria-label="anchor" href="#page-enrichment"></a>
</h3>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create PAGE matrix</span></span>
<span><span class="co"># PAGE matrix should be a binary matrix with each row represent a gene marker and each column represent a cell type</span></span>
<span><span class="co"># markers_scran is generated from single cell analysis ()</span></span>
<span><span class="va">markers_scran</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">giotto_SC</span>, </span>
<span>                                       method <span class="op">=</span> <span class="st">"scran"</span>,</span>
<span>                                       expression_values <span class="op">=</span> <span class="st">"normalized"</span>, </span>
<span>                                       cluster_column <span class="op">=</span> <span class="st">"prostate_labels"</span>,</span>
<span>                                       min_feats <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">topgenes_scran</span> <span class="op">&lt;-</span> <span class="va">markers_scran</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">10</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">]</span></span>
<span></span>
<span><span class="va">celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">markers_scran</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sign_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">celltypes</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">sign_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">topgenes_scran</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">topgenes_scran</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="va">celltypes</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">feats</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">PAGE_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignMatrixPAGE.html">makeSignMatrixPAGE</a></span><span class="op">(</span>sign_names <span class="op">=</span> <span class="va">celltypes</span>,</span>
<span>                                  sign_list <span class="op">=</span> <span class="va">sign_list</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPAGEEnrich.html">runPAGEEnrich</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>                           sign_matrix <span class="op">=</span> <span class="va">PAGE_matrix</span>,</span>
<span>                           min_overlap_genes <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">PAGE_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot PAGE enrichment result</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html" class="external-link">spatCellPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>             spat_enr_names <span class="op">=</span> <span class="st">"PAGE"</span>,</span>
<span>             cell_annotation_values <span class="op">=</span> <span class="va">cell_types</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,</span>
<span>             cow_n_col <span class="op">=</span> <span class="fl">2</span>,</span>
<span>             coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>             point_size <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F15-spatCellPlot2D.png" width="90%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="7.2">
<h3 number="7.2" id="hypergeometric-test">
<span class="header-section-number">7.2</span> Hypergeometric test<a class="anchor" aria-label="anchor" href="#hypergeometric-test"></a>
</h3>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runHyperGeometricEnrich.html">runHyperGeometricEnrich</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>                                     expression_values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                                     sign_matrix <span class="op">=</span> <span class="va">PAGE_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">PAGE_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html" class="external-link">spatCellPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>             spat_enr_names <span class="op">=</span> <span class="st">"hypergeometric"</span>,</span>
<span>             cell_annotation_values <span class="op">=</span> <span class="va">cell_types</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,</span>
<span>             cow_n_col <span class="op">=</span> <span class="fl">2</span>,</span>
<span>             coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>             point_size <span class="op">=</span> <span class="fl">1.75</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F16-spatCellPlot2D.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="7.3">
<h3 number="7.3" id="rank-enrichment">
<span class="header-section-number">7.3</span> Rank Enrichment<a class="anchor" aria-label="anchor" href="#rank-enrichment"></a>
</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create rank matrix, not that rank matrix is different from PAGE</span></span>
<span><span class="co"># A count matrix and a vector for all cell labels will be needed</span></span>
<span><span class="va">sc_expression_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/getExpression.html" class="external-link">getExpression</a></span><span class="op">(</span><span class="va">giotto_SC</span>,</span>
<span>                                    values <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>                                    output <span class="op">=</span> <span class="st">"matrix"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">prostate_feats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">giotto_SC</span><span class="op">)</span><span class="op">$</span><span class="va">prostate_label</span></span>
<span></span>
<span><span class="va">rank_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignMatrixRank.html">makeSignMatrixRank</a></span><span class="op">(</span>sc_matrix <span class="op">=</span> <span class="va">sc_expression_norm</span>,</span>
<span>                                  sc_cluster_ids <span class="op">=</span> <span class="va">prostate_feats</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/row-plus-colnames-generic.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rank_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">prostate_feats</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runRankEnrich.html">runRankEnrich</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>                           sign_matrix <span class="op">=</span> <span class="va">rank_matrix</span>,</span>
<span>                           expression_values <span class="op">=</span> <span class="st">"normalized"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot Rank enrichment result</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html" class="external-link">spatCellPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>               spat_enr_names <span class="op">=</span> <span class="st">"rank"</span>,</span>
<span>               cell_annotation_values <span class="op">=</span> <span class="va">cell_types</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">2</span>,</span>
<span>               coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>               point_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F17-spatCellPlot2D.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3" number="7.4">
<h3 number="7.4" id="dwls-deconvolution">
<span class="header-section-number">7.4</span> DWLS Deconvolution<a class="anchor" aria-label="anchor" href="#dwls-deconvolution"></a>
</h3>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create DWLS matrix, not that DWLS matrix is different from PAGE and rank</span></span>
<span><span class="co"># A count matrix a vector for a list of gene signatures and a vector for all cell labels will be needed</span></span>
<span><span class="va">DWLS_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignMatrixDWLSfromMatrix.html">makeSignMatrixDWLSfromMatrix</a></span><span class="op">(</span>matrix <span class="op">=</span> <span class="va">sc_expression_norm</span>,</span>
<span>                                            cell_type <span class="op">=</span> <span class="va">prostate_feats</span>,</span>
<span>                                            sign_gene <span class="op">=</span> <span class="va">topgenes_scran</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testcombo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runDWLSDeconv.html">runDWLSDeconv</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>, </span>
<span>                           sign_matrix <span class="op">=</span> <span class="va">DWLS_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot DWLS deconvolution result</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatCellPlot.html" class="external-link">spatCellPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testcombo</span>,</span>
<span>               spat_enr_names <span class="op">=</span> <span class="st">"DWLS"</span>,</span>
<span>               cell_annotation_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">prostate_feats</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,</span>
<span>               cow_n_col <span class="op">=</span> <span class="fl">2</span>,</span>
<span>               coord_fix_ratio <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>               point_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fvisium_prostate_integration%2F18-spatCellPlot2D.png" width="100%"></p>
</div>
</div>
<div class="section level2" number="8">
<h2 number="8" id="session-info">
<span class="header-section-number">8</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true"></a>R version <span class="dv">4</span>.<span class="fl">4.0</span> (<span class="dv">2024-04-24</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true"></a>Platform<span class="op">:</span><span class="st"> </span>x86_<span class="dv">64</span><span class="op">-</span>apple<span class="op">-</span>darwin20</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true"></a>Running under<span class="op">:</span><span class="st"> </span>macOS Sonoma <span class="dv">14</span>.<span class="fl">6.1</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true"></a></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true"></a>Matrix products<span class="op">:</span><span class="st"> </span>default</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true"></a>BLAS<span class="op">:</span><span class="st">   </span><span class="er">/</span>System<span class="op">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>Accelerate.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>Frameworks<span class="op">/</span>vecLib.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>libBLAS.dylib </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true"></a>LAPACK<span class="op">:</span><span class="st"> </span><span class="er">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>R.framework<span class="op">/</span>Versions<span class="op">/</span><span class="fl">4.4</span><span class="op">-</span>x86_<span class="dv">64</span><span class="op">/</span>Resources<span class="op">/</span>lib<span class="op">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true"></a></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true"></a>locale<span class="op">:</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>C<span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true"></a></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true"></a>time zone<span class="op">:</span><span class="st"> </span>America<span class="op">/</span>New_York</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true"></a>tzcode source<span class="op">:</span><span class="st"> </span>internal</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true"></a></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true"></a>attached base packages<span class="op">:</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true"></a></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true"></a>other attached packages<span class="op">:</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true"></a>[<span class="dv">1</span>] ggplot2_<span class="dv">3</span>.<span class="fl">5.1</span>     Giotto_<span class="dv">4</span>.<span class="fl">1.1</span>      GiottoClass_<span class="dv">0</span>.<span class="fl">3.5</span></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true"></a></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true"></a>loaded via a <span class="kw">namespace</span> (and not attached)<span class="op">:</span></span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true"></a><span class="st">  </span>[<span class="dv">1</span>] RColorBrewer_<span class="fl">1.1</span><span class="dv">-3</span>          rstudioapi_<span class="dv">0</span>.<span class="fl">16.0</span>          </span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true"></a>  [<span class="dv">3</span>] jsonlite_<span class="dv">1</span>.<span class="fl">8.8</span>              magrittr_<span class="dv">2</span>.<span class="fl">0.3</span>             </span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true"></a>  [<span class="dv">5</span>] magick_<span class="dv">2</span>.<span class="fl">8.4</span>                farver_<span class="dv">2</span>.<span class="fl">1.2</span>               </span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true"></a>  [<span class="dv">7</span>] rmarkdown_<span class="fl">2.27</span>              zlibbioc_<span class="dv">1</span>.<span class="fl">50.0</span>            </span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true"></a>  [<span class="dv">9</span>] ragg_<span class="dv">1</span>.<span class="fl">3.2</span>                  vctrs_<span class="dv">0</span>.<span class="fl">6.5</span>                </span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true"></a> [<span class="dv">11</span>] DelayedMatrixStats_<span class="dv">1</span>.<span class="fl">26.0</span>   GiottoUtils_<span class="dv">0</span>.<span class="fl">1.11</span>         </span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true"></a> [<span class="dv">13</span>] terra_<span class="fl">1.7</span><span class="dv">-78</span>                htmltools_<span class="dv">0</span>.<span class="dv">5</span>.<span class="fl">8.1</span>          </span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true"></a> [<span class="dv">15</span>] S4Arrays_<span class="dv">1</span>.<span class="fl">4.1</span>              BiocNeighbors_<span class="dv">1</span>.<span class="fl">22.0</span>       </span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true"></a> [<span class="dv">17</span>] SparseArray_<span class="dv">1</span>.<span class="fl">4.8</span>           parallelly_<span class="dv">1</span>.<span class="fl">38.0</span>          </span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true"></a> [<span class="dv">19</span>] htmlwidgets_<span class="dv">1</span>.<span class="fl">6.4</span>           plyr_<span class="dv">1</span>.<span class="fl">8.9</span>                 </span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true"></a> [<span class="dv">21</span>] plotly_<span class="dv">4</span>.<span class="fl">10.4</span>               igraph_<span class="dv">2</span>.<span class="fl">0.3</span>               </span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true"></a> [<span class="dv">23</span>] lifecycle_<span class="dv">1</span>.<span class="fl">0.4</span>             pkgconfig_<span class="dv">2</span>.<span class="fl">0.3</span>            </span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true"></a> [<span class="dv">25</span>] rsvd_<span class="dv">1</span>.<span class="fl">0.5</span>                  Matrix_<span class="fl">1.7</span><span class="dv">-0</span>               </span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true"></a> [<span class="dv">27</span>] R6_<span class="dv">2</span>.<span class="fl">5.1</span>                    fastmap_<span class="dv">1</span>.<span class="fl">2.0</span>              </span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true"></a> [<span class="dv">29</span>] GenomeInfoDbData_<span class="dv">1</span>.<span class="fl">2.12</span>     MatrixGenerics_<span class="dv">1</span>.<span class="fl">16.0</span>      </span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true"></a> [<span class="dv">31</span>] future_<span class="dv">1</span>.<span class="fl">34.0</span>               digest_<span class="dv">0</span>.<span class="fl">6.36</span>              </span>
<span id="cb45-38"><a href="#cb45-38" aria-hidden="true"></a> [<span class="dv">33</span>] colorspace_<span class="fl">2.1</span><span class="dv">-1</span>            S4Vectors_<span class="dv">0</span>.<span class="fl">42.1</span>           </span>
<span id="cb45-39"><a href="#cb45-39" aria-hidden="true"></a> [<span class="dv">35</span>] dqrng_<span class="dv">0</span>.<span class="fl">4.1</span>                 irlba_<span class="dv">2</span>.<span class="dv">3</span>.<span class="fl">5.1</span>              </span>
<span id="cb45-40"><a href="#cb45-40" aria-hidden="true"></a> [<span class="dv">37</span>] textshaping_<span class="dv">0</span>.<span class="fl">4.0</span>           GenomicRanges_<span class="dv">1</span>.<span class="fl">56.1</span>       </span>
<span id="cb45-41"><a href="#cb45-41" aria-hidden="true"></a> [<span class="dv">39</span>] beachmat_<span class="dv">2</span>.<span class="fl">20.0</span>             labeling_<span class="dv">0</span>.<span class="fl">4.3</span>             </span>
<span id="cb45-42"><a href="#cb45-42" aria-hidden="true"></a> [<span class="dv">41</span>] RcppZiggurat_<span class="dv">0</span>.<span class="fl">1.6</span>          progressr_<span class="dv">0</span>.<span class="fl">14.0</span>           </span>
<span id="cb45-43"><a href="#cb45-43" aria-hidden="true"></a> [<span class="dv">43</span>] fansi_<span class="dv">1</span>.<span class="fl">0.6</span>                 httr_<span class="dv">1</span>.<span class="fl">4.7</span>                 </span>
<span id="cb45-44"><a href="#cb45-44" aria-hidden="true"></a> [<span class="dv">45</span>] abind_<span class="fl">1.4</span><span class="dv">-5</span>                 compiler_<span class="dv">4</span>.<span class="fl">4.0</span>             </span>
<span id="cb45-45"><a href="#cb45-45" aria-hidden="true"></a> [<span class="dv">47</span>] withr_<span class="dv">3</span>.<span class="fl">0.0</span>                 backports_<span class="dv">1</span>.<span class="fl">5.0</span>            </span>
<span id="cb45-46"><a href="#cb45-46" aria-hidden="true"></a> [<span class="dv">49</span>] BiocParallel_<span class="dv">1</span>.<span class="fl">38.0</span>         R.utils_<span class="dv">2</span>.<span class="fl">12.3</span>             </span>
<span id="cb45-47"><a href="#cb45-47" aria-hidden="true"></a> [<span class="dv">51</span>] DelayedArray_<span class="dv">0</span>.<span class="fl">30.1</span>         rjson_<span class="dv">0</span>.<span class="fl">2.21</span>               </span>
<span id="cb45-48"><a href="#cb45-48" aria-hidden="true"></a> [<span class="dv">53</span>] bluster_<span class="dv">1</span>.<span class="fl">14.0</span>              gtools_<span class="dv">3</span>.<span class="fl">9.5</span>               </span>
<span id="cb45-49"><a href="#cb45-49" aria-hidden="true"></a> [<span class="dv">55</span>] GiottoVisuals_<span class="dv">0</span>.<span class="fl">2.5</span>         tools_<span class="dv">4</span>.<span class="fl">4.0</span>                </span>
<span id="cb45-50"><a href="#cb45-50" aria-hidden="true"></a> [<span class="dv">57</span>] future.apply_<span class="dv">1</span>.<span class="fl">11.2</span>         quadprog_<span class="fl">1.5</span><span class="dv">-8</span>             </span>
<span id="cb45-51"><a href="#cb45-51" aria-hidden="true"></a> [<span class="dv">59</span>] R.oo_<span class="dv">1</span>.<span class="fl">26.0</span>                 glue_<span class="dv">1</span>.<span class="fl">7.0</span>                 </span>
<span id="cb45-52"><a href="#cb45-52" aria-hidden="true"></a> [<span class="dv">61</span>] dbscan_<span class="fl">1.2</span><span class="dv">-0</span>                grid_<span class="dv">4</span>.<span class="fl">4.0</span>                 </span>
<span id="cb45-53"><a href="#cb45-53" aria-hidden="true"></a> [<span class="dv">63</span>] checkmate_<span class="dv">2</span>.<span class="fl">3.2</span>             cluster_<span class="dv">2</span>.<span class="fl">1.6</span>              </span>
<span id="cb45-54"><a href="#cb45-54" aria-hidden="true"></a> [<span class="dv">65</span>] reshape2_<span class="dv">1</span>.<span class="fl">4.4</span>              generics_<span class="dv">0</span>.<span class="fl">1.3</span>             </span>
<span id="cb45-55"><a href="#cb45-55" aria-hidden="true"></a> [<span class="dv">67</span>] gtable_<span class="dv">0</span>.<span class="fl">3.5</span>                R.methodsS3_<span class="dv">1</span>.<span class="fl">8.2</span>          </span>
<span id="cb45-56"><a href="#cb45-56" aria-hidden="true"></a> [<span class="dv">69</span>] tidyr_<span class="dv">1</span>.<span class="fl">3.1</span>                 data.table_<span class="dv">1</span>.<span class="fl">15.4</span>          </span>
<span id="cb45-57"><a href="#cb45-57" aria-hidden="true"></a> [<span class="dv">71</span>] BiocSingular_<span class="dv">1</span>.<span class="fl">20.0</span>         ScaledMatrix_<span class="dv">1</span>.<span class="fl">12.0</span>        </span>
<span id="cb45-58"><a href="#cb45-58" aria-hidden="true"></a> [<span class="dv">73</span>] metapod_<span class="dv">1</span>.<span class="fl">12.0</span>              sp_<span class="fl">2.1</span><span class="dv">-4</span>                   </span>
<span id="cb45-59"><a href="#cb45-59" aria-hidden="true"></a> [<span class="dv">75</span>] utf8_<span class="dv">1</span>.<span class="fl">2.4</span>                  XVector_<span class="dv">0</span>.<span class="fl">44.0</span>             </span>
<span id="cb45-60"><a href="#cb45-60" aria-hidden="true"></a> [<span class="dv">77</span>] BiocGenerics_<span class="dv">0</span>.<span class="fl">50.0</span>         RcppAnnoy_<span class="dv">0</span>.<span class="fl">0.22</span>           </span>
<span id="cb45-61"><a href="#cb45-61" aria-hidden="true"></a> [<span class="dv">79</span>] ggrepel_<span class="dv">0</span>.<span class="fl">9.5</span>               pillar_<span class="dv">1</span>.<span class="fl">9.0</span>               </span>
<span id="cb45-62"><a href="#cb45-62" aria-hidden="true"></a> [<span class="dv">81</span>] stringr_<span class="dv">1</span>.<span class="fl">5.1</span>               limma_<span class="dv">3</span>.<span class="fl">60.4</span>               </span>
<span id="cb45-63"><a href="#cb45-63" aria-hidden="true"></a> [<span class="dv">83</span>] dplyr_<span class="dv">1</span>.<span class="fl">1.4</span>                 lattice_<span class="fl">0.22</span><span class="dv">-6</span>             </span>
<span id="cb45-64"><a href="#cb45-64" aria-hidden="true"></a> [<span class="dv">85</span>] deldir_<span class="fl">2.0</span><span class="dv">-4</span>                tidyselect_<span class="dv">1</span>.<span class="fl">2.1</span>           </span>
<span id="cb45-65"><a href="#cb45-65" aria-hidden="true"></a> [<span class="dv">87</span>] SingleCellExperiment_<span class="dv">1</span>.<span class="fl">26.0</span> locfit_<span class="fl">1.5-9.10</span>            </span>
<span id="cb45-66"><a href="#cb45-66" aria-hidden="true"></a> [<span class="dv">89</span>] scuttle_<span class="dv">1</span>.<span class="fl">14.0</span>              knitr_<span class="fl">1.48</span>                 </span>
<span id="cb45-67"><a href="#cb45-67" aria-hidden="true"></a> [<span class="dv">91</span>] IRanges_<span class="dv">2</span>.<span class="fl">38.1</span>              edgeR_<span class="dv">4</span>.<span class="fl">2.1</span>                </span>
<span id="cb45-68"><a href="#cb45-68" aria-hidden="true"></a> [<span class="dv">93</span>] SummarizedExperiment_<span class="dv">1</span>.<span class="fl">34.0</span> scattermore_<span class="fl">1.2</span>            </span>
<span id="cb45-69"><a href="#cb45-69" aria-hidden="true"></a> [<span class="dv">95</span>] RhpcBLASctl_<span class="fl">0.23</span><span class="dv">-42</span>         stats4_<span class="dv">4</span>.<span class="fl">4.0</span>               </span>
<span id="cb45-70"><a href="#cb45-70" aria-hidden="true"></a> [<span class="dv">97</span>] xfun_<span class="fl">0.46</span>                   Biobase_<span class="dv">2</span>.<span class="fl">64.0</span>             </span>
<span id="cb45-71"><a href="#cb45-71" aria-hidden="true"></a> [<span class="dv">99</span>] statmod_<span class="dv">1</span>.<span class="fl">5.0</span>               matrixStats_<span class="dv">1</span>.<span class="fl">3.0</span>          </span>
<span id="cb45-72"><a href="#cb45-72" aria-hidden="true"></a>[<span class="dv">101</span>] stringi_<span class="dv">1</span>.<span class="fl">8.4</span>               UCSC.utils_<span class="dv">1</span>.<span class="fl">0.0</span>           </span>
<span id="cb45-73"><a href="#cb45-73" aria-hidden="true"></a>[<span class="dv">103</span>] lazyeval_<span class="dv">0</span>.<span class="fl">2.2</span>              yaml_<span class="dv">2</span>.<span class="fl">3.10</span>                </span>
<span id="cb45-74"><a href="#cb45-74" aria-hidden="true"></a>[<span class="dv">105</span>] evaluate_<span class="dv">0</span>.<span class="fl">24.0</span>             codetools_<span class="fl">0.2</span><span class="dv">-20</span>           </span>
<span id="cb45-75"><a href="#cb45-75" aria-hidden="true"></a>[<span class="dv">107</span>] GiottoData_<span class="dv">0</span>.<span class="fl">2.13</span>           tibble_<span class="dv">3</span>.<span class="fl">2.1</span>               </span>
<span id="cb45-76"><a href="#cb45-76" aria-hidden="true"></a>[<span class="dv">109</span>] colorRamp2_<span class="dv">0</span>.<span class="fl">1.0</span>            cli_<span class="dv">3</span>.<span class="fl">6.3</span>                  </span>
<span id="cb45-77"><a href="#cb45-77" aria-hidden="true"></a>[<span class="dv">111</span>] RcppParallel_<span class="dv">5</span>.<span class="fl">1.8</span>          uwot_<span class="dv">0</span>.<span class="fl">2.2</span>                 </span>
<span id="cb45-78"><a href="#cb45-78" aria-hidden="true"></a>[<span class="dv">113</span>] reticulate_<span class="dv">1</span>.<span class="fl">38.0</span>           systemfonts_<span class="dv">1</span>.<span class="fl">1.0</span>          </span>
<span id="cb45-79"><a href="#cb45-79" aria-hidden="true"></a>[<span class="dv">115</span>] munsell_<span class="dv">0</span>.<span class="fl">5.1</span>               harmony_<span class="dv">1</span>.<span class="fl">2.0</span>              </span>
<span id="cb45-80"><a href="#cb45-80" aria-hidden="true"></a>[<span class="dv">117</span>] Rcpp_<span class="dv">1</span>.<span class="fl">0.13</span>                 GenomeInfoDb_<span class="dv">1</span>.<span class="fl">40.1</span>        </span>
<span id="cb45-81"><a href="#cb45-81" aria-hidden="true"></a>[<span class="dv">119</span>] globals_<span class="dv">0</span>.<span class="fl">16.3</span>              png_<span class="fl">0.1</span><span class="dv">-8</span>                  </span>
<span id="cb45-82"><a href="#cb45-82" aria-hidden="true"></a>[<span class="dv">121</span>] Rfast_<span class="dv">2</span>.<span class="fl">1.0</span>                 parallel_<span class="dv">4</span>.<span class="fl">4.0</span>             </span>
<span id="cb45-83"><a href="#cb45-83" aria-hidden="true"></a>[<span class="dv">123</span>] scran_<span class="dv">1</span>.<span class="fl">32.0</span>                sparseMatrixStats_<span class="dv">1</span>.<span class="fl">16.0</span>   </span>
<span id="cb45-84"><a href="#cb45-84" aria-hidden="true"></a>[<span class="dv">125</span>] listenv_<span class="dv">0</span>.<span class="fl">9.1</span>               SpatialExperiment_<span class="dv">1</span>.<span class="fl">14.0</span>   </span>
<span id="cb45-85"><a href="#cb45-85" aria-hidden="true"></a>[<span class="dv">127</span>] viridisLite_<span class="dv">0</span>.<span class="fl">4.2</span>           scales_<span class="dv">1</span>.<span class="fl">3.0</span>               </span>
<span id="cb45-86"><a href="#cb45-86" aria-hidden="true"></a>[<span class="dv">129</span>] purrr_<span class="dv">1</span>.<span class="fl">0.2</span>                 crayon_<span class="dv">1</span>.<span class="fl">5.3</span>               </span>
<span id="cb45-87"><a href="#cb45-87" aria-hidden="true"></a>[<span class="dv">131</span>] rlang_<span class="dv">1</span>.<span class="fl">1.4</span>                 cowplot_<span class="dv">1</span>.<span class="fl">1.3</span>   </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. ChÃ¡vez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew Oâ€™Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
