<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visium CytAssist Multi-omics Human Glioblastoma • Giotto</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visium CytAssist Multi-omics Human Glioblastoma">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-2V1K0KPLJC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2V1K0KPLJC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Get started

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/installation.html">Giotto Installation</a>
    </li>
    <li>
      <a href="../articles/ecosystem.html">Giotto Ecosystem</a>
    </li>
    <li>
      <a href="../articles/structure.html">Giotto Class Structure</a>
    </li>
    <li>
      <a href="../articles/faqs.html">FAQs</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">How to report a bug</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Examples

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Seq-Scope</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/seqscope_mouse_liver.html">Mouse liver</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visium_mouse_kidney.html">Visium Mouse Kidney</a>
        </li>
        <li>
          <a href="../articles/visium_mouse_brain.html">Visium Mouse Brain</a>
        </li>
        <li>
          <a href="../articles/visium_prostate_integration.html">Visium Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_lungcancer.html">CytAssist Human Lung Cancer</a>
        </li>
        <li>
          <a href="../articles/visium_cytassist_human_glioblastoma.html">CytAssist Multi-omics Human Glioblastoma</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Slide-seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/slideseq_mouse_brain.html">Slide-seq Mouse brain</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial Genomics</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_genomics_mouse_kidney.html">Mouse Kidney</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">DBiT-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mouse_embryo_dbitseq.html">Multi-omics Mouse Embryo DBiT-Seq</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial CITE-Seq</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_citeseq_human_skin.html">Multi-omics Spatial CITE-Seq Human skin</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Xenium</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/xenium_breast_cancer.html">Xenium Human breast cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Nanostring CosMx</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/nanostring_cosmx_lung_cancer.html">Nanostring CosMx Subcellular Lung Cancer</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">MERFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/merfish_mouse_hypothalamic.html">MERFISH Mouse Hypothalamic Preoptic Region</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Vizgen</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/vizgen_mouse_brain.html">Vizgen Mouse Brain Receptor Map</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">seqFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/mini_seqfish.html">mini seqFISH</a>
        </li>
        <li>
          <a href="../articles/seqfish_cortex.html">SeqFish+ Mouse Cortex Example</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">osmFISH</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/osmfish_mouse_ss_cortex.html">osmFISH Mouse SS Cortex</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Resolve Bioscience</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/resolve_biosciences.html">Resolve Bioscience Breast Cancer Subcellular</a>
        </li>
        <li>
          <a href="../articles/resolve_biosciences_human_lung.html">Resolve Bioscience Human lung</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">CODEX</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/codex_mouse_spleen.html">Codex Mouse Spleen</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Single Cell RNA Sequencing</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/singlecell_prostate_standard.html">Prostate</a>
        </li>
        <li>
          <a href="../articles/singlecell_prostate_integration.html">Prostate Integration</a>
        </li>
        <li>
          <a href="../articles/singlecell_lung_adenocarcinoma.html">Human Lung Adenocarcinoma</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Object Creation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/object_creation.html">Giotto Object Creation</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Giotto Configuration</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/configuration.html">Giotto Configuration</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Spatial manipulation</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/spatial_transformations.html">Spatial transformations</a>
        </li>
        <li>
          <a href="https://drieslab.github.io/GiottoClass/articles/spatial_geometries.html" class="external-link">Spatial geometries</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Image alignment</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/image_alignment.html">Image alignment</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interactive selection</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interactive_selection.html">Interactive selection</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Visualizations</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/visualizations.html">Visualizations</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Saving and Loading</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/saving_object.html">Saving a Giotto object</a>
        </li>
        <li>
          <a href="../articles/saving_plots.html">Saving plots</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Interoperability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/interoperability_anndata.html">AnnData</a>
        </li>
        <li>
          <a href="../articles/interoperability_spatialexperiment.html">SpatialExperiment</a>
        </li>
        <li>
          <a href="../articles/interoperability_seurat.html">Seurat</a>
        </li>
        <li>
          <a href="../articles/bento.html">Integrate Bento Analysis</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Try Giotto in the Cloud</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/terra.html">Giotto on Terra.bio</a>
        </li>
        <li>
          <a href="../articles/nemo_slideseq.html">NeMO Slide-seq on Terra.bio</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Containers</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/docker.html">Docker</a>
        </li>
        <li>
          <a href="../articles/singularity.html">Singularity</a>
        </li>
      </ul>
</li>
    <li class="divider">
    </li>
<li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Scalability</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/dbMatrix.html">Using dbMatrix with Giotto</a>
        </li>
      </ul>
</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Changes</a>
    </li>
    <li>
      <a href="../articles/presentations.html">Presentations</a>
    </li>
    <li>
      <a href="../articles/branches.html">Branches</a>
    </li>
  </ul>
</li>
<li>
  <a></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drieslab/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Visium CytAssist Multi-omics Human Glioblastoma</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drieslab/Giotto/blob/suite/vignettes/visium_cytassist_human_glioblastoma.Rmd" class="external-link"><code>vignettes/visium_cytassist_human_glioblastoma.Rmd</code></a></small>
      <div class="hidden name"><code>visium_cytassist_human_glioblastoma.Rmd</code></div>

    </div>

    
    
<div class="section level2" number="1">
<h2 number="1" id="dataset-explanation">
<span class="header-section-number">1</span> Dataset explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p>The <a href="https://www.10xgenomics.com/resources/datasets/gene-and-protein-expression-library-of-human-glioblastoma-cytassist-ffpe-2-standard/" class="external-link">Human glioblastoma (FFPE) dataset</a> was obtained from 10x Genomics. The tissue was sectioned as described in Visium CytAssist Spatial Gene Expression for FFPE – Tissue Preparation Guide Demonstrated Protocol (CG000518). 5 µm tissue sections were placed on Superfrost glass slides, then IF stained following deparaffinization, then hard coverslipped. Sections were imaged, decoverslipped, followed by Demonstrated Protocol (CG000494).</p>
<p>More information about this dataset can be found <a href="https://www.10xgenomics.com/resources/datasets/gene-and-protein-expression-library-of-human-glioblastoma-cytassist-ffpe-2-standard" class="external-link">here</a>.</p>
</div>
<div class="section level2" number="2">
<h2 number="2" id="start-giotto">
<span class="header-section-number">2</span> Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure Giotto Suite is installed</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"Giotto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure the Python environment for Giotto has been installed</span></span>
<span><span class="va">genv_exists</span> <span class="op">&lt;-</span> <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/checkGiottoEnvironment.html" class="external-link">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment</span></span>
<span>  <span class="fu">Giotto</span><span class="fu">::</span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/installGiottoEnvironment.html" class="external-link">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drieslab.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 1. set results directory</span></span>
<span><span class="va">results_folder</span> <span class="op">&lt;-</span> <span class="st">"/path/to/results/"</span></span>
<span></span>
<span><span class="co"># 2. set giotto python path</span></span>
<span><span class="co"># set python path to your preferred python version path</span></span>
<span><span class="co"># set python path to NULL if you want to automatically install (only the 1st time) and use the giotto miniconda environment</span></span>
<span><span class="va">python_path</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="co"># 3. create giotto instructions</span></span>
<span><span class="va">instructions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createGiottoInstructions.html" class="external-link">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">results_folder</span>, </span>
<span>                                         save_plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                         show_plot <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                         return_plot <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                         python_path <span class="op">=</span> <span class="va">python_path</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 number="3" id="create-giotto-object">
<span class="header-section-number">3</span> Create Giotto object<a class="anchor" aria-label="anchor" href="#create-giotto-object"></a>
</h2>
<p>The minimum requirements are</p>
<ul>
<li>matrix with expression information (or path to)</li>
<li>x,y(,z) coordinates for cells or spots (or path to)</li>
</ul>
<p>createGiottoVisiumObject() will automatically detect both rna and protein modalities within the expression matrix creating a multi-omics Giotto object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Provide path to visium folder</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="st">"/path/to/data/"</span></span>
<span></span>
<span><span class="co"># Create Giotto object</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createGiottoVisiumObject.html">createGiottoVisiumObject</a></span><span class="op">(</span>visium_dir <span class="op">=</span> <span class="va">data_path</span>,</span>
<span>                                  expr_data <span class="op">=</span> <span class="st">"raw"</span>,</span>
<span>                                  png_name <span class="op">=</span> <span class="st">"tissue_lowres_image.png"</span>,</span>
<span>                                  gene_column_index <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                  instructions <span class="op">=</span> <span class="va">instructions</span><span class="op">)</span></span></code></pre></div>
<p>How to work with Giotto instructions that are part of your Giotto object:</p>
<ul>
<li>Show the instructions associated with your Giotto object with showGiottoInstructions()</li>
<li>Change one or more instructions with changeGiottoInstructions()</li>
<li>Replace all instructions at once with replaceGiottoInstructions()</li>
<li>Read or get a specific Giotto instruction with readGiottoInstructions()</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show instructions associated with the giotto object</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/showGiottoInstructions.html" class="external-link">showGiottoInstructions</a></span><span class="op">(</span><span class="va">visium</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 number="4" id="processing">
<span class="header-section-number">4</span> Processing<a class="anchor" aria-label="anchor" href="#processing"></a>
</h2>
<ul>
<li>Filter features and cells based on detection frequencies</li>
<li>Normalize expression matrix (log transformation, scaling factor and/or z-scores)</li>
<li>Add cell and feature statistics (optional)</li>
<li>Adjust expression matrix for technical covariates or batches (optional).</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Subset on spots that were covered by tissue</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/pDataDT.html" class="external-link">pDataDT</a></span><span class="op">(</span><span class="va">visium</span><span class="op">)</span></span>
<span><span class="va">in_tissue_barcodes</span> <span class="op">&lt;-</span> <span class="va">metadata</span><span class="op">[</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span></span>
<span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/subsetGiotto.html" class="external-link">subsetGiotto</a></span><span class="op">(</span><span class="va">visium</span>, </span>
<span>                       cell_ids <span class="op">=</span> <span class="va">in_tissue_barcodes</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Visualize aligned tissue</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/0-spatPlot2D.png" width="60%"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filtering, normalization, and statistics</span></span>
<span></span>
<span><span class="co">## RNA feature</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                       expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                       feat_det_in_min_cells <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                       min_det_feats_per_cell <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>                       expression_values <span class="op">=</span> <span class="st">"raw"</span>,</span>
<span>                       verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                          scalefactor <span class="op">=</span> <span class="fl">6000</span>,</span>
<span>                          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Visualize number of features after processing</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"nr_feats"</span>,</span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/1-spatPlot2D.png" width="60%"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Protein feature </span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>, </span>
<span>                       spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                       feat_type <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                       expression_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                       feat_det_in_min_cells <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                       min_det_feats_per_cell <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                       expression_values <span class="op">=</span> <span class="st">"raw"</span>, </span>
<span>                       verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                          spat_unit <span class="op">=</span> <span class="st">"cell"</span>, </span>
<span>                          feat_type <span class="op">=</span> <span class="st">"protein"</span>, </span>
<span>                          scalefactor <span class="op">=</span> <span class="fl">6000</span>, </span>
<span>                          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                        spat_unit <span class="op">=</span> <span class="st">"cell"</span>, </span>
<span>                        feat_type <span class="op">=</span> <span class="st">"protein"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Visualize number of features after processing </span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>           spat_unit <span class="op">=</span> <span class="st">"cell"</span>, </span>
<span>           feat_type <span class="op">=</span> <span class="st">"protein"</span>, </span>
<span>           point_alpha <span class="op">=</span> <span class="fl">0.7</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">"nr_feats"</span>, </span>
<span>           color_as_factor <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/2-spatPlot2D.png" width="60%"></p>
</div>
<div class="section level2" number="5">
<h2 number="5" id="dimension-reduction">
<span class="header-section-number">5</span> Dimension Reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify highly variable features (HVF)</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateHVF.html">calculateHVF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># PCA</span></span>
<span></span>
<span><span class="co">## RNA</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">visium</span>, ncp <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/3-screePlot.png" width="60%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Visualize RNA PCA</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/4-PCA.png" width="60%"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Protein</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                 spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                 feat_type <span class="op">=</span> <span class="st">"protein"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">visium</span>,</span>
<span>          spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>          feat_type <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>          ncp <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/5-screePlot.png" width="60%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Visualize Protein PCA</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>        spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>        feat_type <span class="op">=</span> <span class="st">"protein"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/6-PCA.png" width="60%"></p>
</div>
<div class="section level2" number="6">
<h2 number="6" id="clustering">
<span class="header-section-number">6</span> Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cluster and run UMAP</span></span>
<span><span class="co"># sNN network (default)</span></span>
<span></span>
<span><span class="co">## RNA feature</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                               dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                               k <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Protein feature</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                               spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                               feat_type <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                               dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                               k <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Leiden clustering</span></span>
<span></span>
<span><span class="co">## RNA feature</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                          resolution <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                          n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Protein feature</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                          spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                          feat_type <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                          resolution <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                          n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># UMAP</span></span>
<span></span>
<span><span class="co">## RNA feature</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">visium</span>,</span>
<span>                  dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/7-UMAP.png" width="60%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Protein feature </span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">visium</span>,</span>
<span>                  spat_unit <span class="op">=</span> <span class="st">"cell"</span>, </span>
<span>                  feat_type <span class="op">=</span> <span class="st">"protein"</span>, </span>
<span>                  dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>         spat_unit <span class="op">=</span> <span class="st">"cell"</span>, </span>
<span>         feat_type <span class="op">=</span> <span class="st">"protein"</span>, </span>
<span>         cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>, </span>
<span>         show_NN_network <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>         point_size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/8-UMAP.png" width="60%"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize spatial plot</span></span>
<span></span>
<span><span class="co">## RNA feature</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/9-spatPlot2D.png" width="60%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Protein feature</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>           spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>           feat_type <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"leiden_clus"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/10-spatPlot2D.png" width="60%"></p>
</div>
<div class="section level2" number="7">
<h2 number="7" id="multi-omics-integration">
<span class="header-section-number">7</span> Multi-omics integration<a class="anchor" aria-label="anchor" href="#multi-omics-integration"></a>
</h2>
<p>The Weighted Nearest Neighbors allows to integrate two or more modalities acquired from the same sample. WNN will re-calculate the clustering to provide an integrated umap and leiden clustering. For running WNN, the Giotto object must contain the results of running PCA calculation for each modality.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate kNN</span></span>
<span></span>
<span><span class="co">## RNA modality</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                               type <span class="op">=</span> <span class="st">"kNN"</span>,</span>
<span>                               dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                               k <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Protein modality</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://drieslab.github.io/GiottoClass/reference/createNearestNetwork.html" class="external-link">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                               spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                               feat_type <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                               type <span class="op">=</span> <span class="st">"kNN"</span>,</span>
<span>                               dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                               k <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run WNN</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runWNN.html">runWNN</a></span><span class="op">(</span><span class="va">visium</span>,</span>
<span>                 spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                 modality_1 <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                 modality_2 <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                 pca_name_modality_1 <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>                 pca_name_modality_2 <span class="op">=</span> <span class="st">"protein.pca"</span>,</span>
<span>                 k <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                 integrated_feat_type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 matrix_result_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 w_name_modality_1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 w_name_modality_2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run Integrated umap</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runIntegratedUMAP.html">runIntegratedUMAP</a></span><span class="op">(</span><span class="va">visium</span>,</span>
<span>                            modality1 <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                            modality2 <span class="op">=</span> <span class="st">"protein"</span>,</span>
<span>                            spread <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                            min_dist <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                            force <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate integrated clusters</span></span>
<span><span class="va">visium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>                          spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                          feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>                          nn_network_to_use <span class="op">=</span> <span class="st">"kNN"</span>,</span>
<span>                          network_name <span class="op">=</span> <span class="st">"integrated_kNN"</span>,</span>
<span>                          name <span class="op">=</span> <span class="st">"integrated_leiden_clus"</span>,</span>
<span>                          resolution <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize integrated umap</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/plotUMAP.html" class="external-link">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">visium</span>,</span>
<span>         spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>         feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>         cell_color <span class="op">=</span> <span class="st">"integrated_leiden_clus"</span>,</span>
<span>         dim_reduction_name <span class="op">=</span> <span class="st">"integrated.umap"</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>         title <span class="op">=</span> <span class="st">"Integrated UMAP using Integrated Leiden clusters"</span>,</span>
<span>         axis_title <span class="op">=</span> <span class="fl">12</span>,</span>
<span>         axis_text <span class="op">=</span> <span class="fl">10</span> <span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/11-UMAP.png" width="60%"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize spatial plot with integrated clusters</span></span>
<span><span class="fu"><a href="https://drieslab.github.io/GiottoVisuals/reference/spatPlot.html" class="external-link">spatPlot2D</a></span><span class="op">(</span><span class="va">visium</span>,</span>
<span>           spat_unit <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>           feat_type <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">"integrated_leiden_clus"</span>,</span>
<span>           point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>           show_image <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           title <span class="op">=</span> <span class="st">"Integrated Leiden clustering"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/visium_cytassist_human_glioblastoma/12-spatPlot2D.png" width="60%"></p>
</div>
<div class="section level2" number="8">
<h2 number="8" id="session-info">
<span class="header-section-number">8</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true"></a>R version <span class="dv">4</span>.<span class="fl">4.0</span> (<span class="dv">2024-04-24</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true"></a>Platform<span class="op">:</span><span class="st"> </span>x86_<span class="dv">64</span><span class="op">-</span>apple<span class="op">-</span>darwin20</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true"></a>Running under<span class="op">:</span><span class="st"> </span>macOS Sonoma <span class="fl">14.5</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true"></a>Matrix products<span class="op">:</span><span class="st"> </span>default</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true"></a>BLAS<span class="op">:</span><span class="st">   </span><span class="er">/</span>System<span class="op">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>Accelerate.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>Frameworks<span class="op">/</span>vecLib.framework<span class="op">/</span>Versions<span class="op">/</span>A<span class="op">/</span>libBLAS.dylib </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true"></a>LAPACK<span class="op">:</span><span class="st"> </span><span class="er">/</span>Library<span class="op">/</span>Frameworks<span class="op">/</span>R.framework<span class="op">/</span>Versions<span class="op">/</span><span class="fl">4.4</span><span class="op">-</span>x86_<span class="dv">64</span><span class="op">/</span>Resources<span class="op">/</span>lib<span class="op">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true"></a>locale<span class="op">:</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>C<span class="op">/</span>en_US.UTF<span class="dv">-8</span><span class="op">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true"></a>time zone<span class="op">:</span><span class="st"> </span>America<span class="op">/</span>New_York</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true"></a>tzcode source<span class="op">:</span><span class="st"> </span>internal</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true"></a></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true"></a>attached base packages<span class="op">:</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true"></a></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true"></a>other attached packages<span class="op">:</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true"></a>[<span class="dv">1</span>] Giotto_<span class="dv">4</span>.<span class="fl">1.0</span>      GiottoClass_<span class="dv">0</span>.<span class="fl">3.4</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true"></a></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true"></a>loaded via a <span class="kw">namespace</span> (and not attached)<span class="op">:</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true"></a><span class="st">  </span>[<span class="dv">1</span>] colorRamp2_<span class="dv">0</span>.<span class="fl">1.0</span>            deldir_<span class="fl">2.0</span><span class="dv">-4</span>                rlang_<span class="dv">1</span>.<span class="fl">1.4</span>                </span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true"></a>  [<span class="dv">4</span>] magrittr_<span class="dv">2</span>.<span class="fl">0.3</span>              RcppAnnoy_<span class="dv">0</span>.<span class="fl">0.22</span>            GiottoUtils_<span class="dv">0</span>.<span class="fl">1.10</span>         </span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true"></a>  [<span class="dv">7</span>] matrixStats_<span class="dv">1</span>.<span class="fl">3.0</span>           compiler_<span class="dv">4</span>.<span class="fl">4.0</span>              png_<span class="fl">0.1</span><span class="dv">-8</span>                  </span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true"></a> [<span class="dv">10</span>] systemfonts_<span class="dv">1</span>.<span class="fl">1.0</span>           vctrs_<span class="dv">0</span>.<span class="fl">6.5</span>                 reshape2_<span class="dv">1</span>.<span class="fl">4.4</span>             </span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true"></a> [<span class="dv">13</span>] stringr_<span class="dv">1</span>.<span class="fl">5.1</span>               pkgconfig_<span class="dv">2</span>.<span class="fl">0.3</span>             SpatialExperiment_<span class="dv">1</span>.<span class="fl">14.0</span>   </span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true"></a> [<span class="dv">16</span>] crayon_<span class="dv">1</span>.<span class="fl">5.3</span>                fastmap_<span class="dv">1</span>.<span class="fl">2.0</span>               backports_<span class="dv">1</span>.<span class="fl">5.0</span>            </span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true"></a> [<span class="dv">19</span>] magick_<span class="dv">2</span>.<span class="fl">8.4</span>                XVector_<span class="dv">0</span>.<span class="fl">44.0</span>              labeling_<span class="dv">0</span>.<span class="fl">4.3</span>             </span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true"></a> [<span class="dv">22</span>] utf8_<span class="dv">1</span>.<span class="fl">2.4</span>                  rmarkdown_<span class="fl">2.27</span>              UCSC.utils_<span class="dv">1</span>.<span class="fl">0.0</span>           </span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true"></a> [<span class="dv">25</span>] ragg_<span class="dv">1</span>.<span class="fl">3.2</span>                  purrr_<span class="dv">1</span>.<span class="fl">0.2</span>                 xfun_<span class="fl">0.46</span>                  </span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true"></a> [<span class="dv">28</span>] beachmat_<span class="dv">2</span>.<span class="fl">20.0</span>             zlibbioc_<span class="dv">1</span>.<span class="fl">50.0</span>             GenomeInfoDb_<span class="dv">1</span>.<span class="fl">40.1</span>        </span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true"></a> [<span class="dv">31</span>] jsonlite_<span class="dv">1</span>.<span class="fl">8.8</span>              DelayedArray_<span class="dv">0</span>.<span class="fl">30.1</span>         BiocParallel_<span class="dv">1</span>.<span class="fl">38.0</span>        </span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true"></a> [<span class="dv">34</span>] terra_<span class="fl">1.7</span><span class="dv">-78</span>                irlba_<span class="dv">2</span>.<span class="dv">3</span>.<span class="fl">5.1</span>               parallel_<span class="dv">4</span>.<span class="fl">4.0</span>             </span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true"></a> [<span class="dv">37</span>] R6_<span class="dv">2</span>.<span class="fl">5.1</span>                    stringi_<span class="dv">1</span>.<span class="fl">8.4</span>               RColorBrewer_<span class="fl">1.1</span><span class="dv">-3</span>         </span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true"></a> [<span class="dv">40</span>] reticulate_<span class="dv">1</span>.<span class="fl">38.0</span>           parallelly_<span class="dv">1</span>.<span class="fl">38.0</span>           GenomicRanges_<span class="dv">1</span>.<span class="fl">56.1</span>       </span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true"></a> [<span class="dv">43</span>] scattermore_<span class="fl">1.2</span>             Rcpp_<span class="dv">1</span>.<span class="fl">0.13</span>                 SummarizedExperiment_<span class="dv">1</span>.<span class="fl">34.0</span></span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true"></a> [<span class="dv">46</span>] knitr_<span class="fl">1.48</span>                  future.apply_<span class="dv">1</span>.<span class="fl">11.2</span>         R.utils_<span class="dv">2</span>.<span class="fl">12.3</span>             </span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true"></a> [<span class="dv">49</span>] IRanges_<span class="dv">2</span>.<span class="fl">38.1</span>              Matrix_<span class="fl">1.7</span><span class="dv">-0</span>                igraph_<span class="dv">2</span>.<span class="fl">0.3</span>               </span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true"></a> [<span class="dv">52</span>] tidyselect_<span class="dv">1</span>.<span class="fl">2.1</span>            rstudioapi_<span class="dv">0</span>.<span class="fl">16.0</span>           abind_<span class="fl">1.4</span><span class="dv">-5</span>                </span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true"></a> [<span class="dv">55</span>] yaml_<span class="dv">2</span>.<span class="fl">3.10</span>                 codetools_<span class="fl">0.2</span><span class="dv">-20</span>            listenv_<span class="dv">0</span>.<span class="fl">9.1</span>              </span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true"></a> [<span class="dv">58</span>] lattice_<span class="fl">0.22</span><span class="dv">-6</span>              tibble_<span class="dv">3</span>.<span class="fl">2.1</span>                plyr_<span class="dv">1</span>.<span class="fl">8.9</span>                 </span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true"></a> [<span class="dv">61</span>] Biobase_<span class="dv">2</span>.<span class="fl">64.0</span>              withr_<span class="dv">3</span>.<span class="fl">0.0</span>                 evaluate_<span class="dv">0</span>.<span class="fl">24.0</span>            </span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true"></a> [<span class="dv">64</span>] future_<span class="dv">1</span>.<span class="fl">34.0</span>               pillar_<span class="dv">1</span>.<span class="fl">9.0</span>                MatrixGenerics_<span class="dv">1</span>.<span class="fl">16.0</span>      </span>
<span id="cb20-44"><a href="#cb20-44" aria-hidden="true"></a> [<span class="dv">67</span>] checkmate_<span class="dv">2</span>.<span class="fl">3.2</span>             stats4_<span class="dv">4</span>.<span class="fl">4.0</span>                plotly_<span class="dv">4</span>.<span class="fl">10.4</span>              </span>
<span id="cb20-45"><a href="#cb20-45" aria-hidden="true"></a> [<span class="dv">70</span>] generics_<span class="dv">0</span>.<span class="fl">1.3</span>              dbscan_<span class="fl">1.2</span><span class="dv">-0</span>                sp_<span class="fl">2.1</span><span class="dv">-4</span>                   </span>
<span id="cb20-46"><a href="#cb20-46" aria-hidden="true"></a> [<span class="dv">73</span>] S4Vectors_<span class="dv">0</span>.<span class="fl">42.1</span>            ggplot2_<span class="dv">3</span>.<span class="fl">5.1</span>               munsell_<span class="dv">0</span>.<span class="fl">5.1</span>              </span>
<span id="cb20-47"><a href="#cb20-47" aria-hidden="true"></a> [<span class="dv">76</span>] scales_<span class="dv">1</span>.<span class="fl">3.0</span>                globals_<span class="dv">0</span>.<span class="fl">16.3</span>              gtools_<span class="dv">3</span>.<span class="fl">9.5</span>               </span>
<span id="cb20-48"><a href="#cb20-48" aria-hidden="true"></a> [<span class="dv">79</span>] glue_<span class="dv">1</span>.<span class="fl">7.0</span>                  lazyeval_<span class="dv">0</span>.<span class="fl">2.2</span>              tools_<span class="dv">4</span>.<span class="fl">4.0</span>                </span>
<span id="cb20-49"><a href="#cb20-49" aria-hidden="true"></a> [<span class="dv">82</span>] GiottoVisuals_<span class="dv">0</span>.<span class="fl">2.4</span>         data.table_<span class="dv">1</span>.<span class="fl">15.4</span>           ScaledMatrix_<span class="dv">1</span>.<span class="fl">12.0</span>        </span>
<span id="cb20-50"><a href="#cb20-50" aria-hidden="true"></a> [<span class="dv">85</span>] cowplot_<span class="dv">1</span>.<span class="fl">1.3</span>               grid_<span class="dv">4</span>.<span class="fl">4.0</span>                  tidyr_<span class="dv">1</span>.<span class="fl">3.1</span>                </span>
<span id="cb20-51"><a href="#cb20-51" aria-hidden="true"></a> [<span class="dv">88</span>] colorspace_<span class="fl">2.1</span><span class="dv">-1</span>            SingleCellExperiment_<span class="dv">1</span>.<span class="fl">26.0</span> GenomeInfoDbData_<span class="dv">1</span>.<span class="fl">2.12</span>    </span>
<span id="cb20-52"><a href="#cb20-52" aria-hidden="true"></a> [<span class="dv">91</span>] BiocSingular_<span class="dv">1</span>.<span class="fl">20.0</span>         rsvd_<span class="dv">1</span>.<span class="fl">0.5</span>                  cli_<span class="dv">3</span>.<span class="fl">6.3</span>                  </span>
<span id="cb20-53"><a href="#cb20-53" aria-hidden="true"></a> [<span class="dv">94</span>] textshaping_<span class="dv">0</span>.<span class="fl">4.0</span>           fansi_<span class="dv">1</span>.<span class="fl">0.6</span>                 S4Arrays_<span class="dv">1</span>.<span class="fl">4.1</span>             </span>
<span id="cb20-54"><a href="#cb20-54" aria-hidden="true"></a> [<span class="dv">97</span>] viridisLite_<span class="dv">0</span>.<span class="fl">4.2</span>           dplyr_<span class="dv">1</span>.<span class="fl">1.4</span>                 uwot_<span class="dv">0</span>.<span class="fl">2.2</span>                 </span>
<span id="cb20-55"><a href="#cb20-55" aria-hidden="true"></a>[<span class="dv">100</span>] gtable_<span class="dv">0</span>.<span class="fl">3.5</span>                R.methodsS3_<span class="dv">1</span>.<span class="fl">8.2</span>           digest_<span class="dv">0</span>.<span class="fl">6.36</span>              </span>
<span id="cb20-56"><a href="#cb20-56" aria-hidden="true"></a>[<span class="dv">103</span>] BiocGenerics_<span class="dv">0</span>.<span class="fl">50.0</span>         SparseArray_<span class="dv">1</span>.<span class="fl">4.8</span>           ggrepel_<span class="dv">0</span>.<span class="fl">9.5</span>              </span>
<span id="cb20-57"><a href="#cb20-57" aria-hidden="true"></a>[<span class="dv">106</span>] rjson_<span class="dv">0</span>.<span class="fl">2.21</span>                htmlwidgets_<span class="dv">1</span>.<span class="fl">6.4</span>           farver_<span class="dv">2</span>.<span class="fl">1.2</span>               </span>
<span id="cb20-58"><a href="#cb20-58" aria-hidden="true"></a>[<span class="dv">109</span>] htmltools_<span class="dv">0</span>.<span class="dv">5</span>.<span class="fl">8.1</span>           R.oo_<span class="dv">1</span>.<span class="fl">26.0</span>                 lifecycle_<span class="dv">1</span>.<span class="fl">0.4</span>            </span>
<span id="cb20-59"><a href="#cb20-59" aria-hidden="true"></a>[<span class="dv">112</span>] httr_<span class="dv">1</span>.<span class="fl">4.7</span> </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com" class="external-link">Ruben Dries</a>, Jiaji George Chen, <a href="https://josschavezf.github.io" class="external-link">Joselyn C. Chávez-Fuentes</a>, <a href="https://labs.icahn.mssm.edu/yuanlab/" class="external-link">Guo-Cheng Yuan</a>, Matthew O’Brien, Edward Ruiz, Wen Wang, Natalie Del Rossi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
